<!doctype html><html lang=en-us dir=ltr><head><meta charset=UTF-8><meta name=viewport content="width=device-width,initial-scale=1"><meta name=description content='
  methylKit: DMR 분석
  #

#2025-07-28


  1. Load packages
  #

library("methylKit")
library("genomation")
library("GenomicRanges")

  2. Set path
  #

setwd("/data/home/ysh980101/2309_5-aza/Bismark/sorted_n")
getwd()
&#39;/data1/home/ysh980101/2309_5-aza/Bismark/sorted_n&#39;

  3. Load data
  #

# Define the list containing the bismark coverage files.
covlist <- list(
   "KEB1/KEB01_1_bismark_bt2_pe.sorted_n.deduplicated.bismark.cov.gz",
   "KEB2/KEB02_1_bismark_bt2_pe.sorted_n.deduplicated.bismark.cov.gz",
   "KEB4/KEB04_1_bismark_bt2_pe.sorted_n.deduplicated.bismark.cov.gz")
myobj_lowCov <- methRead(covlist,
           sample.id=list("KEB01","KEB02","KEB04"),
           pipeline = "bismarkCoverage",
           assembly="hg38",
           treatment=c(0,1,2),
           mincov = 3
           )
tiles <- tileMethylCounts(myobj_lowCov,win.size=1000,step.size=1000,cov.bases = 3

tiles.norm <- normalizeCoverage(tiles, method = "median")
meth.tiles <- unite(tiles.norm, destrand=FALSE)
meth.tiles

meth.tilesDf = getData(meth.tiles)
meth.tilesDf_filtered <- meth.tilesDf[!grepl("\\.|\\_|MT", meth.tilesDf$chr), ]
meth.tilesDf_filtered


  4. Test: keb01 vs keb02
  #

meth.tilesDf.1vs2 <- meth.tilesDf_filtered[, 1:10]
meth.tilesDf.1vs2

meth.tiles.1vs2=new("methylBase",meth.tilesDf.1vs2,sample.ids=c("KEB01","KEB02"),
                       assembly="hg38",treatment=c(0,1))

myDiff <- calculateDiffMeth(meth.tiles.1vs2,
                            overdispersion = "MN",
                            adjust="BH")
myDiff <- myDiff[order(myDiff$qvalue), ]
myDiff


  5. Save
  #

myDiffDf = getData(myDiff)
write.table(myDiffDf, file = "/data/home/ysh980101/2310/res_total/myDiff1vs2.tsv", sep = "\t", quote = FALSE, row.names = FALSE)

  6. CpG filtering
  #

차등 메틸화된 모든 region에 대한 test 결과가 myDiff에 있는데 CpG island 부분만 남겨서 downstream 분석을 진행한다.'><meta name=theme-color media="(prefers-color-scheme: light)" content="#ffffff"><meta name=theme-color media="(prefers-color-scheme: dark)" content="#343a40"><meta name=color-scheme content="light dark"><meta property="og:url" content="https://yshghid.github.io/docs/study/bioinformatics/bi31/"><meta property="og:site_name" content=" "><meta property="og:title" content="methylKit: DMR 분석"><meta property="og:description" content='methylKit: DMR 분석 # #2025-07-28
1. Load packages # library("methylKit") library("genomation") library("GenomicRanges") 2. Set path # setwd("/data/home/ysh980101/2309_5-aza/Bismark/sorted_n") getwd() &#39;/data1/home/ysh980101/2309_5-aza/Bismark/sorted_n&#39; 3. Load data # # Define the list containing the bismark coverage files. covlist <- list( "KEB1/KEB01_1_bismark_bt2_pe.sorted_n.deduplicated.bismark.cov.gz", "KEB2/KEB02_1_bismark_bt2_pe.sorted_n.deduplicated.bismark.cov.gz", "KEB4/KEB04_1_bismark_bt2_pe.sorted_n.deduplicated.bismark.cov.gz") myobj_lowCov <- methRead(covlist, sample.id=list("KEB01","KEB02","KEB04"), pipeline = "bismarkCoverage", assembly="hg38", treatment=c(0,1,2), mincov = 3 ) tiles <- tileMethylCounts(myobj_lowCov,win.size=1000,step.size=1000,cov.bases = 3 tiles.norm <- normalizeCoverage(tiles, method = "median") meth.tiles <- unite(tiles.norm, destrand=FALSE) meth.tiles meth.tilesDf = getData(meth.tiles) meth.tilesDf_filtered <- meth.tilesDf[!grepl("\\.|\\_|MT", meth.tilesDf$chr), ] meth.tilesDf_filtered 4. Test: keb01 vs keb02 # meth.tilesDf.1vs2 <- meth.tilesDf_filtered[, 1:10] meth.tilesDf.1vs2 meth.tiles.1vs2=new("methylBase",meth.tilesDf.1vs2,sample.ids=c("KEB01","KEB02"), assembly="hg38",treatment=c(0,1)) myDiff <- calculateDiffMeth(meth.tiles.1vs2, overdispersion = "MN", adjust="BH") myDiff <- myDiff[order(myDiff$qvalue), ] myDiff 5. Save # myDiffDf = getData(myDiff) write.table(myDiffDf, file = "/data/home/ysh980101/2310/res_total/myDiff1vs2.tsv", sep = "\t", quote = FALSE, row.names = FALSE) 6. CpG filtering # 차등 메틸화된 모든 region에 대한 test 결과가 myDiff에 있는데 CpG island 부분만 남겨서 downstream 분석을 진행한다.'><meta property="og:locale" content="en_us"><meta property="og:type" content="article"><meta property="article:section" content="docs"><meta property="article:published_time" content="2025-07-28T00:00:00+00:00"><meta property="article:modified_time" content="2025-07-28T00:00:00+00:00"><meta property="article:tag" content="2025-07"><title>methylKit: DMR 분석 |</title><link rel=icon href=/favicon.png><link rel=manifest href=/manifest.json><link rel=canonical href=https://yshghid.github.io/docs/study/bioinformatics/bi31/><link rel=stylesheet href=/book.min.6217d077edb4189fd0578345e84bca1a884dfdee121ff8dc9a0f55cfe0852bc9.css integrity="sha256-YhfQd+20GJ/QV4NF6EvKGohN/e4SH/jcmg9Vz+CFK8k=" crossorigin=anonymous><script defer src=/fuse.min.js></script><script defer src=/en.search.min.578a2d8e6e30e0a0ae3682797882d89ca0cbbf1734d206705396ea76cf57bbb6.js integrity="sha256-V4otjm4w4KCuNoJ5eILYnKDLvxc00gZwU5bqds9Xu7Y=" crossorigin=anonymous></script></head><body dir=ltr><input type=checkbox class="hidden toggle" id=menu-control>
<input type=checkbox class="hidden toggle" id=toc-control><main class="container flex"><aside class=book-menu><div class=book-menu-content><nav><h2 class=book-brand><a class="flex align-center" href=/><img src=/logo.png alt=Logo class=book-icon><span></span></a></h2><div class="book-search hidden"><input type=text id=book-search-input placeholder=Search aria-label=Search maxlength=64 data-hotkeys=s/><div class="book-search-spinner hidden"></div><ul id=book-search-results></ul></div><script>document.querySelector(".book-search").classList.remove("hidden")</script><ul><li class=book-section-flat><span>기록</span><ul><li><a href=/docs/hobby/book/>글</a><ul></ul></li><li><a href=/docs/hobby/daily/>일상</a><ul></ul></li></ul></li><li class=book-section-flat><span>공부</span><ul><li><a href=/docs/study/bioinformatics/>Bioinformatics</a><ul></ul></li><li><a href=/docs/study/ai/>AI</a><ul></ul></li><li><a href=/docs/study/sw/>SW</a><ul></ul></li><li><a href=/docs/study/algorithm/>알고리즘</a><ul></ul></li><li><a href=/docs/study/career/>취업</a><ul></ul></li></ul></li></ul></nav><script>(function(){var e=document.querySelector("aside .book-menu-content");addEventListener("beforeunload",function(){localStorage.setItem("menu.scrollTop",e.scrollTop)}),e.scrollTop=localStorage.getItem("menu.scrollTop")})()</script></div></aside><div class=book-page><header class=book-header><div class="flex align-center justify-between"><label for=menu-control><img src=/svg/menu.svg class=book-icon alt=Menu></label><h3>methylKit: DMR 분석</h3><label for=toc-control><img src=/svg/toc.svg class=book-icon alt="Table of Contents"></label></div><aside class="hidden clearfix"><nav id=TableOfContents><ul><li><ul><li><a href=#1-load-packages>1. Load packages</a></li><li><a href=#2-set-path>2. Set path</a></li><li><a href=#3-load-data>3. Load data</a></li><li><a href=#4-test-keb01-vs-keb02>4. Test: keb01 vs keb02</a></li><li><a href=#5-save>5. Save</a></li><li><a href=#6-cpg-filtering>6. CpG filtering</a></li><li><a href=#7-gene-region-annotation>7. Gene region annotation</a></li><li><a href=#8-save-result>8. Save result</a></li></ul></li></ul></nav></aside></header><article class="markdown book-article"><h1 id=methylkit-dmr-분석>methylKit: DMR 분석
<a class=anchor href=#methylkit-dmr-%eb%b6%84%ec%84%9d>#</a></h1><p>#2025-07-28</p><hr><h3 id=1-load-packages>1. Load packages
<a class=anchor href=#1-load-packages>#</a></h3><div class=highlight><pre tabindex=0 style=color:#e5e5e5;background-color:#000;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-R data-lang=R><span style=display:flex><span>library(<span style=color:#0ff;font-weight:700>&#34;methylKit&#34;</span>)
</span></span><span style=display:flex><span>library(<span style=color:#0ff;font-weight:700>&#34;genomation&#34;</span>)
</span></span><span style=display:flex><span>library(<span style=color:#0ff;font-weight:700>&#34;GenomicRanges&#34;</span>)
</span></span></code></pre></div><h3 id=2-set-path>2. Set path
<a class=anchor href=#2-set-path>#</a></h3><div class=highlight><pre tabindex=0 style=color:#e5e5e5;background-color:#000;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-R data-lang=R><span style=display:flex><span>setwd(<span style=color:#0ff;font-weight:700>&#34;/data/home/ysh980101/2309_5-aza/Bismark/sorted_n&#34;</span>)
</span></span><span style=display:flex><span>getwd()
</span></span></code></pre></div><div class=highlight><pre tabindex=0 style=color:#e5e5e5;background-color:#000;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-plain data-lang=plain><span style=display:flex><span>&#39;/data1/home/ysh980101/2309_5-aza/Bismark/sorted_n&#39;
</span></span></code></pre></div><h3 id=3-load-data>3. Load data
<a class=anchor href=#3-load-data>#</a></h3><div class=highlight><pre tabindex=0 style=color:#e5e5e5;background-color:#000;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-R data-lang=R><span style=display:flex><span><span style=color:#007f7f># Define the list containing the bismark coverage files.</span>
</span></span><span style=display:flex><span>covlist &lt;- list(
</span></span><span style=display:flex><span>   <span style=color:#0ff;font-weight:700>&#34;KEB1/KEB01_1_bismark_bt2_pe.sorted_n.deduplicated.bismark.cov.gz&#34;</span>,
</span></span><span style=display:flex><span>   <span style=color:#0ff;font-weight:700>&#34;KEB2/KEB02_1_bismark_bt2_pe.sorted_n.deduplicated.bismark.cov.gz&#34;</span>,
</span></span><span style=display:flex><span>   <span style=color:#0ff;font-weight:700>&#34;KEB4/KEB04_1_bismark_bt2_pe.sorted_n.deduplicated.bismark.cov.gz&#34;</span>)
</span></span><span style=display:flex><span>myobj_lowCov &lt;- methRead(covlist,
</span></span><span style=display:flex><span>           sample.id=list(<span style=color:#0ff;font-weight:700>&#34;KEB01&#34;</span>,<span style=color:#0ff;font-weight:700>&#34;KEB02&#34;</span>,<span style=color:#0ff;font-weight:700>&#34;KEB04&#34;</span>),
</span></span><span style=display:flex><span>           pipeline = <span style=color:#0ff;font-weight:700>&#34;bismarkCoverage&#34;</span>,
</span></span><span style=display:flex><span>           assembly=<span style=color:#0ff;font-weight:700>&#34;hg38&#34;</span>,
</span></span><span style=display:flex><span>           treatment=c(<span style=color:#ff0;font-weight:700>0</span>,<span style=color:#ff0;font-weight:700>1</span>,<span style=color:#ff0;font-weight:700>2</span>),
</span></span><span style=display:flex><span>           mincov = <span style=color:#ff0;font-weight:700>3</span>
</span></span><span style=display:flex><span>           )
</span></span><span style=display:flex><span>tiles &lt;- tileMethylCounts(myobj_lowCov,win.size=<span style=color:#ff0;font-weight:700>1000</span>,step.size=<span style=color:#ff0;font-weight:700>1000</span>,cov.bases = <span style=color:#ff0;font-weight:700>3</span>
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span>tiles.norm &lt;- normalizeCoverage(tiles, method = <span style=color:#0ff;font-weight:700>&#34;median&#34;</span>)
</span></span><span style=display:flex><span>meth.tiles &lt;- unite(tiles.norm, destrand=<span style=color:#fff;font-weight:700>FALSE</span>)
</span></span><span style=display:flex><span>meth.tiles
</span></span></code></pre></div><img width=1062 height=786 alt=image src=https://github.com/user-attachments/assets/2ab50f79-7f69-4b35-a150-b40beb54709c><div class=highlight><pre tabindex=0 style=color:#e5e5e5;background-color:#000;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-R data-lang=R><span style=display:flex><span>meth.tilesDf = getData(meth.tiles)
</span></span><span style=display:flex><span>meth.tilesDf_filtered &lt;- meth.tilesDf[!grepl(<span style=color:#0ff;font-weight:700>&#34;\\.|\\_|MT&#34;</span>, meth.tilesDf$chr), ]
</span></span><span style=display:flex><span>meth.tilesDf_filtered
</span></span></code></pre></div><img width=1043 height=861 alt=image src=https://github.com/user-attachments/assets/4e79725b-da91-4be6-a71f-f62d4401e883><h3 id=4-test-keb01-vs-keb02>4. Test: keb01 vs keb02
<a class=anchor href=#4-test-keb01-vs-keb02>#</a></h3><div class=highlight><pre tabindex=0 style=color:#e5e5e5;background-color:#000;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-R data-lang=R><span style=display:flex><span>meth.tilesDf.1vs2 &lt;- meth.tilesDf_filtered[, <span style=color:#ff0;font-weight:700>1</span>:<span style=color:#ff0;font-weight:700>10</span>]
</span></span><span style=display:flex><span>meth.tilesDf.1vs2
</span></span></code></pre></div><img width=841 height=744 alt=image src=https://github.com/user-attachments/assets/0f2f6a50-877a-4c77-9e64-3308f42e8361><div class=highlight><pre tabindex=0 style=color:#e5e5e5;background-color:#000;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-R data-lang=R><span style=display:flex><span>meth.tiles.1vs2=new(<span style=color:#0ff;font-weight:700>&#34;methylBase&#34;</span>,meth.tilesDf.1vs2,sample.ids=c(<span style=color:#0ff;font-weight:700>&#34;KEB01&#34;</span>,<span style=color:#0ff;font-weight:700>&#34;KEB02&#34;</span>),
</span></span><span style=display:flex><span>                       assembly=<span style=color:#0ff;font-weight:700>&#34;hg38&#34;</span>,treatment=c(<span style=color:#ff0;font-weight:700>0</span>,<span style=color:#ff0;font-weight:700>1</span>))
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span>myDiff &lt;- calculateDiffMeth(meth.tiles.1vs2,
</span></span><span style=display:flex><span>                            overdispersion = <span style=color:#0ff;font-weight:700>&#34;MN&#34;</span>,
</span></span><span style=display:flex><span>                            adjust=<span style=color:#0ff;font-weight:700>&#34;BH&#34;</span>)
</span></span><span style=display:flex><span>myDiff &lt;- myDiff[order(myDiff$qvalue), ]
</span></span><span style=display:flex><span>myDiff
</span></span></code></pre></div><img width=766 height=807 alt=image src=https://github.com/user-attachments/assets/12ffce09-8aa4-47c7-be34-36090dfdc74c><h3 id=5-save>5. Save
<a class=anchor href=#5-save>#</a></h3><div class=highlight><pre tabindex=0 style=color:#e5e5e5;background-color:#000;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-R data-lang=R><span style=display:flex><span>myDiffDf = getData(myDiff)
</span></span><span style=display:flex><span>write.table(myDiffDf, file = <span style=color:#0ff;font-weight:700>&#34;/data/home/ysh980101/2310/res_total/myDiff1vs2.tsv&#34;</span>, sep = <span style=color:#0ff;font-weight:700>&#34;\t&#34;</span>, quote = <span style=color:#fff;font-weight:700>FALSE</span>, row.names = <span style=color:#fff;font-weight:700>FALSE</span>)
</span></span></code></pre></div><h3 id=6-cpg-filtering>6. CpG filtering
<a class=anchor href=#6-cpg-filtering>#</a></h3><p>차등 메틸화된 모든 region에 대한 test 결과가 myDiff에 있는데 CpG island 부분만 남겨서 downstream 분석을 진행한다.</p><div class=highlight><pre tabindex=0 style=color:#e5e5e5;background-color:#000;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-R data-lang=R><span style=display:flex><span>cpgi.df = read.table(<span style=color:#0ff;font-weight:700>&#34;/data/home/ysh980101/2310/data/hg38_CpGIslands.chr.bed.txt&#34;</span>, header = <span style=color:#fff;font-weight:700>FALSE</span>, stringsAsFactors=<span style=color:#fff;font-weight:700>FALSE</span>)
</span></span><span style=display:flex><span>cpgi.df = cpgi.df[grep(<span style=color:#0ff;font-weight:700>&#34;_&#34;</span>,cpgi.df[,<span style=color:#ff0;font-weight:700>1</span>],invert=<span style=color:#fff;font-weight:700>TRUE</span>),]
</span></span><span style=display:flex><span>cpgi.gr = GRanges(seqnames=cpgi.df[,<span style=color:#ff0;font-weight:700>1</span>],ranges=IRanges(start=cpgi.df[,<span style=color:#ff0;font-weight:700>2</span>],end=cpgi.df[,<span style=color:#ff0;font-weight:700>3</span>]))
</span></span><span style=display:flex><span>myDiff.sbo &lt;- selectByOverlap(myDiff,cpgi.gr)
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span>myDiff.sboDf &lt;- getData(myDiff.sbo)
</span></span><span style=display:flex><span>myDiff.sboDf &lt;- myDiff.sboDf[order(myDiff.sboDf$qvalue), ]
</span></span><span style=display:flex><span>write.table(myDiff.sboDf, file = <span style=color:#0ff;font-weight:700>&#34;/data/home/ysh980101/2310/res_total_cpgfilt/myDiff.sbo1vs2.tsv&#34;</span>, sep = <span style=color:#0ff;font-weight:700>&#34;\t&#34;</span>, quote = <span style=color:#fff;font-weight:700>FALSE</span>, row.names = <span style=color:#fff;font-weight:700>FALSE</span>)
</span></span></code></pre></div><h3 id=7-gene-region-annotation>7. Gene region annotation
<a class=anchor href=#7-gene-region-annotation>#</a></h3><div class=highlight><pre tabindex=0 style=color:#e5e5e5;background-color:#000;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-R data-lang=R><span style=display:flex><span>myDiff.sbo25p &lt;- getMethylDiff(myDiff.sbo,
</span></span><span style=display:flex><span>                        difference=<span style=color:#ff0;font-weight:700>0</span>,
</span></span><span style=display:flex><span>                        qvalue=<span style=color:#ff0;font-weight:700>0.05</span>)
</span></span><span style=display:flex><span>myDiff.sbo25p &lt;- myDiff.sbo25p[order(myDiff.sbo25p$qvalue),]
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span><span style=color:#007f7f>#gene region annotation</span>
</span></span><span style=display:flex><span>refseq_anot &lt;- readTranscriptFeatures(<span style=color:#0ff;font-weight:700>&#34;/data/home/ysh980101/2310/data/hg38_NCBI_RefSeq.chr.bed.txt&#34;</span>)
</span></span><span style=display:flex><span>myDiff.sbo25p.anot &lt;- annotateWithGeneParts(target = as(myDiff.sbo25p,<span style=color:#0ff;font-weight:700>&#34;GRanges&#34;</span>),feature = refseq_anot)
</span></span><span style=display:flex><span>dist_tss &lt;- getAssociationWithTSS(myDiff.sbo25p.anot)
</span></span><span style=display:flex><span>dist_tss
</span></span></code></pre></div><img width=979 height=747 alt=image src=https://github.com/user-attachments/assets/19d08218-583f-4550-a2ed-ec370e593a2f><div class=highlight><pre tabindex=0 style=color:#e5e5e5;background-color:#000;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-R data-lang=R><span style=display:flex><span>plotTargetAnnotation(myDiff.sbo25p.anot, main = <span style=color:#0ff;font-weight:700>&#34;Differential Methylation Annotation&#34;</span>)
</span></span></code></pre></div><img width=1081 height=417 alt=image src=https://github.com/user-attachments/assets/36f8d789-4d0f-4dea-95a6-09c8940041ae><div class=highlight><pre tabindex=0 style=color:#e5e5e5;background-color:#000;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-R data-lang=R><span style=display:flex><span>cpg_anot &lt;- readFeatureFlank(<span style=color:#0ff;font-weight:700>&#34;/data/home/ysh980101/2310/data/hg38_CpGIslands.chr.bed.txt&#34;</span>, feature.flank.name = c(<span style=color:#0ff;font-weight:700>&#34;CpGi&#34;</span>, <span style=color:#0ff;font-weight:700>&#34;shores&#34;</span>), flank=<span style=color:#ff0;font-weight:700>2000</span>)
</span></span><span style=display:flex><span>diffCpGann &lt;- annotateWithFeatureFlank(as(myDiff.sbo25p,<span style=color:#0ff;font-weight:700>&#34;GRanges&#34;</span>), feature = cpg_anot$CpGi, flank = cpg_anot$shores, feature.name = <span style=color:#0ff;font-weight:700>&#34;CpGi&#34;</span>, flank.name = <span style=color:#0ff;font-weight:700>&#34;shores&#34;</span>)
</span></span><span style=display:flex><span>plotTargetAnnotation(diffCpGann, main = <span style=color:#0ff;font-weight:700>&#34;Differential Methylation Annotation&#34;</span>)
</span></span></code></pre></div><img width=963 height=374 alt=image src=https://github.com/user-attachments/assets/0398d795-068f-4695-82a5-3a521d38f0d9><p>(는 cpg island만 필터링했기때문에 당연한 결과)</p><h3 id=8-save-result>8. Save result
<a class=anchor href=#8-save-result>#</a></h3><div class=highlight><pre tabindex=0 style=color:#e5e5e5;background-color:#000;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-R data-lang=R><span style=display:flex><span>write.table(dist_tss, file = <span style=color:#0ff;font-weight:700>&#34;/data/home/ysh980101/2310/res_total_cpgfilt/dist_tss.1vs2.tsv&#34;</span>, sep = <span style=color:#0ff;font-weight:700>&#34;\t&#34;</span>, quote = <span style=color:#fff;font-weight:700>FALSE</span>, row.names = <span style=color:#fff;font-weight:700>FALSE</span>)
</span></span></code></pre></div></article><footer class=book-footer><div class="flex flex-wrap justify-between"></div><script>(function(){function e(e){const t=window.getSelection(),n=document.createRange();n.selectNodeContents(e),t.removeAllRanges(),t.addRange(n)}document.querySelectorAll("pre code").forEach(t=>{t.addEventListener("click",function(){if(window.getSelection().toString())return;e(t.parentElement),navigator.clipboard&&navigator.clipboard.writeText(t.parentElement.textContent)})})})()</script></footer><div class=book-comments><script src=https://giscus.app/client.js data-repo=yshghid/yshghid.github.io data-repo-id=R_kgDONkMkNg data-category-id=DIC_kwDONkMkNs4CloJh data-mapping=pathname data-strict=0 data-reactions-enabled=1 data-emit-metadata=0 data-input-position=bottom data-theme=preferred_color_scheme data-lang=ko crossorigin=anonymous async></script></div><label for=menu-control class="hidden book-menu-overlay"></label></div><aside class=book-toc><div class=book-toc-content><nav id=TableOfContents><ul><li><ul><li><a href=#1-load-packages>1. Load packages</a></li><li><a href=#2-set-path>2. Set path</a></li><li><a href=#3-load-data>3. Load data</a></li><li><a href=#4-test-keb01-vs-keb02>4. Test: keb01 vs keb02</a></li><li><a href=#5-save>5. Save</a></li><li><a href=#6-cpg-filtering>6. CpG filtering</a></li><li><a href=#7-gene-region-annotation>7. Gene region annotation</a></li><li><a href=#8-save-result>8. Save result</a></li></ul></li></ul></nav></div></aside></main></body></html>