<!doctype html><html lang=en-us dir=ltr><head><meta charset=UTF-8><meta name=viewport content="width=device-width,initial-scale=1"><meta name=description content="
  Bioinformatics
  #
"><meta name=theme-color media="(prefers-color-scheme: light)" content="#ffffff"><meta name=theme-color media="(prefers-color-scheme: dark)" content="#343a40"><meta name=color-scheme content="light dark"><meta property="og:url" content="https://yshghid.github.io/docs/study/bioinformatics/"><meta property="og:site_name" content=" "><meta property="og:title" content="Bioinformatics"><meta property="og:description" content="Bioinformatics #"><meta property="og:locale" content="en_us"><meta property="og:type" content="website"><title>Bioinformatics |</title><link rel=icon href=/favicon.png><link rel=manifest href=/manifest.json><link rel=canonical href=https://yshghid.github.io/docs/study/bioinformatics/><link rel=stylesheet href=/book.min.30a7836b6a89342da3b88e7afd1036166aeced16c8de12df060ded2031837886.css integrity="sha256-MKeDa2qJNC2juI56/RA2Fmrs7RbI3hLfBg3tIDGDeIY=" crossorigin=anonymous><script defer src=/fuse.min.js></script><script defer src=/en.search.min.b4b18c56b5773a611c5e8fa1b4891806a62a1d935229d490e91d2cbb8f21a47d.js integrity="sha256-tLGMVrV3OmEcXo+htIkYBqYqHZNSKdSQ6R0su48hpH0=" crossorigin=anonymous></script><link rel=alternate type=application/rss+xml href=https://yshghid.github.io/docs/study/bioinformatics/index.xml title=" "></head><body dir=ltr><input type=checkbox class="hidden toggle" id=menu-control>
<input type=checkbox class="hidden toggle" id=toc-control><main class="container flex"><aside class=book-menu><div class=book-menu-content><nav><h2 class=book-brand><a class="flex align-center" href=/><img src=/logo.png alt=Logo class=book-icon><span></span></a></h2><div class="book-search hidden"><input type=text id=book-search-input placeholder=Search aria-label=Search maxlength=64 data-hotkeys=s/><div class="book-search-spinner hidden"></div><ul id=book-search-results></ul></div><script>document.querySelector(".book-search").classList.remove("hidden")</script><ul><li class=book-section-flat><span>기록</span><ul><li><a href=/docs/hobby/daily/>일상</a><ul></ul></li><li><a href=/docs/hobby/book/>글</a><ul></ul></li></ul></li><li class=book-section-flat><span>공부</span><ul><li><a href=/docs/study/bioinformatics/ class=active>Bioinformatics</a><ul></ul></li><li><a href=/docs/study/ai/>AI</a><ul></ul></li><li><a href=/docs/study/be/>BE</a><ul></ul></li><li><a href=/docs/study/fe/>FE</a><ul></ul></li><li><a href=/docs/study/career/>취업</a><ul></ul></li></ul></li></ul></nav><script>(function(){var e=document.querySelector("aside .book-menu-content");addEventListener("beforeunload",function(){localStorage.setItem("menu.scrollTop",e.scrollTop)}),e.scrollTop=localStorage.getItem("menu.scrollTop")})()</script></div></aside><div class=book-page><header class=book-header><div class="flex align-center justify-between"><label for=menu-control><img src=/svg/menu.svg class=book-icon alt=Menu></label><h3>Bioinformatics</h3><label for=toc-control><img src=/svg/toc.svg class=book-icon alt="Table of Contents"></label></div><aside class="hidden clearfix"><nav id=TableOfContents></nav></aside></header><article class="markdown book-article"><h1 id=bioinformatics>Bioinformatics
<a class=anchor href=#bioinformatics>#</a></h1><p><em>2025-04-21</em> ⋯ RNA-seq 전처리 파이프라인 비교</p><p style=height:4.5em;overflow:hidden><a href=/docs/study/bioinformatics/bi9/>1 Methods 비교 의의 - Traditional 방법은 TopHat2+HTseq 조합이지만 오류도 넘 많이나고 Rsubread를 쓰면 빠르고 깔끔한데 왜 써야하지..? 싶어서 동일한 데이터(pair-end fastq)로 돌려봄. - HTseq에서 아래 코드를 수행 ⋯</a></p><hr><p><em>2025-04-21</em> ⋯ RNA-seq 전처리 (TopHat, SAMtools, HTSeq)</p><p style=height:4.5em;overflow:hidden><a href=/docs/study/bioinformatics/bi7/>1 TopHat 실행 ```bash
$ tophatpy -o tophat_out_33-1 --no-mixed -p 40 \
$ ⋯</a></p><hr><p><em>2025-04-21</em> ⋯ RNA-seq 전처리 (Rsubread, edgeR)</p><p style=height:4.5em;overflow:hidden><a href=/docs/study/bioinformatics/bi8/>가장 오류 적게나는 조합! 1. Align RNA-seq Load Packages ```R library(Rsubread) library(org.Mm.eg.db) library(gridExtra) library(reshape2) ``` Set Path ```R ⋯</a></p><hr><p><em>2025-04-21</em> ⋯ Kallisto Pseudoalignment 작업</p><p style=height:4.5em;overflow:hidden><a href=/docs/study/bioinformatics/bi4/>1. Build Index ```bash
$ kallisto index -i transcripts_cDNA.idx Homo_sapiens.GRCh38.cdna.all.fa.gz
``` 2. Pseudoalign ```bash
$ kallisto quant -i ⋯</a></p><hr><p><em>2025-04-21</em> ⋯ Enrichment 분석, 버블 플롯 (gProfiler/ggplot2)</p><p style=height:4.5em;overflow:hidden><a href=/docs/study/bioinformatics/bi3/>1 Load Package ```r library(ggplot2) ``` 2 Set Path ```r setwd("/data-blog/bi3") getwd() ``` ```plain text '/data-blog/bi3' ``` 3 Functional ⋯</a></p><hr><p><em>2025-04-21</em> ⋯ ChIP-seq 전처리</p><p style=height:4.5em;overflow:hidden><a href=/docs/study/bioinformatics/bi5/>1. Trimming chipseq_trimming.sh ```bash !/bin/bash setting envs export bdir="/data3/projects/2022_KNU_EBV" export ⋯</a></p><hr><p><em>2024-12-31</em> ⋯ RNA-seq 전처리: EBV genome</p><p style=height:4.5em;overflow:hidden><a href=/docs/study/bioinformatics/bi11/>0 분석 목적 - 제공받은 fastq를 human genome에 매핑해서 전처리, 분석 후 DE 결과 보냄 - DE 분석시에 EBV 유전자도 포함해달라는 요청 해야하는것 - fastq를 EBV genome에 매핑해서 전처리, EBV count 생성 - human ⋯</a></p><hr><p><em>2024-12-31</em> ⋯ EndNote 사용법</p><p style=height:4.5em;overflow:hidden><a href=/docs/study/bioinformatics/bi16/>1. EndNote 설치 및 계정 설정 - 계정 설정: [공식 웹사이트](https://endnote.com/?srsltid=AfmBOopL9WuN7IRp_CVgVcEHXZAd8eJWpK1YIMA9ZEccRbFk-2r-y9d4)에서 End note 계정을 생성한다. - ⋯</a></p><hr><p><em>2024-12-31</em> ⋯ DE 분석 (DESeq2)</p><p style=height:4.5em;overflow:hidden><a href=/docs/study/bioinformatics/bi1/>> Tool > > Bioconductor - DESeq2 https://bioconductor.org/packages/release/bioc/html/DESeq2.html 1. Load package ```R suppressMessages({ ⋯</a></p><hr><div class=pagination style=margin-top:2rem;display:flex;justify-content:center;align-items:center;gap:1rem><a href=/docs/study/bioinformatics/ style="padding:.5rem 1rem;text-decoration:none">←</a>
<span style=color:#666>2 / 2</span></div></article><footer class=book-footer><div class="flex flex-wrap justify-between"></div><script>(function(){function e(e){const t=window.getSelection(),n=document.createRange();n.selectNodeContents(e),t.removeAllRanges(),t.addRange(n)}document.querySelectorAll("pre code").forEach(t=>{t.addEventListener("click",function(){if(window.getSelection().toString())return;e(t.parentElement),navigator.clipboard&&navigator.clipboard.writeText(t.parentElement.textContent)})})})()</script></footer><label for=menu-control class="hidden book-menu-overlay"></label></div><aside class=book-toc><div class=book-toc-content><nav id=TableOfContents></nav></div></aside></main></body></html>