<!doctype html><html lang=en-us dir=ltr><head><meta charset=UTF-8><meta name=viewport content="width=device-width,initial-scale=1"><meta name=description content="
  Kallisto Pseudoalignment ì‘ì—…
  #

#2025-04-21


  1. Build Index
  #

$ kallisto index -i transcripts_cDNA.idx Homo_sapiens.GRCh38.cdna.all.fa.gz

  2. Pseudoalign
  #

$ kallisto quant -i transcripts_cDNA.idx -o output_150-1 -t 40 ../2306_tophat/data/Bowtie2Index/5-AZA_150-1_1_edited.fastq ../2306_tophat/data/Bowtie2Index/5-AZA_150-1_2_edited.fastq

3ê°œ íŒŒì¼ ìƒì„±

abundance.h5 - HDF5 binary file containing run info, abundance esimates, bootstrap estimates, and transcript length information length. This file can be read in by sleuth
abundance.tsv - plaintext file of the abundance estimates. It does not contains bootstrap estimates. Please use the &ndash;plaintext mode to output plaintext abundance estimates. Alternatively, kallisto h5dump can be used to output an HDF5 file to plaintext. The first line contains a header for each column, including estimated counts, TPM, effective length.
run_info.json - json file containing information about the run




  3. Downstream ë¶„ì„
  #


KallistoëŠ” ì¼ë°˜ì ì¸ Alignment ë„êµ¬ì™€ ë‹¬ë¦¬ bam íŒŒì¼ì„ outputìœ¼ë¡œ ìƒì„±í•˜ì§€ ì•Šê¸° ë•Œë¬¸ì— HTSeq-countë¥¼ ì“°ëŠ” ëŒ€ì‹  abundance.tsv ë˜ëŠ” .h5 íŒŒì¼ì„ Sleuthì—ì„œ ì§ì ‘ ë¶ˆëŸ¬ì™€ì„œ í†µê³„ ë¶„ì„ì„ ìˆ˜í–‰í•˜ëŠ” ê²ƒì´ í‘œì¤€ ì›Œí¬í”Œë¡œìš°ì´ë‹¤.
"><meta name=theme-color media="(prefers-color-scheme: light)" content="#ffffff"><meta name=theme-color media="(prefers-color-scheme: dark)" content="#343a40"><meta name=color-scheme content="light dark"><meta property="og:url" content="https://yshghid.github.io/docs/study/bioinformatics/bi4/"><meta property="og:site_name" content=" "><meta property="og:title" content="Kallisto Pseudoalignment ì‘ì—…"><meta property="og:description" content="Kallisto Pseudoalignment ì‘ì—… # #2025-04-21
1. Build Index # $ kallisto index -i transcripts_cDNA.idx Homo_sapiens.GRCh38.cdna.all.fa.gz 2. Pseudoalign # $ kallisto quant -i transcripts_cDNA.idx -o output_150-1 -t 40 ../2306_tophat/data/Bowtie2Index/5-AZA_150-1_1_edited.fastq ../2306_tophat/data/Bowtie2Index/5-AZA_150-1_2_edited.fastq 3ê°œ íŒŒì¼ ìƒì„± abundance.h5 - HDF5 binary file containing run info, abundance esimates, bootstrap estimates, and transcript length information length. This file can be read in by sleuth abundance.tsv - plaintext file of the abundance estimates. It does not contains bootstrap estimates. Please use the â€“plaintext mode to output plaintext abundance estimates. Alternatively, kallisto h5dump can be used to output an HDF5 file to plaintext. The first line contains a header for each column, including estimated counts, TPM, effective length. run_info.json - json file containing information about the run 3. Downstream ë¶„ì„ # KallistoëŠ” ì¼ë°˜ì ì¸ Alignment ë„êµ¬ì™€ ë‹¬ë¦¬ bam íŒŒì¼ì„ outputìœ¼ë¡œ ìƒì„±í•˜ì§€ ì•Šê¸° ë•Œë¬¸ì— HTSeq-countë¥¼ ì“°ëŠ” ëŒ€ì‹  abundance.tsv ë˜ëŠ” .h5 íŒŒì¼ì„ Sleuthì—ì„œ ì§ì ‘ ë¶ˆëŸ¬ì™€ì„œ í†µê³„ ë¶„ì„ì„ ìˆ˜í–‰í•˜ëŠ” ê²ƒì´ í‘œì¤€ ì›Œí¬í”Œë¡œìš°ì´ë‹¤."><meta property="og:locale" content="en_us"><meta property="og:type" content="article"><meta property="article:section" content="docs"><meta property="article:published_time" content="2025-04-21T00:00:00+00:00"><meta property="article:modified_time" content="2025-04-21T00:00:00+00:00"><meta property="article:tag" content="2025-04"><title>Kallisto Pseudoalignment ì‘ì—… |</title><link rel=icon href=/favicon.png><link rel=manifest href=/manifest.json><link rel=canonical href=https://yshghid.github.io/docs/study/bioinformatics/bi4/><link rel=stylesheet href=/book.min.6217d077edb4189fd0578345e84bca1a884dfdee121ff8dc9a0f55cfe0852bc9.css integrity="sha256-YhfQd+20GJ/QV4NF6EvKGohN/e4SH/jcmg9Vz+CFK8k=" crossorigin=anonymous><script defer src=/fuse.min.js></script><script defer src=/en.search.min.a3034e7944743feafde97a6054e4f2d7af6ff24a0ef0b929be8fbc5e238ab4ec.js integrity="sha256-owNOeUR0P+r96XpgVOTy169v8koO8Lkpvo+8XiOKtOw=" crossorigin=anonymous></script></head><body dir=ltr><input type=checkbox class="hidden toggle" id=menu-control>
<input type=checkbox class="hidden toggle" id=toc-control><main class="container flex"><aside class=book-menu><div class=book-menu-content><nav><h2 class=book-brand><a class="flex align-center" href=/><img src=/logo.png alt=Logo class=book-icon><span></span></a></h2><div class="book-search hidden"><input type=text id=book-search-input placeholder=Search aria-label=Search maxlength=64 data-hotkeys=s/><div class="book-search-spinner hidden"></div><ul id=book-search-results></ul></div><script>document.querySelector(".book-search").classList.remove("hidden")</script><ul><li class=book-section-flat><span>ê¸°ë¡</span><ul><li><a href=/docs/hobby/book/>ê¸€</a><ul></ul></li><li><a href=/docs/hobby/daily/>ì¼ìƒ</a><ul></ul></li><li><a href=/docs/hobby/baking/>ë² ì´í‚¹</a><ul></ul></li><li><a href=/docs/hobby/favorite/>ğŸŒ¸</a><ul></ul></li></ul></li><li class=book-section-flat><span>ê³µë¶€</span><ul><li><a href=/docs/study/bioinformatics/>ìƒë¬¼ì •ë³´í•™</a><ul></ul></li><li><a href=/docs/study/algorithm/>ì½”í…Œ</a><ul></ul></li><li><a href=/docs/study/career/>ì·¨ì—…</a><ul></ul></li><li><a href=/docs/study/etc/>ê¸°íƒ€</a><ul></ul></li></ul></li></ul></nav><script>(function(){var e=document.querySelector("aside .book-menu-content");addEventListener("beforeunload",function(){localStorage.setItem("menu.scrollTop",e.scrollTop)}),e.scrollTop=localStorage.getItem("menu.scrollTop")})()</script></div></aside><div class=book-page><header class=book-header><div class="flex align-center justify-between"><label for=menu-control><img src=/svg/menu.svg class=book-icon alt=Menu></label><h3>Kallisto Pseudoalignment ì‘ì—…</h3><label for=toc-control><img src=/svg/toc.svg class=book-icon alt="Table of Contents"></label></div><aside class="hidden clearfix"><nav id=TableOfContents><ul><li><a href=#1-build-index>1. Build Index</a></li><li><a href=#2-pseudoalign>2. Pseudoalign</a></li><li><a href=#3-downstream-ë¶„ì„>3. Downstream ë¶„ì„</a></li></ul></nav></aside></header><article class="markdown book-article"><h1 id=kallisto-pseudoalignment-ì‘ì—…>Kallisto Pseudoalignment ì‘ì—…
<a class=anchor href=#kallisto-pseudoalignment-%ec%9e%91%ec%97%85>#</a></h1><p>#2025-04-21</p><hr><h2 id=1-build-index>1. Build Index
<a class=anchor href=#1-build-index>#</a></h2><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-bash data-lang=bash><span style=display:flex><span>$ kallisto index -i transcripts_cDNA.idx Homo_sapiens.GRCh38.cdna.all.fa.gz
</span></span></code></pre></div><h2 id=2-pseudoalign>2. Pseudoalign
<a class=anchor href=#2-pseudoalign>#</a></h2><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-bash data-lang=bash><span style=display:flex><span>$ kallisto quant -i transcripts_cDNA.idx -o output_150-1 -t <span style=color:#ae81ff>40</span> ../2306_tophat/data/Bowtie2Index/5-AZA_150-1_1_edited.fastq ../2306_tophat/data/Bowtie2Index/5-AZA_150-1_2_edited.fastq
</span></span></code></pre></div><ul><li>3ê°œ íŒŒì¼ ìƒì„±<ul><li>abundance.h5 - HDF5 binary file containing run info, abundance esimates, bootstrap estimates, and transcript length information length. This file can be read in by sleuth</li><li>abundance.tsv - plaintext file of the abundance estimates. It does not contains bootstrap estimates. Please use the &ndash;plaintext mode to output plaintext abundance estimates. Alternatively, kallisto h5dump can be used to output an HDF5 file to plaintext. The first line contains a header for each column, including estimated counts, TPM, effective length.</li><li>run_info.json - json file containing information about the run</li></ul></li></ul><h2 id=3-downstream-ë¶„ì„>3. Downstream ë¶„ì„
<a class=anchor href=#3-downstream-%eb%b6%84%ec%84%9d>#</a></h2><ul><li>KallistoëŠ” ì¼ë°˜ì ì¸ Alignment ë„êµ¬ì™€ ë‹¬ë¦¬ bam íŒŒì¼ì„ outputìœ¼ë¡œ ìƒì„±í•˜ì§€ ì•Šê¸° ë•Œë¬¸ì— HTSeq-countë¥¼ ì“°ëŠ” ëŒ€ì‹  abundance.tsv ë˜ëŠ” .h5 íŒŒì¼ì„ Sleuthì—ì„œ ì§ì ‘ ë¶ˆëŸ¬ì™€ì„œ í†µê³„ ë¶„ì„ì„ ìˆ˜í–‰í•˜ëŠ” ê²ƒì´ í‘œì¤€ ì›Œí¬í”Œë¡œìš°ì´ë‹¤.</li></ul></article><footer class=book-footer><div class="flex flex-wrap justify-between"></div><script>(function(){function e(e){const t=window.getSelection(),n=document.createRange();n.selectNodeContents(e),t.removeAllRanges(),t.addRange(n)}document.querySelectorAll("pre code").forEach(t=>{t.addEventListener("click",function(){if(window.getSelection().toString())return;e(t.parentElement),navigator.clipboard&&navigator.clipboard.writeText(t.parentElement.textContent)})})})()</script></footer><div class=book-comments><script src=https://giscus.app/client.js data-repo=yshghid/yshghid.github.io data-repo-id=R_kgDONkMkNg data-category-id=DIC_kwDONkMkNs4CloJh data-mapping=pathname data-strict=0 data-reactions-enabled=1 data-emit-metadata=0 data-input-position=bottom data-theme=preferred_color_scheme data-lang=ko crossorigin=anonymous async></script></div><label for=menu-control class="hidden book-menu-overlay"></label></div><aside class=book-toc><div class=book-toc-content><nav id=TableOfContents><ul><li><a href=#1-build-index>1. Build Index</a></li><li><a href=#2-pseudoalign>2. Pseudoalign</a></li><li><a href=#3-downstream-ë¶„ì„>3. Downstream ë¶„ì„</a></li></ul></nav></div></aside></main></body></html>