<!doctype html><html lang=en-us dir=ltr><head><meta charset=UTF-8><meta name=viewport content="width=device-width,initial-scale=1"><meta name=description content='
  Rsubread, edgeR: RNA-seq 전처리
  #

#2025-04-21

가장 오류 적게나는 조합!

  1. Align RNA-seq
  #

#Load Packages
library(Rsubread)
library(org.Mm.eg.db)
library(gridExtra)
library(reshape2)
#Set Path
indir = "/data/home/ysh980101/2504/mirna/data"
outdir = "/data/home/ysh980101/2504/mirna/data"
refpath = "/data/home/ysh980101/2406/data-gne/mm39.fa"

setwd(indir)
getwd()
&#39;/data/home/ysh980101/2504/mirna/data&#39;
#Build Index
buildindex(basename = "mm39", reference = refpath)
#Read Alignment
files <- list.files(pattern="\\.fastq\\.gz$", full.names=TRUE)
bams <- sub("\\.fastq\\.gz$", ".bam", files)
samples <- gsub("^\\.\\/|\\.fastq\\.gz$", "", files)
targets <- read.delim("target.txt", header=TRUE)

align(index="mm39", readfile1=files, input_format="gzFASTQ", output_file=bams, nthreads=50)
#Quantification
fc = featureCounts(bams, isGTFAnnotationFile=TRUE, GTF.featureType="exon", GTF.attrType="gene_id", isPairedEnd=FALSE, annot.ext="mm39.knownGene.gtf", useMetaFeatures=FALSE, allowMultiOverlap=TRUE, nthreads=50)
#Save Countdata
colnames(fc$counts) <- samples
y <- DGEList(fc$counts, group=group)
write.csv(as.data.frame(y$counts), file = paste0(outdir,"/count.csv", row.names = TRUE))

  2. Gene ID Annotation
  #

#Load Packages
import pandas as pd
import numpy as np
import os
#Set Path
indir = "/data/home/ysh980101/2504/mirna/data"
outdir = "/data/home/ysh980101/2504/mirna/data"
annotpath = "/data/home/ysh980101/2406/data-gne/MRK_ENSEMBL.rpt"

os.chdir(indir)
os.getcwd()
&#39;/data1/home/ysh980101/2504/mirna/result&#39;
#Load Annotation'><meta name=theme-color media="(prefers-color-scheme: light)" content="#ffffff"><meta name=theme-color media="(prefers-color-scheme: dark)" content="#343a40"><meta name=color-scheme content="light dark"><meta property="og:url" content="https://yshghid.github.io/docs/study/bioinformatics/bi8/"><meta property="og:site_name" content=" "><meta property="og:title" content="Rsubread, edgeR: RNA-seq 전처리"><meta property="og:description" content='Rsubread, edgeR: RNA-seq 전처리 # #2025-04-21
가장 오류 적게나는 조합!
1. Align RNA-seq # #Load Packages
library(Rsubread) library(org.Mm.eg.db) library(gridExtra) library(reshape2) #Set Path
indir = "/data/home/ysh980101/2504/mirna/data" outdir = "/data/home/ysh980101/2504/mirna/data" refpath = "/data/home/ysh980101/2406/data-gne/mm39.fa" setwd(indir) getwd() &#39;/data/home/ysh980101/2504/mirna/data&#39; #Build Index
buildindex(basename = "mm39", reference = refpath) #Read Alignment
files <- list.files(pattern="\\.fastq\\.gz$", full.names=TRUE) bams <- sub("\\.fastq\\.gz$", ".bam", files) samples <- gsub("^\\.\\/|\\.fastq\\.gz$", "", files) targets <- read.delim("target.txt", header=TRUE) align(index="mm39", readfile1=files, input_format="gzFASTQ", output_file=bams, nthreads=50) #Quantification
fc = featureCounts(bams, isGTFAnnotationFile=TRUE, GTF.featureType="exon", GTF.attrType="gene_id", isPairedEnd=FALSE, annot.ext="mm39.knownGene.gtf", useMetaFeatures=FALSE, allowMultiOverlap=TRUE, nthreads=50) #Save Countdata
colnames(fc$counts) <- samples y <- DGEList(fc$counts, group=group) write.csv(as.data.frame(y$counts), file = paste0(outdir,"/count.csv", row.names = TRUE)) 2. Gene ID Annotation # #Load Packages
import pandas as pd import numpy as np import os #Set Path
indir = "/data/home/ysh980101/2504/mirna/data" outdir = "/data/home/ysh980101/2504/mirna/data" annotpath = "/data/home/ysh980101/2406/data-gne/MRK_ENSEMBL.rpt" os.chdir(indir) os.getcwd() &#39;/data1/home/ysh980101/2504/mirna/result&#39; #Load Annotation'><meta property="og:locale" content="en_us"><meta property="og:type" content="article"><meta property="article:section" content="docs"><meta property="article:published_time" content="2025-04-21T00:00:00+00:00"><meta property="article:modified_time" content="2025-04-21T00:00:00+00:00"><meta property="article:tag" content="2025-04"><title>Rsubread, edgeR: RNA-seq 전처리 |</title><link rel=icon href=/favicon.png><link rel=manifest href=/manifest.json><link rel=canonical href=https://yshghid.github.io/docs/study/bioinformatics/bi8/><link rel=stylesheet href=/book.min.6217d077edb4189fd0578345e84bca1a884dfdee121ff8dc9a0f55cfe0852bc9.css integrity="sha256-YhfQd+20GJ/QV4NF6EvKGohN/e4SH/jcmg9Vz+CFK8k=" crossorigin=anonymous><script defer src=/fuse.min.js></script><script defer src=/en.search.min.578a2d8e6e30e0a0ae3682797882d89ca0cbbf1734d206705396ea76cf57bbb6.js integrity="sha256-V4otjm4w4KCuNoJ5eILYnKDLvxc00gZwU5bqds9Xu7Y=" crossorigin=anonymous></script></head><body dir=ltr><input type=checkbox class="hidden toggle" id=menu-control>
<input type=checkbox class="hidden toggle" id=toc-control><main class="container flex"><aside class=book-menu><div class=book-menu-content><nav><h2 class=book-brand><a class="flex align-center" href=/><img src=/logo.png alt=Logo class=book-icon><span></span></a></h2><div class="book-search hidden"><input type=text id=book-search-input placeholder=Search aria-label=Search maxlength=64 data-hotkeys=s/><div class="book-search-spinner hidden"></div><ul id=book-search-results></ul></div><script>document.querySelector(".book-search").classList.remove("hidden")</script><ul><li class=book-section-flat><span>기록</span><ul><li><a href=/docs/hobby/book/>글</a><ul></ul></li><li><a href=/docs/hobby/daily/>일상</a><ul></ul></li></ul></li><li class=book-section-flat><span>공부</span><ul><li><a href=/docs/study/bioinformatics/>Bioinformatics</a><ul></ul></li><li><a href=/docs/study/ai/>AI</a><ul></ul></li><li><a href=/docs/study/sw/>SW</a><ul></ul></li><li><a href=/docs/study/algorithm/>알고리즘</a><ul></ul></li><li><a href=/docs/study/career/>취업</a><ul></ul></li></ul></li></ul></nav><script>(function(){var e=document.querySelector("aside .book-menu-content");addEventListener("beforeunload",function(){localStorage.setItem("menu.scrollTop",e.scrollTop)}),e.scrollTop=localStorage.getItem("menu.scrollTop")})()</script></div></aside><div class=book-page><header class=book-header><div class="flex align-center justify-between"><label for=menu-control><img src=/svg/menu.svg class=book-icon alt=Menu></label><h3>Rsubread, edgeR: RNA-seq 전처리</h3><label for=toc-control><img src=/svg/toc.svg class=book-icon alt="Table of Contents"></label></div><aside class="hidden clearfix"><nav id=TableOfContents><ul><li><ul><li><a href=#1-align-rna-seq>1. Align RNA-seq</a></li><li><a href=#2-gene-id-annotation>2. Gene ID Annotation</a></li><li><a href=#3-deg-analysis>3. DEG Analysis</a></li></ul></li></ul></nav></aside></header><article class="markdown book-article"><h1 id=rsubread-edger-rna-seq-전처리>Rsubread, edgeR: RNA-seq 전처리
<a class=anchor href=#rsubread-edger-rna-seq-%ec%a0%84%ec%b2%98%eb%a6%ac>#</a></h1><p>#2025-04-21</p><hr><p>가장 오류 적게나는 조합!</p><h3 id=1-align-rna-seq>1. Align RNA-seq
<a class=anchor href=#1-align-rna-seq>#</a></h3><p>#Load Packages</p><div class=highlight><pre tabindex=0 class=chroma><code class=language-R data-lang=R><span class=line><span class=cl><span class=nf>library</span><span class=p>(</span><span class=n>Rsubread</span><span class=p>)</span>
</span></span><span class=line><span class=cl><span class=nf>library</span><span class=p>(</span><span class=n>org.Mm.eg.db</span><span class=p>)</span>
</span></span><span class=line><span class=cl><span class=nf>library</span><span class=p>(</span><span class=n>gridExtra</span><span class=p>)</span>
</span></span><span class=line><span class=cl><span class=nf>library</span><span class=p>(</span><span class=n>reshape2</span><span class=p>)</span>
</span></span></code></pre></div><p>#Set Path</p><div class=highlight><pre tabindex=0 class=chroma><code class=language-R data-lang=R><span class=line><span class=cl><span class=n>indir</span> <span class=o>=</span> <span class=s>&#34;/data/home/ysh980101/2504/mirna/data&#34;</span>
</span></span><span class=line><span class=cl><span class=n>outdir</span> <span class=o>=</span> <span class=s>&#34;/data/home/ysh980101/2504/mirna/data&#34;</span>
</span></span><span class=line><span class=cl><span class=n>refpath</span> <span class=o>=</span> <span class=s>&#34;/data/home/ysh980101/2406/data-gne/mm39.fa&#34;</span>
</span></span><span class=line><span class=cl>
</span></span><span class=line><span class=cl><span class=nf>setwd</span><span class=p>(</span><span class=n>indir</span><span class=p>)</span>
</span></span><span class=line><span class=cl><span class=nf>getwd</span><span class=p>()</span>
</span></span></code></pre></div><div class=highlight><pre tabindex=0 class=chroma><code class=language-plain data-lang=plain><span class=line><span class=cl>&#39;/data/home/ysh980101/2504/mirna/data&#39;
</span></span></code></pre></div><p>#Build Index</p><div class=highlight><pre tabindex=0 class=chroma><code class=language-R data-lang=R><span class=line><span class=cl><span class=nf>buildindex</span><span class=p>(</span><span class=n>basename</span> <span class=o>=</span> <span class=s>&#34;mm39&#34;</span><span class=p>,</span> <span class=n>reference</span> <span class=o>=</span> <span class=n>refpath</span><span class=p>)</span>
</span></span></code></pre></div><p>#Read Alignment</p><div class=highlight><pre tabindex=0 class=chroma><code class=language-R data-lang=R><span class=line><span class=cl><span class=n>files</span> <span class=o>&lt;-</span> <span class=nf>list.files</span><span class=p>(</span><span class=n>pattern</span><span class=o>=</span><span class=s>&#34;\\.fastq\\.gz$&#34;</span><span class=p>,</span> <span class=n>full.names</span><span class=o>=</span><span class=kc>TRUE</span><span class=p>)</span>
</span></span><span class=line><span class=cl><span class=n>bams</span> <span class=o>&lt;-</span> <span class=nf>sub</span><span class=p>(</span><span class=s>&#34;\\.fastq\\.gz$&#34;</span><span class=p>,</span> <span class=s>&#34;.bam&#34;</span><span class=p>,</span> <span class=n>files</span><span class=p>)</span>
</span></span><span class=line><span class=cl><span class=n>samples</span> <span class=o>&lt;-</span> <span class=nf>gsub</span><span class=p>(</span><span class=s>&#34;^\\.\\/|\\.fastq\\.gz$&#34;</span><span class=p>,</span> <span class=s>&#34;&#34;</span><span class=p>,</span> <span class=n>files</span><span class=p>)</span>
</span></span><span class=line><span class=cl><span class=n>targets</span> <span class=o>&lt;-</span> <span class=nf>read.delim</span><span class=p>(</span><span class=s>&#34;target.txt&#34;</span><span class=p>,</span> <span class=n>header</span><span class=o>=</span><span class=kc>TRUE</span><span class=p>)</span>
</span></span><span class=line><span class=cl>
</span></span><span class=line><span class=cl><span class=nf>align</span><span class=p>(</span><span class=n>index</span><span class=o>=</span><span class=s>&#34;mm39&#34;</span><span class=p>,</span> <span class=n>readfile1</span><span class=o>=</span><span class=n>files</span><span class=p>,</span> <span class=n>input_format</span><span class=o>=</span><span class=s>&#34;gzFASTQ&#34;</span><span class=p>,</span> <span class=n>output_file</span><span class=o>=</span><span class=n>bams</span><span class=p>,</span> <span class=n>nthreads</span><span class=o>=</span><span class=m>50</span><span class=p>)</span>
</span></span></code></pre></div><p>#Quantification</p><div class=highlight><pre tabindex=0 class=chroma><code class=language-R data-lang=R><span class=line><span class=cl><span class=n>fc</span> <span class=o>=</span> <span class=nf>featureCounts</span><span class=p>(</span><span class=n>bams</span><span class=p>,</span> <span class=n>isGTFAnnotationFile</span><span class=o>=</span><span class=kc>TRUE</span><span class=p>,</span> <span class=n>GTF.featureType</span><span class=o>=</span><span class=s>&#34;exon&#34;</span><span class=p>,</span> <span class=n>GTF.attrType</span><span class=o>=</span><span class=s>&#34;gene_id&#34;</span><span class=p>,</span> <span class=n>isPairedEnd</span><span class=o>=</span><span class=kc>FALSE</span><span class=p>,</span> <span class=n>annot.ext</span><span class=o>=</span><span class=s>&#34;mm39.knownGene.gtf&#34;</span><span class=p>,</span> <span class=n>useMetaFeatures</span><span class=o>=</span><span class=kc>FALSE</span><span class=p>,</span> <span class=n>allowMultiOverlap</span><span class=o>=</span><span class=kc>TRUE</span><span class=p>,</span> <span class=n>nthreads</span><span class=o>=</span><span class=m>50</span><span class=p>)</span>
</span></span></code></pre></div><p>#Save Countdata</p><div class=highlight><pre tabindex=0 class=chroma><code class=language-R data-lang=R><span class=line><span class=cl><span class=nf>colnames</span><span class=p>(</span><span class=n>fc</span><span class=o>$</span><span class=n>counts</span><span class=p>)</span> <span class=o>&lt;-</span> <span class=n>samples</span>
</span></span><span class=line><span class=cl><span class=n>y</span> <span class=o>&lt;-</span> <span class=nf>DGEList</span><span class=p>(</span><span class=n>fc</span><span class=o>$</span><span class=n>counts</span><span class=p>,</span> <span class=n>group</span><span class=o>=</span><span class=n>group</span><span class=p>)</span>
</span></span><span class=line><span class=cl><span class=nf>write.csv</span><span class=p>(</span><span class=nf>as.data.frame</span><span class=p>(</span><span class=n>y</span><span class=o>$</span><span class=n>counts</span><span class=p>),</span> <span class=n>file</span> <span class=o>=</span> <span class=nf>paste0</span><span class=p>(</span><span class=n>outdir</span><span class=p>,</span><span class=s>&#34;/count.csv&#34;</span><span class=p>,</span> <span class=n>row.names</span> <span class=o>=</span> <span class=kc>TRUE</span><span class=p>))</span>
</span></span></code></pre></div><h3 id=2-gene-id-annotation>2. Gene ID Annotation
<a class=anchor href=#2-gene-id-annotation>#</a></h3><p>#Load Packages</p><div class=highlight><pre tabindex=0 class=chroma><code class=language-python data-lang=python><span class=line><span class=cl><span class=kn>import</span> <span class=nn>pandas</span> <span class=k>as</span> <span class=nn>pd</span>
</span></span><span class=line><span class=cl><span class=kn>import</span> <span class=nn>numpy</span> <span class=k>as</span> <span class=nn>np</span>
</span></span><span class=line><span class=cl><span class=kn>import</span> <span class=nn>os</span>
</span></span></code></pre></div><p>#Set Path</p><div class=highlight><pre tabindex=0 class=chroma><code class=language-python data-lang=python><span class=line><span class=cl><span class=n>indir</span> <span class=o>=</span> <span class=s2>&#34;/data/home/ysh980101/2504/mirna/data&#34;</span>
</span></span><span class=line><span class=cl><span class=n>outdir</span> <span class=o>=</span> <span class=s2>&#34;/data/home/ysh980101/2504/mirna/data&#34;</span>
</span></span><span class=line><span class=cl><span class=n>annotpath</span> <span class=o>=</span> <span class=s2>&#34;/data/home/ysh980101/2406/data-gne/MRK_ENSEMBL.rpt&#34;</span>
</span></span><span class=line><span class=cl>
</span></span><span class=line><span class=cl><span class=n>os</span><span class=o>.</span><span class=n>chdir</span><span class=p>(</span><span class=n>indir</span><span class=p>)</span>
</span></span><span class=line><span class=cl><span class=n>os</span><span class=o>.</span><span class=n>getcwd</span><span class=p>()</span>
</span></span></code></pre></div><div class=highlight><pre tabindex=0 class=chroma><code class=language-plain data-lang=plain><span class=line><span class=cl>&#39;/data1/home/ysh980101/2504/mirna/result&#39;
</span></span></code></pre></div><p>#Load Annotation</p><div class=highlight><pre tabindex=0 class=chroma><code class=language-python data-lang=python><span class=line><span class=cl><span class=n>annotation</span> <span class=o>=</span> <span class=n>pd</span><span class=o>.</span><span class=n>read_csv</span><span class=p>(</span><span class=n>annotpath</span><span class=p>,</span> <span class=n>sep</span><span class=o>=</span><span class=s2>&#34;</span><span class=se>\t</span><span class=s2>&#34;</span><span class=p>,</span> <span class=n>names</span><span class=o>=</span><span class=p>[</span><span class=nb>str</span><span class=p>(</span><span class=n>i</span><span class=p>)</span> <span class=k>for</span> <span class=n>i</span> <span class=ow>in</span> <span class=nb>range</span><span class=p>(</span><span class=mi>13</span><span class=p>)])</span>
</span></span><span class=line><span class=cl><span class=n>annotation</span> <span class=o>=</span> <span class=n>annotation</span><span class=o>.</span><span class=n>dropna</span><span class=p>(</span><span class=n>subset</span><span class=o>=</span><span class=p>[</span><span class=s1>&#39;6&#39;</span><span class=p>])</span>
</span></span><span class=line><span class=cl><span class=n>annotation</span> <span class=o>=</span> <span class=n>annotation</span><span class=p>[</span><span class=n>annotation</span><span class=p>[</span><span class=s1>&#39;8&#39;</span><span class=p>]</span> <span class=o>==</span> <span class=s1>&#39;protein coding gene&#39;</span><span class=p>]</span>
</span></span></code></pre></div><p>#Load Count & Gene ID Mapping</p><div class=highlight><pre tabindex=0 class=chroma><code class=language-python data-lang=python><span class=line><span class=cl><span class=n>count_mm39</span> <span class=o>=</span> <span class=n>pd</span><span class=o>.</span><span class=n>read_csv</span><span class=p>(</span><span class=s2>&#34;count.csv&#34;</span><span class=p>)</span>
</span></span><span class=line><span class=cl><span class=n>count_mm39</span><span class=o>.</span><span class=n>rename</span><span class=p>(</span><span class=n>columns</span><span class=o>=</span><span class=p>{</span><span class=n>count_mm39</span><span class=o>.</span><span class=n>columns</span><span class=p>[</span><span class=mi>0</span><span class=p>]:</span> <span class=s1>&#39;ens_id&#39;</span><span class=p>},</span> <span class=n>inplace</span><span class=o>=</span><span class=kc>True</span><span class=p>)</span>
</span></span><span class=line><span class=cl><span class=n>count_mm39</span><span class=p>[</span><span class=s1>&#39;ens_id&#39;</span><span class=p>]</span> <span class=o>=</span> <span class=n>count_mm39</span><span class=p>[</span><span class=s1>&#39;ens_id&#39;</span><span class=p>]</span><span class=o>.</span><span class=n>str</span><span class=o>.</span><span class=n>split</span><span class=p>(</span><span class=s1>&#39;.&#39;</span><span class=p>)</span><span class=o>.</span><span class=n>str</span><span class=p>[</span><span class=mi>0</span><span class=p>]</span>
</span></span><span class=line><span class=cl>
</span></span><span class=line><span class=cl><span class=k>for</span> <span class=n>index</span><span class=p>,</span> <span class=n>row</span> <span class=ow>in</span> <span class=n>annotation</span><span class=o>.</span><span class=n>iterrows</span><span class=p>():</span>
</span></span><span class=line><span class=cl>    <span class=n>ens_ids</span> <span class=o>=</span> <span class=n>row</span><span class=p>[</span><span class=s1>&#39;6&#39;</span><span class=p>]</span><span class=o>.</span><span class=n>split</span><span class=p>()</span>
</span></span><span class=line><span class=cl>    <span class=n>gene_id</span> <span class=o>=</span> <span class=n>row</span><span class=p>[</span><span class=s1>&#39;1&#39;</span><span class=p>]</span>
</span></span><span class=line><span class=cl>    <span class=n>count_mm39</span><span class=o>.</span><span class=n>loc</span><span class=p>[</span><span class=n>count_mm39</span><span class=p>[</span><span class=s1>&#39;ens_id&#39;</span><span class=p>]</span><span class=o>.</span><span class=n>isin</span><span class=p>(</span><span class=n>ens_ids</span><span class=p>),</span> <span class=s1>&#39;gene_id&#39;</span><span class=p>]</span> <span class=o>=</span> <span class=n>gene_id</span>
</span></span></code></pre></div><p>#Transcript Filtering</p><div class=highlight><pre tabindex=0 class=chroma><code class=language-python data-lang=python><span class=line><span class=cl><span class=n>count_mm39</span><span class=p>[</span><span class=s1>&#39;sum&#39;</span><span class=p>]</span> <span class=o>=</span> <span class=n>count_mm39</span><span class=o>.</span><span class=n>iloc</span><span class=p>[:,</span> <span class=mi>2</span><span class=p>:]</span><span class=o>.</span><span class=n>sum</span><span class=p>(</span><span class=n>axis</span><span class=o>=</span><span class=mi>1</span><span class=p>)</span>
</span></span><span class=line><span class=cl><span class=n>count_mm39</span> <span class=o>=</span> <span class=n>count_mm39</span><span class=o>.</span><span class=n>sort_values</span><span class=p>(</span><span class=n>by</span><span class=o>=</span><span class=p>[</span><span class=s1>&#39;gene_id&#39;</span><span class=p>,</span> <span class=s1>&#39;sum&#39;</span><span class=p>],</span> <span class=n>ascending</span><span class=o>=</span><span class=p>[</span><span class=kc>True</span><span class=p>,</span> <span class=kc>False</span><span class=p>])</span>
</span></span><span class=line><span class=cl><span class=n>count_mm39</span> <span class=o>=</span> <span class=n>count_mm39</span><span class=o>.</span><span class=n>drop_duplicates</span><span class=p>(</span><span class=n>subset</span><span class=o>=</span><span class=p>[</span><span class=s1>&#39;gene_id&#39;</span><span class=p>],</span> <span class=n>keep</span><span class=o>=</span><span class=s1>&#39;first&#39;</span><span class=p>)</span>
</span></span><span class=line><span class=cl><span class=n>count_mm39</span> <span class=o>=</span> <span class=n>count_mm39</span><span class=o>.</span><span class=n>dropna</span><span class=p>(</span><span class=n>subset</span><span class=o>=</span><span class=p>[</span><span class=s1>&#39;gene_id&#39;</span><span class=p>])</span>
</span></span><span class=line><span class=cl>
</span></span><span class=line><span class=cl><span class=n>count_mm39</span> <span class=o>=</span> <span class=n>count_mm39</span><span class=o>.</span><span class=n>drop</span><span class=p>(</span><span class=n>columns</span><span class=o>=</span><span class=p>[</span><span class=s1>&#39;sum&#39;</span><span class=p>,</span> <span class=s1>&#39;ens_id&#39;</span><span class=p>])</span>
</span></span><span class=line><span class=cl><span class=n>gene_id_column</span> <span class=o>=</span> <span class=n>count_mm39</span><span class=p>[</span><span class=s1>&#39;gene_id&#39;</span><span class=p>]</span>
</span></span><span class=line><span class=cl><span class=n>count_mm39</span><span class=o>.</span><span class=n>drop</span><span class=p>(</span><span class=n>columns</span><span class=o>=</span><span class=p>[</span><span class=s1>&#39;gene_id&#39;</span><span class=p>],</span> <span class=n>inplace</span><span class=o>=</span><span class=kc>True</span><span class=p>)</span>
</span></span><span class=line><span class=cl><span class=n>count_mm39</span><span class=o>.</span><span class=n>insert</span><span class=p>(</span><span class=mi>0</span><span class=p>,</span> <span class=s1>&#39;gene_id&#39;</span><span class=p>,</span> <span class=n>gene_id_column</span><span class=p>)</span>
</span></span></code></pre></div><p>#Save</p><div class=highlight><pre tabindex=0 class=chroma><code class=language-python data-lang=python><span class=line><span class=cl><span class=n>count_mm39</span><span class=o>.</span><span class=n>rename</span><span class=p>(</span><span class=n>columns</span><span class=o>=</span><span class=p>{</span><span class=s1>&#39;gene_id&#39;</span><span class=p>:</span> <span class=s1>&#39;GeneID&#39;</span><span class=p>},</span> <span class=n>inplace</span><span class=o>=</span><span class=kc>True</span><span class=p>)</span>
</span></span><span class=line><span class=cl>
</span></span><span class=line><span class=cl><span class=k>def</span> <span class=nf>rename_columns</span><span class=p>(</span><span class=n>col</span><span class=p>):</span>
</span></span><span class=line><span class=cl>    <span class=n>parts</span> <span class=o>=</span> <span class=n>col</span><span class=o>.</span><span class=n>split</span><span class=p>(</span><span class=s1>&#39;_&#39;</span><span class=p>)</span>
</span></span><span class=line><span class=cl>    <span class=k>if</span> <span class=nb>len</span><span class=p>(</span><span class=n>parts</span><span class=p>)</span> <span class=o>&gt;=</span> <span class=mi>3</span><span class=p>:</span>
</span></span><span class=line><span class=cl>        <span class=n>new_col</span> <span class=o>=</span> <span class=n>parts</span><span class=p>[</span><span class=mi>0</span><span class=p>]</span> <span class=o>+</span> <span class=n>parts</span><span class=p>[</span><span class=mi>2</span><span class=p>]</span> <span class=o>+</span> <span class=s1>&#39;_&#39;</span> <span class=o>+</span> <span class=n>parts</span><span class=p>[</span><span class=mi>1</span><span class=p>]</span>
</span></span><span class=line><span class=cl>    <span class=k>else</span><span class=p>:</span>
</span></span><span class=line><span class=cl>        <span class=n>new_col</span> <span class=o>=</span> <span class=n>col</span>
</span></span><span class=line><span class=cl>    <span class=k>return</span> <span class=n>new_col</span>
</span></span><span class=line><span class=cl>
</span></span><span class=line><span class=cl><span class=n>count_mm39</span><span class=o>.</span><span class=n>columns</span> <span class=o>=</span> <span class=p>[</span><span class=n>rename_columns</span><span class=p>(</span><span class=n>col</span><span class=p>)</span> <span class=k>for</span> <span class=n>col</span> <span class=ow>in</span> <span class=n>count_mm39</span><span class=o>.</span><span class=n>columns</span><span class=p>]</span>
</span></span><span class=line><span class=cl><span class=n>count_mm39</span><span class=o>.</span><span class=n>to_csv</span><span class=p>(</span><span class=sa>f</span><span class=s2>&#34;</span><span class=si>{</span><span class=n>outdir</span><span class=si>}</span><span class=s2>/count_processed.csv&#34;</span><span class=p>,</span> <span class=n>index</span><span class=o>=</span><span class=kc>False</span><span class=p>)</span>
</span></span></code></pre></div><h3 id=3-deg-analysis>3. DEG Analysis
<a class=anchor href=#3-deg-analysis>#</a></h3><p>#Library & Set Path</p><div class=highlight><pre tabindex=0 class=chroma><code class=language-R data-lang=R><span class=line><span class=cl><span class=nf>library</span><span class=p>(</span><span class=n>edgeR</span><span class=p>)</span>
</span></span><span class=line><span class=cl>
</span></span><span class=line><span class=cl><span class=n>indir</span> <span class=o>=</span> <span class=s>&#34;/data/home/ysh980101/2504/mirna/data&#34;</span>
</span></span><span class=line><span class=cl><span class=n>outdir</span> <span class=o>=</span> <span class=s>&#34;/data/home/ysh980101/2504/mirna/result&#34;</span>
</span></span><span class=line><span class=cl>
</span></span><span class=line><span class=cl><span class=nf>setwd</span><span class=p>(</span><span class=n>indir</span><span class=p>)</span>
</span></span><span class=line><span class=cl><span class=nf>getwd</span><span class=p>()</span>
</span></span></code></pre></div><div class=highlight><pre tabindex=0 class=chroma><code class=language-plain data-lang=plain><span class=line><span class=cl>&#39;/data1/home/ysh980101/2504/mirna/data&#39;
</span></span></code></pre></div><p>#Set variables & Load Data</p><div class=highlight><pre tabindex=0 class=chroma><code class=language-R data-lang=R><span class=line><span class=cl><span class=n>tissue</span> <span class=o>&lt;-</span> <span class=s>&#34;G&#34;</span>
</span></span><span class=line><span class=cl><span class=n>S1</span> <span class=o>&lt;-</span> <span class=s>&#34;WT&#34;</span> 
</span></span><span class=line><span class=cl><span class=n>S2</span> <span class=o>&lt;-</span> <span class=s>&#34;GneKI&#34;</span>
</span></span><span class=line><span class=cl>
</span></span><span class=line><span class=cl><span class=n>counts</span> <span class=o>&lt;-</span> <span class=nf>read.csv</span><span class=p>(</span><span class=s>&#34;count_processed.csv&#34;</span><span class=p>)</span>
</span></span><span class=line><span class=cl><span class=n>meta</span> <span class=o>&lt;-</span> <span class=nf>read.csv</span><span class=p>(</span><span class=nf>paste0</span><span class=p>(</span><span class=s>&#34;mouse_meta_&#34;</span><span class=p>,</span><span class=n>tissue</span><span class=p>,</span><span class=s>&#34;.csv&#34;</span><span class=p>))</span>
</span></span><span class=line><span class=cl><span class=n>meta</span> <span class=o>&lt;-</span> <span class=n>meta[meta</span><span class=o>$</span><span class=n>Group</span> <span class=o>%in%</span> <span class=nf>c</span><span class=p>(</span><span class=n>S1</span><span class=p>,</span> <span class=n>S2</span><span class=p>),</span> <span class=n>]</span>
</span></span><span class=line><span class=cl>
</span></span><span class=line><span class=cl><span class=n>counts</span> <span class=o>&lt;-</span> <span class=n>counts[</span><span class=p>,</span> <span class=nf>c</span><span class=p>(</span><span class=s>&#34;GeneID&#34;</span><span class=p>,</span> <span class=nf>unique</span><span class=p>(</span><span class=n>meta</span><span class=o>$</span><span class=n>SampleID</span><span class=p>))</span><span class=n>]</span>
</span></span></code></pre></div><p>#Create DGElist & Normalization</p><div class=highlight><pre tabindex=0 class=chroma><code class=language-R data-lang=R><span class=line><span class=cl><span class=n>Group</span> <span class=o>&lt;-</span> <span class=nf>factor</span><span class=p>(</span><span class=n>meta</span><span class=o>$</span><span class=n>Group</span><span class=p>)</span>
</span></span><span class=line><span class=cl><span class=n>Group</span> <span class=o>&lt;-</span> <span class=nf>relevel</span><span class=p>(</span><span class=n>Group</span><span class=p>,</span> <span class=n>ref</span><span class=o>=</span><span class=n>S1</span><span class=p>)</span>
</span></span><span class=line><span class=cl>
</span></span><span class=line><span class=cl><span class=n>y</span> <span class=o>&lt;-</span> <span class=nf>DGEList</span><span class=p>(</span><span class=n>counts</span><span class=o>=</span><span class=n>counts[</span><span class=p>,</span><span class=m>2</span><span class=o>:</span><span class=nf>ncol</span><span class=p>(</span><span class=n>counts</span><span class=p>)</span><span class=n>]</span><span class=p>,</span> <span class=n>group</span><span class=o>=</span><span class=n>Group</span><span class=p>,</span> <span class=n>genes</span> <span class=o>=</span> <span class=n>counts[</span><span class=p>,</span><span class=m>1</span><span class=n>]</span><span class=p>)</span>
</span></span><span class=line><span class=cl><span class=n>y</span> <span class=o>&lt;-</span> <span class=nf>calcNormFactors</span><span class=p>(</span><span class=n>y</span><span class=p>)</span>
</span></span></code></pre></div><p>#Run DEG</p><div class=highlight><pre tabindex=0 class=chroma><code class=language-R data-lang=R><span class=line><span class=cl><span class=n>design</span> <span class=o>&lt;-</span> <span class=nf>model.matrix</span><span class=p>(</span><span class=o>~</span><span class=n>Group</span><span class=p>)</span>
</span></span><span class=line><span class=cl><span class=n>y</span> <span class=o>&lt;-</span> <span class=nf>estimateDisp</span><span class=p>(</span><span class=n>y</span><span class=p>,</span> <span class=n>design</span><span class=p>)</span>
</span></span><span class=line><span class=cl><span class=n>y</span> <span class=o>&lt;-</span> <span class=nf>estimateGLMRobustDisp</span><span class=p>(</span><span class=n>y</span><span class=p>,</span><span class=n>design</span><span class=p>)</span>
</span></span><span class=line><span class=cl>
</span></span><span class=line><span class=cl><span class=n>fit</span> <span class=o>&lt;-</span> <span class=nf>glmFit</span><span class=p>(</span><span class=n>y</span><span class=p>,</span> <span class=n>design</span><span class=p>)</span>
</span></span><span class=line><span class=cl><span class=n>lrt</span> <span class=o>&lt;-</span> <span class=nf>glmLRT</span><span class=p>(</span><span class=n>fit</span><span class=p>)</span>
</span></span><span class=line><span class=cl>
</span></span><span class=line><span class=cl><span class=nf>plotMD</span><span class=p>(</span><span class=n>lrt</span><span class=p>)</span>
</span></span><span class=line><span class=cl><span class=nf>abline</span><span class=p>(</span><span class=n>h</span><span class=o>=</span><span class=nf>c</span><span class=p>(</span><span class=m>-1</span><span class=p>,</span><span class=m>1</span><span class=p>),</span> <span class=n>col</span><span class=o>=</span><span class=s>&#34;blue&#34;</span><span class=p>)</span>
</span></span></code></pre></div><p><img src=https://github.com/user-attachments/assets/339f2e97-6b71-45c1-8092-b801b68c9f23 alt=image></p><p>#Save</p><div class=highlight><pre tabindex=0 class=chroma><code class=language-R data-lang=R><span class=line><span class=cl><span class=n>result_table</span> <span class=o>&lt;-</span> <span class=nf>topTags</span><span class=p>(</span><span class=n>lrt</span><span class=p>,</span> <span class=n>n</span> <span class=o>=</span> <span class=nf>nrow</span><span class=p>(</span><span class=n>lrt</span><span class=o>$</span><span class=n>table</span><span class=p>))</span>
</span></span><span class=line><span class=cl><span class=n>sorted_result_table</span> <span class=o>&lt;-</span> <span class=n>result_table</span><span class=nf>[order</span><span class=p>(</span><span class=n>result_table</span><span class=o>$</span><span class=n>table</span><span class=o>$</span><span class=n>FDR</span><span class=p>),</span> <span class=n>]</span>
</span></span><span class=line><span class=cl><span class=n>filtered_result_table</span> <span class=o>&lt;-</span> <span class=n>sorted_result_table[sorted_result_table</span><span class=o>$</span><span class=n>table</span><span class=o>$</span><span class=n>FDR</span> <span class=o>&lt;</span> <span class=m>0.05</span><span class=p>,</span> <span class=n>]</span>
</span></span><span class=line><span class=cl>
</span></span><span class=line><span class=cl><span class=nf>write.csv</span><span class=p>(</span><span class=n>sorted_result_table</span><span class=p>,</span> <span class=n>file</span> <span class=o>=</span> <span class=nf>paste0</span><span class=p>(</span><span class=n>outdir</span><span class=p>,</span><span class=s>&#34;/de-&#34;</span><span class=p>,</span><span class=n>tissue</span><span class=p>,</span><span class=s>&#34;_&#34;</span><span class=p>,</span><span class=n>S1</span><span class=p>,</span><span class=s>&#34;-&#34;</span><span class=p>,</span><span class=n>S2</span><span class=p>,</span><span class=s>&#34;.csv&#34;</span><span class=p>))</span>
</span></span></code></pre></div></article><footer class=book-footer><div class="flex flex-wrap justify-between"></div><script>(function(){function e(e){const t=window.getSelection(),n=document.createRange();n.selectNodeContents(e),t.removeAllRanges(),t.addRange(n)}document.querySelectorAll("pre code").forEach(t=>{t.addEventListener("click",function(){if(window.getSelection().toString())return;e(t.parentElement),navigator.clipboard&&navigator.clipboard.writeText(t.parentElement.textContent)})})})()</script></footer><div class=book-comments><script src=https://giscus.app/client.js data-repo=yshghid/yshghid.github.io data-repo-id=R_kgDONkMkNg data-category-id=DIC_kwDONkMkNs4CloJh data-mapping=pathname data-strict=0 data-reactions-enabled=1 data-emit-metadata=0 data-input-position=bottom data-theme=preferred_color_scheme data-lang=ko crossorigin=anonymous async></script></div><label for=menu-control class="hidden book-menu-overlay"></label></div><aside class=book-toc><div class=book-toc-content><nav id=TableOfContents><ul><li><ul><li><a href=#1-align-rna-seq>1. Align RNA-seq</a></li><li><a href=#2-gene-id-annotation>2. Gene ID Annotation</a></li><li><a href=#3-deg-analysis>3. DEG Analysis</a></li></ul></li></ul></nav></div></aside></main></body></html>