<!doctype html><html lang=en-us dir=ltr><head><meta charset=UTF-8><meta name=viewport content="width=device-width,initial-scale=1"><meta name=description content='
  Rsubread, edgeR: RNA-seq 전처리
  #

#2025-04-21

가장 오류 적게나는 조합!

  1. Align RNA-seq
  #

#Load Packages
library(Rsubread)
library(org.Mm.eg.db)
library(gridExtra)
library(reshape2)
#Set Path
indir = "/data/home/ysh980101/2504/mirna/data"
outdir = "/data/home/ysh980101/2504/mirna/data"
refpath = "/data/home/ysh980101/2406/data-gne/mm39.fa"

setwd(indir)
getwd()
&#39;/data/home/ysh980101/2504/mirna/data&#39;
#Build Index
buildindex(basename = "mm39", reference = refpath)
#Read Alignment
files <- list.files(pattern="\\.fastq\\.gz$", full.names=TRUE)
bams <- sub("\\.fastq\\.gz$", ".bam", files)
samples <- gsub("^\\.\\/|\\.fastq\\.gz$", "", files)
targets <- read.delim("target.txt", header=TRUE)

align(index="mm39", readfile1=files, input_format="gzFASTQ", output_file=bams, nthreads=50)
#Quantification
fc = featureCounts(bams, isGTFAnnotationFile=TRUE, GTF.featureType="exon", GTF.attrType="gene_id", isPairedEnd=FALSE, annot.ext="mm39.knownGene.gtf", useMetaFeatures=FALSE, allowMultiOverlap=TRUE, nthreads=50)
#Save Countdata'><meta name=theme-color media="(prefers-color-scheme: light)" content="#ffffff"><meta name=theme-color media="(prefers-color-scheme: dark)" content="#343a40"><meta name=color-scheme content="light dark"><meta property="og:url" content="https://yshghid.github.io/docs/study/bioinformatics/bi8/"><meta property="og:site_name" content=" "><meta property="og:title" content="Rsubread, edgeR: RNA-seq 전처리"><meta property="og:description" content='Rsubread, edgeR: RNA-seq 전처리 # #2025-04-21
가장 오류 적게나는 조합!
1. Align RNA-seq # #Load Packages
library(Rsubread) library(org.Mm.eg.db) library(gridExtra) library(reshape2) #Set Path
indir = "/data/home/ysh980101/2504/mirna/data" outdir = "/data/home/ysh980101/2504/mirna/data" refpath = "/data/home/ysh980101/2406/data-gne/mm39.fa" setwd(indir) getwd() &#39;/data/home/ysh980101/2504/mirna/data&#39; #Build Index
buildindex(basename = "mm39", reference = refpath) #Read Alignment
files <- list.files(pattern="\\.fastq\\.gz$", full.names=TRUE) bams <- sub("\\.fastq\\.gz$", ".bam", files) samples <- gsub("^\\.\\/|\\.fastq\\.gz$", "", files) targets <- read.delim("target.txt", header=TRUE) align(index="mm39", readfile1=files, input_format="gzFASTQ", output_file=bams, nthreads=50) #Quantification
fc = featureCounts(bams, isGTFAnnotationFile=TRUE, GTF.featureType="exon", GTF.attrType="gene_id", isPairedEnd=FALSE, annot.ext="mm39.knownGene.gtf", useMetaFeatures=FALSE, allowMultiOverlap=TRUE, nthreads=50) #Save Countdata'><meta property="og:locale" content="en_us"><meta property="og:type" content="article"><meta property="article:section" content="docs"><meta property="article:published_time" content="2025-04-21T00:00:00+00:00"><meta property="article:modified_time" content="2025-04-21T00:00:00+00:00"><meta property="article:tag" content="2025-04"><title>Rsubread, edgeR: RNA-seq 전처리 |</title><link rel=icon href=/favicon.png><link rel=manifest href=/manifest.json><link rel=canonical href=https://yshghid.github.io/docs/study/bioinformatics/bi8/><link rel=stylesheet href=/book.min.6217d077edb4189fd0578345e84bca1a884dfdee121ff8dc9a0f55cfe0852bc9.css integrity="sha256-YhfQd+20GJ/QV4NF6EvKGohN/e4SH/jcmg9Vz+CFK8k=" crossorigin=anonymous><script defer src=/fuse.min.js></script><script defer src=/en.search.min.578a2d8e6e30e0a0ae3682797882d89ca0cbbf1734d206705396ea76cf57bbb6.js integrity="sha256-V4otjm4w4KCuNoJ5eILYnKDLvxc00gZwU5bqds9Xu7Y=" crossorigin=anonymous></script></head><body dir=ltr><input type=checkbox class="hidden toggle" id=menu-control>
<input type=checkbox class="hidden toggle" id=toc-control><main class="container flex"><aside class=book-menu><div class=book-menu-content><nav><h2 class=book-brand><a class="flex align-center" href=/><img src=/logo.png alt=Logo class=book-icon><span></span></a></h2><div class="book-search hidden"><input type=text id=book-search-input placeholder=Search aria-label=Search maxlength=64 data-hotkeys=s/><div class="book-search-spinner hidden"></div><ul id=book-search-results></ul></div><script>document.querySelector(".book-search").classList.remove("hidden")</script><ul><li class=book-section-flat><span>기록</span><ul><li><a href=/docs/hobby/book/>글</a><ul></ul></li><li><a href=/docs/hobby/daily/>일상</a><ul></ul></li></ul></li><li class=book-section-flat><span>공부</span><ul><li><a href=/docs/study/bioinformatics/>Bioinformatics</a><ul></ul></li><li><a href=/docs/study/ai/>AI</a><ul></ul></li><li><a href=/docs/study/sw/>SW</a><ul></ul></li><li><a href=/docs/study/algorithm/>알고리즘</a><ul></ul></li><li><a href=/docs/study/career/>취업</a><ul></ul></li></ul></li></ul></nav><script>(function(){var e=document.querySelector("aside .book-menu-content");addEventListener("beforeunload",function(){localStorage.setItem("menu.scrollTop",e.scrollTop)}),e.scrollTop=localStorage.getItem("menu.scrollTop")})()</script></div></aside><div class=book-page><header class=book-header><div class="flex align-center justify-between"><label for=menu-control><img src=/svg/menu.svg class=book-icon alt=Menu></label><h3>Rsubread, edgeR: RNA-seq 전처리</h3><label for=toc-control><img src=/svg/toc.svg class=book-icon alt="Table of Contents"></label></div><aside class="hidden clearfix"><nav id=TableOfContents><ul><li><ul><li><a href=#1-align-rna-seq>1. Align RNA-seq</a></li><li><a href=#2-gene-id-annotation>2. Gene ID Annotation</a></li><li><a href=#3-deg-analysis>3. DEG Analysis</a></li></ul></li></ul></nav></aside></header><article class="markdown book-article"><h1 id=rsubread-edger-rna-seq-전처리>Rsubread, edgeR: RNA-seq 전처리
<a class=anchor href=#rsubread-edger-rna-seq-%ec%a0%84%ec%b2%98%eb%a6%ac>#</a></h1><p>#2025-04-21</p><hr><p>가장 오류 적게나는 조합!</p><h3 id=1-align-rna-seq>1. Align RNA-seq
<a class=anchor href=#1-align-rna-seq>#</a></h3><p>#Load Packages</p><div class=highlight><pre tabindex=0 style=color:#e5e5e5;background-color:#000;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-R data-lang=R><span style=display:flex><span>library(Rsubread)
</span></span><span style=display:flex><span>library(org.Mm.eg.db)
</span></span><span style=display:flex><span>library(gridExtra)
</span></span><span style=display:flex><span>library(reshape2)
</span></span></code></pre></div><p>#Set Path</p><div class=highlight><pre tabindex=0 style=color:#e5e5e5;background-color:#000;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-R data-lang=R><span style=display:flex><span>indir = <span style=color:#0ff;font-weight:700>&#34;/data/home/ysh980101/2504/mirna/data&#34;</span>
</span></span><span style=display:flex><span>outdir = <span style=color:#0ff;font-weight:700>&#34;/data/home/ysh980101/2504/mirna/data&#34;</span>
</span></span><span style=display:flex><span>refpath = <span style=color:#0ff;font-weight:700>&#34;/data/home/ysh980101/2406/data-gne/mm39.fa&#34;</span>
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span>setwd(indir)
</span></span><span style=display:flex><span>getwd()
</span></span></code></pre></div><div class=highlight><pre tabindex=0 style=color:#e5e5e5;background-color:#000;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-plain data-lang=plain><span style=display:flex><span>&#39;/data/home/ysh980101/2504/mirna/data&#39;
</span></span></code></pre></div><p>#Build Index</p><div class=highlight><pre tabindex=0 style=color:#e5e5e5;background-color:#000;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-R data-lang=R><span style=display:flex><span>buildindex(basename = <span style=color:#0ff;font-weight:700>&#34;mm39&#34;</span>, reference = refpath)
</span></span></code></pre></div><p>#Read Alignment</p><div class=highlight><pre tabindex=0 style=color:#e5e5e5;background-color:#000;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-R data-lang=R><span style=display:flex><span>files &lt;- list.files(pattern=<span style=color:#0ff;font-weight:700>&#34;\\.fastq\\.gz$&#34;</span>, full.names=<span style=color:#fff;font-weight:700>TRUE</span>)
</span></span><span style=display:flex><span>bams &lt;- sub(<span style=color:#0ff;font-weight:700>&#34;\\.fastq\\.gz$&#34;</span>, <span style=color:#0ff;font-weight:700>&#34;.bam&#34;</span>, files)
</span></span><span style=display:flex><span>samples &lt;- gsub(<span style=color:#0ff;font-weight:700>&#34;^\\.\\/|\\.fastq\\.gz$&#34;</span>, <span style=color:#0ff;font-weight:700>&#34;&#34;</span>, files)
</span></span><span style=display:flex><span>targets &lt;- read.delim(<span style=color:#0ff;font-weight:700>&#34;target.txt&#34;</span>, header=<span style=color:#fff;font-weight:700>TRUE</span>)
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span>align(index=<span style=color:#0ff;font-weight:700>&#34;mm39&#34;</span>, readfile1=files, input_format=<span style=color:#0ff;font-weight:700>&#34;gzFASTQ&#34;</span>, output_file=bams, nthreads=<span style=color:#ff0;font-weight:700>50</span>)
</span></span></code></pre></div><p>#Quantification</p><div class=highlight><pre tabindex=0 style=color:#e5e5e5;background-color:#000;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-R data-lang=R><span style=display:flex><span>fc = featureCounts(bams, isGTFAnnotationFile=<span style=color:#fff;font-weight:700>TRUE</span>, GTF.featureType=<span style=color:#0ff;font-weight:700>&#34;exon&#34;</span>, GTF.attrType=<span style=color:#0ff;font-weight:700>&#34;gene_id&#34;</span>, isPairedEnd=<span style=color:#fff;font-weight:700>FALSE</span>, annot.ext=<span style=color:#0ff;font-weight:700>&#34;mm39.knownGene.gtf&#34;</span>, useMetaFeatures=<span style=color:#fff;font-weight:700>FALSE</span>, allowMultiOverlap=<span style=color:#fff;font-weight:700>TRUE</span>, nthreads=<span style=color:#ff0;font-weight:700>50</span>)
</span></span></code></pre></div><p>#Save Countdata</p><div class=highlight><pre tabindex=0 style=color:#e5e5e5;background-color:#000;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-R data-lang=R><span style=display:flex><span>colnames(fc$counts) &lt;- samples
</span></span><span style=display:flex><span>y &lt;- DGEList(fc$counts, group=group)
</span></span><span style=display:flex><span>write.csv(as.data.frame(y$counts), file = paste0(outdir,<span style=color:#0ff;font-weight:700>&#34;/count.csv&#34;</span>, row.names = <span style=color:#fff;font-weight:700>TRUE</span>))
</span></span></code></pre></div><h3 id=2-gene-id-annotation>2. Gene ID Annotation
<a class=anchor href=#2-gene-id-annotation>#</a></h3><p>#Load Packages</p><div class=highlight><pre tabindex=0 style=color:#e5e5e5;background-color:#000;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-python data-lang=python><span style=display:flex><span><span style=color:#fff;font-weight:700>import</span> pandas <span style=color:#fff;font-weight:700>as</span> pd
</span></span><span style=display:flex><span><span style=color:#fff;font-weight:700>import</span> numpy <span style=color:#fff;font-weight:700>as</span> np
</span></span><span style=display:flex><span><span style=color:#fff;font-weight:700>import</span> os
</span></span></code></pre></div><p>#Set Path</p><div class=highlight><pre tabindex=0 style=color:#e5e5e5;background-color:#000;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-python data-lang=python><span style=display:flex><span>indir = <span style=color:#0ff;font-weight:700>&#34;/data/home/ysh980101/2504/mirna/data&#34;</span>
</span></span><span style=display:flex><span>outdir = <span style=color:#0ff;font-weight:700>&#34;/data/home/ysh980101/2504/mirna/data&#34;</span>
</span></span><span style=display:flex><span>annotpath = <span style=color:#0ff;font-weight:700>&#34;/data/home/ysh980101/2406/data-gne/MRK_ENSEMBL.rpt&#34;</span>
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span>os.chdir(indir)
</span></span><span style=display:flex><span>os.getcwd()
</span></span></code></pre></div><div class=highlight><pre tabindex=0 style=color:#e5e5e5;background-color:#000;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-plain data-lang=plain><span style=display:flex><span>&#39;/data1/home/ysh980101/2504/mirna/result&#39;
</span></span></code></pre></div><p>#Load Annotation</p><div class=highlight><pre tabindex=0 style=color:#e5e5e5;background-color:#000;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-python data-lang=python><span style=display:flex><span>annotation = pd.read_csv(annotpath, sep=<span style=color:#0ff;font-weight:700>&#34;</span><span style=color:#0ff;font-weight:700>\t</span><span style=color:#0ff;font-weight:700>&#34;</span>, names=[<span style=color:#fff;font-weight:700>str</span>(i) <span style=color:#fff;font-weight:700>for</span> i in <span style=color:#fff;font-weight:700>range</span>(<span style=color:#ff0;font-weight:700>13</span>)])
</span></span><span style=display:flex><span>annotation = annotation.dropna(subset=[<span style=color:#0ff;font-weight:700>&#39;6&#39;</span>])
</span></span><span style=display:flex><span>annotation = annotation[annotation[<span style=color:#0ff;font-weight:700>&#39;8&#39;</span>] == <span style=color:#0ff;font-weight:700>&#39;protein coding gene&#39;</span>]
</span></span></code></pre></div><p>#Load Count & Gene ID Mapping</p><div class=highlight><pre tabindex=0 style=color:#e5e5e5;background-color:#000;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-python data-lang=python><span style=display:flex><span>count_mm39 = pd.read_csv(<span style=color:#0ff;font-weight:700>&#34;count.csv&#34;</span>)
</span></span><span style=display:flex><span>count_mm39.rename(columns={count_mm39.columns[<span style=color:#ff0;font-weight:700>0</span>]: <span style=color:#0ff;font-weight:700>&#39;ens_id&#39;</span>}, inplace=<span style=color:#fff;font-weight:700>True</span>)
</span></span><span style=display:flex><span>count_mm39[<span style=color:#0ff;font-weight:700>&#39;ens_id&#39;</span>] = count_mm39[<span style=color:#0ff;font-weight:700>&#39;ens_id&#39;</span>].str.split(<span style=color:#0ff;font-weight:700>&#39;.&#39;</span>).str[<span style=color:#ff0;font-weight:700>0</span>]
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span><span style=color:#fff;font-weight:700>for</span> index, row in annotation.iterrows():
</span></span><span style=display:flex><span>    ens_ids = row[<span style=color:#0ff;font-weight:700>&#39;6&#39;</span>].split()
</span></span><span style=display:flex><span>    gene_id = row[<span style=color:#0ff;font-weight:700>&#39;1&#39;</span>]
</span></span><span style=display:flex><span>    count_mm39.loc[count_mm39[<span style=color:#0ff;font-weight:700>&#39;ens_id&#39;</span>].isin(ens_ids), <span style=color:#0ff;font-weight:700>&#39;gene_id&#39;</span>] = gene_id
</span></span></code></pre></div><p>#Transcript Filtering</p><div class=highlight><pre tabindex=0 style=color:#e5e5e5;background-color:#000;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-python data-lang=python><span style=display:flex><span>count_mm39[<span style=color:#0ff;font-weight:700>&#39;sum&#39;</span>] = count_mm39.iloc[:, <span style=color:#ff0;font-weight:700>2</span>:].sum(axis=<span style=color:#ff0;font-weight:700>1</span>)
</span></span><span style=display:flex><span>count_mm39 = count_mm39.sort_values(by=[<span style=color:#0ff;font-weight:700>&#39;gene_id&#39;</span>, <span style=color:#0ff;font-weight:700>&#39;sum&#39;</span>], ascending=[<span style=color:#fff;font-weight:700>True</span>, <span style=color:#fff;font-weight:700>False</span>])
</span></span><span style=display:flex><span>count_mm39 = count_mm39.drop_duplicates(subset=[<span style=color:#0ff;font-weight:700>&#39;gene_id&#39;</span>], keep=<span style=color:#0ff;font-weight:700>&#39;first&#39;</span>)
</span></span><span style=display:flex><span>count_mm39 = count_mm39.dropna(subset=[<span style=color:#0ff;font-weight:700>&#39;gene_id&#39;</span>])
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span>count_mm39 = count_mm39.drop(columns=[<span style=color:#0ff;font-weight:700>&#39;sum&#39;</span>, <span style=color:#0ff;font-weight:700>&#39;ens_id&#39;</span>])
</span></span><span style=display:flex><span>gene_id_column = count_mm39[<span style=color:#0ff;font-weight:700>&#39;gene_id&#39;</span>]
</span></span><span style=display:flex><span>count_mm39.drop(columns=[<span style=color:#0ff;font-weight:700>&#39;gene_id&#39;</span>], inplace=<span style=color:#fff;font-weight:700>True</span>)
</span></span><span style=display:flex><span>count_mm39.insert(<span style=color:#ff0;font-weight:700>0</span>, <span style=color:#0ff;font-weight:700>&#39;gene_id&#39;</span>, gene_id_column)
</span></span></code></pre></div><p>#Save</p><div class=highlight><pre tabindex=0 style=color:#e5e5e5;background-color:#000;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-python data-lang=python><span style=display:flex><span>count_mm39.rename(columns={<span style=color:#0ff;font-weight:700>&#39;gene_id&#39;</span>: <span style=color:#0ff;font-weight:700>&#39;GeneID&#39;</span>}, inplace=<span style=color:#fff;font-weight:700>True</span>)
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span><span style=color:#fff;font-weight:700>def</span> rename_columns(col):
</span></span><span style=display:flex><span>    parts = col.split(<span style=color:#0ff;font-weight:700>&#39;_&#39;</span>)
</span></span><span style=display:flex><span>    <span style=color:#fff;font-weight:700>if</span> <span style=color:#fff;font-weight:700>len</span>(parts) &gt;= <span style=color:#ff0;font-weight:700>3</span>:
</span></span><span style=display:flex><span>        new_col = parts[<span style=color:#ff0;font-weight:700>0</span>] + parts[<span style=color:#ff0;font-weight:700>2</span>] + <span style=color:#0ff;font-weight:700>&#39;_&#39;</span> + parts[<span style=color:#ff0;font-weight:700>1</span>]
</span></span><span style=display:flex><span>    <span style=color:#fff;font-weight:700>else</span>:
</span></span><span style=display:flex><span>        new_col = col
</span></span><span style=display:flex><span>    <span style=color:#fff;font-weight:700>return</span> new_col
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span>count_mm39.columns = [rename_columns(col) <span style=color:#fff;font-weight:700>for</span> col in count_mm39.columns]
</span></span><span style=display:flex><span>count_mm39.to_csv(<span style=color:#0ff;font-weight:700>f</span><span style=color:#0ff;font-weight:700>&#34;</span><span style=color:#0ff;font-weight:700>{</span>outdir<span style=color:#0ff;font-weight:700>}</span><span style=color:#0ff;font-weight:700>/count_processed.csv&#34;</span>, index=<span style=color:#fff;font-weight:700>False</span>)
</span></span></code></pre></div><h3 id=3-deg-analysis>3. DEG Analysis
<a class=anchor href=#3-deg-analysis>#</a></h3><p>#Library & Set Path</p><div class=highlight><pre tabindex=0 style=color:#e5e5e5;background-color:#000;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-R data-lang=R><span style=display:flex><span>library(edgeR)
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span>indir = <span style=color:#0ff;font-weight:700>&#34;/data/home/ysh980101/2504/mirna/data&#34;</span>
</span></span><span style=display:flex><span>outdir = <span style=color:#0ff;font-weight:700>&#34;/data/home/ysh980101/2504/mirna/result&#34;</span>
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span>setwd(indir)
</span></span><span style=display:flex><span>getwd()
</span></span></code></pre></div><div class=highlight><pre tabindex=0 style=color:#e5e5e5;background-color:#000;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-plain data-lang=plain><span style=display:flex><span>&#39;/data1/home/ysh980101/2504/mirna/data&#39;
</span></span></code></pre></div><p>#Set variables & Load Data</p><div class=highlight><pre tabindex=0 style=color:#e5e5e5;background-color:#000;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-R data-lang=R><span style=display:flex><span>tissue &lt;- <span style=color:#0ff;font-weight:700>&#34;G&#34;</span>
</span></span><span style=display:flex><span>S1 &lt;- <span style=color:#0ff;font-weight:700>&#34;WT&#34;</span> 
</span></span><span style=display:flex><span>S2 &lt;- <span style=color:#0ff;font-weight:700>&#34;GneKI&#34;</span>
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span>counts &lt;- read.csv(<span style=color:#0ff;font-weight:700>&#34;count_processed.csv&#34;</span>)
</span></span><span style=display:flex><span>meta &lt;- read.csv(paste0(<span style=color:#0ff;font-weight:700>&#34;mouse_meta_&#34;</span>,tissue,<span style=color:#0ff;font-weight:700>&#34;.csv&#34;</span>))
</span></span><span style=display:flex><span>meta &lt;- meta[meta$Group %in% c(S1, S2), ]
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span>counts &lt;- counts[, c(<span style=color:#0ff;font-weight:700>&#34;GeneID&#34;</span>, unique(meta$SampleID))]
</span></span></code></pre></div><p>#Create DGElist & Normalization</p><div class=highlight><pre tabindex=0 style=color:#e5e5e5;background-color:#000;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-R data-lang=R><span style=display:flex><span>Group &lt;- factor(meta$Group)
</span></span><span style=display:flex><span>Group &lt;- relevel(Group, ref=S1)
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span>y &lt;- DGEList(counts=counts[,<span style=color:#ff0;font-weight:700>2</span>:ncol(counts)], group=Group, genes = counts[,<span style=color:#ff0;font-weight:700>1</span>])
</span></span><span style=display:flex><span>y &lt;- calcNormFactors(y)
</span></span></code></pre></div><p>#Run DEG</p><div class=highlight><pre tabindex=0 style=color:#e5e5e5;background-color:#000;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-R data-lang=R><span style=display:flex><span>design &lt;- model.matrix(~Group)
</span></span><span style=display:flex><span>y &lt;- estimateDisp(y, design)
</span></span><span style=display:flex><span>y &lt;- estimateGLMRobustDisp(y,design)
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span>fit &lt;- glmFit(y, design)
</span></span><span style=display:flex><span>lrt &lt;- glmLRT(fit)
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span>plotMD(lrt)
</span></span><span style=display:flex><span>abline(h=c(<span style=color:#ff0;font-weight:700>-1</span>,<span style=color:#ff0;font-weight:700>1</span>), col=<span style=color:#0ff;font-weight:700>&#34;blue&#34;</span>)
</span></span></code></pre></div><p><img src=https://github.com/user-attachments/assets/339f2e97-6b71-45c1-8092-b801b68c9f23 alt=image></p><p>#Save</p><div class=highlight><pre tabindex=0 style=color:#e5e5e5;background-color:#000;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-R data-lang=R><span style=display:flex><span>result_table &lt;- topTags(lrt, n = nrow(lrt$table))
</span></span><span style=display:flex><span>sorted_result_table &lt;- result_table[order(result_table$table$FDR), ]
</span></span><span style=display:flex><span>filtered_result_table &lt;- sorted_result_table[sorted_result_table$table$FDR &lt; <span style=color:#ff0;font-weight:700>0.05</span>, ]
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span>write.csv(sorted_result_table, file = paste0(outdir,<span style=color:#0ff;font-weight:700>&#34;/de-&#34;</span>,tissue,<span style=color:#0ff;font-weight:700>&#34;_&#34;</span>,S1,<span style=color:#0ff;font-weight:700>&#34;-&#34;</span>,S2,<span style=color:#0ff;font-weight:700>&#34;.csv&#34;</span>))
</span></span></code></pre></div></article><footer class=book-footer><div class="flex flex-wrap justify-between"></div><script>(function(){function e(e){const t=window.getSelection(),n=document.createRange();n.selectNodeContents(e),t.removeAllRanges(),t.addRange(n)}document.querySelectorAll("pre code").forEach(t=>{t.addEventListener("click",function(){if(window.getSelection().toString())return;e(t.parentElement),navigator.clipboard&&navigator.clipboard.writeText(t.parentElement.textContent)})})})()</script></footer><div class=book-comments><script src=https://giscus.app/client.js data-repo=yshghid/yshghid.github.io data-repo-id=R_kgDONkMkNg data-category-id=DIC_kwDONkMkNs4CloJh data-mapping=pathname data-strict=0 data-reactions-enabled=1 data-emit-metadata=0 data-input-position=bottom data-theme=preferred_color_scheme data-lang=ko crossorigin=anonymous async></script></div><label for=menu-control class="hidden book-menu-overlay"></label></div><aside class=book-toc><div class=book-toc-content><nav id=TableOfContents><ul><li><ul><li><a href=#1-align-rna-seq>1. Align RNA-seq</a></li><li><a href=#2-gene-id-annotation>2. Gene ID Annotation</a></li><li><a href=#3-deg-analysis>3. DEG Analysis</a></li></ul></li></ul></nav></div></aside></main></body></html>