<!doctype html><html lang=en-us dir=ltr><head><meta charset=UTF-8><meta name=viewport content="width=device-width,initial-scale=1"><meta name=description content='
  RNA-seq 전처리 꿀조합 (Rsubread, edgeR)
  #

#2025-04-21

가장 오류 적게나는 조합!

  1. Align RNA-seq
  #


  Load Packages
  #

library(Rsubread)
library(org.Mm.eg.db)
library(gridExtra)
library(reshape2)

  Set Path
  #

indir = "/data/home/ysh980101/2504/mirna/data"
outdir = "/data/home/ysh980101/2504/mirna/data"
refpath = "/data/home/ysh980101/2406/data-gne/mm39.fa"

setwd(indir)
getwd()
&#39;/data/home/ysh980101/2504/mirna/data&#39;

  Build Index
  #

buildindex(basename = "mm39", reference = refpath)

  Read Alignment
  #

files <- list.files(pattern="\\.fastq\\.gz$", full.names=TRUE)
bams <- sub("\\.fastq\\.gz$", ".bam", files)
samples <- gsub("^\\.\\/|\\.fastq\\.gz$", "", files)
targets <- read.delim("target.txt", header=TRUE)

align(index="mm39", readfile1=files, input_format="gzFASTQ", output_file=bams, nthreads=50)

  Quantification
  #

fc = featureCounts(bams, isGTFAnnotationFile=TRUE, GTF.featureType="exon", GTF.attrType="gene_id", isPairedEnd=FALSE, annot.ext="mm39.knownGene.gtf", useMetaFeatures=FALSE, allowMultiOverlap=TRUE, nthreads=50)

  Save Countdata
  #

colnames(fc$counts) <- samples
y <- DGEList(fc$counts, group=group)
write.csv(as.data.frame(y$counts), file = paste0(outdir,"/count.csv", row.names = TRUE))

  2. Gene ID Annotation
  #


  Load Packages
  #

import pandas as pd
import numpy as np
import os

  Set Path
  #

indir = "/data/home/ysh980101/2504/mirna/data"
outdir = "/data/home/ysh980101/2504/mirna/data"
annotpath = "/data/home/ysh980101/2406/data-gne/MRK_ENSEMBL.rpt"

os.chdir(indir)
os.getcwd()
&#39;/data1/home/ysh980101/2504/mirna/result&#39;

  Load Annotation
  #

annotation = pd.read_csv(annotpath, sep="\t", names=[str(i) for i in range(13)])
annotation = annotation.dropna(subset=[&#39;6&#39;])
annotation = annotation[annotation[&#39;8&#39;] == &#39;protein coding gene&#39;]

  Load Count & Gene ID Mapping
  #

count_mm39 = pd.read_csv("count.csv")
count_mm39.rename(columns={count_mm39.columns[0]: &#39;ens_id&#39;}, inplace=True)
count_mm39[&#39;ens_id&#39;] = count_mm39[&#39;ens_id&#39;].str.split(&#39;.&#39;).str[0]

for index, row in annotation.iterrows():
    ens_ids = row[&#39;6&#39;].split()
    gene_id = row[&#39;1&#39;]
    count_mm39.loc[count_mm39[&#39;ens_id&#39;].isin(ens_ids), &#39;gene_id&#39;] = gene_id

  Transcript Filtering
  #

count_mm39[&#39;sum&#39;] = count_mm39.iloc[:, 2:].sum(axis=1)
count_mm39 = count_mm39.sort_values(by=[&#39;gene_id&#39;, &#39;sum&#39;], ascending=[True, False])
count_mm39 = count_mm39.drop_duplicates(subset=[&#39;gene_id&#39;], keep=&#39;first&#39;)
count_mm39 = count_mm39.dropna(subset=[&#39;gene_id&#39;])

count_mm39 = count_mm39.drop(columns=[&#39;sum&#39;, &#39;ens_id&#39;])
gene_id_column = count_mm39[&#39;gene_id&#39;]
count_mm39.drop(columns=[&#39;gene_id&#39;], inplace=True)
count_mm39.insert(0, &#39;gene_id&#39;, gene_id_column)

  Save
  #

count_mm39.rename(columns={&#39;gene_id&#39;: &#39;GeneID&#39;}, inplace=True)

def rename_columns(col):
    parts = col.split(&#39;_&#39;)
    if len(parts) >= 3:
        new_col = parts[0] + parts[2] + &#39;_&#39; + parts[1]
    else:
        new_col = col
    return new_col

count_mm39.columns = [rename_columns(col) for col in count_mm39.columns]
count_mm39.to_csv(f"{outdir}/count_processed.csv", index=False)

  3. DEG Analysis
  #


  Library & Set Path
  #

library(edgeR)

indir = "/data/home/ysh980101/2504/mirna/data"
outdir = "/data/home/ysh980101/2504/mirna/result"

setwd(indir)
getwd()
&#39;/data1/home/ysh980101/2504/mirna/data&#39;

  Set variables & Load Data
  #

tissue <- "G"
S1 <- "WT" 
S2 <- "GneKI"

counts <- read.csv("count_processed.csv")
meta <- read.csv(paste0("mouse_meta_",tissue,".csv"))
meta <- meta[meta$Group %in% c(S1, S2), ]

counts <- counts[, c("GeneID", unique(meta$SampleID))]

  Create DGElist & Normalization
  #

Group <- factor(meta$Group)
Group <- relevel(Group, ref=S1)

y <- DGEList(counts=counts[,2:ncol(counts)], group=Group, genes = counts[,1])
y <- calcNormFactors(y)

  Run DEG
  #

design <- model.matrix(~Group)
y <- estimateDisp(y, design)
y <- estimateGLMRobustDisp(y,design)

fit <- glmFit(y, design)
lrt <- glmLRT(fit)

plotMD(lrt)
abline(h=c(-1,1), col="blue")
'><meta name=theme-color media="(prefers-color-scheme: light)" content="#ffffff"><meta name=theme-color media="(prefers-color-scheme: dark)" content="#343a40"><meta name=color-scheme content="light dark"><meta property="og:url" content="https://yshghid.github.io/docs/study/bioinformatics/bi8/"><meta property="og:site_name" content=" "><meta property="og:title" content="RNA-seq 전처리 꿀조합 (Rsubread, edgeR)"><meta property="og:description" content='RNA-seq 전처리 꿀조합 (Rsubread, edgeR) # #2025-04-21
가장 오류 적게나는 조합!
1. Align RNA-seq # Load Packages # library(Rsubread) library(org.Mm.eg.db) library(gridExtra) library(reshape2) Set Path # indir = "/data/home/ysh980101/2504/mirna/data" outdir = "/data/home/ysh980101/2504/mirna/data" refpath = "/data/home/ysh980101/2406/data-gne/mm39.fa" setwd(indir) getwd() &#39;/data/home/ysh980101/2504/mirna/data&#39; Build Index # buildindex(basename = "mm39", reference = refpath) Read Alignment # files <- list.files(pattern="\\.fastq\\.gz$", full.names=TRUE) bams <- sub("\\.fastq\\.gz$", ".bam", files) samples <- gsub("^\\.\\/|\\.fastq\\.gz$", "", files) targets <- read.delim("target.txt", header=TRUE) align(index="mm39", readfile1=files, input_format="gzFASTQ", output_file=bams, nthreads=50) Quantification # fc = featureCounts(bams, isGTFAnnotationFile=TRUE, GTF.featureType="exon", GTF.attrType="gene_id", isPairedEnd=FALSE, annot.ext="mm39.knownGene.gtf", useMetaFeatures=FALSE, allowMultiOverlap=TRUE, nthreads=50) Save Countdata # colnames(fc$counts) <- samples y <- DGEList(fc$counts, group=group) write.csv(as.data.frame(y$counts), file = paste0(outdir,"/count.csv", row.names = TRUE)) 2. Gene ID Annotation # Load Packages # import pandas as pd import numpy as np import os Set Path # indir = "/data/home/ysh980101/2504/mirna/data" outdir = "/data/home/ysh980101/2504/mirna/data" annotpath = "/data/home/ysh980101/2406/data-gne/MRK_ENSEMBL.rpt" os.chdir(indir) os.getcwd() &#39;/data1/home/ysh980101/2504/mirna/result&#39; Load Annotation # annotation = pd.read_csv(annotpath, sep="\t", names=[str(i) for i in range(13)]) annotation = annotation.dropna(subset=[&#39;6&#39;]) annotation = annotation[annotation[&#39;8&#39;] == &#39;protein coding gene&#39;] Load Count & Gene ID Mapping # count_mm39 = pd.read_csv("count.csv") count_mm39.rename(columns={count_mm39.columns[0]: &#39;ens_id&#39;}, inplace=True) count_mm39[&#39;ens_id&#39;] = count_mm39[&#39;ens_id&#39;].str.split(&#39;.&#39;).str[0] for index, row in annotation.iterrows(): ens_ids = row[&#39;6&#39;].split() gene_id = row[&#39;1&#39;] count_mm39.loc[count_mm39[&#39;ens_id&#39;].isin(ens_ids), &#39;gene_id&#39;] = gene_id Transcript Filtering # count_mm39[&#39;sum&#39;] = count_mm39.iloc[:, 2:].sum(axis=1) count_mm39 = count_mm39.sort_values(by=[&#39;gene_id&#39;, &#39;sum&#39;], ascending=[True, False]) count_mm39 = count_mm39.drop_duplicates(subset=[&#39;gene_id&#39;], keep=&#39;first&#39;) count_mm39 = count_mm39.dropna(subset=[&#39;gene_id&#39;]) count_mm39 = count_mm39.drop(columns=[&#39;sum&#39;, &#39;ens_id&#39;]) gene_id_column = count_mm39[&#39;gene_id&#39;] count_mm39.drop(columns=[&#39;gene_id&#39;], inplace=True) count_mm39.insert(0, &#39;gene_id&#39;, gene_id_column) Save # count_mm39.rename(columns={&#39;gene_id&#39;: &#39;GeneID&#39;}, inplace=True) def rename_columns(col): parts = col.split(&#39;_&#39;) if len(parts) >= 3: new_col = parts[0] + parts[2] + &#39;_&#39; + parts[1] else: new_col = col return new_col count_mm39.columns = [rename_columns(col) for col in count_mm39.columns] count_mm39.to_csv(f"{outdir}/count_processed.csv", index=False) 3. DEG Analysis # Library & Set Path # library(edgeR) indir = "/data/home/ysh980101/2504/mirna/data" outdir = "/data/home/ysh980101/2504/mirna/result" setwd(indir) getwd() &#39;/data1/home/ysh980101/2504/mirna/data&#39; Set variables & Load Data # tissue <- "G" S1 <- "WT" S2 <- "GneKI" counts <- read.csv("count_processed.csv") meta <- read.csv(paste0("mouse_meta_",tissue,".csv")) meta <- meta[meta$Group %in% c(S1, S2), ] counts <- counts[, c("GeneID", unique(meta$SampleID))] Create DGElist & Normalization # Group <- factor(meta$Group) Group <- relevel(Group, ref=S1) y <- DGEList(counts=counts[,2:ncol(counts)], group=Group, genes = counts[,1]) y <- calcNormFactors(y) Run DEG # design <- model.matrix(~Group) y <- estimateDisp(y, design) y <- estimateGLMRobustDisp(y,design) fit <- glmFit(y, design) lrt <- glmLRT(fit) plotMD(lrt) abline(h=c(-1,1), col="blue")'><meta property="og:locale" content="en_us"><meta property="og:type" content="article"><meta property="article:section" content="docs"><meta property="article:published_time" content="2025-04-21T00:00:00+00:00"><meta property="article:modified_time" content="2025-04-21T00:00:00+00:00"><meta property="article:tag" content="2025-04"><title>RNA-seq 전처리 꿀조합 (Rsubread, edgeR) |</title><link rel=icon href=/favicon.png><link rel=manifest href=/manifest.json><link rel=canonical href=https://yshghid.github.io/docs/study/bioinformatics/bi8/><link rel=stylesheet href=/book.min.6217d077edb4189fd0578345e84bca1a884dfdee121ff8dc9a0f55cfe0852bc9.css integrity="sha256-YhfQd+20GJ/QV4NF6EvKGohN/e4SH/jcmg9Vz+CFK8k=" crossorigin=anonymous><script defer src=/fuse.min.js></script><script defer src=/en.search.min.1af1399442a5d15136bdfd8afc4f303a44682e2c22b7432af6dea949f1d6d0a4.js integrity="sha256-GvE5lEKl0VE2vf2K/E8wOkRoLiwit0Mq9t6pSfHW0KQ=" crossorigin=anonymous></script></head><body dir=ltr><input type=checkbox class="hidden toggle" id=menu-control>
<input type=checkbox class="hidden toggle" id=toc-control><main class="container flex"><aside class=book-menu><div class=book-menu-content><nav><h2 class=book-brand><a class="flex align-center" href=/><img src=/logo.png alt=Logo class=book-icon><span></span></a></h2><div class="book-search hidden"><input type=text id=book-search-input placeholder=Search aria-label=Search maxlength=64 data-hotkeys=s/><div class="book-search-spinner hidden"></div><ul id=book-search-results></ul></div><script>document.querySelector(".book-search").classList.remove("hidden")</script><ul><li class=book-section-flat><span>기록</span><ul><li><a href=/docs/hobby/book/>글</a><ul></ul></li><li><a href=/docs/hobby/daily/>일상</a><ul></ul></li><li><a href=/docs/hobby/baking/>베이킹</a><ul></ul></li><li><a href=/docs/hobby/favorite/>🌸</a><ul></ul></li></ul></li><li class=book-section-flat><span>공부</span><ul><li><a href=/docs/study/bioinformatics/>생물정보학</a><ul></ul></li><li><a href=/docs/study/algorithm/>코테</a><ul></ul></li><li><a href=/docs/study/career/>취업</a><ul></ul></li><li><a href=/docs/study/etc/>기타</a><ul></ul></li></ul></li></ul></nav><script>(function(){var e=document.querySelector("aside .book-menu-content");addEventListener("beforeunload",function(){localStorage.setItem("menu.scrollTop",e.scrollTop)}),e.scrollTop=localStorage.getItem("menu.scrollTop")})()</script></div></aside><div class=book-page><header class=book-header><div class="flex align-center justify-between"><label for=menu-control><img src=/svg/menu.svg class=book-icon alt=Menu></label><h3>RNA-seq 전처리 꿀조합 (Rsubread, edgeR)</h3><label for=toc-control><img src=/svg/toc.svg class=book-icon alt="Table of Contents"></label></div><aside class="hidden clearfix"><nav id=TableOfContents><ul><li><a href=#1-align-rna-seq>1. Align RNA-seq</a><ul><li><a href=#load-packages>Load Packages</a></li><li><a href=#set-path>Set Path</a></li><li><a href=#build-index>Build Index</a></li><li><a href=#read-alignment>Read Alignment</a></li><li><a href=#quantification>Quantification</a></li><li><a href=#save-countdata>Save Countdata</a></li></ul></li><li><a href=#2-gene-id-annotation>2. Gene ID Annotation</a><ul><li><a href=#load-packages-1>Load Packages</a></li><li><a href=#set-path-1>Set Path</a></li><li><a href=#load-annotation>Load Annotation</a></li><li><a href=#load-count--gene-id-mapping>Load Count & Gene ID Mapping</a></li><li><a href=#transcript-filtering>Transcript Filtering</a></li><li><a href=#save>Save</a></li></ul></li><li><a href=#3-deg-analysis>3. DEG Analysis</a><ul><li><a href=#library--set-path>Library & Set Path</a></li><li><a href=#set-variables--load-data>Set variables & Load Data</a></li><li><a href=#create-dgelist--normalization>Create DGElist & Normalization</a></li><li><a href=#run-deg>Run DEG</a></li><li><a href=#save-1>Save</a></li></ul></li></ul></nav></aside></header><article class="markdown book-article"><h1 id=rna-seq-전처리-꿀조합-rsubread-edger>RNA-seq 전처리 꿀조합 (Rsubread, edgeR)
<a class=anchor href=#rna-seq-%ec%a0%84%ec%b2%98%eb%a6%ac-%ea%bf%80%ec%a1%b0%ed%95%a9-rsubread-edger>#</a></h1><p>#2025-04-21</p><hr><p>가장 오류 적게나는 조합!</p><h2 id=1-align-rna-seq>1. Align RNA-seq
<a class=anchor href=#1-align-rna-seq>#</a></h2><h3 id=load-packages>Load Packages
<a class=anchor href=#load-packages>#</a></h3><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-R data-lang=R><span style=display:flex><span><span style=color:#a6e22e>library</span>(Rsubread)
</span></span><span style=display:flex><span><span style=color:#a6e22e>library</span>(org.Mm.eg.db)
</span></span><span style=display:flex><span><span style=color:#a6e22e>library</span>(gridExtra)
</span></span><span style=display:flex><span><span style=color:#a6e22e>library</span>(reshape2)
</span></span></code></pre></div><h3 id=set-path>Set Path
<a class=anchor href=#set-path>#</a></h3><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-R data-lang=R><span style=display:flex><span>indir <span style=color:#f92672>=</span> <span style=color:#e6db74>&#34;/data/home/ysh980101/2504/mirna/data&#34;</span>
</span></span><span style=display:flex><span>outdir <span style=color:#f92672>=</span> <span style=color:#e6db74>&#34;/data/home/ysh980101/2504/mirna/data&#34;</span>
</span></span><span style=display:flex><span>refpath <span style=color:#f92672>=</span> <span style=color:#e6db74>&#34;/data/home/ysh980101/2406/data-gne/mm39.fa&#34;</span>
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span><span style=color:#a6e22e>setwd</span>(indir)
</span></span><span style=display:flex><span><span style=color:#a6e22e>getwd</span>()
</span></span></code></pre></div><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-plain data-lang=plain><span style=display:flex><span>&#39;/data/home/ysh980101/2504/mirna/data&#39;
</span></span></code></pre></div><h3 id=build-index>Build Index
<a class=anchor href=#build-index>#</a></h3><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-R data-lang=R><span style=display:flex><span><span style=color:#a6e22e>buildindex</span>(basename <span style=color:#f92672>=</span> <span style=color:#e6db74>&#34;mm39&#34;</span>, reference <span style=color:#f92672>=</span> refpath)
</span></span></code></pre></div><h3 id=read-alignment>Read Alignment
<a class=anchor href=#read-alignment>#</a></h3><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-R data-lang=R><span style=display:flex><span>files <span style=color:#f92672>&lt;-</span> <span style=color:#a6e22e>list.files</span>(pattern<span style=color:#f92672>=</span><span style=color:#e6db74>&#34;\\.fastq\\.gz$&#34;</span>, full.names<span style=color:#f92672>=</span><span style=color:#66d9ef>TRUE</span>)
</span></span><span style=display:flex><span>bams <span style=color:#f92672>&lt;-</span> <span style=color:#a6e22e>sub</span>(<span style=color:#e6db74>&#34;\\.fastq\\.gz$&#34;</span>, <span style=color:#e6db74>&#34;.bam&#34;</span>, files)
</span></span><span style=display:flex><span>samples <span style=color:#f92672>&lt;-</span> <span style=color:#a6e22e>gsub</span>(<span style=color:#e6db74>&#34;^\\.\\/|\\.fastq\\.gz$&#34;</span>, <span style=color:#e6db74>&#34;&#34;</span>, files)
</span></span><span style=display:flex><span>targets <span style=color:#f92672>&lt;-</span> <span style=color:#a6e22e>read.delim</span>(<span style=color:#e6db74>&#34;target.txt&#34;</span>, header<span style=color:#f92672>=</span><span style=color:#66d9ef>TRUE</span>)
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span><span style=color:#a6e22e>align</span>(index<span style=color:#f92672>=</span><span style=color:#e6db74>&#34;mm39&#34;</span>, readfile1<span style=color:#f92672>=</span>files, input_format<span style=color:#f92672>=</span><span style=color:#e6db74>&#34;gzFASTQ&#34;</span>, output_file<span style=color:#f92672>=</span>bams, nthreads<span style=color:#f92672>=</span><span style=color:#ae81ff>50</span>)
</span></span></code></pre></div><h3 id=quantification>Quantification
<a class=anchor href=#quantification>#</a></h3><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-R data-lang=R><span style=display:flex><span>fc <span style=color:#f92672>=</span> <span style=color:#a6e22e>featureCounts</span>(bams, isGTFAnnotationFile<span style=color:#f92672>=</span><span style=color:#66d9ef>TRUE</span>, GTF.featureType<span style=color:#f92672>=</span><span style=color:#e6db74>&#34;exon&#34;</span>, GTF.attrType<span style=color:#f92672>=</span><span style=color:#e6db74>&#34;gene_id&#34;</span>, isPairedEnd<span style=color:#f92672>=</span><span style=color:#66d9ef>FALSE</span>, annot.ext<span style=color:#f92672>=</span><span style=color:#e6db74>&#34;mm39.knownGene.gtf&#34;</span>, useMetaFeatures<span style=color:#f92672>=</span><span style=color:#66d9ef>FALSE</span>, allowMultiOverlap<span style=color:#f92672>=</span><span style=color:#66d9ef>TRUE</span>, nthreads<span style=color:#f92672>=</span><span style=color:#ae81ff>50</span>)
</span></span></code></pre></div><h3 id=save-countdata>Save Countdata
<a class=anchor href=#save-countdata>#</a></h3><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-R data-lang=R><span style=display:flex><span><span style=color:#a6e22e>colnames</span>(fc<span style=color:#f92672>$</span>counts) <span style=color:#f92672>&lt;-</span> samples
</span></span><span style=display:flex><span>y <span style=color:#f92672>&lt;-</span> <span style=color:#a6e22e>DGEList</span>(fc<span style=color:#f92672>$</span>counts, group<span style=color:#f92672>=</span>group)
</span></span><span style=display:flex><span><span style=color:#a6e22e>write.csv</span>(<span style=color:#a6e22e>as.data.frame</span>(y<span style=color:#f92672>$</span>counts), file <span style=color:#f92672>=</span> <span style=color:#a6e22e>paste0</span>(outdir,<span style=color:#e6db74>&#34;/count.csv&#34;</span>, row.names <span style=color:#f92672>=</span> <span style=color:#66d9ef>TRUE</span>))
</span></span></code></pre></div><h2 id=2-gene-id-annotation>2. Gene ID Annotation
<a class=anchor href=#2-gene-id-annotation>#</a></h2><h3 id=load-packages-1>Load Packages
<a class=anchor href=#load-packages-1>#</a></h3><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-python data-lang=python><span style=display:flex><span><span style=color:#f92672>import</span> pandas <span style=color:#66d9ef>as</span> pd
</span></span><span style=display:flex><span><span style=color:#f92672>import</span> numpy <span style=color:#66d9ef>as</span> np
</span></span><span style=display:flex><span><span style=color:#f92672>import</span> os
</span></span></code></pre></div><h3 id=set-path-1>Set Path
<a class=anchor href=#set-path-1>#</a></h3><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-python data-lang=python><span style=display:flex><span>indir <span style=color:#f92672>=</span> <span style=color:#e6db74>&#34;/data/home/ysh980101/2504/mirna/data&#34;</span>
</span></span><span style=display:flex><span>outdir <span style=color:#f92672>=</span> <span style=color:#e6db74>&#34;/data/home/ysh980101/2504/mirna/data&#34;</span>
</span></span><span style=display:flex><span>annotpath <span style=color:#f92672>=</span> <span style=color:#e6db74>&#34;/data/home/ysh980101/2406/data-gne/MRK_ENSEMBL.rpt&#34;</span>
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span>os<span style=color:#f92672>.</span>chdir(indir)
</span></span><span style=display:flex><span>os<span style=color:#f92672>.</span>getcwd()
</span></span></code></pre></div><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-plain data-lang=plain><span style=display:flex><span>&#39;/data1/home/ysh980101/2504/mirna/result&#39;
</span></span></code></pre></div><h3 id=load-annotation>Load Annotation
<a class=anchor href=#load-annotation>#</a></h3><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-python data-lang=python><span style=display:flex><span>annotation <span style=color:#f92672>=</span> pd<span style=color:#f92672>.</span>read_csv(annotpath, sep<span style=color:#f92672>=</span><span style=color:#e6db74>&#34;</span><span style=color:#ae81ff>\t</span><span style=color:#e6db74>&#34;</span>, names<span style=color:#f92672>=</span>[str(i) <span style=color:#66d9ef>for</span> i <span style=color:#f92672>in</span> range(<span style=color:#ae81ff>13</span>)])
</span></span><span style=display:flex><span>annotation <span style=color:#f92672>=</span> annotation<span style=color:#f92672>.</span>dropna(subset<span style=color:#f92672>=</span>[<span style=color:#e6db74>&#39;6&#39;</span>])
</span></span><span style=display:flex><span>annotation <span style=color:#f92672>=</span> annotation[annotation[<span style=color:#e6db74>&#39;8&#39;</span>] <span style=color:#f92672>==</span> <span style=color:#e6db74>&#39;protein coding gene&#39;</span>]
</span></span></code></pre></div><h3 id=load-count--gene-id-mapping>Load Count & Gene ID Mapping
<a class=anchor href=#load-count--gene-id-mapping>#</a></h3><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-python data-lang=python><span style=display:flex><span>count_mm39 <span style=color:#f92672>=</span> pd<span style=color:#f92672>.</span>read_csv(<span style=color:#e6db74>&#34;count.csv&#34;</span>)
</span></span><span style=display:flex><span>count_mm39<span style=color:#f92672>.</span>rename(columns<span style=color:#f92672>=</span>{count_mm39<span style=color:#f92672>.</span>columns[<span style=color:#ae81ff>0</span>]: <span style=color:#e6db74>&#39;ens_id&#39;</span>}, inplace<span style=color:#f92672>=</span><span style=color:#66d9ef>True</span>)
</span></span><span style=display:flex><span>count_mm39[<span style=color:#e6db74>&#39;ens_id&#39;</span>] <span style=color:#f92672>=</span> count_mm39[<span style=color:#e6db74>&#39;ens_id&#39;</span>]<span style=color:#f92672>.</span>str<span style=color:#f92672>.</span>split(<span style=color:#e6db74>&#39;.&#39;</span>)<span style=color:#f92672>.</span>str[<span style=color:#ae81ff>0</span>]
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span><span style=color:#66d9ef>for</span> index, row <span style=color:#f92672>in</span> annotation<span style=color:#f92672>.</span>iterrows():
</span></span><span style=display:flex><span>    ens_ids <span style=color:#f92672>=</span> row[<span style=color:#e6db74>&#39;6&#39;</span>]<span style=color:#f92672>.</span>split()
</span></span><span style=display:flex><span>    gene_id <span style=color:#f92672>=</span> row[<span style=color:#e6db74>&#39;1&#39;</span>]
</span></span><span style=display:flex><span>    count_mm39<span style=color:#f92672>.</span>loc[count_mm39[<span style=color:#e6db74>&#39;ens_id&#39;</span>]<span style=color:#f92672>.</span>isin(ens_ids), <span style=color:#e6db74>&#39;gene_id&#39;</span>] <span style=color:#f92672>=</span> gene_id
</span></span></code></pre></div><h3 id=transcript-filtering>Transcript Filtering
<a class=anchor href=#transcript-filtering>#</a></h3><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-python data-lang=python><span style=display:flex><span>count_mm39[<span style=color:#e6db74>&#39;sum&#39;</span>] <span style=color:#f92672>=</span> count_mm39<span style=color:#f92672>.</span>iloc[:, <span style=color:#ae81ff>2</span>:]<span style=color:#f92672>.</span>sum(axis<span style=color:#f92672>=</span><span style=color:#ae81ff>1</span>)
</span></span><span style=display:flex><span>count_mm39 <span style=color:#f92672>=</span> count_mm39<span style=color:#f92672>.</span>sort_values(by<span style=color:#f92672>=</span>[<span style=color:#e6db74>&#39;gene_id&#39;</span>, <span style=color:#e6db74>&#39;sum&#39;</span>], ascending<span style=color:#f92672>=</span>[<span style=color:#66d9ef>True</span>, <span style=color:#66d9ef>False</span>])
</span></span><span style=display:flex><span>count_mm39 <span style=color:#f92672>=</span> count_mm39<span style=color:#f92672>.</span>drop_duplicates(subset<span style=color:#f92672>=</span>[<span style=color:#e6db74>&#39;gene_id&#39;</span>], keep<span style=color:#f92672>=</span><span style=color:#e6db74>&#39;first&#39;</span>)
</span></span><span style=display:flex><span>count_mm39 <span style=color:#f92672>=</span> count_mm39<span style=color:#f92672>.</span>dropna(subset<span style=color:#f92672>=</span>[<span style=color:#e6db74>&#39;gene_id&#39;</span>])
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span>count_mm39 <span style=color:#f92672>=</span> count_mm39<span style=color:#f92672>.</span>drop(columns<span style=color:#f92672>=</span>[<span style=color:#e6db74>&#39;sum&#39;</span>, <span style=color:#e6db74>&#39;ens_id&#39;</span>])
</span></span><span style=display:flex><span>gene_id_column <span style=color:#f92672>=</span> count_mm39[<span style=color:#e6db74>&#39;gene_id&#39;</span>]
</span></span><span style=display:flex><span>count_mm39<span style=color:#f92672>.</span>drop(columns<span style=color:#f92672>=</span>[<span style=color:#e6db74>&#39;gene_id&#39;</span>], inplace<span style=color:#f92672>=</span><span style=color:#66d9ef>True</span>)
</span></span><span style=display:flex><span>count_mm39<span style=color:#f92672>.</span>insert(<span style=color:#ae81ff>0</span>, <span style=color:#e6db74>&#39;gene_id&#39;</span>, gene_id_column)
</span></span></code></pre></div><h3 id=save>Save
<a class=anchor href=#save>#</a></h3><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-python data-lang=python><span style=display:flex><span>count_mm39<span style=color:#f92672>.</span>rename(columns<span style=color:#f92672>=</span>{<span style=color:#e6db74>&#39;gene_id&#39;</span>: <span style=color:#e6db74>&#39;GeneID&#39;</span>}, inplace<span style=color:#f92672>=</span><span style=color:#66d9ef>True</span>)
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span><span style=color:#66d9ef>def</span> <span style=color:#a6e22e>rename_columns</span>(col):
</span></span><span style=display:flex><span>    parts <span style=color:#f92672>=</span> col<span style=color:#f92672>.</span>split(<span style=color:#e6db74>&#39;_&#39;</span>)
</span></span><span style=display:flex><span>    <span style=color:#66d9ef>if</span> len(parts) <span style=color:#f92672>&gt;=</span> <span style=color:#ae81ff>3</span>:
</span></span><span style=display:flex><span>        new_col <span style=color:#f92672>=</span> parts[<span style=color:#ae81ff>0</span>] <span style=color:#f92672>+</span> parts[<span style=color:#ae81ff>2</span>] <span style=color:#f92672>+</span> <span style=color:#e6db74>&#39;_&#39;</span> <span style=color:#f92672>+</span> parts[<span style=color:#ae81ff>1</span>]
</span></span><span style=display:flex><span>    <span style=color:#66d9ef>else</span>:
</span></span><span style=display:flex><span>        new_col <span style=color:#f92672>=</span> col
</span></span><span style=display:flex><span>    <span style=color:#66d9ef>return</span> new_col
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span>count_mm39<span style=color:#f92672>.</span>columns <span style=color:#f92672>=</span> [rename_columns(col) <span style=color:#66d9ef>for</span> col <span style=color:#f92672>in</span> count_mm39<span style=color:#f92672>.</span>columns]
</span></span><span style=display:flex><span>count_mm39<span style=color:#f92672>.</span>to_csv(<span style=color:#e6db74>f</span><span style=color:#e6db74>&#34;</span><span style=color:#e6db74>{</span>outdir<span style=color:#e6db74>}</span><span style=color:#e6db74>/count_processed.csv&#34;</span>, index<span style=color:#f92672>=</span><span style=color:#66d9ef>False</span>)
</span></span></code></pre></div><h2 id=3-deg-analysis>3. DEG Analysis
<a class=anchor href=#3-deg-analysis>#</a></h2><h3 id=library--set-path>Library & Set Path
<a class=anchor href=#library--set-path>#</a></h3><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-R data-lang=R><span style=display:flex><span><span style=color:#a6e22e>library</span>(edgeR)
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span>indir <span style=color:#f92672>=</span> <span style=color:#e6db74>&#34;/data/home/ysh980101/2504/mirna/data&#34;</span>
</span></span><span style=display:flex><span>outdir <span style=color:#f92672>=</span> <span style=color:#e6db74>&#34;/data/home/ysh980101/2504/mirna/result&#34;</span>
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span><span style=color:#a6e22e>setwd</span>(indir)
</span></span><span style=display:flex><span><span style=color:#a6e22e>getwd</span>()
</span></span></code></pre></div><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-plain data-lang=plain><span style=display:flex><span>&#39;/data1/home/ysh980101/2504/mirna/data&#39;
</span></span></code></pre></div><h3 id=set-variables--load-data>Set variables & Load Data
<a class=anchor href=#set-variables--load-data>#</a></h3><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-R data-lang=R><span style=display:flex><span>tissue <span style=color:#f92672>&lt;-</span> <span style=color:#e6db74>&#34;G&#34;</span>
</span></span><span style=display:flex><span>S1 <span style=color:#f92672>&lt;-</span> <span style=color:#e6db74>&#34;WT&#34;</span> 
</span></span><span style=display:flex><span>S2 <span style=color:#f92672>&lt;-</span> <span style=color:#e6db74>&#34;GneKI&#34;</span>
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span>counts <span style=color:#f92672>&lt;-</span> <span style=color:#a6e22e>read.csv</span>(<span style=color:#e6db74>&#34;count_processed.csv&#34;</span>)
</span></span><span style=display:flex><span>meta <span style=color:#f92672>&lt;-</span> <span style=color:#a6e22e>read.csv</span>(<span style=color:#a6e22e>paste0</span>(<span style=color:#e6db74>&#34;mouse_meta_&#34;</span>,tissue,<span style=color:#e6db74>&#34;.csv&#34;</span>))
</span></span><span style=display:flex><span>meta <span style=color:#f92672>&lt;-</span> meta[meta<span style=color:#f92672>$</span>Group <span style=color:#f92672>%in%</span> <span style=color:#a6e22e>c</span>(S1, S2), ]
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span>counts <span style=color:#f92672>&lt;-</span> counts[, <span style=color:#a6e22e>c</span>(<span style=color:#e6db74>&#34;GeneID&#34;</span>, <span style=color:#a6e22e>unique</span>(meta<span style=color:#f92672>$</span>SampleID))]
</span></span></code></pre></div><h3 id=create-dgelist--normalization>Create DGElist & Normalization
<a class=anchor href=#create-dgelist--normalization>#</a></h3><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-R data-lang=R><span style=display:flex><span>Group <span style=color:#f92672>&lt;-</span> <span style=color:#a6e22e>factor</span>(meta<span style=color:#f92672>$</span>Group)
</span></span><span style=display:flex><span>Group <span style=color:#f92672>&lt;-</span> <span style=color:#a6e22e>relevel</span>(Group, ref<span style=color:#f92672>=</span>S1)
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span>y <span style=color:#f92672>&lt;-</span> <span style=color:#a6e22e>DGEList</span>(counts<span style=color:#f92672>=</span>counts[,<span style=color:#ae81ff>2</span><span style=color:#f92672>:</span><span style=color:#a6e22e>ncol</span>(counts)], group<span style=color:#f92672>=</span>Group, genes <span style=color:#f92672>=</span> counts[,<span style=color:#ae81ff>1</span>])
</span></span><span style=display:flex><span>y <span style=color:#f92672>&lt;-</span> <span style=color:#a6e22e>calcNormFactors</span>(y)
</span></span></code></pre></div><h3 id=run-deg>Run DEG
<a class=anchor href=#run-deg>#</a></h3><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-R data-lang=R><span style=display:flex><span>design <span style=color:#f92672>&lt;-</span> <span style=color:#a6e22e>model.matrix</span>(<span style=color:#f92672>~</span>Group)
</span></span><span style=display:flex><span>y <span style=color:#f92672>&lt;-</span> <span style=color:#a6e22e>estimateDisp</span>(y, design)
</span></span><span style=display:flex><span>y <span style=color:#f92672>&lt;-</span> <span style=color:#a6e22e>estimateGLMRobustDisp</span>(y,design)
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span>fit <span style=color:#f92672>&lt;-</span> <span style=color:#a6e22e>glmFit</span>(y, design)
</span></span><span style=display:flex><span>lrt <span style=color:#f92672>&lt;-</span> <span style=color:#a6e22e>glmLRT</span>(fit)
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span><span style=color:#a6e22e>plotMD</span>(lrt)
</span></span><span style=display:flex><span><span style=color:#a6e22e>abline</span>(h<span style=color:#f92672>=</span><span style=color:#a6e22e>c</span>(<span style=color:#ae81ff>-1</span>,<span style=color:#ae81ff>1</span>), col<span style=color:#f92672>=</span><span style=color:#e6db74>&#34;blue&#34;</span>)
</span></span></code></pre></div><p><img src=https://github.com/user-attachments/assets/339f2e97-6b71-45c1-8092-b801b68c9f23 alt=image></p><h3 id=save-1>Save
<a class=anchor href=#save-1>#</a></h3><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-R data-lang=R><span style=display:flex><span>result_table <span style=color:#f92672>&lt;-</span> <span style=color:#a6e22e>topTags</span>(lrt, n <span style=color:#f92672>=</span> <span style=color:#a6e22e>nrow</span>(lrt<span style=color:#f92672>$</span>table))
</span></span><span style=display:flex><span>sorted_result_table <span style=color:#f92672>&lt;-</span> result_table<span style=color:#a6e22e>[order</span>(result_table<span style=color:#f92672>$</span>table<span style=color:#f92672>$</span>FDR), ]
</span></span><span style=display:flex><span>filtered_result_table <span style=color:#f92672>&lt;-</span> sorted_result_table[sorted_result_table<span style=color:#f92672>$</span>table<span style=color:#f92672>$</span>FDR <span style=color:#f92672>&lt;</span> <span style=color:#ae81ff>0.05</span>, ]
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span><span style=color:#a6e22e>write.csv</span>(sorted_result_table, file <span style=color:#f92672>=</span> <span style=color:#a6e22e>paste0</span>(outdir,<span style=color:#e6db74>&#34;/de-&#34;</span>,tissue,<span style=color:#e6db74>&#34;_&#34;</span>,S1,<span style=color:#e6db74>&#34;-&#34;</span>,S2,<span style=color:#e6db74>&#34;.csv&#34;</span>))
</span></span></code></pre></div></article><footer class=book-footer><div class="flex flex-wrap justify-between"></div><script>(function(){function e(e){const t=window.getSelection(),n=document.createRange();n.selectNodeContents(e),t.removeAllRanges(),t.addRange(n)}document.querySelectorAll("pre code").forEach(t=>{t.addEventListener("click",function(){if(window.getSelection().toString())return;e(t.parentElement),navigator.clipboard&&navigator.clipboard.writeText(t.parentElement.textContent)})})})()</script></footer><div class=book-comments><script src=https://giscus.app/client.js data-repo=yshghid/yshghid.github.io data-repo-id=R_kgDONkMkNg data-category-id=DIC_kwDONkMkNs4CloJh data-mapping=pathname data-strict=0 data-reactions-enabled=1 data-emit-metadata=0 data-input-position=bottom data-theme=preferred_color_scheme data-lang=ko crossorigin=anonymous async></script></div><label for=menu-control class="hidden book-menu-overlay"></label></div><aside class=book-toc><div class=book-toc-content><nav id=TableOfContents><ul><li><a href=#1-align-rna-seq>1. Align RNA-seq</a><ul><li><a href=#load-packages>Load Packages</a></li><li><a href=#set-path>Set Path</a></li><li><a href=#build-index>Build Index</a></li><li><a href=#read-alignment>Read Alignment</a></li><li><a href=#quantification>Quantification</a></li><li><a href=#save-countdata>Save Countdata</a></li></ul></li><li><a href=#2-gene-id-annotation>2. Gene ID Annotation</a><ul><li><a href=#load-packages-1>Load Packages</a></li><li><a href=#set-path-1>Set Path</a></li><li><a href=#load-annotation>Load Annotation</a></li><li><a href=#load-count--gene-id-mapping>Load Count & Gene ID Mapping</a></li><li><a href=#transcript-filtering>Transcript Filtering</a></li><li><a href=#save>Save</a></li></ul></li><li><a href=#3-deg-analysis>3. DEG Analysis</a><ul><li><a href=#library--set-path>Library & Set Path</a></li><li><a href=#set-variables--load-data>Set variables & Load Data</a></li><li><a href=#create-dgelist--normalization>Create DGElist & Normalization</a></li><li><a href=#run-deg>Run DEG</a></li><li><a href=#save-1>Save</a></li></ul></li></ul></nav></div></aside></main></body></html>