<!doctype html><html lang=en-us dir=ltr><head><meta charset=UTF-8><meta name=viewport content="width=device-width,initial-scale=1"><meta name=description content='
  ìƒë¬¼ì •ë³´í•™
  #


  2025
  #

07-12 â‹¯ EdgeR: DE ë¶„ì„
Load package suppressMessages({ library(&ldquo;DESeq2&rdquo;) library(pheatmap) library(withr) #library(tidyverse) library(RColorBrewer) library(gplots) library(dplyr) }) Set path setwd("/data/home/ysh980101/2307_EBV") getwd() &lsquo;/data1/home/ysh980101/2307_EBV&rsquo; â‹¯

05-28 â‹¯ Bash ì‰˜ ìŠ¤í¬ë¦½íŠ¸ (ë¡œì»¬, ì„œë²„)
#local #alias cobi2=&lsquo;ssh -p 5290 ysh980101@155.230.28.211&rsquo; alias cobi2=&ldquo;ssh -p 3160 ysh980101@155.230.110.91&rdquo; alias cobi3=&ldquo;ssh -p 7777 ysh980101@155.230.110.92&rdquo; alias cobi4=&ldquo;ssh -p 4712 ysh980101@155.230.110.93&rdquo; # &#187;> conda initialize &#187;> # !! Contents â‹¯

04-21 â‹¯ TopHat2, HTSeq, Rsubread: RNA-seq ì „ì²˜ë¦¬ íŒŒì´í”„ë¼ì¸ ë¹„êµ'><meta name=theme-color media="(prefers-color-scheme: light)" content="#ffffff"><meta name=theme-color media="(prefers-color-scheme: dark)" content="#343a40"><meta name=color-scheme content="light dark"><meta property="og:url" content="https://yshghid.github.io/docs/study/bioinformatics/"><meta property="og:site_name" content=" "><meta property="og:title" content="ìƒë¬¼ì •ë³´í•™"><meta property="og:description" content='ìƒë¬¼ì •ë³´í•™ # 2025 # 07-12 â‹¯ EdgeR: DE ë¶„ì„
Load package suppressMessages({ library(â€œDESeq2â€) library(pheatmap) library(withr) #library(tidyverse) library(RColorBrewer) library(gplots) library(dplyr) }) Set path setwd("/data/home/ysh980101/2307_EBV") getwd() â€˜/data1/home/ysh980101/2307_EBVâ€™ â‹¯
05-28 â‹¯ Bash ì‰˜ ìŠ¤í¬ë¦½íŠ¸ (ë¡œì»¬, ì„œë²„)
#local #alias cobi2=â€˜ssh -p 5290 ysh980101@155.230.28.211â€™ alias cobi2=â€œssh -p 3160 ysh980101@155.230.110.91â€ alias cobi3=â€œssh -p 7777 ysh980101@155.230.110.92â€ alias cobi4=â€œssh -p 4712 ysh980101@155.230.110.93â€ # Â»> conda initialize Â»> # !! Contents â‹¯
04-21 â‹¯ TopHat2, HTSeq, Rsubread: RNA-seq ì „ì²˜ë¦¬ íŒŒì´í”„ë¼ì¸ ë¹„êµ'><meta property="og:locale" content="en_us"><meta property="og:type" content="website"><title>ìƒë¬¼ì •ë³´í•™ |</title><link rel=icon href=/favicon.png><link rel=manifest href=/manifest.json><link rel=canonical href=https://yshghid.github.io/docs/study/bioinformatics/><link rel=stylesheet href=/book.min.6217d077edb4189fd0578345e84bca1a884dfdee121ff8dc9a0f55cfe0852bc9.css integrity="sha256-YhfQd+20GJ/QV4NF6EvKGohN/e4SH/jcmg9Vz+CFK8k=" crossorigin=anonymous><script defer src=/fuse.min.js></script><script defer src=/en.search.min.b40d1da87bde558170d8fe8b2a5c55a9878fcc2437f37fae11672f2fb9372c10.js integrity="sha256-tA0dqHveVYFw2P6LKlxVqYePzCQ383+uEWcvL7k3LBA=" crossorigin=anonymous></script><link rel=alternate type=application/rss+xml href=https://yshghid.github.io/docs/study/bioinformatics/index.xml title=" "></head><body dir=ltr><input type=checkbox class="hidden toggle" id=menu-control>
<input type=checkbox class="hidden toggle" id=toc-control><main class="container flex"><aside class=book-menu><div class=book-menu-content><nav><h2 class=book-brand><a class="flex align-center" href=/><img src=/logo.png alt=Logo class=book-icon><span></span></a></h2><div class="book-search hidden"><input type=text id=book-search-input placeholder=Search aria-label=Search maxlength=64 data-hotkeys=s/><div class="book-search-spinner hidden"></div><ul id=book-search-results></ul></div><script>document.querySelector(".book-search").classList.remove("hidden")</script><ul><li class=book-section-flat><span>ê¸°ë¡</span><ul><li><a href=/docs/hobby/daily/>ì¼ìƒ</a><ul></ul></li><li><a href=/docs/hobby/book/>ê¸€</a><ul></ul></li><li><a href=/docs/hobby/baking/>ë² ì´í‚¹</a><ul></ul></li><li><a href=/docs/hobby/favorite/>ğŸŒ¸</a><ul></ul></li></ul></li><li class=book-section-flat><span>ê³µë¶€</span><ul><li><a href=/docs/study/dl/>ë”¥ëŸ¬ë‹</a><ul></ul></li><li><a href=/docs/study/algorithm/>ì½”í…Œ</a><ul></ul></li><li><a href=/docs/study/career/>ì·¨ì—…</a><ul></ul></li><li><a href=/docs/study/bioinformatics/ class=active>ìƒë¬¼ì •ë³´í•™</a><ul></ul></li><li><a href=/docs/study/etc/>ê¸°íƒ€</a><ul></ul></li></ul></li></ul></nav><script>(function(){var e=document.querySelector("aside .book-menu-content");addEventListener("beforeunload",function(){localStorage.setItem("menu.scrollTop",e.scrollTop)}),e.scrollTop=localStorage.getItem("menu.scrollTop")})()</script></div></aside><div class=book-page><header class=book-header><div class="flex align-center justify-between"><label for=menu-control><img src=/svg/menu.svg class=book-icon alt=Menu></label><h3>ìƒë¬¼ì •ë³´í•™</h3><label for=toc-control><img src=/svg/toc.svg class=book-icon alt="Table of Contents"></label></div><aside class="hidden clearfix"><nav id=TableOfContents><ul><li><a href=#2025>2025</a></li><li><a href=#2024>2024</a></li></ul><ul><li><ul><li><a href=#ìƒë¬¼ì •ë³´í•™-ë©”ëª¨ì¥>ìƒë¬¼ì •ë³´í•™ ë©”ëª¨ì¥</a></li></ul></li></ul></nav></aside></header><article class="markdown book-article"><h1 id=ìƒë¬¼ì •ë³´í•™>ìƒë¬¼ì •ë³´í•™
<a class=anchor href=#%ec%83%9d%eb%ac%bc%ec%a0%95%eb%b3%b4%ed%95%99>#</a></h1><h2 id=2025>2025
<a class=anchor href=#2025>#</a></h2><p><em>07-12</em> â‹¯ EdgeR: DE ë¶„ì„</p><p><a href=https://yshghid.github.io/docs/study/bioinformatics/bi23/>Load package suppressMessages({ library(&ldquo;DESeq2&rdquo;) library(pheatmap) library(withr) #library(tidyverse) library(RColorBrewer) library(gplots) library(dplyr) }) Set path setwd("/data/home/ysh980101/2307_EBV") getwd() &lsquo;/data1/home/ysh980101/2307_EBV&rsquo; â‹¯</a></p><hr><p><em>05-28</em> â‹¯ Bash ì‰˜ ìŠ¤í¬ë¦½íŠ¸ (ë¡œì»¬, ì„œë²„)</p><p><a href=https://yshghid.github.io/docs/study/tech/tech11/>#local #alias cobi2=&lsquo;ssh -p 5290 ysh980101@155.230.28.211&rsquo; alias cobi2=&ldquo;ssh -p 3160 ysh980101@155.230.110.91&rdquo; alias cobi3=&ldquo;ssh -p 7777 ysh980101@155.230.110.92&rdquo; alias cobi4=&ldquo;ssh -p 4712 ysh980101@155.230.110.93&rdquo; # &#187;> conda initialize &#187;> # !! Contents â‹¯</a></p><hr><p><em>04-21</em> â‹¯ TopHat2, HTSeq, Rsubread: RNA-seq ì „ì²˜ë¦¬ íŒŒì´í”„ë¼ì¸ ë¹„êµ</p><p><a href=https://yshghid.github.io/docs/study/bioinformatics/bi9/>ë¹„êµ ì˜ì˜ Traditional ë°©ë²•ì€ TopHat2+HTseq ì¡°í•©ì´ì§€ë§Œ ì˜¤ë¥˜ë„ ë„˜ ë§ì´ë‚˜ê³  Rsubreadë¥¼ ì“°ë©´ ë¹ ë¥´ê³  ê¹”ë”í•œë° ì™œ ì¨ì•¼í•˜ì§€..? ì‹¶ì–´ì„œ ë™ì¼í•œ ë°ì´í„°(pair-end fastq)ë¡œ ëŒë ¤ë´„. HTseqì—ì„œ ì•„ë˜ ì½”ë“œë¥¼ ìˆ˜í–‰í• ë•Œ íŒŒë¼ë¯¸í„°ê°€ ë§ì€ë° ë­ê°€ ë‹¤ë¥´ê²Œë‚˜ì˜¤ëŠ”ì§€ ëª¨ë¥´ê² ì–´ì„œ ì‹¤í—˜í•´ë´„. Cases
Rsubread ì‚¬ìš© HTSeq ì‚¬ìš©, â‹¯</a></p><hr><p><em>04-21</em> â‹¯ Bismark: WGBS ì „ì²˜ë¦¬</p><p><a href=https://yshghid.github.io/docs/study/bioinformatics/bi6/>1. Build Index $ bowtie2-build Homo_sapiens.GRCh38.dna.toplevel.fa GRCh38 -p 40 2. Bam Sorting & Indexing $ samtools sort KEB01_1_bismark_bt2_pe.bam -o KEB01_1_bismark_bt2_pe.sorted.bam $ samtools index KEB01_1_bismark_bt2_pe.sorted.bam â‹¯</a></p><hr><p><em>04-21</em> â‹¯ ChIP-seq ì „ì²˜ë¦¬</p><p><a href=https://yshghid.github.io/docs/study/bioinformatics/bi5/>1. Trimming #chipseq_trimming.sh #!/bin/bash setting envs export bdir="/data3/projects/2022_KNU_EBV" export hg38_bowtieidx="/data3/PUBLIC_DATA/ref_genomes/homo_sapiens/hg38/hg38_bowtie_idx/hg38.fa" export hg38_bwaidx="/data3/PUBLIC_DATA/ â‹¯</a></p><hr><p><em>04-21</em> â‹¯ Rsubread, edgeR: RNA-seq ì „ì²˜ë¦¬</p><p><a href=https://yshghid.github.io/docs/study/bioinformatics/bi8/>ê°€ì¥ ì˜¤ë¥˜ ì ê²Œë‚˜ëŠ” ì¡°í•©! 1. Align RNA-seq #Load Packages library(Rsubread) library(org.Mm.eg.db) library(gridExtra) library(reshape2) #Set Path indir = &ldquo;/data/home/ysh980101/2504/mirna/data&rdquo; outdir = &ldquo;/data/home/ysh980101/2504/mirna/data&rdquo; â‹¯</a></p><hr><p><em>04-21</em> â‹¯ TopHat, SAMtools, HTSeq: RNA-seq ì „ì²˜ë¦¬</p><p><a href=https://yshghid.github.io/docs/study/bioinformatics/bi7/>1. TopHat ì‹¤í–‰ $ tophatpy -o tophat_out_33-1 &ndash;no-mixed -p 40 \ $ /data3/PUBLIC_DATA/ref_genomes/homo_sapiens/GRCh38/Homo_sapiens.GRCh38.dna.toplevel \ $ /data/home/ysh980101/2306_tophat/data/Bowtie2Index/5-AZA_33-1_1.fastq \ $ /data/home/ysh980101/ â‹¯</a></p><hr><p><em>04-21</em> â‹¯ Gprofiler/ggplot2: Enrichment ë¶„ì„, ë²„ë¸” í”Œë¡¯</p><p><a href=https://yshghid.github.io/docs/study/bioinformatics/bi3/>Load Package library(ggplot2) Set Path setwd("/data-blog/bi3") getwd() &lsquo;/data-blog/bi3&rsquo; Functional Enrichment Bubble Plot condition &lt;- &lsquo;150_con&rsquo; gpsource &lt;- &lsquo;GO:BP&rsquo; #gpsource &lt;- &lsquo;REAC&rsquo; df_c1 &lt;- read.csv(paste0("./sleuth_ward/ â‹¯</a></p><hr><p><em>04-21</em> â‹¯ Sleuth ì‘ì—…</p><p><a href=https://yshghid.github.io/docs/study/bioinformatics/bi2/>Load Package, Run Sleuth require(&ldquo;sleuth&rdquo;) packageVersion(&ldquo;sleuth&rdquo;) library(&ldquo;gridExtra&rdquo;) library(&ldquo;cowplot&rdquo;) library(&ldquo;biomaRt&rdquo;) library(readr) setwd("/data/home/ysh980101/2307_kallisto") getwd() sample_id &lt;- dir(file.path("./")) â‹¯</a></p><hr><p><em>04-21</em> â‹¯ Kallisto Pseudoalignment ì‘ì—…</p><p><a href=https://yshghid.github.io/docs/study/bioinformatics/bi4/>1. Build Index $ kallisto index -i transcripts_cDNA.idx Homo_sapiens.GRCh38.cdna.all.fa.gz 2. Pseudoalign $ kallisto quant -i transcripts_cDNA.idx -o output_150-1 -t 40 ../2306_tophat/data/Bowtie2Index/5-AZA_150-1_1_edited.fastq ../2306 â‹¯</a></p><h2 id=2024>2024
<a class=anchor href=#2024>#</a></h2><p><em>12-31</em> â‹¯ EndNote ì‚¬ìš©ë²•</p><p><a href=https://yshghid.github.io/docs/study/bioinformatics/bi16/>1. EndNote ì„¤ì¹˜ ë° ê³„ì • ì„¤ì • ê³„ì • ì„¤ì •: ê³µì‹ ì›¹ì‚¬ì´íŠ¸ì—ì„œ End note ê³„ì •ì„ ìƒì„±í•œë‹¤. ì„¤ì¹˜: ë‚˜ì˜ ê²½ìš° ì—¬ê¸°ì—ì„œ ë‹¤ìš´ë¡œë“œí•´ì¤¬ë‹¤. 2. ë ˆí¼ëŸ°ìŠ¤ ì¶”ê°€ ë°©ë²• Google Scholarì— ë…¼ë¬¸ ì œëª©ì„ ê²€ìƒ‰í•´ì„œ ì¸ìš©>EndNoteë¥¼ í´ë¦­í•˜ë©´ .enw íŒŒì¼ì´ ë‹¤ìš´ë¡œë“œëœë‹¤. 3. ë ˆí¼ëŸ°ìŠ¤ ê´€ë¦¬ Endnoteì— ì ‘ì†í•œë‹¤. Collect>Import Referencesë¡œ ë“¤ì–´ê°„ë‹¤ â‹¯</a></p><hr><p><em>12-31</em> â‹¯ EBV RNA-seq ì „ì²˜ë¦¬</p><p><a href=https://yshghid.github.io/docs/study/bioinformatics/bi11/>ë¶„ì„ ëª©ì  ì œê³µë°›ì€ fastqë¥¼ human genomeì— ë§¤í•‘í•´ì„œ ì „ì²˜ë¦¬, ë¶„ì„ í›„ DE ê²°ê³¼ ë³´ëƒ„ DE ë¶„ì„ì‹œì— EBV ìœ ì „ìë„ í¬í•¨í•´ë‹¬ë¼ëŠ” ìš”ì²­ í•´ì•¼í•˜ëŠ”ê²ƒ fastqë¥¼ EBV genomeì— ë§¤í•‘í•´ì„œ ì „ì²˜ë¦¬, EBV count ìƒì„± human countì— EBV countë¥¼ ë¶™ì´ê¸° í†µí•© countë¡œ DE ë¶„ì„ ì¬ìˆ˜í–‰ 1. Alignment Load package, Set Path â‹¯</a></p><hr><p><em>12-31</em> â‹¯ DESeq2: DE ë¶„ì„</p><p><a href=https://yshghid.github.io/docs/study/bioinformatics/bi1/>suppressMessages({ library(&ldquo;DESeq2&rdquo;) library(pheatmap) library(withr) #library(tidyverse) library(RColorBrewer) library(gplots) library(dplyr) }) Set path setwd("/data-blog/bi1") getwd() &lsquo;/data-blog/bi1&rsquo; Run DESeq2 S1 &lt;- &lsquo;33&rsquo; S2 &lt;- &lsquo;150&rsquo; â‹¯</a></p><h1><a class=anchor href=#>#</a></h1><h3 id=ìƒë¬¼ì •ë³´í•™-ë©”ëª¨ì¥>ìƒë¬¼ì •ë³´í•™ ë©”ëª¨ì¥
<a class=anchor href=#%ec%83%9d%eb%ac%bc%ec%a0%95%eb%b3%b4%ed%95%99-%eb%a9%94%eb%aa%a8%ec%9e%a5>#</a></h3><p><a href=https://yshghid.github.io/docs/study/tech/tech17/>ï¹‚ë°€ë„ ê¸°ë°˜ í´ëŸ¬ìŠ¤í„°ë§ ì—°êµ¬</a></p><p><a href=https://yshghid.github.io/docs/study/tech/tech19/>ï¹‚í•­ìƒì œ ì˜ˆí›„ ì˜ˆì¸¡ ì—°êµ¬</a></p><h1><a class=anchor href=#>#</a></h1></article><footer class=book-footer><div class="flex flex-wrap justify-between"></div><script>(function(){function e(e){const t=window.getSelection(),n=document.createRange();n.selectNodeContents(e),t.removeAllRanges(),t.addRange(n)}document.querySelectorAll("pre code").forEach(t=>{t.addEventListener("click",function(){if(window.getSelection().toString())return;e(t.parentElement),navigator.clipboard&&navigator.clipboard.writeText(t.parentElement.textContent)})})})()</script></footer><label for=menu-control class="hidden book-menu-overlay"></label></div><aside class=book-toc><div class=book-toc-content><nav id=TableOfContents><ul><li><a href=#2025>2025</a></li><li><a href=#2024>2024</a></li></ul><ul><li><ul><li><a href=#ìƒë¬¼ì •ë³´í•™-ë©”ëª¨ì¥>ìƒë¬¼ì •ë³´í•™ ë©”ëª¨ì¥</a></li></ul></li></ul></nav></div></aside></main></body></html>