<!doctype html><html lang=en-us dir=ltr><head><meta charset=UTF-8><meta name=viewport content="width=device-width,initial-scale=1"><meta name=description content='
  생물정보학
  #


  2025
  #

07-12 ⋯ EdgeR: DE 분석
Load package suppressMessages({ library(&ldquo;DESeq2&rdquo;) library(pheatmap) library(withr) #library(tidyverse) library(RColorBrewer) library(gplots) library(dplyr) }) Set path setwd("/data/home/ysh980101/2307_EBV") getwd() &lsquo;/data1/home/ysh980101/2307_EBV&rsquo; ⋯

05-28 ⋯ Bash 쉘 스크립트 (로컬, 서버)
#local #alias cobi2=&lsquo;ssh -p 5290 ysh980101@155.230.28.211&rsquo; alias cobi2=&ldquo;ssh -p 3160 ysh980101@155.230.110.91&rdquo; alias cobi3=&ldquo;ssh -p 7777 ysh980101@155.230.110.92&rdquo; alias cobi4=&ldquo;ssh -p 4712 ysh980101@155.230.110.93&rdquo; # &#187;> conda initialize &#187;> # !! Contents ⋯

04-21 ⋯ TopHat2, HTSeq, Rsubread: RNA-seq 전처리 파이프라인 비교'><meta name=theme-color media="(prefers-color-scheme: light)" content="#ffffff"><meta name=theme-color media="(prefers-color-scheme: dark)" content="#343a40"><meta name=color-scheme content="light dark"><meta property="og:url" content="https://yshghid.github.io/docs/study/bioinformatics/"><meta property="og:site_name" content=" "><meta property="og:title" content="생물정보학"><meta property="og:description" content='생물정보학 # 2025 # 07-12 ⋯ EdgeR: DE 분석
Load package suppressMessages({ library(“DESeq2”) library(pheatmap) library(withr) #library(tidyverse) library(RColorBrewer) library(gplots) library(dplyr) }) Set path setwd("/data/home/ysh980101/2307_EBV") getwd() ‘/data1/home/ysh980101/2307_EBV’ ⋯
05-28 ⋯ Bash 쉘 스크립트 (로컬, 서버)
#local #alias cobi2=‘ssh -p 5290 ysh980101@155.230.28.211’ alias cobi2=“ssh -p 3160 ysh980101@155.230.110.91” alias cobi3=“ssh -p 7777 ysh980101@155.230.110.92” alias cobi4=“ssh -p 4712 ysh980101@155.230.110.93” # »> conda initialize »> # !! Contents ⋯
04-21 ⋯ TopHat2, HTSeq, Rsubread: RNA-seq 전처리 파이프라인 비교'><meta property="og:locale" content="en_us"><meta property="og:type" content="website"><title>생물정보학 |</title><link rel=icon href=/favicon.png><link rel=manifest href=/manifest.json><link rel=canonical href=https://yshghid.github.io/docs/study/bioinformatics/><link rel=stylesheet href=/book.min.6217d077edb4189fd0578345e84bca1a884dfdee121ff8dc9a0f55cfe0852bc9.css integrity="sha256-YhfQd+20GJ/QV4NF6EvKGohN/e4SH/jcmg9Vz+CFK8k=" crossorigin=anonymous><script defer src=/fuse.min.js></script><script defer src=/en.search.min.b40d1da87bde558170d8fe8b2a5c55a9878fcc2437f37fae11672f2fb9372c10.js integrity="sha256-tA0dqHveVYFw2P6LKlxVqYePzCQ383+uEWcvL7k3LBA=" crossorigin=anonymous></script><link rel=alternate type=application/rss+xml href=https://yshghid.github.io/docs/study/bioinformatics/index.xml title=" "></head><body dir=ltr><input type=checkbox class="hidden toggle" id=menu-control>
<input type=checkbox class="hidden toggle" id=toc-control><main class="container flex"><aside class=book-menu><div class=book-menu-content><nav><h2 class=book-brand><a class="flex align-center" href=/><img src=/logo.png alt=Logo class=book-icon><span></span></a></h2><div class="book-search hidden"><input type=text id=book-search-input placeholder=Search aria-label=Search maxlength=64 data-hotkeys=s/><div class="book-search-spinner hidden"></div><ul id=book-search-results></ul></div><script>document.querySelector(".book-search").classList.remove("hidden")</script><ul><li class=book-section-flat><span>기록</span><ul><li><a href=/docs/hobby/daily/>일상</a><ul></ul></li><li><a href=/docs/hobby/book/>글</a><ul></ul></li><li><a href=/docs/hobby/baking/>베이킹</a><ul></ul></li><li><a href=/docs/hobby/favorite/>🌸</a><ul></ul></li></ul></li><li class=book-section-flat><span>공부</span><ul><li><a href=/docs/study/dl/>딥러닝</a><ul></ul></li><li><a href=/docs/study/algorithm/>코테</a><ul></ul></li><li><a href=/docs/study/career/>취업</a><ul></ul></li><li><a href=/docs/study/bioinformatics/ class=active>생물정보학</a><ul></ul></li><li><a href=/docs/study/etc/>기타</a><ul></ul></li></ul></li></ul></nav><script>(function(){var e=document.querySelector("aside .book-menu-content");addEventListener("beforeunload",function(){localStorage.setItem("menu.scrollTop",e.scrollTop)}),e.scrollTop=localStorage.getItem("menu.scrollTop")})()</script></div></aside><div class=book-page><header class=book-header><div class="flex align-center justify-between"><label for=menu-control><img src=/svg/menu.svg class=book-icon alt=Menu></label><h3>생물정보학</h3><label for=toc-control><img src=/svg/toc.svg class=book-icon alt="Table of Contents"></label></div><aside class="hidden clearfix"><nav id=TableOfContents><ul><li><a href=#2025>2025</a></li><li><a href=#2024>2024</a></li></ul><ul><li><ul><li><a href=#생물정보학-메모장>생물정보학 메모장</a></li></ul></li></ul></nav></aside></header><article class="markdown book-article"><h1 id=생물정보학>생물정보학
<a class=anchor href=#%ec%83%9d%eb%ac%bc%ec%a0%95%eb%b3%b4%ed%95%99>#</a></h1><h2 id=2025>2025
<a class=anchor href=#2025>#</a></h2><p><em>07-12</em> ⋯ EdgeR: DE 분석</p><p><a href=https://yshghid.github.io/docs/study/bioinformatics/bi23/>Load package suppressMessages({ library(&ldquo;DESeq2&rdquo;) library(pheatmap) library(withr) #library(tidyverse) library(RColorBrewer) library(gplots) library(dplyr) }) Set path setwd("/data/home/ysh980101/2307_EBV") getwd() &lsquo;/data1/home/ysh980101/2307_EBV&rsquo; ⋯</a></p><hr><p><em>05-28</em> ⋯ Bash 쉘 스크립트 (로컬, 서버)</p><p><a href=https://yshghid.github.io/docs/study/tech/tech11/>#local #alias cobi2=&lsquo;ssh -p 5290 ysh980101@155.230.28.211&rsquo; alias cobi2=&ldquo;ssh -p 3160 ysh980101@155.230.110.91&rdquo; alias cobi3=&ldquo;ssh -p 7777 ysh980101@155.230.110.92&rdquo; alias cobi4=&ldquo;ssh -p 4712 ysh980101@155.230.110.93&rdquo; # &#187;> conda initialize &#187;> # !! Contents ⋯</a></p><hr><p><em>04-21</em> ⋯ TopHat2, HTSeq, Rsubread: RNA-seq 전처리 파이프라인 비교</p><p><a href=https://yshghid.github.io/docs/study/bioinformatics/bi9/>비교 의의 Traditional 방법은 TopHat2+HTseq 조합이지만 오류도 넘 많이나고 Rsubread를 쓰면 빠르고 깔끔한데 왜 써야하지..? 싶어서 동일한 데이터(pair-end fastq)로 돌려봄. HTseq에서 아래 코드를 수행할때 파라미터가 많은데 뭐가 다르게나오는지 모르겠어서 실험해봄. Cases
Rsubread 사용 HTSeq 사용, ⋯</a></p><hr><p><em>04-21</em> ⋯ Bismark: WGBS 전처리</p><p><a href=https://yshghid.github.io/docs/study/bioinformatics/bi6/>1. Build Index $ bowtie2-build Homo_sapiens.GRCh38.dna.toplevel.fa GRCh38 -p 40 2. Bam Sorting & Indexing $ samtools sort KEB01_1_bismark_bt2_pe.bam -o KEB01_1_bismark_bt2_pe.sorted.bam $ samtools index KEB01_1_bismark_bt2_pe.sorted.bam ⋯</a></p><hr><p><em>04-21</em> ⋯ ChIP-seq 전처리</p><p><a href=https://yshghid.github.io/docs/study/bioinformatics/bi5/>1. Trimming #chipseq_trimming.sh #!/bin/bash setting envs export bdir="/data3/projects/2022_KNU_EBV" export hg38_bowtieidx="/data3/PUBLIC_DATA/ref_genomes/homo_sapiens/hg38/hg38_bowtie_idx/hg38.fa" export hg38_bwaidx="/data3/PUBLIC_DATA/ ⋯</a></p><hr><p><em>04-21</em> ⋯ Rsubread, edgeR: RNA-seq 전처리</p><p><a href=https://yshghid.github.io/docs/study/bioinformatics/bi8/>가장 오류 적게나는 조합! 1. Align RNA-seq #Load Packages library(Rsubread) library(org.Mm.eg.db) library(gridExtra) library(reshape2) #Set Path indir = &ldquo;/data/home/ysh980101/2504/mirna/data&rdquo; outdir = &ldquo;/data/home/ysh980101/2504/mirna/data&rdquo; ⋯</a></p><hr><p><em>04-21</em> ⋯ TopHat, SAMtools, HTSeq: RNA-seq 전처리</p><p><a href=https://yshghid.github.io/docs/study/bioinformatics/bi7/>1. TopHat 실행 $ tophatpy -o tophat_out_33-1 &ndash;no-mixed -p 40 \ $ /data3/PUBLIC_DATA/ref_genomes/homo_sapiens/GRCh38/Homo_sapiens.GRCh38.dna.toplevel \ $ /data/home/ysh980101/2306_tophat/data/Bowtie2Index/5-AZA_33-1_1.fastq \ $ /data/home/ysh980101/ ⋯</a></p><hr><p><em>04-21</em> ⋯ Gprofiler/ggplot2: Enrichment 분석, 버블 플롯</p><p><a href=https://yshghid.github.io/docs/study/bioinformatics/bi3/>Load Package library(ggplot2) Set Path setwd("/data-blog/bi3") getwd() &lsquo;/data-blog/bi3&rsquo; Functional Enrichment Bubble Plot condition &lt;- &lsquo;150_con&rsquo; gpsource &lt;- &lsquo;GO:BP&rsquo; #gpsource &lt;- &lsquo;REAC&rsquo; df_c1 &lt;- read.csv(paste0("./sleuth_ward/ ⋯</a></p><hr><p><em>04-21</em> ⋯ Sleuth 작업</p><p><a href=https://yshghid.github.io/docs/study/bioinformatics/bi2/>Load Package, Run Sleuth require(&ldquo;sleuth&rdquo;) packageVersion(&ldquo;sleuth&rdquo;) library(&ldquo;gridExtra&rdquo;) library(&ldquo;cowplot&rdquo;) library(&ldquo;biomaRt&rdquo;) library(readr) setwd("/data/home/ysh980101/2307_kallisto") getwd() sample_id &lt;- dir(file.path("./")) ⋯</a></p><hr><p><em>04-21</em> ⋯ Kallisto Pseudoalignment 작업</p><p><a href=https://yshghid.github.io/docs/study/bioinformatics/bi4/>1. Build Index $ kallisto index -i transcripts_cDNA.idx Homo_sapiens.GRCh38.cdna.all.fa.gz 2. Pseudoalign $ kallisto quant -i transcripts_cDNA.idx -o output_150-1 -t 40 ../2306_tophat/data/Bowtie2Index/5-AZA_150-1_1_edited.fastq ../2306 ⋯</a></p><h2 id=2024>2024
<a class=anchor href=#2024>#</a></h2><p><em>12-31</em> ⋯ EndNote 사용법</p><p><a href=https://yshghid.github.io/docs/study/bioinformatics/bi16/>1. EndNote 설치 및 계정 설정 계정 설정: 공식 웹사이트에서 End note 계정을 생성한다. 설치: 나의 경우 여기에서 다운로드해줬다. 2. 레퍼런스 추가 방법 Google Scholar에 논문 제목을 검색해서 인용>EndNote를 클릭하면 .enw 파일이 다운로드된다. 3. 레퍼런스 관리 Endnote에 접속한다. Collect>Import References로 들어간다 ⋯</a></p><hr><p><em>12-31</em> ⋯ EBV RNA-seq 전처리</p><p><a href=https://yshghid.github.io/docs/study/bioinformatics/bi11/>분석 목적 제공받은 fastq를 human genome에 매핑해서 전처리, 분석 후 DE 결과 보냄 DE 분석시에 EBV 유전자도 포함해달라는 요청 해야하는것 fastq를 EBV genome에 매핑해서 전처리, EBV count 생성 human count에 EBV count를 붙이기 통합 count로 DE 분석 재수행 1. Alignment Load package, Set Path ⋯</a></p><hr><p><em>12-31</em> ⋯ DESeq2: DE 분석</p><p><a href=https://yshghid.github.io/docs/study/bioinformatics/bi1/>suppressMessages({ library(&ldquo;DESeq2&rdquo;) library(pheatmap) library(withr) #library(tidyverse) library(RColorBrewer) library(gplots) library(dplyr) }) Set path setwd("/data-blog/bi1") getwd() &lsquo;/data-blog/bi1&rsquo; Run DESeq2 S1 &lt;- &lsquo;33&rsquo; S2 &lt;- &lsquo;150&rsquo; ⋯</a></p><h1><a class=anchor href=#>#</a></h1><h3 id=생물정보학-메모장>생물정보학 메모장
<a class=anchor href=#%ec%83%9d%eb%ac%bc%ec%a0%95%eb%b3%b4%ed%95%99-%eb%a9%94%eb%aa%a8%ec%9e%a5>#</a></h3><p><a href=https://yshghid.github.io/docs/study/tech/tech17/>﹂밀도 기반 클러스터링 연구</a></p><p><a href=https://yshghid.github.io/docs/study/tech/tech19/>﹂항생제 예후 예측 연구</a></p><h1><a class=anchor href=#>#</a></h1></article><footer class=book-footer><div class="flex flex-wrap justify-between"></div><script>(function(){function e(e){const t=window.getSelection(),n=document.createRange();n.selectNodeContents(e),t.removeAllRanges(),t.addRange(n)}document.querySelectorAll("pre code").forEach(t=>{t.addEventListener("click",function(){if(window.getSelection().toString())return;e(t.parentElement),navigator.clipboard&&navigator.clipboard.writeText(t.parentElement.textContent)})})})()</script></footer><label for=menu-control class="hidden book-menu-overlay"></label></div><aside class=book-toc><div class=book-toc-content><nav id=TableOfContents><ul><li><a href=#2025>2025</a></li><li><a href=#2024>2024</a></li></ul><ul><li><ul><li><a href=#생물정보학-메모장>생물정보학 메모장</a></li></ul></li></ul></nav></div></aside></main></body></html>