<!doctype html><html lang=en-us dir=ltr><head><meta charset=UTF-8><meta name=viewport content="width=device-width,initial-scale=1"><meta name=description content='
  EBV RNA-seq 전처리
  #

#2024-12-31



분석 목적

제공받은 fastq를 human genome에 매핑해서 전처리, 분석 후 DE 결과 보냄
DE 분석시에 EBV 유전자도 포함해달라는 요청



해야하는것

fastq를 EBV genome에 매핑해서 전처리, EBV count 생성
human count에 EBV count를 붙이기
통합 count로 DE 분석 재수행




  1. Alignment
  #


  Load package, Set Path
  #

library(edgeR)
library(Rsubread)
library(org.Hs.eg.db)

setwd("/data/home/ysh980101/2311/RNA-seq_ebv/Rsubread")
getwd()
&#39;/data1/home/ysh980101/2311/RNA-seq_ebv/Rsubread&#39;

  Build Index
  #

# build index
ref <- "/data3/PUBLIC_DATA/ref_genomes/Human_gammaherpesvirus_4_EBV/NC_007605.1.fa"
output_basename <- "NC_007605.1_idx"
buildindex(basename = output_basename, reference = ref)

  Feature Count
  #

# feature.count
targets <- read.delim("targets.txt", header=TRUE)
output.files <- c(targets$FileName)

fc <- featureCounts(output.files, annot.inbuilt=FALSE, annot.ext = "/data3/PUBLIC_DATA/ref_genomes/Human_gammaherpesvirus_4_EBV/NC_007605.1.gtf", isGTFAnnotationFile = TRUE, 
                    GTF.featureType = "exon", GTF.attrType = "transcript_id", GTF.attrType.extra = NULL, isPairedEnd=TRUE, countReadPairs=TRUE)# nthreads=30)

  Save
  #

colnames(fc$counts) <- c(&#39;33-1&#39;,&#39;33-2&#39;,&#39;33-3&#39;,&#39;150-1&#39;,&#39;150-2&#39;,&#39;150-3&#39;,&#39;con-1&#39;,&#39;con-2&#39;,&#39;con-3&#39;)
head(fc$counts)

group <- factor(targets$Status)
y <- DGEList(fc$counts, group=group)
fc$sort.counts <- fc$counts[order(rownames(fc$counts)), ]

sort.counts.df <- as.data.frame(fc$sort.counts)
sort.counts.df$trans_id <- rownames(fc$sort.counts)
sort.counts.df <- sort.counts.df[, c("trans_id", "33-1", "33-2", "33-3", "150-1", "150-2", "150-3", "con-1", "con-2", "con-3")]

write.table(sort.counts.df, file = "count.tsv", sep = "\t", quote = FALSE, row.names = FALSE)

  2. Annotation
  #


  Load Package
  #

import os
import pandas as pd

  Set Path
  #

os.chdir("/data/home/ysh980101/2311/RNA-seq_ebv/Rsubread")
os.getcwd()
&#39;/data1/home/ysh980101/2311/RNA-seq_ebv/Rsubread&#39;

  Load GTF file
  #


annotation 파일이 없어서 gtf reference 파일을 사용해서 직접 만들어줫다.

gtf_df = pd.read_csv("/data3/PUBLIC_DATA/ref_genomes/Human_gammaherpesvirus_4_EBV/NC_007605.1.gtf", sep=&#39;\t&#39;, header=None, comment="#", names=["seqname", "source", "feature", "start", "end", "score", "strand", "frame", "attribute"])
gtf_df
'><meta name=theme-color media="(prefers-color-scheme: light)" content="#ffffff"><meta name=theme-color media="(prefers-color-scheme: dark)" content="#343a40"><meta name=color-scheme content="light dark"><meta property="og:url" content="https://yshghid.github.io/docs/study/bioinformatics/bi11/"><meta property="og:site_name" content=" "><meta property="og:title" content="EBV RNA-seq 전처리"><meta property="og:description" content='EBV RNA-seq 전처리 # #2024-12-31
분석 목적
제공받은 fastq를 human genome에 매핑해서 전처리, 분석 후 DE 결과 보냄 DE 분석시에 EBV 유전자도 포함해달라는 요청 해야하는것
fastq를 EBV genome에 매핑해서 전처리, EBV count 생성 human count에 EBV count를 붙이기 통합 count로 DE 분석 재수행 1. Alignment # Load package, Set Path # library(edgeR) library(Rsubread) library(org.Hs.eg.db) setwd("/data/home/ysh980101/2311/RNA-seq_ebv/Rsubread") getwd() &#39;/data1/home/ysh980101/2311/RNA-seq_ebv/Rsubread&#39; Build Index # # build index ref <- "/data3/PUBLIC_DATA/ref_genomes/Human_gammaherpesvirus_4_EBV/NC_007605.1.fa" output_basename <- "NC_007605.1_idx" buildindex(basename = output_basename, reference = ref) Feature Count # # feature.count targets <- read.delim("targets.txt", header=TRUE) output.files <- c(targets$FileName) fc <- featureCounts(output.files, annot.inbuilt=FALSE, annot.ext = "/data3/PUBLIC_DATA/ref_genomes/Human_gammaherpesvirus_4_EBV/NC_007605.1.gtf", isGTFAnnotationFile = TRUE, GTF.featureType = "exon", GTF.attrType = "transcript_id", GTF.attrType.extra = NULL, isPairedEnd=TRUE, countReadPairs=TRUE)# nthreads=30) Save # colnames(fc$counts) <- c(&#39;33-1&#39;,&#39;33-2&#39;,&#39;33-3&#39;,&#39;150-1&#39;,&#39;150-2&#39;,&#39;150-3&#39;,&#39;con-1&#39;,&#39;con-2&#39;,&#39;con-3&#39;) head(fc$counts) group <- factor(targets$Status) y <- DGEList(fc$counts, group=group) fc$sort.counts <- fc$counts[order(rownames(fc$counts)), ] sort.counts.df <- as.data.frame(fc$sort.counts) sort.counts.df$trans_id <- rownames(fc$sort.counts) sort.counts.df <- sort.counts.df[, c("trans_id", "33-1", "33-2", "33-3", "150-1", "150-2", "150-3", "con-1", "con-2", "con-3")] write.table(sort.counts.df, file = "count.tsv", sep = "\t", quote = FALSE, row.names = FALSE) 2. Annotation # Load Package # import os import pandas as pd Set Path # os.chdir("/data/home/ysh980101/2311/RNA-seq_ebv/Rsubread") os.getcwd() &#39;/data1/home/ysh980101/2311/RNA-seq_ebv/Rsubread&#39; Load GTF file # annotation 파일이 없어서 gtf reference 파일을 사용해서 직접 만들어줫다. gtf_df = pd.read_csv("/data3/PUBLIC_DATA/ref_genomes/Human_gammaherpesvirus_4_EBV/NC_007605.1.gtf", sep=&#39;\t&#39;, header=None, comment="#", names=["seqname", "source", "feature", "start", "end", "score", "strand", "frame", "attribute"]) gtf_df'><meta property="og:locale" content="en_us"><meta property="og:type" content="article"><meta property="article:section" content="docs"><meta property="article:published_time" content="2024-12-31T00:00:00+00:00"><meta property="article:modified_time" content="2024-12-31T00:00:00+00:00"><meta property="article:tag" content="2024-12"><title>EBV RNA-seq 전처리 |</title><link rel=icon href=/favicon.png><link rel=manifest href=/manifest.json><link rel=canonical href=https://yshghid.github.io/docs/study/bioinformatics/bi11/><link rel=stylesheet href=/book.min.6217d077edb4189fd0578345e84bca1a884dfdee121ff8dc9a0f55cfe0852bc9.css integrity="sha256-YhfQd+20GJ/QV4NF6EvKGohN/e4SH/jcmg9Vz+CFK8k=" crossorigin=anonymous><script defer src=/fuse.min.js></script><script defer src=/en.search.min.578a2d8e6e30e0a0ae3682797882d89ca0cbbf1734d206705396ea76cf57bbb6.js integrity="sha256-V4otjm4w4KCuNoJ5eILYnKDLvxc00gZwU5bqds9Xu7Y=" crossorigin=anonymous></script></head><body dir=ltr><input type=checkbox class="hidden toggle" id=menu-control>
<input type=checkbox class="hidden toggle" id=toc-control><main class="container flex"><aside class=book-menu><div class=book-menu-content><nav><h2 class=book-brand><a class="flex align-center" href=/><img src=/logo.png alt=Logo class=book-icon><span></span></a></h2><div class="book-search hidden"><input type=text id=book-search-input placeholder=Search aria-label=Search maxlength=64 data-hotkeys=s/><div class="book-search-spinner hidden"></div><ul id=book-search-results></ul></div><script>document.querySelector(".book-search").classList.remove("hidden")</script><ul><li class=book-section-flat><span>기록</span><ul><li><a href=/docs/hobby/book/>글</a><ul></ul></li><li><a href=/docs/hobby/daily/>일상</a><ul></ul></li></ul></li><li class=book-section-flat><span>공부</span><ul><li><a href=/docs/study/bioinformatics/>Bioinformatics</a><ul></ul></li><li><a href=/docs/study/ai/>AI</a><ul></ul></li><li><a href=/docs/study/sw/>SW</a><ul></ul></li><li><a href=/docs/study/algorithm/>알고리즘</a><ul></ul></li><li><a href=/docs/study/career/>취업</a><ul></ul></li></ul></li></ul></nav><script>(function(){var e=document.querySelector("aside .book-menu-content");addEventListener("beforeunload",function(){localStorage.setItem("menu.scrollTop",e.scrollTop)}),e.scrollTop=localStorage.getItem("menu.scrollTop")})()</script></div></aside><div class=book-page><header class=book-header><div class="flex align-center justify-between"><label for=menu-control><img src=/svg/menu.svg class=book-icon alt=Menu></label><h3>EBV RNA-seq 전처리</h3><label for=toc-control><img src=/svg/toc.svg class=book-icon alt="Table of Contents"></label></div><aside class="hidden clearfix"><nav id=TableOfContents><ul><li><a href=#1-alignment>1. Alignment</a><ul><li><a href=#load-package-set-path>Load package, Set Path</a></li><li><a href=#build-index>Build Index</a></li><li><a href=#feature-count>Feature Count</a></li><li><a href=#save>Save</a></li></ul></li><li><a href=#2-annotation>2. Annotation</a><ul><li><a href=#load-package>Load Package</a></li><li><a href=#set-path>Set Path</a></li><li><a href=#load-gtf-file>Load GTF file</a></li><li><a href=#make-annotation-file>Make annotation file</a></li><li><a href=#save-1>Save</a></li><li><a href=#load-count>Load Count</a></li><li><a href=#annotation>Annotation</a></li><li><a href=#merge>Merge</a></li></ul></li><li><a href=#3-deg-analysis>3. DEG Analysis</a><ul><li><a href=#load-packages>Load Packages</a></li><li><a href=#set-path-1>Set Path</a></li><li><a href=#run-deg>Run DEG</a></li></ul></li></ul></nav></aside></header><article class="markdown book-article"><h1 id=ebv-rna-seq-전처리>EBV RNA-seq 전처리
<a class=anchor href=#ebv-rna-seq-%ec%a0%84%ec%b2%98%eb%a6%ac>#</a></h1><p>#2024-12-31</p><hr><ul><li><p>분석 목적</p><ul><li>제공받은 fastq를 human genome에 매핑해서 전처리, 분석 후 DE 결과 보냄</li><li>DE 분석시에 EBV 유전자도 포함해달라는 요청</li></ul></li><li><p>해야하는것</p><ul><li>fastq를 EBV genome에 매핑해서 전처리, EBV count 생성</li><li>human count에 EBV count를 붙이기</li><li>통합 count로 DE 분석 재수행</li></ul></li></ul><h2 id=1-alignment>1. Alignment
<a class=anchor href=#1-alignment>#</a></h2><h3 id=load-package-set-path>Load package, Set Path
<a class=anchor href=#load-package-set-path>#</a></h3><div class=highlight><pre tabindex=0 class=chroma><code class=language-R data-lang=R><span class=line><span class=cl><span class=nf>library</span><span class=p>(</span><span class=n>edgeR</span><span class=p>)</span>
</span></span><span class=line><span class=cl><span class=nf>library</span><span class=p>(</span><span class=n>Rsubread</span><span class=p>)</span>
</span></span><span class=line><span class=cl><span class=nf>library</span><span class=p>(</span><span class=n>org.Hs.eg.db</span><span class=p>)</span>
</span></span><span class=line><span class=cl>
</span></span><span class=line><span class=cl><span class=nf>setwd</span><span class=p>(</span><span class=s>&#34;/data/home/ysh980101/2311/RNA-seq_ebv/Rsubread&#34;</span><span class=p>)</span>
</span></span><span class=line><span class=cl><span class=nf>getwd</span><span class=p>()</span>
</span></span></code></pre></div><div class=highlight><pre tabindex=0 class=chroma><code class=language-plain data-lang=plain><span class=line><span class=cl>&#39;/data1/home/ysh980101/2311/RNA-seq_ebv/Rsubread&#39;
</span></span></code></pre></div><h3 id=build-index>Build Index
<a class=anchor href=#build-index>#</a></h3><div class=highlight><pre tabindex=0 class=chroma><code class=language-R data-lang=R><span class=line><span class=cl><span class=c1># build index</span>
</span></span><span class=line><span class=cl><span class=n>ref</span> <span class=o>&lt;-</span> <span class=s>&#34;/data3/PUBLIC_DATA/ref_genomes/Human_gammaherpesvirus_4_EBV/NC_007605.1.fa&#34;</span>
</span></span><span class=line><span class=cl><span class=n>output_basename</span> <span class=o>&lt;-</span> <span class=s>&#34;NC_007605.1_idx&#34;</span>
</span></span><span class=line><span class=cl><span class=nf>buildindex</span><span class=p>(</span><span class=n>basename</span> <span class=o>=</span> <span class=n>output_basename</span><span class=p>,</span> <span class=n>reference</span> <span class=o>=</span> <span class=n>ref</span><span class=p>)</span>
</span></span></code></pre></div><h3 id=feature-count>Feature Count
<a class=anchor href=#feature-count>#</a></h3><div class=highlight><pre tabindex=0 class=chroma><code class=language-R data-lang=R><span class=line><span class=cl><span class=c1># feature.count</span>
</span></span><span class=line><span class=cl><span class=n>targets</span> <span class=o>&lt;-</span> <span class=nf>read.delim</span><span class=p>(</span><span class=s>&#34;targets.txt&#34;</span><span class=p>,</span> <span class=n>header</span><span class=o>=</span><span class=kc>TRUE</span><span class=p>)</span>
</span></span><span class=line><span class=cl><span class=n>output.files</span> <span class=o>&lt;-</span> <span class=nf>c</span><span class=p>(</span><span class=n>targets</span><span class=o>$</span><span class=n>FileName</span><span class=p>)</span>
</span></span><span class=line><span class=cl>
</span></span><span class=line><span class=cl><span class=n>fc</span> <span class=o>&lt;-</span> <span class=nf>featureCounts</span><span class=p>(</span><span class=n>output.files</span><span class=p>,</span> <span class=n>annot.inbuilt</span><span class=o>=</span><span class=kc>FALSE</span><span class=p>,</span> <span class=n>annot.ext</span> <span class=o>=</span> <span class=s>&#34;/data3/PUBLIC_DATA/ref_genomes/Human_gammaherpesvirus_4_EBV/NC_007605.1.gtf&#34;</span><span class=p>,</span> <span class=n>isGTFAnnotationFile</span> <span class=o>=</span> <span class=kc>TRUE</span><span class=p>,</span> 
</span></span><span class=line><span class=cl>                    <span class=n>GTF.featureType</span> <span class=o>=</span> <span class=s>&#34;exon&#34;</span><span class=p>,</span> <span class=n>GTF.attrType</span> <span class=o>=</span> <span class=s>&#34;transcript_id&#34;</span><span class=p>,</span> <span class=n>GTF.attrType.extra</span> <span class=o>=</span> <span class=kc>NULL</span><span class=p>,</span> <span class=n>isPairedEnd</span><span class=o>=</span><span class=kc>TRUE</span><span class=p>,</span> <span class=n>countReadPairs</span><span class=o>=</span><span class=kc>TRUE</span><span class=p>)</span><span class=c1># nthreads=30)</span>
</span></span></code></pre></div><h3 id=save>Save
<a class=anchor href=#save>#</a></h3><div class=highlight><pre tabindex=0 class=chroma><code class=language-R data-lang=R><span class=line><span class=cl><span class=nf>colnames</span><span class=p>(</span><span class=n>fc</span><span class=o>$</span><span class=n>counts</span><span class=p>)</span> <span class=o>&lt;-</span> <span class=nf>c</span><span class=p>(</span><span class=s>&#39;33-1&#39;</span><span class=p>,</span><span class=s>&#39;33-2&#39;</span><span class=p>,</span><span class=s>&#39;33-3&#39;</span><span class=p>,</span><span class=s>&#39;150-1&#39;</span><span class=p>,</span><span class=s>&#39;150-2&#39;</span><span class=p>,</span><span class=s>&#39;150-3&#39;</span><span class=p>,</span><span class=s>&#39;con-1&#39;</span><span class=p>,</span><span class=s>&#39;con-2&#39;</span><span class=p>,</span><span class=s>&#39;con-3&#39;</span><span class=p>)</span>
</span></span><span class=line><span class=cl><span class=nf>head</span><span class=p>(</span><span class=n>fc</span><span class=o>$</span><span class=n>counts</span><span class=p>)</span>
</span></span><span class=line><span class=cl>
</span></span><span class=line><span class=cl><span class=n>group</span> <span class=o>&lt;-</span> <span class=nf>factor</span><span class=p>(</span><span class=n>targets</span><span class=o>$</span><span class=n>Status</span><span class=p>)</span>
</span></span><span class=line><span class=cl><span class=n>y</span> <span class=o>&lt;-</span> <span class=nf>DGEList</span><span class=p>(</span><span class=n>fc</span><span class=o>$</span><span class=n>counts</span><span class=p>,</span> <span class=n>group</span><span class=o>=</span><span class=n>group</span><span class=p>)</span>
</span></span><span class=line><span class=cl><span class=n>fc</span><span class=o>$</span><span class=n>sort.counts</span> <span class=o>&lt;-</span> <span class=n>fc</span><span class=o>$</span><span class=n>counts</span><span class=nf>[order</span><span class=p>(</span><span class=nf>rownames</span><span class=p>(</span><span class=n>fc</span><span class=o>$</span><span class=n>counts</span><span class=p>)),</span> <span class=n>]</span>
</span></span><span class=line><span class=cl>
</span></span><span class=line><span class=cl><span class=n>sort.counts.df</span> <span class=o>&lt;-</span> <span class=nf>as.data.frame</span><span class=p>(</span><span class=n>fc</span><span class=o>$</span><span class=n>sort.counts</span><span class=p>)</span>
</span></span><span class=line><span class=cl><span class=n>sort.counts.df</span><span class=o>$</span><span class=n>trans_id</span> <span class=o>&lt;-</span> <span class=nf>rownames</span><span class=p>(</span><span class=n>fc</span><span class=o>$</span><span class=n>sort.counts</span><span class=p>)</span>
</span></span><span class=line><span class=cl><span class=n>sort.counts.df</span> <span class=o>&lt;-</span> <span class=n>sort.counts.df[</span><span class=p>,</span> <span class=nf>c</span><span class=p>(</span><span class=s>&#34;trans_id&#34;</span><span class=p>,</span> <span class=s>&#34;33-1&#34;</span><span class=p>,</span> <span class=s>&#34;33-2&#34;</span><span class=p>,</span> <span class=s>&#34;33-3&#34;</span><span class=p>,</span> <span class=s>&#34;150-1&#34;</span><span class=p>,</span> <span class=s>&#34;150-2&#34;</span><span class=p>,</span> <span class=s>&#34;150-3&#34;</span><span class=p>,</span> <span class=s>&#34;con-1&#34;</span><span class=p>,</span> <span class=s>&#34;con-2&#34;</span><span class=p>,</span> <span class=s>&#34;con-3&#34;</span><span class=p>)</span><span class=n>]</span>
</span></span><span class=line><span class=cl>
</span></span><span class=line><span class=cl><span class=nf>write.table</span><span class=p>(</span><span class=n>sort.counts.df</span><span class=p>,</span> <span class=n>file</span> <span class=o>=</span> <span class=s>&#34;count.tsv&#34;</span><span class=p>,</span> <span class=n>sep</span> <span class=o>=</span> <span class=s>&#34;\t&#34;</span><span class=p>,</span> <span class=n>quote</span> <span class=o>=</span> <span class=kc>FALSE</span><span class=p>,</span> <span class=n>row.names</span> <span class=o>=</span> <span class=kc>FALSE</span><span class=p>)</span>
</span></span></code></pre></div><h2 id=2-annotation>2. Annotation
<a class=anchor href=#2-annotation>#</a></h2><h3 id=load-package>Load Package
<a class=anchor href=#load-package>#</a></h3><div class=highlight><pre tabindex=0 class=chroma><code class=language-python data-lang=python><span class=line><span class=cl><span class=kn>import</span> <span class=nn>os</span>
</span></span><span class=line><span class=cl><span class=kn>import</span> <span class=nn>pandas</span> <span class=k>as</span> <span class=nn>pd</span>
</span></span></code></pre></div><h3 id=set-path>Set Path
<a class=anchor href=#set-path>#</a></h3><div class=highlight><pre tabindex=0 class=chroma><code class=language-python data-lang=python><span class=line><span class=cl><span class=n>os</span><span class=o>.</span><span class=n>chdir</span><span class=p>(</span><span class=s2>&#34;/data/home/ysh980101/2311/RNA-seq_ebv/Rsubread&#34;</span><span class=p>)</span>
</span></span><span class=line><span class=cl><span class=n>os</span><span class=o>.</span><span class=n>getcwd</span><span class=p>()</span>
</span></span></code></pre></div><div class=highlight><pre tabindex=0 class=chroma><code class=language-plain data-lang=plain><span class=line><span class=cl>&#39;/data1/home/ysh980101/2311/RNA-seq_ebv/Rsubread&#39;
</span></span></code></pre></div><h3 id=load-gtf-file>Load GTF file
<a class=anchor href=#load-gtf-file>#</a></h3><ul><li>annotation 파일이 없어서 gtf reference 파일을 사용해서 직접 만들어줫다.</li></ul><div class=highlight><pre tabindex=0 class=chroma><code class=language-python data-lang=python><span class=line><span class=cl><span class=n>gtf_df</span> <span class=o>=</span> <span class=n>pd</span><span class=o>.</span><span class=n>read_csv</span><span class=p>(</span><span class=s2>&#34;/data3/PUBLIC_DATA/ref_genomes/Human_gammaherpesvirus_4_EBV/NC_007605.1.gtf&#34;</span><span class=p>,</span> <span class=n>sep</span><span class=o>=</span><span class=s1>&#39;</span><span class=se>\t</span><span class=s1>&#39;</span><span class=p>,</span> <span class=n>header</span><span class=o>=</span><span class=kc>None</span><span class=p>,</span> <span class=n>comment</span><span class=o>=</span><span class=s2>&#34;#&#34;</span><span class=p>,</span> <span class=n>names</span><span class=o>=</span><span class=p>[</span><span class=s2>&#34;seqname&#34;</span><span class=p>,</span> <span class=s2>&#34;source&#34;</span><span class=p>,</span> <span class=s2>&#34;feature&#34;</span><span class=p>,</span> <span class=s2>&#34;start&#34;</span><span class=p>,</span> <span class=s2>&#34;end&#34;</span><span class=p>,</span> <span class=s2>&#34;score&#34;</span><span class=p>,</span> <span class=s2>&#34;strand&#34;</span><span class=p>,</span> <span class=s2>&#34;frame&#34;</span><span class=p>,</span> <span class=s2>&#34;attribute&#34;</span><span class=p>])</span>
</span></span><span class=line><span class=cl><span class=n>gtf_df</span>
</span></span></code></pre></div><p><img src=https://github.com/user-attachments/assets/26526e85-9631-47b0-bcf8-a3e37da66d22 alt=image></p><h3 id=make-annotation-file>Make annotation file
<a class=anchor href=#make-annotation-file>#</a></h3><div class=highlight><pre tabindex=0 class=chroma><code class=language-python data-lang=python><span class=line><span class=cl><span class=n>split_attributes</span> <span class=o>=</span> <span class=n>gtf_df</span><span class=p>[</span><span class=s1>&#39;attribute&#39;</span><span class=p>]</span><span class=o>.</span><span class=n>str</span><span class=o>.</span><span class=n>split</span><span class=p>(</span><span class=s1>&#39;;&#39;</span><span class=p>)</span>
</span></span><span class=line><span class=cl>
</span></span><span class=line><span class=cl><span class=n>attribute_1</span> <span class=o>=</span> <span class=p>[]</span>
</span></span><span class=line><span class=cl><span class=n>attribute_2</span> <span class=o>=</span> <span class=p>[]</span>
</span></span><span class=line><span class=cl><span class=n>attribute_3</span> <span class=o>=</span> <span class=p>[]</span>
</span></span><span class=line><span class=cl>
</span></span><span class=line><span class=cl><span class=k>for</span> <span class=n>attributes</span> <span class=ow>in</span> <span class=n>split_attributes</span><span class=p>:</span>
</span></span><span class=line><span class=cl>    <span class=n>attr_1</span> <span class=o>=</span> <span class=s1>&#39;&#39;</span>
</span></span><span class=line><span class=cl>    <span class=n>attr_2</span> <span class=o>=</span> <span class=s1>&#39;&#39;</span>
</span></span><span class=line><span class=cl>    <span class=n>attr_3</span> <span class=o>=</span> <span class=s1>&#39;&#39;</span>
</span></span><span class=line><span class=cl>
</span></span><span class=line><span class=cl>    <span class=k>for</span> <span class=n>attribute</span> <span class=ow>in</span> <span class=n>attributes</span><span class=p>:</span>
</span></span><span class=line><span class=cl>        <span class=n>attribute</span> <span class=o>=</span> <span class=n>attribute</span><span class=o>.</span><span class=n>strip</span><span class=p>()</span>
</span></span><span class=line><span class=cl>        <span class=k>if</span> <span class=s1>&#39;transcript_id&#39;</span> <span class=ow>in</span> <span class=n>attribute</span><span class=p>:</span>
</span></span><span class=line><span class=cl>            <span class=n>attr_1</span> <span class=o>=</span> <span class=n>attribute</span>
</span></span><span class=line><span class=cl>        <span class=k>elif</span> <span class=s1>&#39;gene_id&#39;</span> <span class=ow>in</span> <span class=n>attribute</span><span class=p>:</span>
</span></span><span class=line><span class=cl>            <span class=n>attr_2</span> <span class=o>=</span> <span class=n>attribute</span>
</span></span><span class=line><span class=cl>        <span class=k>elif</span> <span class=s1>&#39;gene_name&#39;</span> <span class=ow>in</span> <span class=n>attribute</span><span class=p>:</span>
</span></span><span class=line><span class=cl>            <span class=n>attr_3</span> <span class=o>=</span> <span class=n>attribute</span>
</span></span><span class=line><span class=cl>
</span></span><span class=line><span class=cl>    <span class=n>attribute_1</span><span class=o>.</span><span class=n>append</span><span class=p>(</span><span class=n>attr_1</span><span class=p>)</span>
</span></span><span class=line><span class=cl>    <span class=n>attribute_2</span><span class=o>.</span><span class=n>append</span><span class=p>(</span><span class=n>attr_2</span><span class=p>)</span>
</span></span><span class=line><span class=cl>    <span class=n>attribute_3</span><span class=o>.</span><span class=n>append</span><span class=p>(</span><span class=n>attr_3</span><span class=p>)</span>
</span></span><span class=line><span class=cl>
</span></span><span class=line><span class=cl><span class=n>gtf_df</span><span class=p>[</span><span class=s1>&#39;attribute.1&#39;</span><span class=p>]</span> <span class=o>=</span> <span class=n>attribute_1</span>
</span></span><span class=line><span class=cl><span class=n>gtf_df</span><span class=p>[</span><span class=s1>&#39;attribute.2&#39;</span><span class=p>]</span> <span class=o>=</span> <span class=n>attribute_2</span>
</span></span><span class=line><span class=cl><span class=n>gtf_df</span><span class=p>[</span><span class=s1>&#39;attribute.3&#39;</span><span class=p>]</span> <span class=o>=</span> <span class=n>attribute_3</span>
</span></span><span class=line><span class=cl>
</span></span><span class=line><span class=cl><span class=n>annot_df</span> <span class=o>=</span> <span class=n>gtf_df</span><span class=o>.</span><span class=n>iloc</span><span class=p>[:,</span> <span class=o>-</span><span class=mi>4</span><span class=p>:]</span><span class=o>.</span><span class=n>copy</span><span class=p>()</span>
</span></span><span class=line><span class=cl><span class=n>annot_df</span><span class=p>[</span><span class=s1>&#39;transcript_id&#39;</span><span class=p>]</span> <span class=o>=</span> <span class=n>annot_df</span><span class=p>[</span><span class=s1>&#39;attribute.1&#39;</span><span class=p>]</span><span class=o>.</span><span class=n>str</span><span class=o>.</span><span class=n>split</span><span class=p>(</span><span class=s1>&#39;&#34;&#39;</span><span class=p>)</span><span class=o>.</span><span class=n>str</span><span class=p>[</span><span class=mi>1</span><span class=p>]</span>
</span></span><span class=line><span class=cl><span class=n>annot_df</span><span class=p>[</span><span class=s1>&#39;gene_id&#39;</span><span class=p>]</span> <span class=o>=</span> <span class=n>annot_df</span><span class=p>[</span><span class=s1>&#39;attribute.2&#39;</span><span class=p>]</span><span class=o>.</span><span class=n>str</span><span class=o>.</span><span class=n>split</span><span class=p>(</span><span class=s1>&#39;&#34;&#39;</span><span class=p>)</span><span class=o>.</span><span class=n>str</span><span class=p>[</span><span class=mi>1</span><span class=p>]</span>
</span></span><span class=line><span class=cl><span class=n>annot_df</span><span class=p>[</span><span class=s1>&#39;gene_name&#39;</span><span class=p>]</span> <span class=o>=</span> <span class=n>annot_df</span><span class=p>[</span><span class=s1>&#39;attribute.3&#39;</span><span class=p>]</span><span class=o>.</span><span class=n>str</span><span class=o>.</span><span class=n>split</span><span class=p>(</span><span class=s1>&#39;&#34;&#39;</span><span class=p>)</span><span class=o>.</span><span class=n>str</span><span class=p>[</span><span class=mi>1</span><span class=p>]</span>
</span></span><span class=line><span class=cl>
</span></span><span class=line><span class=cl><span class=n>annot_df</span>
</span></span></code></pre></div><p><img src=https://github.com/user-attachments/assets/b172aa0a-c287-4d88-a27e-4483c9906cf4 alt=image></p><div class=highlight><pre tabindex=0 class=chroma><code class=language-python data-lang=python><span class=line><span class=cl><span class=n>annot_table</span> <span class=o>=</span> <span class=n>annot_df</span><span class=o>.</span><span class=n>iloc</span><span class=p>[:,</span> <span class=o>-</span><span class=mi>3</span><span class=p>:]</span><span class=o>.</span><span class=n>copy</span><span class=p>()</span>
</span></span><span class=line><span class=cl><span class=n>annot_table_dedup</span> <span class=o>=</span> <span class=n>annot_table</span><span class=p>[</span><span class=o>~</span><span class=n>annot_table</span><span class=p>[</span><span class=s1>&#39;transcript_id&#39;</span><span class=p>]</span><span class=o>.</span><span class=n>duplicated</span><span class=p>(</span><span class=n>keep</span><span class=o>=</span><span class=s1>&#39;first&#39;</span><span class=p>)]</span>
</span></span><span class=line><span class=cl><span class=n>annot</span> <span class=o>=</span> <span class=n>annot_table_dedup</span><span class=o>.</span><span class=n>iloc</span><span class=p>[:,</span> <span class=o>-</span><span class=mi>3</span><span class=p>:]</span><span class=o>.</span><span class=n>copy</span><span class=p>()</span>
</span></span><span class=line><span class=cl><span class=n>annot</span> <span class=o>=</span> <span class=n>annot</span><span class=o>.</span><span class=n>sort_values</span><span class=p>(</span><span class=n>by</span><span class=o>=</span><span class=s1>&#39;transcript_id&#39;</span><span class=p>)</span>
</span></span><span class=line><span class=cl><span class=n>annot</span><span class=o>.</span><span class=n>head</span><span class=p>(</span><span class=mi>50</span><span class=p>)</span>
</span></span></code></pre></div><p><img src=https://github.com/user-attachments/assets/42bca1cb-51be-4486-83dc-c6b31a73fc5c alt=image></p><h3 id=save-1>Save
<a class=anchor href=#save-1>#</a></h3><div class=highlight><pre tabindex=0 class=chroma><code class=language-python data-lang=python><span class=line><span class=cl><span class=c1>#annot.to_csv(&#34;annot.tsv&#34;,sep=&#39;\t&#39;,index=False)</span>
</span></span></code></pre></div><ul><li>나중에 또 써야대니깐 저장.</li></ul><h3 id=load-count>Load Count
<a class=anchor href=#load-count>#</a></h3><div class=highlight><pre tabindex=0 class=chroma><code class=language-python data-lang=python><span class=line><span class=cl><span class=n>count</span> <span class=o>=</span> <span class=n>pd</span><span class=o>.</span><span class=n>read_csv</span><span class=p>(</span><span class=s2>&#34;count.tsv&#34;</span><span class=p>,</span><span class=n>sep</span><span class=o>=</span><span class=s1>&#39;</span><span class=se>\t</span><span class=s1>&#39;</span><span class=p>,</span><span class=n>encoding</span><span class=o>=</span><span class=s1>&#39;cp949&#39;</span><span class=p>)</span>
</span></span><span class=line><span class=cl><span class=n>count</span>
</span></span></code></pre></div><p><img src=https://github.com/user-attachments/assets/a0b6d6b0-682a-44af-9f30-20029fc91608 alt=image></p><ul><li>위에서 Rsubread로 만든 EBV count 가져옴</li></ul><h3 id=annotation>Annotation
<a class=anchor href=#annotation>#</a></h3><div class=highlight><pre tabindex=0 class=chroma><code class=language-python data-lang=python><span class=line><span class=cl><span class=n>merged</span> <span class=o>=</span> <span class=n>count</span><span class=o>.</span><span class=n>merge</span><span class=p>(</span><span class=n>annot</span><span class=p>,</span> <span class=n>left_on</span><span class=o>=</span><span class=s1>&#39;trans_id&#39;</span><span class=p>,</span> <span class=n>right_on</span><span class=o>=</span><span class=s1>&#39;transcript_id&#39;</span><span class=p>,</span> <span class=n>how</span><span class=o>=</span><span class=s1>&#39;left&#39;</span><span class=p>)</span>
</span></span><span class=line><span class=cl><span class=n>res_df</span> <span class=o>=</span> <span class=n>merged</span><span class=o>.</span><span class=n>drop</span><span class=p>(</span><span class=n>columns</span><span class=o>=</span><span class=p>[</span><span class=s1>&#39;trans_id&#39;</span><span class=p>])</span>
</span></span><span class=line><span class=cl><span class=n>res_df</span> <span class=o>=</span> <span class=n>res_df</span><span class=p>[[</span><span class=s1>&#39;transcript_id&#39;</span><span class=p>,</span> <span class=s1>&#39;gene_id&#39;</span><span class=p>,</span> <span class=s1>&#39;gene_name&#39;</span><span class=p>,</span> <span class=s1>&#39;33-1&#39;</span><span class=p>,</span> <span class=s1>&#39;33-2&#39;</span><span class=p>,</span> <span class=s1>&#39;33-3&#39;</span><span class=p>,</span> <span class=s1>&#39;150-1&#39;</span><span class=p>,</span> <span class=s1>&#39;150-2&#39;</span><span class=p>,</span> <span class=s1>&#39;150-3&#39;</span><span class=p>,</span> <span class=s1>&#39;con-1&#39;</span><span class=p>,</span> <span class=s1>&#39;con-2&#39;</span><span class=p>,</span> <span class=s1>&#39;con-3&#39;</span><span class=p>]]</span>
</span></span><span class=line><span class=cl><span class=n>res_df</span>
</span></span></code></pre></div><p><img src=https://github.com/user-attachments/assets/581ad2f0-9ec1-42d1-a150-3a36ea774fe3 alt=image></p><div class=highlight><pre tabindex=0 class=chroma><code class=language-python data-lang=python><span class=line><span class=cl><span class=n>res_df</span><span class=p>[</span><span class=s1>&#39;gene_name&#39;</span><span class=p>]</span><span class=o>.</span><span class=n>value_counts</span><span class=p>()</span>
</span></span></code></pre></div><p><img src=https://github.com/user-attachments/assets/ad94f072-1cfa-4a30-bbfc-a083b9efc252 alt=image></p><ul><li>중복인 transcript가 엄청많음</li></ul><div class=highlight><pre tabindex=0 class=chroma><code class=language-python data-lang=python><span class=line><span class=cl><span class=n>res_df</span><span class=p>[</span><span class=s1>&#39;sum&#39;</span><span class=p>]</span> <span class=o>=</span> <span class=n>res_df</span><span class=o>.</span><span class=n>iloc</span><span class=p>[:,</span> <span class=mi>3</span><span class=p>:</span><span class=mi>12</span><span class=p>]</span><span class=o>.</span><span class=n>sum</span><span class=p>(</span><span class=n>axis</span><span class=o>=</span><span class=mi>1</span><span class=p>)</span>
</span></span><span class=line><span class=cl><span class=n>res_df</span> <span class=o>=</span> <span class=n>res_df</span><span class=o>.</span><span class=n>sort_values</span><span class=p>(</span><span class=n>by</span><span class=o>=</span><span class=s1>&#39;sum&#39;</span><span class=p>,</span> <span class=n>ascending</span><span class=o>=</span><span class=kc>False</span><span class=p>)</span>
</span></span><span class=line><span class=cl><span class=n>res_df</span> <span class=o>=</span> <span class=n>res_df</span><span class=o>.</span><span class=n>drop_duplicates</span><span class=p>(</span><span class=n>subset</span><span class=o>=</span><span class=s1>&#39;gene_name&#39;</span><span class=p>,</span> <span class=n>keep</span><span class=o>=</span><span class=s1>&#39;first&#39;</span><span class=p>)</span>
</span></span><span class=line><span class=cl>
</span></span><span class=line><span class=cl><span class=n>res_df</span> <span class=o>=</span> <span class=n>res_df</span><span class=o>.</span><span class=n>iloc</span><span class=p>[:,</span> <span class=p>:</span><span class=o>-</span><span class=mi>1</span><span class=p>]</span>
</span></span><span class=line><span class=cl><span class=n>res_df</span><span class=o>.</span><span class=n>reset_index</span><span class=p>(</span><span class=n>drop</span><span class=o>=</span><span class=kc>True</span><span class=p>,</span> <span class=n>inplace</span><span class=o>=</span><span class=kc>True</span><span class=p>)</span>
</span></span><span class=line><span class=cl><span class=n>res_df</span> <span class=o>=</span> <span class=n>res_df</span><span class=o>.</span><span class=n>dropna</span><span class=p>(</span><span class=n>subset</span><span class=o>=</span><span class=p>[</span><span class=s1>&#39;gene_name&#39;</span><span class=p>])</span>
</span></span><span class=line><span class=cl><span class=nb>print</span><span class=p>(</span><span class=n>res_df</span><span class=o>.</span><span class=n>shape</span><span class=p>)</span>
</span></span><span class=line><span class=cl><span class=n>res_df</span>
</span></span></code></pre></div><p><img src=https://github.com/user-attachments/assets/f6412998-0d28-4dfd-8d10-263c610c40fb alt=image></p><ul><li>(63, 12)짜리에서 deduplication 결과 (15, 12)가 됏다.</li></ul><div class=highlight><pre tabindex=0 class=chroma><code class=language-python data-lang=python><span class=line><span class=cl><span class=n>res_df</span><span class=o>.</span><span class=n>to_csv</span><span class=p>(</span><span class=s2>&#34;count.annot.tsv&#34;</span><span class=p>,</span> <span class=n>sep</span><span class=o>=</span><span class=s1>&#39;</span><span class=se>\t</span><span class=s1>&#39;</span><span class=p>,</span><span class=n>index</span><span class=o>=</span><span class=kc>False</span><span class=p>)</span>
</span></span></code></pre></div><h3 id=merge>Merge
<a class=anchor href=#merge>#</a></h3><div class=highlight><pre tabindex=0 class=chroma><code class=language-python data-lang=python><span class=line><span class=cl><span class=n>count_hs</span> <span class=o>=</span> <span class=n>pd</span><span class=o>.</span><span class=n>read_csv</span><span class=p>(</span><span class=s2>&#34;/data/home/ysh980101/2311/RNA-seq/count.csv&#34;</span><span class=p>,</span> <span class=n>encoding</span><span class=o>=</span><span class=s1>&#39;cp949&#39;</span><span class=p>)</span>
</span></span><span class=line><span class=cl><span class=n>count_hs</span><span class=o>.</span><span class=n>columns</span> <span class=o>=</span> <span class=p>[</span><span class=s2>&#34;transcript_id&#34;</span><span class=p>,</span> <span class=s2>&#34;gene_id&#34;</span><span class=p>,</span> <span class=s2>&#34;gene_name&#34;</span><span class=p>,</span> <span class=s2>&#34;33-1&#34;</span><span class=p>,</span> <span class=s2>&#34;33-2&#34;</span><span class=p>,</span> <span class=s2>&#34;33-3&#34;</span><span class=p>,</span> <span class=s2>&#34;150-1&#34;</span><span class=p>,</span> <span class=s2>&#34;150-2&#34;</span><span class=p>,</span> <span class=s2>&#34;150-3&#34;</span><span class=p>,</span> <span class=s2>&#34;con-1&#34;</span><span class=p>,</span> <span class=s2>&#34;con-2&#34;</span><span class=p>,</span> <span class=s2>&#34;con-3&#34;</span><span class=p>]</span>
</span></span><span class=line><span class=cl>
</span></span><span class=line><span class=cl><span class=n>combined_df</span> <span class=o>=</span> <span class=n>pd</span><span class=o>.</span><span class=n>concat</span><span class=p>([</span><span class=n>res_df</span><span class=p>,</span> <span class=n>count_hs</span><span class=p>],</span> <span class=n>axis</span><span class=o>=</span><span class=mi>0</span><span class=p>,</span> <span class=n>ignore_index</span><span class=o>=</span><span class=kc>True</span><span class=p>)</span>
</span></span><span class=line><span class=cl><span class=n>combined_df</span>
</span></span><span class=line><span class=cl><span class=n>combined_drop_df</span> <span class=o>=</span> <span class=n>combined_df</span><span class=p>[</span><span class=o>~</span><span class=p>(</span><span class=n>combined_df</span><span class=o>.</span><span class=n>iloc</span><span class=p>[:,</span> <span class=mi>4</span><span class=p>:</span><span class=mi>13</span><span class=p>]</span> <span class=o>==</span> <span class=mi>0</span><span class=p>)</span><span class=o>.</span><span class=n>all</span><span class=p>(</span><span class=n>axis</span><span class=o>=</span><span class=mi>1</span><span class=p>)]</span> <span class=c1>#remove 0 in all samples </span>
</span></span><span class=line><span class=cl><span class=n>combined_drop_df</span>
</span></span></code></pre></div><p><img src=https://github.com/user-attachments/assets/873ee0af-9bfc-4d11-bae7-ad61df069553 alt=image></p><p><img src=https://github.com/user-attachments/assets/f6613d7c-464a-48f7-9c51-eb4b461c354d alt=image></p><div class=highlight><pre tabindex=0 class=chroma><code class=language-python data-lang=python><span class=line><span class=cl><span class=n>combined_drop_df</span><span class=o>.</span><span class=n>to_csv</span><span class=p>(</span><span class=s2>&#34;count.annot.combined.drop.tsv&#34;</span><span class=p>,</span><span class=n>sep</span><span class=o>=</span><span class=s1>&#39;</span><span class=se>\t</span><span class=s1>&#39;</span><span class=p>,</span><span class=n>index</span><span class=o>=</span><span class=kc>False</span><span class=p>)</span>
</span></span></code></pre></div><h2 id=3-deg-analysis>3. DEG Analysis
<a class=anchor href=#3-deg-analysis>#</a></h2><h3 id=load-packages>Load Packages
<a class=anchor href=#load-packages>#</a></h3><div class=highlight><pre tabindex=0 class=chroma><code class=language-R data-lang=R><span class=line><span class=cl><span class=nf>suppressMessages</span><span class=p>({</span>
</span></span><span class=line><span class=cl>    <span class=nf>library</span><span class=p>(</span><span class=s>&#34;DESeq2&#34;</span><span class=p>)</span>
</span></span><span class=line><span class=cl>    <span class=nf>library</span><span class=p>(</span><span class=n>pheatmap</span><span class=p>)</span>
</span></span><span class=line><span class=cl>    <span class=nf>library</span><span class=p>(</span><span class=n>withr</span><span class=p>)</span>
</span></span><span class=line><span class=cl>    <span class=nf>library</span><span class=p>(</span><span class=n>RColorBrewer</span><span class=p>)</span>
</span></span><span class=line><span class=cl>    <span class=nf>library</span><span class=p>(</span><span class=n>gplots</span><span class=p>)</span>
</span></span><span class=line><span class=cl>    <span class=nf>library</span><span class=p>(</span><span class=n>ggplot2</span><span class=p>)</span>
</span></span><span class=line><span class=cl>    <span class=nf>library</span><span class=p>(</span><span class=n>dplyr</span><span class=p>)</span>
</span></span><span class=line><span class=cl>    <span class=p>})</span>
</span></span></code></pre></div><h3 id=set-path-1>Set Path
<a class=anchor href=#set-path-1>#</a></h3><div class=highlight><pre tabindex=0 class=chroma><code class=language-R data-lang=R><span class=line><span class=cl><span class=nf>setwd</span><span class=p>(</span><span class=s>&#34;/data/home/ysh980101/2311/RNA-seq_ebv/Rsubread&#34;</span><span class=p>)</span>
</span></span><span class=line><span class=cl><span class=nf>getwd</span><span class=p>()</span>
</span></span></code></pre></div><div class=highlight><pre tabindex=0 class=chroma><code class=language-plain data-lang=plain><span class=line><span class=cl>&#39;/data1/home/ysh980101/2311/RNA-seq_ebv/Rsubread&#39;
</span></span></code></pre></div><h3 id=run-deg>Run DEG
<a class=anchor href=#run-deg>#</a></h3><div class=highlight><pre tabindex=0 class=chroma><code class=language-R data-lang=R><span class=line><span class=cl><span class=n>S1</span> <span class=o>&lt;-</span> <span class=s>&#39;33&#39;</span>
</span></span><span class=line><span class=cl><span class=n>S2</span> <span class=o>&lt;-</span> <span class=s>&#39;150&#39;</span>
</span></span><span class=line><span class=cl>
</span></span><span class=line><span class=cl><span class=n>countdata</span> <span class=o>&lt;-</span> <span class=nf>read.csv</span><span class=p>(</span><span class=s>&#34;count.annot.combined.drop.tsv&#34;</span><span class=p>,</span> <span class=n>header</span><span class=o>=</span><span class=kc>TRUE</span><span class=p>,</span> <span class=n>sep</span><span class=o>=</span><span class=s>&#39;\t&#39;</span><span class=p>)</span>
</span></span><span class=line><span class=cl><span class=nf>colnames</span><span class=p>(</span><span class=n>countdata</span><span class=p>)</span> <span class=o>&lt;-</span> <span class=nf>c</span><span class=p>(</span><span class=s>&#39;transcript_id&#39;</span><span class=p>,</span><span class=s>&#39;gene_id&#39;</span><span class=p>,</span><span class=s>&#39;gene_name&#39;</span><span class=p>,</span><span class=s>&#39;33-1&#39;</span><span class=p>,</span><span class=s>&#39;33-2&#39;</span><span class=p>,</span><span class=s>&#39;33-3&#39;</span><span class=p>,</span><span class=s>&#39;150-1&#39;</span><span class=p>,</span><span class=s>&#39;150-2&#39;</span><span class=p>,</span><span class=s>&#39;150-3&#39;</span><span class=p>,</span><span class=s>&#39;con-1&#39;</span><span class=p>,</span><span class=s>&#39;con-2&#39;</span><span class=p>,</span><span class=s>&#39;con-3&#39;</span><span class=p>)</span>
</span></span><span class=line><span class=cl><span class=n>countdata</span> <span class=o>&lt;-</span> <span class=n>countdata[</span><span class=p>,</span> <span class=nf>c</span><span class=p>(</span><span class=s>&#39;transcript_id&#39;</span><span class=p>,</span><span class=s>&#39;gene_id&#39;</span><span class=p>,</span><span class=s>&#39;gene_name&#39;</span><span class=p>,</span><span class=s>&#39;150-1&#39;</span><span class=p>,</span><span class=s>&#39;150-2&#39;</span><span class=p>,</span><span class=s>&#39;150-3&#39;</span><span class=p>,</span><span class=s>&#39;33-1&#39;</span><span class=p>,</span><span class=s>&#39;33-2&#39;</span><span class=p>,</span><span class=s>&#39;33-3&#39;</span><span class=p>,</span><span class=s>&#39;con-1&#39;</span><span class=p>,</span><span class=s>&#39;con-2&#39;</span><span class=p>,</span><span class=s>&#39;con-3&#39;</span><span class=p>)</span><span class=n>]</span>
</span></span><span class=line><span class=cl><span class=n>countdata</span> <span class=o>&lt;-</span> <span class=n>countdata[</span><span class=p>,</span> <span class=nf>paste</span><span class=p>(</span><span class=nf>c</span><span class=p>(</span><span class=s>&#39;gene_name&#39;</span><span class=p>,</span><span class=nf>paste0</span><span class=p>(</span><span class=n>S2</span><span class=p>,</span><span class=s>&#34;-1&#34;</span><span class=p>),</span> <span class=nf>paste0</span><span class=p>(</span><span class=n>S2</span><span class=p>,</span><span class=s>&#34;-2&#34;</span><span class=p>),</span> <span class=nf>paste0</span><span class=p>(</span><span class=n>S2</span><span class=p>,</span><span class=s>&#34;-3&#34;</span><span class=p>),</span><span class=nf>paste0</span><span class=p>(</span><span class=n>S1</span><span class=p>,</span><span class=s>&#34;-1&#34;</span><span class=p>),</span> <span class=nf>paste0</span><span class=p>(</span><span class=n>S1</span><span class=p>,</span><span class=s>&#34;-2&#34;</span><span class=p>),</span> <span class=nf>paste0</span><span class=p>(</span><span class=n>S1</span><span class=p>,</span><span class=s>&#34;-3&#34;</span><span class=p>)),</span> <span class=n>sep</span><span class=o>=</span><span class=s>&#34;&#34;</span><span class=p>)</span><span class=n>]</span>
</span></span><span class=line><span class=cl><span class=nf>dim</span><span class=p>(</span><span class=n>countdata</span><span class=p>)</span>
</span></span></code></pre></div><div class=highlight><pre tabindex=0 class=chroma><code class=language-plain data-lang=plain><span class=line><span class=cl>16409 * 7
</span></span></code></pre></div><div class=highlight><pre tabindex=0 class=chroma><code class=language-R data-lang=R><span class=line><span class=cl><span class=n>sample.names</span> <span class=o>&lt;-</span> <span class=nf>paste</span><span class=p>(</span><span class=nf>c</span><span class=p>(</span><span class=nf>paste0</span><span class=p>(</span><span class=n>S2</span><span class=p>,</span><span class=s>&#34;-1&#34;</span><span class=p>),</span> <span class=nf>paste0</span><span class=p>(</span><span class=n>S2</span><span class=p>,</span><span class=s>&#34;-2&#34;</span><span class=p>),</span> <span class=nf>paste0</span><span class=p>(</span><span class=n>S2</span><span class=p>,</span><span class=s>&#34;-3&#34;</span><span class=p>),</span><span class=nf>paste0</span><span class=p>(</span><span class=n>S1</span><span class=p>,</span><span class=s>&#34;-1&#34;</span><span class=p>),</span> <span class=nf>paste0</span><span class=p>(</span><span class=n>S1</span><span class=p>,</span><span class=s>&#34;-2&#34;</span><span class=p>),</span> <span class=nf>paste0</span><span class=p>(</span><span class=n>S1</span><span class=p>,</span><span class=s>&#34;-3&#34;</span><span class=p>)),</span> <span class=n>sep</span><span class=o>=</span><span class=s>&#34;&#34;</span><span class=p>)</span>
</span></span><span class=line><span class=cl><span class=n>conditions</span> <span class=o>&lt;-</span> <span class=nf>factor</span><span class=p>(</span><span class=nf>c</span><span class=p>(</span><span class=n>S2</span><span class=p>,</span><span class=n>S2</span><span class=p>,</span><span class=n>S2</span><span class=p>,</span><span class=n>S1</span><span class=p>,</span><span class=n>S1</span><span class=p>,</span><span class=n>S1</span><span class=p>))</span>
</span></span><span class=line><span class=cl><span class=n>metadata</span> <span class=o>&lt;-</span> <span class=nf>data.frame</span><span class=p>(</span><span class=n>Sample</span> <span class=o>=</span> <span class=n>sample.names</span><span class=p>,</span> <span class=n>group</span> <span class=o>=</span> <span class=n>conditions</span><span class=p>)</span>
</span></span><span class=line><span class=cl>
</span></span><span class=line><span class=cl><span class=n>N</span> <span class=o>&lt;-</span> <span class=nf>dim</span><span class=p>(</span><span class=n>countdata</span><span class=p>)</span><span class=n>[[2]]</span>
</span></span><span class=line><span class=cl><span class=n>cData</span> <span class=o>=</span> <span class=n>countdata[</span><span class=p>,</span><span class=m>2</span><span class=o>:</span><span class=n>N]</span>
</span></span><span class=line><span class=cl><span class=n>GeneID</span> <span class=o>=</span> <span class=n>countdata[</span><span class=p>,</span><span class=m>1</span><span class=n>]</span>
</span></span><span class=line><span class=cl><span class=nf>rownames</span><span class=p>(</span><span class=n>cData</span><span class=p>)</span> <span class=o>=</span> <span class=n>GeneID</span>
</span></span><span class=line><span class=cl>
</span></span><span class=line><span class=cl><span class=n>dds</span> <span class=o>&lt;-</span> <span class=nf>DESeqDataSetFromMatrix</span><span class=p>(</span><span class=n>countData</span> <span class=o>=</span> <span class=n>cData</span><span class=p>,</span> <span class=n>colData</span> <span class=o>=</span> <span class=n>metadata</span><span class=p>,</span> <span class=n>design</span> <span class=o>=</span> <span class=o>~</span><span class=n>group</span><span class=p>)</span>
</span></span><span class=line><span class=cl><span class=n>dds</span><span class=o>$</span><span class=n>group</span> <span class=o>&lt;-</span> <span class=nf>relevel</span><span class=p>(</span><span class=n>dds</span><span class=o>$</span><span class=n>group</span><span class=p>,</span> <span class=n>ref</span> <span class=o>=</span> <span class=n>S1</span><span class=p>)</span>
</span></span><span class=line><span class=cl><span class=n>dds</span> <span class=o>&lt;-</span> <span class=nf>DESeq</span><span class=p>(</span><span class=n>dds</span><span class=p>)</span>  
</span></span><span class=line><span class=cl><span class=n>vsd</span> <span class=o>&lt;-</span> <span class=nf>vst</span><span class=p>(</span><span class=n>dds</span><span class=p>,</span> <span class=n>blind</span> <span class=o>=</span> <span class=kc>FALSE</span><span class=p>)</span>
</span></span><span class=line><span class=cl><span class=n>rld</span> <span class=o>&lt;-</span> <span class=nf>rlogTransformation</span><span class=p>(</span><span class=n>dds</span><span class=p>,</span> <span class=n>blind</span> <span class=o>=</span> <span class=kc>FALSE</span><span class=p>)</span>
</span></span><span class=line><span class=cl>
</span></span><span class=line><span class=cl><span class=n>res</span> <span class=o>&lt;-</span> <span class=nf>results</span><span class=p>(</span><span class=n>dds</span><span class=p>,</span> <span class=n>contrast</span> <span class=o>=</span> <span class=nf>c</span><span class=p>(</span><span class=s>&#34;group&#34;</span><span class=p>,</span> <span class=n>S2</span><span class=p>,</span> <span class=n>S1</span><span class=p>))</span>
</span></span><span class=line><span class=cl><span class=n>res_tbl</span> <span class=o>&lt;-</span> <span class=nf>as.data.frame</span><span class=p>(</span><span class=n>res</span><span class=p>)</span>
</span></span><span class=line><span class=cl><span class=n>res_tbl</span><span class=o>$</span><span class=n>GeneID</span> <span class=o>&lt;-</span> <span class=nf>rownames</span><span class=p>(</span><span class=n>res_tbl</span><span class=p>)</span>
</span></span><span class=line><span class=cl><span class=n>res_tbl</span> <span class=o>&lt;-</span> <span class=n>res_tbl</span><span class=nf>[order</span><span class=p>(</span><span class=n>res_tbl</span><span class=o>$</span><span class=n>padj</span><span class=p>),</span> <span class=n>]</span>
</span></span><span class=line><span class=cl>
</span></span><span class=line><span class=cl><span class=n>NM_no_NA</span> <span class=o>&lt;-</span> <span class=nf>na.omit</span><span class=p>(</span><span class=n>res</span><span class=p>)</span>
</span></span><span class=line><span class=cl><span class=n>res_cut</span> <span class=o>&lt;-</span> <span class=n>NM_no_NA[NM_no_NA</span><span class=o>$</span><span class=n>padj</span> <span class=o>&lt;</span> <span class=m>0.05</span><span class=p>,</span> <span class=n>]</span>
</span></span><span class=line><span class=cl>
</span></span><span class=line><span class=cl><span class=n>lfc</span> <span class=o>&lt;-</span> <span class=m>1.5</span>
</span></span><span class=line><span class=cl><span class=n>res_cut_2</span> <span class=o>&lt;-</span> <span class=n>res_cut</span><span class=nf>[abs</span><span class=p>(</span><span class=n>res_cut</span><span class=o>$</span><span class=n>log2FoldChange</span><span class=p>)</span> <span class=o>&gt;</span> <span class=n>lfc</span><span class=p>,</span> <span class=n>]</span>
</span></span><span class=line><span class=cl><span class=n>res_cut_2_pos</span> <span class=o>&lt;-</span> <span class=n>res_cut_2[res_cut_2</span><span class=o>$</span><span class=n>log2FoldChange</span> <span class=o>&gt;</span> <span class=m>0</span><span class=p>,</span> <span class=n>]</span>
</span></span><span class=line><span class=cl><span class=n>res_cut_2_neg</span> <span class=o>&lt;-</span> <span class=n>res_cut_2[res_cut_2</span><span class=o>$</span><span class=n>log2FoldChange</span> <span class=o>&lt;</span> <span class=m>0</span><span class=p>,</span> <span class=n>]</span>
</span></span></code></pre></div><div class=highlight><pre tabindex=0 class=chroma><code class=language-R data-lang=R><span class=line><span class=cl><span class=c1># Volcano Plot</span>
</span></span><span class=line><span class=cl><span class=nf>par</span><span class=p>(</span><span class=n>mfrow</span> <span class=o>=</span> <span class=nf>c</span><span class=p>(</span><span class=m>1</span><span class=p>,</span> <span class=m>1</span><span class=p>))</span>
</span></span><span class=line><span class=cl><span class=nf>with</span><span class=p>(</span><span class=n>NM_no_NA</span><span class=p>,</span> <span class=nf>plot</span><span class=p>(</span><span class=n>log2FoldChange</span><span class=p>,</span> <span class=o>-</span><span class=nf>log10</span><span class=p>(</span><span class=n>pvalue</span><span class=p>),</span> <span class=n>pch</span> <span class=o>=</span> <span class=m>20</span><span class=p>,</span> <span class=n>col</span> <span class=o>=</span> <span class=s>&#39;gray&#39;</span><span class=p>,</span>
</span></span><span class=line><span class=cl>                    <span class=n>main</span> <span class=o>=</span> <span class=nf>paste0</span><span class=p>(</span><span class=s>&#39;Volcano plot_between\n&#39;</span><span class=p>,</span> <span class=n>S2</span><span class=p>,</span> <span class=s>&#39; vs &#39;</span><span class=p>,</span> <span class=n>S1</span><span class=p>),</span>
</span></span><span class=line><span class=cl>                    <span class=n>xlim</span> <span class=o>=</span> <span class=nf>c</span><span class=p>(</span><span class=m>-3</span><span class=p>,</span> <span class=m>3</span><span class=p>)))</span>
</span></span><span class=line><span class=cl><span class=nf>abline</span><span class=p>(</span><span class=n>h</span> <span class=o>=</span> <span class=o>-</span><span class=nf>log10</span><span class=p>(</span><span class=m>0.05</span><span class=p>),</span> <span class=n>v</span> <span class=o>=</span> <span class=nf>c</span><span class=p>(</span><span class=o>-</span><span class=n>lfc</span><span class=p>,</span> <span class=n>lfc</span><span class=p>),</span> <span class=n>col</span> <span class=o>=</span> <span class=s>&#34;gray&#34;</span><span class=p>,</span> <span class=n>lty</span> <span class=o>=</span> <span class=m>2</span><span class=p>)</span>
</span></span><span class=line><span class=cl><span class=nf>with</span><span class=p>(</span><span class=n>res_cut_2_neg</span><span class=p>,</span> <span class=nf>points</span><span class=p>(</span><span class=n>log2FoldChange</span><span class=p>,</span> <span class=o>-</span><span class=nf>log10</span><span class=p>(</span><span class=n>pvalue</span><span class=p>),</span> <span class=n>pch</span> <span class=o>=</span> <span class=m>20</span><span class=p>,</span> <span class=n>col</span> <span class=o>=</span> <span class=s>&#39;blue&#39;</span><span class=p>))</span>
</span></span><span class=line><span class=cl><span class=nf>with</span><span class=p>(</span><span class=n>res_cut_2_pos</span><span class=p>,</span> <span class=nf>points</span><span class=p>(</span><span class=n>log2FoldChange</span><span class=p>,</span> <span class=o>-</span><span class=nf>log10</span><span class=p>(</span><span class=n>pvalue</span><span class=p>),</span> <span class=n>pch</span> <span class=o>=</span> <span class=m>20</span><span class=p>,</span> <span class=n>col</span> <span class=o>=</span> <span class=s>&#39;gold&#39;</span><span class=p>))</span>
</span></span><span class=line><span class=cl><span class=nf>legend</span><span class=p>(</span><span class=s>&#34;topright&#34;</span><span class=p>,</span> <span class=n>legend</span> <span class=o>=</span> <span class=nf>c</span><span class=p>(</span><span class=nf>paste0</span><span class=p>(</span><span class=s>&#34;FC ≥ &#34;</span><span class=p>,</span> <span class=n>lfc</span><span class=p>,</span> <span class=s>&#34; &amp; p &lt; 0.05&#34;</span><span class=p>),</span>
</span></span><span class=line><span class=cl>                              <span class=nf>paste0</span><span class=p>(</span><span class=s>&#34;FC ≤ -&#34;</span><span class=p>,</span> <span class=n>lfc</span><span class=p>,</span> <span class=s>&#34; &amp; p &lt; 0.05&#34;</span><span class=p>)),</span>
</span></span><span class=line><span class=cl>       <span class=n>col</span> <span class=o>=</span> <span class=nf>c</span><span class=p>(</span><span class=s>&#34;gold&#34;</span><span class=p>,</span> <span class=s>&#34;blue&#34;</span><span class=p>),</span> <span class=n>pch</span> <span class=o>=</span> <span class=m>20</span><span class=p>)</span>
</span></span><span class=line><span class=cl>
</span></span><span class=line><span class=cl><span class=c1># Bar Plot</span>
</span></span><span class=line><span class=cl><span class=n>lengths</span> <span class=o>&lt;-</span> <span class=nf>c</span><span class=p>(</span><span class=nf>nrow</span><span class=p>(</span><span class=n>res_cut_2_neg</span><span class=p>),</span> <span class=nf>nrow</span><span class=p>(</span><span class=n>res_cut_2_pos</span><span class=p>))</span>
</span></span><span class=line><span class=cl><span class=n>bp</span> <span class=o>&lt;-</span> <span class=nf>barplot</span><span class=p>(</span><span class=n>lengths</span><span class=p>,</span> <span class=n>space</span> <span class=o>=</span> <span class=m>0</span><span class=p>,</span> <span class=n>col</span> <span class=o>=</span> <span class=nf>c</span><span class=p>(</span><span class=s>&#34;blue&#34;</span><span class=p>,</span> <span class=s>&#34;gold&#34;</span><span class=p>),</span>
</span></span><span class=line><span class=cl>              <span class=n>horiz</span> <span class=o>=</span> <span class=kc>TRUE</span><span class=p>,</span> <span class=n>xlab</span> <span class=o>=</span> <span class=s>&#34;Count of genes&#34;</span><span class=p>,</span> <span class=n>ylab</span> <span class=o>=</span> <span class=nf>paste0</span><span class=p>(</span><span class=n>S2</span><span class=p>,</span> <span class=s>&#34;/&#34;</span><span class=p>,</span> <span class=n>S1</span><span class=p>),</span>
</span></span><span class=line><span class=cl>              <span class=n>xlim</span> <span class=o>=</span> <span class=nf>c</span><span class=p>(</span><span class=m>0</span><span class=p>,</span> <span class=nf>sum</span><span class=p>(</span><span class=n>lengths</span><span class=p>)),</span> <span class=n>width</span> <span class=o>=</span> <span class=m>0.02</span><span class=p>,</span> <span class=n>las</span> <span class=o>=</span> <span class=m>1</span><span class=p>)</span>
</span></span><span class=line><span class=cl><span class=nf>title</span><span class=p>(</span><span class=nf>paste0</span><span class=p>(</span><span class=s>&#34;UP, DOWN regulated count (|FC| ≥ &#34;</span><span class=p>,</span> <span class=n>lfc</span><span class=p>,</span> <span class=s>&#34;)&#34;</span><span class=p>))</span>
</span></span><span class=line><span class=cl><span class=nf>legend</span><span class=p>(</span><span class=s>&#34;topright&#34;</span><span class=p>,</span> <span class=n>legend</span> <span class=o>=</span> <span class=nf>c</span><span class=p>(</span><span class=s>&#34;UP&#34;</span><span class=p>,</span> <span class=s>&#34;DOWN&#34;</span><span class=p>),</span> <span class=n>fill</span> <span class=o>=</span> <span class=nf>c</span><span class=p>(</span><span class=s>&#34;gold&#34;</span><span class=p>,</span> <span class=s>&#34;blue&#34;</span><span class=p>))</span>
</span></span><span class=line><span class=cl><span class=nf>text</span><span class=p>(</span><span class=n>lengths</span><span class=p>,</span> <span class=n>bp</span><span class=p>,</span> <span class=n>labels</span> <span class=o>=</span> <span class=n>lengths</span><span class=p>,</span> <span class=n>pos</span> <span class=o>=</span> <span class=m>4</span><span class=p>,</span> <span class=n>cex</span> <span class=o>=</span> <span class=m>0.9</span><span class=p>)</span>
</span></span><span class=line><span class=cl>
</span></span><span class=line><span class=cl><span class=c1># Save</span>
</span></span><span class=line><span class=cl><span class=n>cutoff</span> <span class=o>&lt;-</span> <span class=m>0.05</span>
</span></span><span class=line><span class=cl><span class=n>sig_res</span> <span class=o>&lt;-</span> <span class=n>dplyr</span><span class=o>::</span><span class=nf>filter</span><span class=p>(</span><span class=n>res_tbl</span><span class=p>,</span> <span class=n>padj</span> <span class=o>&lt;</span> <span class=n>cutoff</span><span class=p>)</span> <span class=o>%&gt;%</span>
</span></span><span class=line><span class=cl>  <span class=n>dplyr</span><span class=o>::</span><span class=nf>arrange</span><span class=p>(</span><span class=n>padj</span><span class=p>)</span> <span class=o>%&gt;%</span>
</span></span><span class=line><span class=cl>  <span class=n>dplyr</span><span class=o>::</span><span class=nf>select</span><span class=p>(</span><span class=n>GeneID</span><span class=p>,</span> <span class=n>baseMean</span><span class=p>,</span> <span class=n>log2FoldChange</span><span class=p>,</span> <span class=n>lfcSE</span><span class=p>,</span> <span class=n>stat</span><span class=p>,</span> <span class=n>pvalue</span><span class=p>,</span> <span class=n>padj</span><span class=p>)</span>
</span></span><span class=line><span class=cl>
</span></span><span class=line><span class=cl><span class=nf>write.csv</span><span class=p>(</span><span class=n>sig_res</span><span class=p>,</span> <span class=n>file</span> <span class=o>=</span> <span class=nf>paste0</span><span class=p>(</span><span class=s>&#39;../DESeq2/res.&#39;</span><span class=p>,</span> <span class=n>S2</span><span class=p>,</span> <span class=s>&#39;vs&#39;</span><span class=p>,</span> <span class=n>S1</span><span class=p>,</span> <span class=s>&#39;.csv&#39;</span><span class=p>),</span> <span class=n>row.names</span> <span class=o>=</span> <span class=kc>FALSE</span><span class=p>)</span>
</span></span></code></pre></div><p><img src=https://github.com/user-attachments/assets/2e32b5e8-7d44-4b2d-80dd-54106bc300cf alt=image></p><p><img src=https://github.com/user-attachments/assets/d3fa3953-36b0-4a7e-946e-3a24a50a4891 alt=image></p><p>cf) Heatmap 시각화</p><div class=highlight><pre tabindex=0 class=chroma><code class=language-R data-lang=R><span class=line><span class=cl><span class=c1># Heatmap</span>
</span></span><span class=line><span class=cl><span class=n>sig_idx</span> <span class=o>&lt;-</span> <span class=n>res</span><span class=o>$</span><span class=n>padj</span> <span class=o>&lt;</span> <span class=n>cutoff</span> <span class=o>&amp;</span> <span class=o>!</span><span class=nf>is.na</span><span class=p>(</span><span class=n>res</span><span class=o>$</span><span class=n>padj</span><span class=p>)</span>
</span></span><span class=line><span class=cl><span class=n>sig_dat</span> <span class=o>&lt;-</span> <span class=nf>assay</span><span class=p>(</span><span class=n>rld</span><span class=p>)</span><span class=n>[sig_idx</span><span class=p>,</span> <span class=n>]</span>
</span></span><span class=line><span class=cl>
</span></span><span class=line><span class=cl><span class=n>annC</span> <span class=o>&lt;-</span> <span class=nf>data.frame</span><span class=p>(</span><span class=n>condition</span> <span class=o>=</span> <span class=n>conditions</span><span class=p>)</span>
</span></span><span class=line><span class=cl><span class=nf>rownames</span><span class=p>(</span><span class=n>annC</span><span class=p>)</span> <span class=o>&lt;-</span> <span class=nf>colnames</span><span class=p>(</span><span class=n>sig_dat</span><span class=p>)</span>
</span></span><span class=line><span class=cl><span class=n>ann_colors</span> <span class=o>&lt;-</span> <span class=nf>list</span><span class=p>(</span><span class=n>condition</span> <span class=o>=</span> <span class=nf>setNames</span><span class=p>(</span><span class=nf>c</span><span class=p>(</span><span class=s>&#34;#F7819F&#34;</span><span class=p>,</span> <span class=s>&#34;#58D3F7&#34;</span><span class=p>),</span> <span class=nf>c</span><span class=p>(</span><span class=n>S2</span><span class=p>,</span> <span class=n>S1</span><span class=p>)))</span>
</span></span><span class=line><span class=cl>
</span></span><span class=line><span class=cl><span class=nf>pheatmap</span><span class=p>(</span><span class=n>sig_dat</span><span class=p>,</span>
</span></span><span class=line><span class=cl>         <span class=n>scale</span> <span class=o>=</span> <span class=s>&#34;row&#34;</span><span class=p>,</span>
</span></span><span class=line><span class=cl>         <span class=n>fontsize_row</span> <span class=o>=</span> <span class=m>9</span><span class=p>,</span>
</span></span><span class=line><span class=cl>         <span class=n>annotation_col</span> <span class=o>=</span> <span class=n>annC</span><span class=p>,</span>
</span></span><span class=line><span class=cl>         <span class=n>color</span> <span class=o>=</span> <span class=nf>rev</span><span class=p>(</span><span class=nf>brewer.pal</span><span class=p>(</span><span class=m>6</span><span class=p>,</span> <span class=s>&#34;RdYlGn&#34;</span><span class=p>)),</span>
</span></span><span class=line><span class=cl>         <span class=n>annotation_colors</span> <span class=o>=</span> <span class=n>ann_colors</span><span class=p>,</span>
</span></span><span class=line><span class=cl>         <span class=n>show_rownames</span> <span class=o>=</span> <span class=kc>FALSE</span><span class=p>,</span>
</span></span><span class=line><span class=cl>         <span class=n>show_colnames</span> <span class=o>=</span> <span class=kc>TRUE</span><span class=p>)</span>
</span></span></code></pre></div><p><img src=https://github.com/user-attachments/assets/67e7e43f-4db6-48c2-9fed-fc03dfb4e892 alt=image></p></article><footer class=book-footer><div class="flex flex-wrap justify-between"></div><script>(function(){function e(e){const t=window.getSelection(),n=document.createRange();n.selectNodeContents(e),t.removeAllRanges(),t.addRange(n)}document.querySelectorAll("pre code").forEach(t=>{t.addEventListener("click",function(){if(window.getSelection().toString())return;e(t.parentElement),navigator.clipboard&&navigator.clipboard.writeText(t.parentElement.textContent)})})})()</script></footer><div class=book-comments><script src=https://giscus.app/client.js data-repo=yshghid/yshghid.github.io data-repo-id=R_kgDONkMkNg data-category-id=DIC_kwDONkMkNs4CloJh data-mapping=pathname data-strict=0 data-reactions-enabled=1 data-emit-metadata=0 data-input-position=bottom data-theme=preferred_color_scheme data-lang=ko crossorigin=anonymous async></script></div><label for=menu-control class="hidden book-menu-overlay"></label></div><aside class=book-toc><div class=book-toc-content><nav id=TableOfContents><ul><li><a href=#1-alignment>1. Alignment</a><ul><li><a href=#load-package-set-path>Load package, Set Path</a></li><li><a href=#build-index>Build Index</a></li><li><a href=#feature-count>Feature Count</a></li><li><a href=#save>Save</a></li></ul></li><li><a href=#2-annotation>2. Annotation</a><ul><li><a href=#load-package>Load Package</a></li><li><a href=#set-path>Set Path</a></li><li><a href=#load-gtf-file>Load GTF file</a></li><li><a href=#make-annotation-file>Make annotation file</a></li><li><a href=#save-1>Save</a></li><li><a href=#load-count>Load Count</a></li><li><a href=#annotation>Annotation</a></li><li><a href=#merge>Merge</a></li></ul></li><li><a href=#3-deg-analysis>3. DEG Analysis</a><ul><li><a href=#load-packages>Load Packages</a></li><li><a href=#set-path-1>Set Path</a></li><li><a href=#run-deg>Run DEG</a></li></ul></li></ul></nav></div></aside></main></body></html>