<!doctype html><html lang=en-us dir=ltr><head><meta charset=UTF-8><meta name=viewport content="width=device-width,initial-scale=1"><meta name=description content='
  EBV 유전체 RNA-seq 전처리
  #

#2024-12-31



분석 목적

제공받은 fastq를 human genome에 매핑해서 전처리, 분석 후 DE 결과 보냄
DE 분석시에 EBV 유전자도 포함해달라는 요청



해야하는것

fastq를 EBV genome에 매핑해서 전처리, EBV count 생성
human count에 EBV count를 붙이기
통합 count로 DE 분석 재수행




  1. Alignment
  #


  Load package, Set Path
  #

library(edgeR)
library(Rsubread)
library(org.Hs.eg.db)

setwd("/data/home/ysh980101/2311/RNA-seq_ebv/Rsubread")
getwd()
&#39;/data1/home/ysh980101/2311/RNA-seq_ebv/Rsubread&#39;

  Build Index
  #

# build index
ref <- "/data3/PUBLIC_DATA/ref_genomes/Human_gammaherpesvirus_4_EBV/NC_007605.1.fa"
output_basename <- "NC_007605.1_idx"
buildindex(basename = output_basename, reference = ref)

  Feature Count
  #

# feature.count
targets <- read.delim("targets.txt", header=TRUE)
output.files <- c(targets$FileName)

fc <- featureCounts(output.files, annot.inbuilt=FALSE, annot.ext = "/data3/PUBLIC_DATA/ref_genomes/Human_gammaherpesvirus_4_EBV/NC_007605.1.gtf", isGTFAnnotationFile = TRUE, 
                    GTF.featureType = "exon", GTF.attrType = "transcript_id", GTF.attrType.extra = NULL, isPairedEnd=TRUE, countReadPairs=TRUE)# nthreads=30)

  Save
  #

colnames(fc$counts) <- c(&#39;33-1&#39;,&#39;33-2&#39;,&#39;33-3&#39;,&#39;150-1&#39;,&#39;150-2&#39;,&#39;150-3&#39;,&#39;con-1&#39;,&#39;con-2&#39;,&#39;con-3&#39;)
head(fc$counts)

group <- factor(targets$Status)
y <- DGEList(fc$counts, group=group)
fc$sort.counts <- fc$counts[order(rownames(fc$counts)), ]

sort.counts.df <- as.data.frame(fc$sort.counts)
sort.counts.df$trans_id <- rownames(fc$sort.counts)
sort.counts.df <- sort.counts.df[, c("trans_id", "33-1", "33-2", "33-3", "150-1", "150-2", "150-3", "con-1", "con-2", "con-3")]

write.table(sort.counts.df, file = "count.tsv", sep = "\t", quote = FALSE, row.names = FALSE)

  2. Annotation
  #


  Load Package
  #

import os
import pandas as pd

  Set Path
  #

os.chdir("/data/home/ysh980101/2311/RNA-seq_ebv/Rsubread")
os.getcwd()
&#39;/data1/home/ysh980101/2311/RNA-seq_ebv/Rsubread&#39;

  Load GTF file
  #


annotation 파일이 없어서 gtf reference 파일을 사용해서 직접 만들어줫다.

gtf_df = pd.read_csv("/data3/PUBLIC_DATA/ref_genomes/Human_gammaherpesvirus_4_EBV/NC_007605.1.gtf", sep=&#39;\t&#39;, header=None, comment="#", names=["seqname", "source", "feature", "start", "end", "score", "strand", "frame", "attribute"])
gtf_df
'><meta name=theme-color media="(prefers-color-scheme: light)" content="#ffffff"><meta name=theme-color media="(prefers-color-scheme: dark)" content="#343a40"><meta name=color-scheme content="light dark"><meta property="og:url" content="https://yshghid.github.io/docs/study/bioinformatics/bi11/"><meta property="og:site_name" content="Lifelog 2025"><meta property="og:title" content="EBV 유전체 RNA-seq 전처리"><meta property="og:description" content='EBV 유전체 RNA-seq 전처리 # #2024-12-31
분석 목적
제공받은 fastq를 human genome에 매핑해서 전처리, 분석 후 DE 결과 보냄 DE 분석시에 EBV 유전자도 포함해달라는 요청 해야하는것
fastq를 EBV genome에 매핑해서 전처리, EBV count 생성 human count에 EBV count를 붙이기 통합 count로 DE 분석 재수행 1. Alignment # Load package, Set Path # library(edgeR) library(Rsubread) library(org.Hs.eg.db) setwd("/data/home/ysh980101/2311/RNA-seq_ebv/Rsubread") getwd() &#39;/data1/home/ysh980101/2311/RNA-seq_ebv/Rsubread&#39; Build Index # # build index ref <- "/data3/PUBLIC_DATA/ref_genomes/Human_gammaherpesvirus_4_EBV/NC_007605.1.fa" output_basename <- "NC_007605.1_idx" buildindex(basename = output_basename, reference = ref) Feature Count # # feature.count targets <- read.delim("targets.txt", header=TRUE) output.files <- c(targets$FileName) fc <- featureCounts(output.files, annot.inbuilt=FALSE, annot.ext = "/data3/PUBLIC_DATA/ref_genomes/Human_gammaherpesvirus_4_EBV/NC_007605.1.gtf", isGTFAnnotationFile = TRUE, GTF.featureType = "exon", GTF.attrType = "transcript_id", GTF.attrType.extra = NULL, isPairedEnd=TRUE, countReadPairs=TRUE)# nthreads=30) Save # colnames(fc$counts) <- c(&#39;33-1&#39;,&#39;33-2&#39;,&#39;33-3&#39;,&#39;150-1&#39;,&#39;150-2&#39;,&#39;150-3&#39;,&#39;con-1&#39;,&#39;con-2&#39;,&#39;con-3&#39;) head(fc$counts) group <- factor(targets$Status) y <- DGEList(fc$counts, group=group) fc$sort.counts <- fc$counts[order(rownames(fc$counts)), ] sort.counts.df <- as.data.frame(fc$sort.counts) sort.counts.df$trans_id <- rownames(fc$sort.counts) sort.counts.df <- sort.counts.df[, c("trans_id", "33-1", "33-2", "33-3", "150-1", "150-2", "150-3", "con-1", "con-2", "con-3")] write.table(sort.counts.df, file = "count.tsv", sep = "\t", quote = FALSE, row.names = FALSE) 2. Annotation # Load Package # import os import pandas as pd Set Path # os.chdir("/data/home/ysh980101/2311/RNA-seq_ebv/Rsubread") os.getcwd() &#39;/data1/home/ysh980101/2311/RNA-seq_ebv/Rsubread&#39; Load GTF file # annotation 파일이 없어서 gtf reference 파일을 사용해서 직접 만들어줫다. gtf_df = pd.read_csv("/data3/PUBLIC_DATA/ref_genomes/Human_gammaherpesvirus_4_EBV/NC_007605.1.gtf", sep=&#39;\t&#39;, header=None, comment="#", names=["seqname", "source", "feature", "start", "end", "score", "strand", "frame", "attribute"]) gtf_df'><meta property="og:locale" content="en_us"><meta property="og:type" content="article"><meta property="article:section" content="docs"><meta property="article:published_time" content="2024-12-31T00:00:00+00:00"><meta property="article:modified_time" content="2024-12-31T00:00:00+00:00"><meta property="article:tag" content="2024-12"><title>EBV 유전체 RNA-seq 전처리 | Lifelog 2025</title><link rel=icon href=/favicon.png><link rel=manifest href=/manifest.json><link rel=canonical href=https://yshghid.github.io/docs/study/bioinformatics/bi11/><link rel=stylesheet href=/book.min.6217d077edb4189fd0578345e84bca1a884dfdee121ff8dc9a0f55cfe0852bc9.css integrity="sha256-YhfQd+20GJ/QV4NF6EvKGohN/e4SH/jcmg9Vz+CFK8k=" crossorigin=anonymous><script defer src=/fuse.min.js></script><script defer src=/en.search.min.5a2cb5bf69e0efa875e2bde3abb41d13655c4123fcd62868fe8b6e41c392d630.js integrity="sha256-Wiy1v2ng76h14r3jq7QdE2VcQSP81iho/otuQcOS1jA=" crossorigin=anonymous></script></head><body dir=ltr><input type=checkbox class="hidden toggle" id=menu-control>
<input type=checkbox class="hidden toggle" id=toc-control><main class="container flex"><aside class=book-menu><div class=book-menu-content><nav><h2 class=book-brand><a class="flex align-center" href=/><img src=/logo.png alt=Logo class=book-icon><span>Lifelog 2025</span></a></h2><div class="book-search hidden"><input type=text id=book-search-input placeholder=Search aria-label=Search maxlength=64 data-hotkeys=s/><div class="book-search-spinner hidden"></div><ul id=book-search-results></ul></div><script>document.querySelector(".book-search").classList.remove("hidden")</script><ul><li class=book-section-flat><span>기록</span><ul><li><a href=/docs/hobby/book/>독서</a><ul></ul></li><li><a href=/docs/hobby/daily/>일상</a><ul></ul></li><li><a href=/docs/hobby/etc/>기타</a><ul></ul></li></ul></li><li class=book-section-flat><span>공부</span><ul><li><a href=/docs/study/tech/>연구실</a><ul></ul></li><li><a href=/docs/study/career/>취업</a><ul></ul></li><li><a href=/docs/study/etc/>기타</a><ul></ul></li></ul></li></ul></nav><script>(function(){var e=document.querySelector("aside .book-menu-content");addEventListener("beforeunload",function(){localStorage.setItem("menu.scrollTop",e.scrollTop)}),e.scrollTop=localStorage.getItem("menu.scrollTop")})()</script></div></aside><div class=book-page><header class=book-header><div class="flex align-center justify-between"><label for=menu-control><img src=/svg/menu.svg class=book-icon alt=Menu></label><h3>EBV 유전체 RNA-seq 전처리</h3><label for=toc-control><img src=/svg/toc.svg class=book-icon alt="Table of Contents"></label></div><aside class="hidden clearfix"><nav id=TableOfContents><ul><li><a href=#1-alignment>1. Alignment</a><ul><li><a href=#load-package-set-path>Load package, Set Path</a></li><li><a href=#build-index>Build Index</a></li><li><a href=#feature-count>Feature Count</a></li><li><a href=#save>Save</a></li></ul></li><li><a href=#2-annotation>2. Annotation</a><ul><li><a href=#load-package>Load Package</a></li><li><a href=#set-path>Set Path</a></li><li><a href=#load-gtf-file>Load GTF file</a></li><li><a href=#make-annotation-file>Make annotation file</a></li><li><a href=#save-1>Save</a></li><li><a href=#load-count>Load Count</a></li><li><a href=#annotation>Annotation</a></li><li><a href=#merge>Merge</a></li></ul></li><li><a href=#3-deg-analysis>3. DEG Analysis</a><ul><li><a href=#load-packages>Load Packages</a></li><li><a href=#set-path-1>Set Path</a></li><li><a href=#run-deg>Run DEG</a></li></ul></li></ul></nav></aside></header><article class="markdown book-article"><h1 id=ebv-유전체-rna-seq-전처리>EBV 유전체 RNA-seq 전처리
<a class=anchor href=#ebv-%ec%9c%a0%ec%a0%84%ec%b2%b4-rna-seq-%ec%a0%84%ec%b2%98%eb%a6%ac>#</a></h1><p>#2024-12-31</p><hr><ul><li><p>분석 목적</p><ul><li>제공받은 fastq를 human genome에 매핑해서 전처리, 분석 후 DE 결과 보냄</li><li>DE 분석시에 EBV 유전자도 포함해달라는 요청</li></ul></li><li><p>해야하는것</p><ul><li>fastq를 EBV genome에 매핑해서 전처리, EBV count 생성</li><li>human count에 EBV count를 붙이기</li><li>통합 count로 DE 분석 재수행</li></ul></li></ul><h2 id=1-alignment>1. Alignment
<a class=anchor href=#1-alignment>#</a></h2><h3 id=load-package-set-path>Load package, Set Path
<a class=anchor href=#load-package-set-path>#</a></h3><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-R data-lang=R><span style=display:flex><span><span style=color:#a6e22e>library</span>(edgeR)
</span></span><span style=display:flex><span><span style=color:#a6e22e>library</span>(Rsubread)
</span></span><span style=display:flex><span><span style=color:#a6e22e>library</span>(org.Hs.eg.db)
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span><span style=color:#a6e22e>setwd</span>(<span style=color:#e6db74>&#34;/data/home/ysh980101/2311/RNA-seq_ebv/Rsubread&#34;</span>)
</span></span><span style=display:flex><span><span style=color:#a6e22e>getwd</span>()
</span></span></code></pre></div><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-plain data-lang=plain><span style=display:flex><span>&#39;/data1/home/ysh980101/2311/RNA-seq_ebv/Rsubread&#39;
</span></span></code></pre></div><h3 id=build-index>Build Index
<a class=anchor href=#build-index>#</a></h3><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-R data-lang=R><span style=display:flex><span><span style=color:#75715e># build index</span>
</span></span><span style=display:flex><span>ref <span style=color:#f92672>&lt;-</span> <span style=color:#e6db74>&#34;/data3/PUBLIC_DATA/ref_genomes/Human_gammaherpesvirus_4_EBV/NC_007605.1.fa&#34;</span>
</span></span><span style=display:flex><span>output_basename <span style=color:#f92672>&lt;-</span> <span style=color:#e6db74>&#34;NC_007605.1_idx&#34;</span>
</span></span><span style=display:flex><span><span style=color:#a6e22e>buildindex</span>(basename <span style=color:#f92672>=</span> output_basename, reference <span style=color:#f92672>=</span> ref)
</span></span></code></pre></div><h3 id=feature-count>Feature Count
<a class=anchor href=#feature-count>#</a></h3><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-R data-lang=R><span style=display:flex><span><span style=color:#75715e># feature.count</span>
</span></span><span style=display:flex><span>targets <span style=color:#f92672>&lt;-</span> <span style=color:#a6e22e>read.delim</span>(<span style=color:#e6db74>&#34;targets.txt&#34;</span>, header<span style=color:#f92672>=</span><span style=color:#66d9ef>TRUE</span>)
</span></span><span style=display:flex><span>output.files <span style=color:#f92672>&lt;-</span> <span style=color:#a6e22e>c</span>(targets<span style=color:#f92672>$</span>FileName)
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span>fc <span style=color:#f92672>&lt;-</span> <span style=color:#a6e22e>featureCounts</span>(output.files, annot.inbuilt<span style=color:#f92672>=</span><span style=color:#66d9ef>FALSE</span>, annot.ext <span style=color:#f92672>=</span> <span style=color:#e6db74>&#34;/data3/PUBLIC_DATA/ref_genomes/Human_gammaherpesvirus_4_EBV/NC_007605.1.gtf&#34;</span>, isGTFAnnotationFile <span style=color:#f92672>=</span> <span style=color:#66d9ef>TRUE</span>, 
</span></span><span style=display:flex><span>                    GTF.featureType <span style=color:#f92672>=</span> <span style=color:#e6db74>&#34;exon&#34;</span>, GTF.attrType <span style=color:#f92672>=</span> <span style=color:#e6db74>&#34;transcript_id&#34;</span>, GTF.attrType.extra <span style=color:#f92672>=</span> <span style=color:#66d9ef>NULL</span>, isPairedEnd<span style=color:#f92672>=</span><span style=color:#66d9ef>TRUE</span>, countReadPairs<span style=color:#f92672>=</span><span style=color:#66d9ef>TRUE</span>)<span style=color:#75715e># nthreads=30)</span>
</span></span></code></pre></div><h3 id=save>Save
<a class=anchor href=#save>#</a></h3><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-R data-lang=R><span style=display:flex><span><span style=color:#a6e22e>colnames</span>(fc<span style=color:#f92672>$</span>counts) <span style=color:#f92672>&lt;-</span> <span style=color:#a6e22e>c</span>(<span style=color:#e6db74>&#39;33-1&#39;</span>,<span style=color:#e6db74>&#39;33-2&#39;</span>,<span style=color:#e6db74>&#39;33-3&#39;</span>,<span style=color:#e6db74>&#39;150-1&#39;</span>,<span style=color:#e6db74>&#39;150-2&#39;</span>,<span style=color:#e6db74>&#39;150-3&#39;</span>,<span style=color:#e6db74>&#39;con-1&#39;</span>,<span style=color:#e6db74>&#39;con-2&#39;</span>,<span style=color:#e6db74>&#39;con-3&#39;</span>)
</span></span><span style=display:flex><span><span style=color:#a6e22e>head</span>(fc<span style=color:#f92672>$</span>counts)
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span>group <span style=color:#f92672>&lt;-</span> <span style=color:#a6e22e>factor</span>(targets<span style=color:#f92672>$</span>Status)
</span></span><span style=display:flex><span>y <span style=color:#f92672>&lt;-</span> <span style=color:#a6e22e>DGEList</span>(fc<span style=color:#f92672>$</span>counts, group<span style=color:#f92672>=</span>group)
</span></span><span style=display:flex><span>fc<span style=color:#f92672>$</span>sort.counts <span style=color:#f92672>&lt;-</span> fc<span style=color:#f92672>$</span>counts<span style=color:#a6e22e>[order</span>(<span style=color:#a6e22e>rownames</span>(fc<span style=color:#f92672>$</span>counts)), ]
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span>sort.counts.df <span style=color:#f92672>&lt;-</span> <span style=color:#a6e22e>as.data.frame</span>(fc<span style=color:#f92672>$</span>sort.counts)
</span></span><span style=display:flex><span>sort.counts.df<span style=color:#f92672>$</span>trans_id <span style=color:#f92672>&lt;-</span> <span style=color:#a6e22e>rownames</span>(fc<span style=color:#f92672>$</span>sort.counts)
</span></span><span style=display:flex><span>sort.counts.df <span style=color:#f92672>&lt;-</span> sort.counts.df[, <span style=color:#a6e22e>c</span>(<span style=color:#e6db74>&#34;trans_id&#34;</span>, <span style=color:#e6db74>&#34;33-1&#34;</span>, <span style=color:#e6db74>&#34;33-2&#34;</span>, <span style=color:#e6db74>&#34;33-3&#34;</span>, <span style=color:#e6db74>&#34;150-1&#34;</span>, <span style=color:#e6db74>&#34;150-2&#34;</span>, <span style=color:#e6db74>&#34;150-3&#34;</span>, <span style=color:#e6db74>&#34;con-1&#34;</span>, <span style=color:#e6db74>&#34;con-2&#34;</span>, <span style=color:#e6db74>&#34;con-3&#34;</span>)]
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span><span style=color:#a6e22e>write.table</span>(sort.counts.df, file <span style=color:#f92672>=</span> <span style=color:#e6db74>&#34;count.tsv&#34;</span>, sep <span style=color:#f92672>=</span> <span style=color:#e6db74>&#34;\t&#34;</span>, quote <span style=color:#f92672>=</span> <span style=color:#66d9ef>FALSE</span>, row.names <span style=color:#f92672>=</span> <span style=color:#66d9ef>FALSE</span>)
</span></span></code></pre></div><h2 id=2-annotation>2. Annotation
<a class=anchor href=#2-annotation>#</a></h2><h3 id=load-package>Load Package
<a class=anchor href=#load-package>#</a></h3><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-python data-lang=python><span style=display:flex><span><span style=color:#f92672>import</span> os
</span></span><span style=display:flex><span><span style=color:#f92672>import</span> pandas <span style=color:#66d9ef>as</span> pd
</span></span></code></pre></div><h3 id=set-path>Set Path
<a class=anchor href=#set-path>#</a></h3><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-python data-lang=python><span style=display:flex><span>os<span style=color:#f92672>.</span>chdir(<span style=color:#e6db74>&#34;/data/home/ysh980101/2311/RNA-seq_ebv/Rsubread&#34;</span>)
</span></span><span style=display:flex><span>os<span style=color:#f92672>.</span>getcwd()
</span></span></code></pre></div><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-plain data-lang=plain><span style=display:flex><span>&#39;/data1/home/ysh980101/2311/RNA-seq_ebv/Rsubread&#39;
</span></span></code></pre></div><h3 id=load-gtf-file>Load GTF file
<a class=anchor href=#load-gtf-file>#</a></h3><ul><li>annotation 파일이 없어서 gtf reference 파일을 사용해서 직접 만들어줫다.</li></ul><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-python data-lang=python><span style=display:flex><span>gtf_df <span style=color:#f92672>=</span> pd<span style=color:#f92672>.</span>read_csv(<span style=color:#e6db74>&#34;/data3/PUBLIC_DATA/ref_genomes/Human_gammaherpesvirus_4_EBV/NC_007605.1.gtf&#34;</span>, sep<span style=color:#f92672>=</span><span style=color:#e6db74>&#39;</span><span style=color:#ae81ff>\t</span><span style=color:#e6db74>&#39;</span>, header<span style=color:#f92672>=</span><span style=color:#66d9ef>None</span>, comment<span style=color:#f92672>=</span><span style=color:#e6db74>&#34;#&#34;</span>, names<span style=color:#f92672>=</span>[<span style=color:#e6db74>&#34;seqname&#34;</span>, <span style=color:#e6db74>&#34;source&#34;</span>, <span style=color:#e6db74>&#34;feature&#34;</span>, <span style=color:#e6db74>&#34;start&#34;</span>, <span style=color:#e6db74>&#34;end&#34;</span>, <span style=color:#e6db74>&#34;score&#34;</span>, <span style=color:#e6db74>&#34;strand&#34;</span>, <span style=color:#e6db74>&#34;frame&#34;</span>, <span style=color:#e6db74>&#34;attribute&#34;</span>])
</span></span><span style=display:flex><span>gtf_df
</span></span></code></pre></div><p><img src=https://github.com/user-attachments/assets/26526e85-9631-47b0-bcf8-a3e37da66d22 alt=image></p><h3 id=make-annotation-file>Make annotation file
<a class=anchor href=#make-annotation-file>#</a></h3><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-python data-lang=python><span style=display:flex><span>split_attributes <span style=color:#f92672>=</span> gtf_df[<span style=color:#e6db74>&#39;attribute&#39;</span>]<span style=color:#f92672>.</span>str<span style=color:#f92672>.</span>split(<span style=color:#e6db74>&#39;;&#39;</span>)
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span>attribute_1 <span style=color:#f92672>=</span> []
</span></span><span style=display:flex><span>attribute_2 <span style=color:#f92672>=</span> []
</span></span><span style=display:flex><span>attribute_3 <span style=color:#f92672>=</span> []
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span><span style=color:#66d9ef>for</span> attributes <span style=color:#f92672>in</span> split_attributes:
</span></span><span style=display:flex><span>    attr_1 <span style=color:#f92672>=</span> <span style=color:#e6db74>&#39;&#39;</span>
</span></span><span style=display:flex><span>    attr_2 <span style=color:#f92672>=</span> <span style=color:#e6db74>&#39;&#39;</span>
</span></span><span style=display:flex><span>    attr_3 <span style=color:#f92672>=</span> <span style=color:#e6db74>&#39;&#39;</span>
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span>    <span style=color:#66d9ef>for</span> attribute <span style=color:#f92672>in</span> attributes:
</span></span><span style=display:flex><span>        attribute <span style=color:#f92672>=</span> attribute<span style=color:#f92672>.</span>strip()
</span></span><span style=display:flex><span>        <span style=color:#66d9ef>if</span> <span style=color:#e6db74>&#39;transcript_id&#39;</span> <span style=color:#f92672>in</span> attribute:
</span></span><span style=display:flex><span>            attr_1 <span style=color:#f92672>=</span> attribute
</span></span><span style=display:flex><span>        <span style=color:#66d9ef>elif</span> <span style=color:#e6db74>&#39;gene_id&#39;</span> <span style=color:#f92672>in</span> attribute:
</span></span><span style=display:flex><span>            attr_2 <span style=color:#f92672>=</span> attribute
</span></span><span style=display:flex><span>        <span style=color:#66d9ef>elif</span> <span style=color:#e6db74>&#39;gene_name&#39;</span> <span style=color:#f92672>in</span> attribute:
</span></span><span style=display:flex><span>            attr_3 <span style=color:#f92672>=</span> attribute
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span>    attribute_1<span style=color:#f92672>.</span>append(attr_1)
</span></span><span style=display:flex><span>    attribute_2<span style=color:#f92672>.</span>append(attr_2)
</span></span><span style=display:flex><span>    attribute_3<span style=color:#f92672>.</span>append(attr_3)
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span>gtf_df[<span style=color:#e6db74>&#39;attribute.1&#39;</span>] <span style=color:#f92672>=</span> attribute_1
</span></span><span style=display:flex><span>gtf_df[<span style=color:#e6db74>&#39;attribute.2&#39;</span>] <span style=color:#f92672>=</span> attribute_2
</span></span><span style=display:flex><span>gtf_df[<span style=color:#e6db74>&#39;attribute.3&#39;</span>] <span style=color:#f92672>=</span> attribute_3
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span>annot_df <span style=color:#f92672>=</span> gtf_df<span style=color:#f92672>.</span>iloc[:, <span style=color:#f92672>-</span><span style=color:#ae81ff>4</span>:]<span style=color:#f92672>.</span>copy()
</span></span><span style=display:flex><span>annot_df[<span style=color:#e6db74>&#39;transcript_id&#39;</span>] <span style=color:#f92672>=</span> annot_df[<span style=color:#e6db74>&#39;attribute.1&#39;</span>]<span style=color:#f92672>.</span>str<span style=color:#f92672>.</span>split(<span style=color:#e6db74>&#39;&#34;&#39;</span>)<span style=color:#f92672>.</span>str[<span style=color:#ae81ff>1</span>]
</span></span><span style=display:flex><span>annot_df[<span style=color:#e6db74>&#39;gene_id&#39;</span>] <span style=color:#f92672>=</span> annot_df[<span style=color:#e6db74>&#39;attribute.2&#39;</span>]<span style=color:#f92672>.</span>str<span style=color:#f92672>.</span>split(<span style=color:#e6db74>&#39;&#34;&#39;</span>)<span style=color:#f92672>.</span>str[<span style=color:#ae81ff>1</span>]
</span></span><span style=display:flex><span>annot_df[<span style=color:#e6db74>&#39;gene_name&#39;</span>] <span style=color:#f92672>=</span> annot_df[<span style=color:#e6db74>&#39;attribute.3&#39;</span>]<span style=color:#f92672>.</span>str<span style=color:#f92672>.</span>split(<span style=color:#e6db74>&#39;&#34;&#39;</span>)<span style=color:#f92672>.</span>str[<span style=color:#ae81ff>1</span>]
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span>annot_df
</span></span></code></pre></div><p><img src=https://github.com/user-attachments/assets/b172aa0a-c287-4d88-a27e-4483c9906cf4 alt=image></p><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-python data-lang=python><span style=display:flex><span>annot_table <span style=color:#f92672>=</span> annot_df<span style=color:#f92672>.</span>iloc[:, <span style=color:#f92672>-</span><span style=color:#ae81ff>3</span>:]<span style=color:#f92672>.</span>copy()
</span></span><span style=display:flex><span>annot_table_dedup <span style=color:#f92672>=</span> annot_table[<span style=color:#f92672>~</span>annot_table[<span style=color:#e6db74>&#39;transcript_id&#39;</span>]<span style=color:#f92672>.</span>duplicated(keep<span style=color:#f92672>=</span><span style=color:#e6db74>&#39;first&#39;</span>)]
</span></span><span style=display:flex><span>annot <span style=color:#f92672>=</span> annot_table_dedup<span style=color:#f92672>.</span>iloc[:, <span style=color:#f92672>-</span><span style=color:#ae81ff>3</span>:]<span style=color:#f92672>.</span>copy()
</span></span><span style=display:flex><span>annot <span style=color:#f92672>=</span> annot<span style=color:#f92672>.</span>sort_values(by<span style=color:#f92672>=</span><span style=color:#e6db74>&#39;transcript_id&#39;</span>)
</span></span><span style=display:flex><span>annot<span style=color:#f92672>.</span>head(<span style=color:#ae81ff>50</span>)
</span></span></code></pre></div><p><img src=https://github.com/user-attachments/assets/42bca1cb-51be-4486-83dc-c6b31a73fc5c alt=image></p><h3 id=save-1>Save
<a class=anchor href=#save-1>#</a></h3><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-python data-lang=python><span style=display:flex><span><span style=color:#75715e>#annot.to_csv(&#34;annot.tsv&#34;,sep=&#39;\t&#39;,index=False)</span>
</span></span></code></pre></div><ul><li>나중에 또 써야대니깐 저장.</li></ul><h3 id=load-count>Load Count
<a class=anchor href=#load-count>#</a></h3><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-python data-lang=python><span style=display:flex><span>count <span style=color:#f92672>=</span> pd<span style=color:#f92672>.</span>read_csv(<span style=color:#e6db74>&#34;count.tsv&#34;</span>,sep<span style=color:#f92672>=</span><span style=color:#e6db74>&#39;</span><span style=color:#ae81ff>\t</span><span style=color:#e6db74>&#39;</span>,encoding<span style=color:#f92672>=</span><span style=color:#e6db74>&#39;cp949&#39;</span>)
</span></span><span style=display:flex><span>count
</span></span></code></pre></div><p><img src=https://github.com/user-attachments/assets/a0b6d6b0-682a-44af-9f30-20029fc91608 alt=image></p><ul><li>위에서 Rsubread로 만든 EBV count 가져옴</li></ul><h3 id=annotation>Annotation
<a class=anchor href=#annotation>#</a></h3><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-python data-lang=python><span style=display:flex><span>merged <span style=color:#f92672>=</span> count<span style=color:#f92672>.</span>merge(annot, left_on<span style=color:#f92672>=</span><span style=color:#e6db74>&#39;trans_id&#39;</span>, right_on<span style=color:#f92672>=</span><span style=color:#e6db74>&#39;transcript_id&#39;</span>, how<span style=color:#f92672>=</span><span style=color:#e6db74>&#39;left&#39;</span>)
</span></span><span style=display:flex><span>res_df <span style=color:#f92672>=</span> merged<span style=color:#f92672>.</span>drop(columns<span style=color:#f92672>=</span>[<span style=color:#e6db74>&#39;trans_id&#39;</span>])
</span></span><span style=display:flex><span>res_df <span style=color:#f92672>=</span> res_df[[<span style=color:#e6db74>&#39;transcript_id&#39;</span>, <span style=color:#e6db74>&#39;gene_id&#39;</span>, <span style=color:#e6db74>&#39;gene_name&#39;</span>, <span style=color:#e6db74>&#39;33-1&#39;</span>, <span style=color:#e6db74>&#39;33-2&#39;</span>, <span style=color:#e6db74>&#39;33-3&#39;</span>, <span style=color:#e6db74>&#39;150-1&#39;</span>, <span style=color:#e6db74>&#39;150-2&#39;</span>, <span style=color:#e6db74>&#39;150-3&#39;</span>, <span style=color:#e6db74>&#39;con-1&#39;</span>, <span style=color:#e6db74>&#39;con-2&#39;</span>, <span style=color:#e6db74>&#39;con-3&#39;</span>]]
</span></span><span style=display:flex><span>res_df
</span></span></code></pre></div><p><img src=https://github.com/user-attachments/assets/581ad2f0-9ec1-42d1-a150-3a36ea774fe3 alt=image></p><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-python data-lang=python><span style=display:flex><span>res_df[<span style=color:#e6db74>&#39;gene_name&#39;</span>]<span style=color:#f92672>.</span>value_counts()
</span></span></code></pre></div><p><img src=https://github.com/user-attachments/assets/ad94f072-1cfa-4a30-bbfc-a083b9efc252 alt=image></p><ul><li>중복인 transcript가 엄청많음</li></ul><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-python data-lang=python><span style=display:flex><span>res_df[<span style=color:#e6db74>&#39;sum&#39;</span>] <span style=color:#f92672>=</span> res_df<span style=color:#f92672>.</span>iloc[:, <span style=color:#ae81ff>3</span>:<span style=color:#ae81ff>12</span>]<span style=color:#f92672>.</span>sum(axis<span style=color:#f92672>=</span><span style=color:#ae81ff>1</span>)
</span></span><span style=display:flex><span>res_df <span style=color:#f92672>=</span> res_df<span style=color:#f92672>.</span>sort_values(by<span style=color:#f92672>=</span><span style=color:#e6db74>&#39;sum&#39;</span>, ascending<span style=color:#f92672>=</span><span style=color:#66d9ef>False</span>)
</span></span><span style=display:flex><span>res_df <span style=color:#f92672>=</span> res_df<span style=color:#f92672>.</span>drop_duplicates(subset<span style=color:#f92672>=</span><span style=color:#e6db74>&#39;gene_name&#39;</span>, keep<span style=color:#f92672>=</span><span style=color:#e6db74>&#39;first&#39;</span>)
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span>res_df <span style=color:#f92672>=</span> res_df<span style=color:#f92672>.</span>iloc[:, :<span style=color:#f92672>-</span><span style=color:#ae81ff>1</span>]
</span></span><span style=display:flex><span>res_df<span style=color:#f92672>.</span>reset_index(drop<span style=color:#f92672>=</span><span style=color:#66d9ef>True</span>, inplace<span style=color:#f92672>=</span><span style=color:#66d9ef>True</span>)
</span></span><span style=display:flex><span>res_df <span style=color:#f92672>=</span> res_df<span style=color:#f92672>.</span>dropna(subset<span style=color:#f92672>=</span>[<span style=color:#e6db74>&#39;gene_name&#39;</span>])
</span></span><span style=display:flex><span>print(res_df<span style=color:#f92672>.</span>shape)
</span></span><span style=display:flex><span>res_df
</span></span></code></pre></div><p><img src=https://github.com/user-attachments/assets/f6412998-0d28-4dfd-8d10-263c610c40fb alt=image></p><ul><li>(63, 12)짜리에서 deduplication 결과 (15, 12)가 됏다.</li></ul><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-python data-lang=python><span style=display:flex><span>res_df<span style=color:#f92672>.</span>to_csv(<span style=color:#e6db74>&#34;count.annot.tsv&#34;</span>, sep<span style=color:#f92672>=</span><span style=color:#e6db74>&#39;</span><span style=color:#ae81ff>\t</span><span style=color:#e6db74>&#39;</span>,index<span style=color:#f92672>=</span><span style=color:#66d9ef>False</span>)
</span></span></code></pre></div><h3 id=merge>Merge
<a class=anchor href=#merge>#</a></h3><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-python data-lang=python><span style=display:flex><span>count_hs <span style=color:#f92672>=</span> pd<span style=color:#f92672>.</span>read_csv(<span style=color:#e6db74>&#34;/data/home/ysh980101/2311/RNA-seq/count.csv&#34;</span>, encoding<span style=color:#f92672>=</span><span style=color:#e6db74>&#39;cp949&#39;</span>)
</span></span><span style=display:flex><span>count_hs<span style=color:#f92672>.</span>columns <span style=color:#f92672>=</span> [<span style=color:#e6db74>&#34;transcript_id&#34;</span>, <span style=color:#e6db74>&#34;gene_id&#34;</span>, <span style=color:#e6db74>&#34;gene_name&#34;</span>, <span style=color:#e6db74>&#34;33-1&#34;</span>, <span style=color:#e6db74>&#34;33-2&#34;</span>, <span style=color:#e6db74>&#34;33-3&#34;</span>, <span style=color:#e6db74>&#34;150-1&#34;</span>, <span style=color:#e6db74>&#34;150-2&#34;</span>, <span style=color:#e6db74>&#34;150-3&#34;</span>, <span style=color:#e6db74>&#34;con-1&#34;</span>, <span style=color:#e6db74>&#34;con-2&#34;</span>, <span style=color:#e6db74>&#34;con-3&#34;</span>]
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span>combined_df <span style=color:#f92672>=</span> pd<span style=color:#f92672>.</span>concat([res_df, count_hs], axis<span style=color:#f92672>=</span><span style=color:#ae81ff>0</span>, ignore_index<span style=color:#f92672>=</span><span style=color:#66d9ef>True</span>)
</span></span><span style=display:flex><span>combined_df
</span></span><span style=display:flex><span>combined_drop_df <span style=color:#f92672>=</span> combined_df[<span style=color:#f92672>~</span>(combined_df<span style=color:#f92672>.</span>iloc[:, <span style=color:#ae81ff>4</span>:<span style=color:#ae81ff>13</span>] <span style=color:#f92672>==</span> <span style=color:#ae81ff>0</span>)<span style=color:#f92672>.</span>all(axis<span style=color:#f92672>=</span><span style=color:#ae81ff>1</span>)] <span style=color:#75715e>#remove 0 in all samples </span>
</span></span><span style=display:flex><span>combined_drop_df
</span></span></code></pre></div><p><img src=https://github.com/user-attachments/assets/873ee0af-9bfc-4d11-bae7-ad61df069553 alt=image></p><p><img src=https://github.com/user-attachments/assets/f6613d7c-464a-48f7-9c51-eb4b461c354d alt=image></p><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-python data-lang=python><span style=display:flex><span>combined_drop_df<span style=color:#f92672>.</span>to_csv(<span style=color:#e6db74>&#34;count.annot.combined.drop.tsv&#34;</span>,sep<span style=color:#f92672>=</span><span style=color:#e6db74>&#39;</span><span style=color:#ae81ff>\t</span><span style=color:#e6db74>&#39;</span>,index<span style=color:#f92672>=</span><span style=color:#66d9ef>False</span>)
</span></span></code></pre></div><h2 id=3-deg-analysis>3. DEG Analysis
<a class=anchor href=#3-deg-analysis>#</a></h2><h3 id=load-packages>Load Packages
<a class=anchor href=#load-packages>#</a></h3><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-R data-lang=R><span style=display:flex><span><span style=color:#a6e22e>suppressMessages</span>({
</span></span><span style=display:flex><span>    <span style=color:#a6e22e>library</span>(<span style=color:#e6db74>&#34;DESeq2&#34;</span>)
</span></span><span style=display:flex><span>    <span style=color:#a6e22e>library</span>(pheatmap)
</span></span><span style=display:flex><span>    <span style=color:#a6e22e>library</span>(withr)
</span></span><span style=display:flex><span>    <span style=color:#a6e22e>library</span>(RColorBrewer)
</span></span><span style=display:flex><span>    <span style=color:#a6e22e>library</span>(gplots)
</span></span><span style=display:flex><span>    <span style=color:#a6e22e>library</span>(ggplot2)
</span></span><span style=display:flex><span>    <span style=color:#a6e22e>library</span>(dplyr)
</span></span><span style=display:flex><span>    })
</span></span></code></pre></div><h3 id=set-path-1>Set Path
<a class=anchor href=#set-path-1>#</a></h3><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-R data-lang=R><span style=display:flex><span><span style=color:#a6e22e>setwd</span>(<span style=color:#e6db74>&#34;/data/home/ysh980101/2311/RNA-seq_ebv/Rsubread&#34;</span>)
</span></span><span style=display:flex><span><span style=color:#a6e22e>getwd</span>()
</span></span></code></pre></div><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-plain data-lang=plain><span style=display:flex><span>&#39;/data1/home/ysh980101/2311/RNA-seq_ebv/Rsubread&#39;
</span></span></code></pre></div><h3 id=run-deg>Run DEG
<a class=anchor href=#run-deg>#</a></h3><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-R data-lang=R><span style=display:flex><span>S1 <span style=color:#f92672>&lt;-</span> <span style=color:#e6db74>&#39;33&#39;</span>
</span></span><span style=display:flex><span>S2 <span style=color:#f92672>&lt;-</span> <span style=color:#e6db74>&#39;150&#39;</span>
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span>countdata <span style=color:#f92672>&lt;-</span> <span style=color:#a6e22e>read.csv</span>(<span style=color:#e6db74>&#34;count.annot.combined.drop.tsv&#34;</span>, header<span style=color:#f92672>=</span><span style=color:#66d9ef>TRUE</span>, sep<span style=color:#f92672>=</span><span style=color:#e6db74>&#39;\t&#39;</span>)
</span></span><span style=display:flex><span><span style=color:#a6e22e>colnames</span>(countdata) <span style=color:#f92672>&lt;-</span> <span style=color:#a6e22e>c</span>(<span style=color:#e6db74>&#39;transcript_id&#39;</span>,<span style=color:#e6db74>&#39;gene_id&#39;</span>,<span style=color:#e6db74>&#39;gene_name&#39;</span>,<span style=color:#e6db74>&#39;33-1&#39;</span>,<span style=color:#e6db74>&#39;33-2&#39;</span>,<span style=color:#e6db74>&#39;33-3&#39;</span>,<span style=color:#e6db74>&#39;150-1&#39;</span>,<span style=color:#e6db74>&#39;150-2&#39;</span>,<span style=color:#e6db74>&#39;150-3&#39;</span>,<span style=color:#e6db74>&#39;con-1&#39;</span>,<span style=color:#e6db74>&#39;con-2&#39;</span>,<span style=color:#e6db74>&#39;con-3&#39;</span>)
</span></span><span style=display:flex><span>countdata <span style=color:#f92672>&lt;-</span> countdata[, <span style=color:#a6e22e>c</span>(<span style=color:#e6db74>&#39;transcript_id&#39;</span>,<span style=color:#e6db74>&#39;gene_id&#39;</span>,<span style=color:#e6db74>&#39;gene_name&#39;</span>,<span style=color:#e6db74>&#39;150-1&#39;</span>,<span style=color:#e6db74>&#39;150-2&#39;</span>,<span style=color:#e6db74>&#39;150-3&#39;</span>,<span style=color:#e6db74>&#39;33-1&#39;</span>,<span style=color:#e6db74>&#39;33-2&#39;</span>,<span style=color:#e6db74>&#39;33-3&#39;</span>,<span style=color:#e6db74>&#39;con-1&#39;</span>,<span style=color:#e6db74>&#39;con-2&#39;</span>,<span style=color:#e6db74>&#39;con-3&#39;</span>)]
</span></span><span style=display:flex><span>countdata <span style=color:#f92672>&lt;-</span> countdata[, <span style=color:#a6e22e>paste</span>(<span style=color:#a6e22e>c</span>(<span style=color:#e6db74>&#39;gene_name&#39;</span>,<span style=color:#a6e22e>paste0</span>(S2,<span style=color:#e6db74>&#34;-1&#34;</span>), <span style=color:#a6e22e>paste0</span>(S2,<span style=color:#e6db74>&#34;-2&#34;</span>), <span style=color:#a6e22e>paste0</span>(S2,<span style=color:#e6db74>&#34;-3&#34;</span>),<span style=color:#a6e22e>paste0</span>(S1,<span style=color:#e6db74>&#34;-1&#34;</span>), <span style=color:#a6e22e>paste0</span>(S1,<span style=color:#e6db74>&#34;-2&#34;</span>), <span style=color:#a6e22e>paste0</span>(S1,<span style=color:#e6db74>&#34;-3&#34;</span>)), sep<span style=color:#f92672>=</span><span style=color:#e6db74>&#34;&#34;</span>)]
</span></span><span style=display:flex><span><span style=color:#a6e22e>dim</span>(countdata)
</span></span></code></pre></div><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-plain data-lang=plain><span style=display:flex><span>16409 * 7
</span></span></code></pre></div><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-R data-lang=R><span style=display:flex><span>sample.names <span style=color:#f92672>&lt;-</span> <span style=color:#a6e22e>paste</span>(<span style=color:#a6e22e>c</span>(<span style=color:#a6e22e>paste0</span>(S2,<span style=color:#e6db74>&#34;-1&#34;</span>), <span style=color:#a6e22e>paste0</span>(S2,<span style=color:#e6db74>&#34;-2&#34;</span>), <span style=color:#a6e22e>paste0</span>(S2,<span style=color:#e6db74>&#34;-3&#34;</span>),<span style=color:#a6e22e>paste0</span>(S1,<span style=color:#e6db74>&#34;-1&#34;</span>), <span style=color:#a6e22e>paste0</span>(S1,<span style=color:#e6db74>&#34;-2&#34;</span>), <span style=color:#a6e22e>paste0</span>(S1,<span style=color:#e6db74>&#34;-3&#34;</span>)), sep<span style=color:#f92672>=</span><span style=color:#e6db74>&#34;&#34;</span>)
</span></span><span style=display:flex><span>conditions <span style=color:#f92672>&lt;-</span> <span style=color:#a6e22e>factor</span>(<span style=color:#a6e22e>c</span>(S2,S2,S2,S1,S1,S1))
</span></span><span style=display:flex><span>metadata <span style=color:#f92672>&lt;-</span> <span style=color:#a6e22e>data.frame</span>(Sample <span style=color:#f92672>=</span> sample.names, group <span style=color:#f92672>=</span> conditions)
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span>N <span style=color:#f92672>&lt;-</span> <span style=color:#a6e22e>dim</span>(countdata)[[2]]
</span></span><span style=display:flex><span>cData <span style=color:#f92672>=</span> countdata[,<span style=color:#ae81ff>2</span><span style=color:#f92672>:</span>N]
</span></span><span style=display:flex><span>GeneID <span style=color:#f92672>=</span> countdata[,<span style=color:#ae81ff>1</span>]
</span></span><span style=display:flex><span><span style=color:#a6e22e>rownames</span>(cData) <span style=color:#f92672>=</span> GeneID
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span>dds <span style=color:#f92672>&lt;-</span> <span style=color:#a6e22e>DESeqDataSetFromMatrix</span>(countData <span style=color:#f92672>=</span> cData, colData <span style=color:#f92672>=</span> metadata, design <span style=color:#f92672>=</span> <span style=color:#f92672>~</span>group)
</span></span><span style=display:flex><span>dds<span style=color:#f92672>$</span>group <span style=color:#f92672>&lt;-</span> <span style=color:#a6e22e>relevel</span>(dds<span style=color:#f92672>$</span>group, ref <span style=color:#f92672>=</span> S1)
</span></span><span style=display:flex><span>dds <span style=color:#f92672>&lt;-</span> <span style=color:#a6e22e>DESeq</span>(dds)  
</span></span><span style=display:flex><span>vsd <span style=color:#f92672>&lt;-</span> <span style=color:#a6e22e>vst</span>(dds, blind <span style=color:#f92672>=</span> <span style=color:#66d9ef>FALSE</span>)
</span></span><span style=display:flex><span>rld <span style=color:#f92672>&lt;-</span> <span style=color:#a6e22e>rlogTransformation</span>(dds, blind <span style=color:#f92672>=</span> <span style=color:#66d9ef>FALSE</span>)
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span>res <span style=color:#f92672>&lt;-</span> <span style=color:#a6e22e>results</span>(dds, contrast <span style=color:#f92672>=</span> <span style=color:#a6e22e>c</span>(<span style=color:#e6db74>&#34;group&#34;</span>, S2, S1))
</span></span><span style=display:flex><span>res_tbl <span style=color:#f92672>&lt;-</span> <span style=color:#a6e22e>as.data.frame</span>(res)
</span></span><span style=display:flex><span>res_tbl<span style=color:#f92672>$</span>GeneID <span style=color:#f92672>&lt;-</span> <span style=color:#a6e22e>rownames</span>(res_tbl)
</span></span><span style=display:flex><span>res_tbl <span style=color:#f92672>&lt;-</span> res_tbl<span style=color:#a6e22e>[order</span>(res_tbl<span style=color:#f92672>$</span>padj), ]
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span>NM_no_NA <span style=color:#f92672>&lt;-</span> <span style=color:#a6e22e>na.omit</span>(res)
</span></span><span style=display:flex><span>res_cut <span style=color:#f92672>&lt;-</span> NM_no_NA[NM_no_NA<span style=color:#f92672>$</span>padj <span style=color:#f92672>&lt;</span> <span style=color:#ae81ff>0.05</span>, ]
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span>lfc <span style=color:#f92672>&lt;-</span> <span style=color:#ae81ff>1.5</span>
</span></span><span style=display:flex><span>res_cut_2 <span style=color:#f92672>&lt;-</span> res_cut<span style=color:#a6e22e>[abs</span>(res_cut<span style=color:#f92672>$</span>log2FoldChange) <span style=color:#f92672>&gt;</span> lfc, ]
</span></span><span style=display:flex><span>res_cut_2_pos <span style=color:#f92672>&lt;-</span> res_cut_2[res_cut_2<span style=color:#f92672>$</span>log2FoldChange <span style=color:#f92672>&gt;</span> <span style=color:#ae81ff>0</span>, ]
</span></span><span style=display:flex><span>res_cut_2_neg <span style=color:#f92672>&lt;-</span> res_cut_2[res_cut_2<span style=color:#f92672>$</span>log2FoldChange <span style=color:#f92672>&lt;</span> <span style=color:#ae81ff>0</span>, ]
</span></span></code></pre></div><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-R data-lang=R><span style=display:flex><span><span style=color:#75715e># Volcano Plot</span>
</span></span><span style=display:flex><span><span style=color:#a6e22e>par</span>(mfrow <span style=color:#f92672>=</span> <span style=color:#a6e22e>c</span>(<span style=color:#ae81ff>1</span>, <span style=color:#ae81ff>1</span>))
</span></span><span style=display:flex><span><span style=color:#a6e22e>with</span>(NM_no_NA, <span style=color:#a6e22e>plot</span>(log2FoldChange, <span style=color:#f92672>-</span><span style=color:#a6e22e>log10</span>(pvalue), pch <span style=color:#f92672>=</span> <span style=color:#ae81ff>20</span>, col <span style=color:#f92672>=</span> <span style=color:#e6db74>&#39;gray&#39;</span>,
</span></span><span style=display:flex><span>                    main <span style=color:#f92672>=</span> <span style=color:#a6e22e>paste0</span>(<span style=color:#e6db74>&#39;Volcano plot_between\n&#39;</span>, S2, <span style=color:#e6db74>&#39; vs &#39;</span>, S1),
</span></span><span style=display:flex><span>                    xlim <span style=color:#f92672>=</span> <span style=color:#a6e22e>c</span>(<span style=color:#ae81ff>-3</span>, <span style=color:#ae81ff>3</span>)))
</span></span><span style=display:flex><span><span style=color:#a6e22e>abline</span>(h <span style=color:#f92672>=</span> <span style=color:#f92672>-</span><span style=color:#a6e22e>log10</span>(<span style=color:#ae81ff>0.05</span>), v <span style=color:#f92672>=</span> <span style=color:#a6e22e>c</span>(<span style=color:#f92672>-</span>lfc, lfc), col <span style=color:#f92672>=</span> <span style=color:#e6db74>&#34;gray&#34;</span>, lty <span style=color:#f92672>=</span> <span style=color:#ae81ff>2</span>)
</span></span><span style=display:flex><span><span style=color:#a6e22e>with</span>(res_cut_2_neg, <span style=color:#a6e22e>points</span>(log2FoldChange, <span style=color:#f92672>-</span><span style=color:#a6e22e>log10</span>(pvalue), pch <span style=color:#f92672>=</span> <span style=color:#ae81ff>20</span>, col <span style=color:#f92672>=</span> <span style=color:#e6db74>&#39;blue&#39;</span>))
</span></span><span style=display:flex><span><span style=color:#a6e22e>with</span>(res_cut_2_pos, <span style=color:#a6e22e>points</span>(log2FoldChange, <span style=color:#f92672>-</span><span style=color:#a6e22e>log10</span>(pvalue), pch <span style=color:#f92672>=</span> <span style=color:#ae81ff>20</span>, col <span style=color:#f92672>=</span> <span style=color:#e6db74>&#39;gold&#39;</span>))
</span></span><span style=display:flex><span><span style=color:#a6e22e>legend</span>(<span style=color:#e6db74>&#34;topright&#34;</span>, legend <span style=color:#f92672>=</span> <span style=color:#a6e22e>c</span>(<span style=color:#a6e22e>paste0</span>(<span style=color:#e6db74>&#34;FC ≥ &#34;</span>, lfc, <span style=color:#e6db74>&#34; &amp; p &lt; 0.05&#34;</span>),
</span></span><span style=display:flex><span>                              <span style=color:#a6e22e>paste0</span>(<span style=color:#e6db74>&#34;FC ≤ -&#34;</span>, lfc, <span style=color:#e6db74>&#34; &amp; p &lt; 0.05&#34;</span>)),
</span></span><span style=display:flex><span>       col <span style=color:#f92672>=</span> <span style=color:#a6e22e>c</span>(<span style=color:#e6db74>&#34;gold&#34;</span>, <span style=color:#e6db74>&#34;blue&#34;</span>), pch <span style=color:#f92672>=</span> <span style=color:#ae81ff>20</span>)
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span><span style=color:#75715e># Bar Plot</span>
</span></span><span style=display:flex><span>lengths <span style=color:#f92672>&lt;-</span> <span style=color:#a6e22e>c</span>(<span style=color:#a6e22e>nrow</span>(res_cut_2_neg), <span style=color:#a6e22e>nrow</span>(res_cut_2_pos))
</span></span><span style=display:flex><span>bp <span style=color:#f92672>&lt;-</span> <span style=color:#a6e22e>barplot</span>(lengths, space <span style=color:#f92672>=</span> <span style=color:#ae81ff>0</span>, col <span style=color:#f92672>=</span> <span style=color:#a6e22e>c</span>(<span style=color:#e6db74>&#34;blue&#34;</span>, <span style=color:#e6db74>&#34;gold&#34;</span>),
</span></span><span style=display:flex><span>              horiz <span style=color:#f92672>=</span> <span style=color:#66d9ef>TRUE</span>, xlab <span style=color:#f92672>=</span> <span style=color:#e6db74>&#34;Count of genes&#34;</span>, ylab <span style=color:#f92672>=</span> <span style=color:#a6e22e>paste0</span>(S2, <span style=color:#e6db74>&#34;/&#34;</span>, S1),
</span></span><span style=display:flex><span>              xlim <span style=color:#f92672>=</span> <span style=color:#a6e22e>c</span>(<span style=color:#ae81ff>0</span>, <span style=color:#a6e22e>sum</span>(lengths)), width <span style=color:#f92672>=</span> <span style=color:#ae81ff>0.02</span>, las <span style=color:#f92672>=</span> <span style=color:#ae81ff>1</span>)
</span></span><span style=display:flex><span><span style=color:#a6e22e>title</span>(<span style=color:#a6e22e>paste0</span>(<span style=color:#e6db74>&#34;UP, DOWN regulated count (|FC| ≥ &#34;</span>, lfc, <span style=color:#e6db74>&#34;)&#34;</span>))
</span></span><span style=display:flex><span><span style=color:#a6e22e>legend</span>(<span style=color:#e6db74>&#34;topright&#34;</span>, legend <span style=color:#f92672>=</span> <span style=color:#a6e22e>c</span>(<span style=color:#e6db74>&#34;UP&#34;</span>, <span style=color:#e6db74>&#34;DOWN&#34;</span>), fill <span style=color:#f92672>=</span> <span style=color:#a6e22e>c</span>(<span style=color:#e6db74>&#34;gold&#34;</span>, <span style=color:#e6db74>&#34;blue&#34;</span>))
</span></span><span style=display:flex><span><span style=color:#a6e22e>text</span>(lengths, bp, labels <span style=color:#f92672>=</span> lengths, pos <span style=color:#f92672>=</span> <span style=color:#ae81ff>4</span>, cex <span style=color:#f92672>=</span> <span style=color:#ae81ff>0.9</span>)
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span><span style=color:#75715e># Save</span>
</span></span><span style=display:flex><span>cutoff <span style=color:#f92672>&lt;-</span> <span style=color:#ae81ff>0.05</span>
</span></span><span style=display:flex><span>sig_res <span style=color:#f92672>&lt;-</span> dplyr<span style=color:#f92672>::</span><span style=color:#a6e22e>filter</span>(res_tbl, padj <span style=color:#f92672>&lt;</span> cutoff) <span style=color:#f92672>%&gt;%</span>
</span></span><span style=display:flex><span>  dplyr<span style=color:#f92672>::</span><span style=color:#a6e22e>arrange</span>(padj) <span style=color:#f92672>%&gt;%</span>
</span></span><span style=display:flex><span>  dplyr<span style=color:#f92672>::</span><span style=color:#a6e22e>select</span>(GeneID, baseMean, log2FoldChange, lfcSE, stat, pvalue, padj)
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span><span style=color:#a6e22e>write.csv</span>(sig_res, file <span style=color:#f92672>=</span> <span style=color:#a6e22e>paste0</span>(<span style=color:#e6db74>&#39;../DESeq2/res.&#39;</span>, S2, <span style=color:#e6db74>&#39;vs&#39;</span>, S1, <span style=color:#e6db74>&#39;.csv&#39;</span>), row.names <span style=color:#f92672>=</span> <span style=color:#66d9ef>FALSE</span>)
</span></span></code></pre></div><p><img src=https://github.com/user-attachments/assets/2e32b5e8-7d44-4b2d-80dd-54106bc300cf alt=image></p><p><img src=https://github.com/user-attachments/assets/d3fa3953-36b0-4a7e-946e-3a24a50a4891 alt=image></p><p>cf) Heatmap 시각화</p><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-R data-lang=R><span style=display:flex><span><span style=color:#75715e># Heatmap</span>
</span></span><span style=display:flex><span>sig_idx <span style=color:#f92672>&lt;-</span> res<span style=color:#f92672>$</span>padj <span style=color:#f92672>&lt;</span> cutoff <span style=color:#f92672>&amp;</span> <span style=color:#f92672>!</span><span style=color:#a6e22e>is.na</span>(res<span style=color:#f92672>$</span>padj)
</span></span><span style=display:flex><span>sig_dat <span style=color:#f92672>&lt;-</span> <span style=color:#a6e22e>assay</span>(rld)[sig_idx, ]
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span>annC <span style=color:#f92672>&lt;-</span> <span style=color:#a6e22e>data.frame</span>(condition <span style=color:#f92672>=</span> conditions)
</span></span><span style=display:flex><span><span style=color:#a6e22e>rownames</span>(annC) <span style=color:#f92672>&lt;-</span> <span style=color:#a6e22e>colnames</span>(sig_dat)
</span></span><span style=display:flex><span>ann_colors <span style=color:#f92672>&lt;-</span> <span style=color:#a6e22e>list</span>(condition <span style=color:#f92672>=</span> <span style=color:#a6e22e>setNames</span>(<span style=color:#a6e22e>c</span>(<span style=color:#e6db74>&#34;#F7819F&#34;</span>, <span style=color:#e6db74>&#34;#58D3F7&#34;</span>), <span style=color:#a6e22e>c</span>(S2, S1)))
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span><span style=color:#a6e22e>pheatmap</span>(sig_dat,
</span></span><span style=display:flex><span>         scale <span style=color:#f92672>=</span> <span style=color:#e6db74>&#34;row&#34;</span>,
</span></span><span style=display:flex><span>         fontsize_row <span style=color:#f92672>=</span> <span style=color:#ae81ff>9</span>,
</span></span><span style=display:flex><span>         annotation_col <span style=color:#f92672>=</span> annC,
</span></span><span style=display:flex><span>         color <span style=color:#f92672>=</span> <span style=color:#a6e22e>rev</span>(<span style=color:#a6e22e>brewer.pal</span>(<span style=color:#ae81ff>6</span>, <span style=color:#e6db74>&#34;RdYlGn&#34;</span>)),
</span></span><span style=display:flex><span>         annotation_colors <span style=color:#f92672>=</span> ann_colors,
</span></span><span style=display:flex><span>         show_rownames <span style=color:#f92672>=</span> <span style=color:#66d9ef>FALSE</span>,
</span></span><span style=display:flex><span>         show_colnames <span style=color:#f92672>=</span> <span style=color:#66d9ef>TRUE</span>)
</span></span></code></pre></div><p><img src=https://github.com/user-attachments/assets/67e7e43f-4db6-48c2-9fed-fc03dfb4e892 alt=image></p></article><footer class=book-footer><div class="flex flex-wrap justify-between"></div><script>(function(){function e(e){const t=window.getSelection(),n=document.createRange();n.selectNodeContents(e),t.removeAllRanges(),t.addRange(n)}document.querySelectorAll("pre code").forEach(t=>{t.addEventListener("click",function(){if(window.getSelection().toString())return;e(t.parentElement),navigator.clipboard&&navigator.clipboard.writeText(t.parentElement.textContent)})})})()</script></footer><div class=book-comments><script src=https://giscus.app/client.js data-repo=yshghid/yshghid.github.io data-repo-id=R_kgDONkMkNg data-category-id=DIC_kwDONkMkNs4CloJh data-mapping=pathname data-strict=0 data-reactions-enabled=1 data-emit-metadata=0 data-input-position=bottom data-theme=preferred_color_scheme data-lang=ko crossorigin=anonymous async></script></div><label for=menu-control class="hidden book-menu-overlay"></label></div><aside class=book-toc><div class=book-toc-content><nav id=TableOfContents><ul><li><a href=#1-alignment>1. Alignment</a><ul><li><a href=#load-package-set-path>Load package, Set Path</a></li><li><a href=#build-index>Build Index</a></li><li><a href=#feature-count>Feature Count</a></li><li><a href=#save>Save</a></li></ul></li><li><a href=#2-annotation>2. Annotation</a><ul><li><a href=#load-package>Load Package</a></li><li><a href=#set-path>Set Path</a></li><li><a href=#load-gtf-file>Load GTF file</a></li><li><a href=#make-annotation-file>Make annotation file</a></li><li><a href=#save-1>Save</a></li><li><a href=#load-count>Load Count</a></li><li><a href=#annotation>Annotation</a></li><li><a href=#merge>Merge</a></li></ul></li><li><a href=#3-deg-analysis>3. DEG Analysis</a><ul><li><a href=#load-packages>Load Packages</a></li><li><a href=#set-path-1>Set Path</a></li><li><a href=#run-deg>Run DEG</a></li></ul></li></ul></nav></div></aside></main></body></html>