<!doctype html><html lang=en-us dir=ltr><head><meta charset=UTF-8><meta name=viewport content="width=device-width,initial-scale=1"><meta name=description content='
  EBV RNA-seq 전처리
  #

#2024-12-31



분석 목적

제공받은 fastq를 human genome에 매핑해서 전처리, 분석 후 DE 결과 보냄
DE 분석시에 EBV 유전자도 포함해달라는 요청



해야하는것

fastq를 EBV genome에 매핑해서 전처리, EBV count 생성
human count에 EBV count를 붙이기
통합 count로 DE 분석 재수행




  1. Alignment
  #


  Load package, Set Path
  #

library(edgeR)
library(Rsubread)
library(org.Hs.eg.db)

setwd("/data/home/ysh980101/2311/RNA-seq_ebv/Rsubread")
getwd()
&#39;/data1/home/ysh980101/2311/RNA-seq_ebv/Rsubread&#39;

  Build Index
  #

# build index
ref <- "/data3/PUBLIC_DATA/ref_genomes/Human_gammaherpesvirus_4_EBV/NC_007605.1.fa"
output_basename <- "NC_007605.1_idx"
buildindex(basename = output_basename, reference = ref)

  Feature Count
  #

# feature.count
targets <- read.delim("targets.txt", header=TRUE)
output.files <- c(targets$FileName)

fc <- featureCounts(output.files, annot.inbuilt=FALSE, annot.ext = "/data3/PUBLIC_DATA/ref_genomes/Human_gammaherpesvirus_4_EBV/NC_007605.1.gtf", isGTFAnnotationFile = TRUE, 
                    GTF.featureType = "exon", GTF.attrType = "transcript_id", GTF.attrType.extra = NULL, isPairedEnd=TRUE, countReadPairs=TRUE)# nthreads=30)

  Save
  #

colnames(fc$counts) <- c(&#39;33-1&#39;,&#39;33-2&#39;,&#39;33-3&#39;,&#39;150-1&#39;,&#39;150-2&#39;,&#39;150-3&#39;,&#39;con-1&#39;,&#39;con-2&#39;,&#39;con-3&#39;)
head(fc$counts)

group <- factor(targets$Status)
y <- DGEList(fc$counts, group=group)
fc$sort.counts <- fc$counts[order(rownames(fc$counts)), ]

sort.counts.df <- as.data.frame(fc$sort.counts)
sort.counts.df$trans_id <- rownames(fc$sort.counts)
sort.counts.df <- sort.counts.df[, c("trans_id", "33-1", "33-2", "33-3", "150-1", "150-2", "150-3", "con-1", "con-2", "con-3")]

write.table(sort.counts.df, file = "count.tsv", sep = "\t", quote = FALSE, row.names = FALSE)

  2. Annotation
  #


  Load Package
  #

import os
import pandas as pd

  Set Path
  #

os.chdir("/data/home/ysh980101/2311/RNA-seq_ebv/Rsubread")
os.getcwd()
&#39;/data1/home/ysh980101/2311/RNA-seq_ebv/Rsubread&#39;

  Load GTF file
  #


annotation 파일이 없어서 gtf reference 파일을 사용해서 직접 만들어줫다.

gtf_df = pd.read_csv("/data3/PUBLIC_DATA/ref_genomes/Human_gammaherpesvirus_4_EBV/NC_007605.1.gtf", sep=&#39;\t&#39;, header=None, comment="#", names=["seqname", "source", "feature", "start", "end", "score", "strand", "frame", "attribute"])
gtf_df
'><meta name=theme-color media="(prefers-color-scheme: light)" content="#ffffff"><meta name=theme-color media="(prefers-color-scheme: dark)" content="#343a40"><meta name=color-scheme content="light dark"><meta property="og:url" content="https://yshghid.github.io/docs/study/bioinformatics/bi11/"><meta property="og:site_name" content=" "><meta property="og:title" content="EBV RNA-seq 전처리"><meta property="og:description" content='EBV RNA-seq 전처리 # #2024-12-31
분석 목적
제공받은 fastq를 human genome에 매핑해서 전처리, 분석 후 DE 결과 보냄 DE 분석시에 EBV 유전자도 포함해달라는 요청 해야하는것
fastq를 EBV genome에 매핑해서 전처리, EBV count 생성 human count에 EBV count를 붙이기 통합 count로 DE 분석 재수행 1. Alignment # Load package, Set Path # library(edgeR) library(Rsubread) library(org.Hs.eg.db) setwd("/data/home/ysh980101/2311/RNA-seq_ebv/Rsubread") getwd() &#39;/data1/home/ysh980101/2311/RNA-seq_ebv/Rsubread&#39; Build Index # # build index ref <- "/data3/PUBLIC_DATA/ref_genomes/Human_gammaherpesvirus_4_EBV/NC_007605.1.fa" output_basename <- "NC_007605.1_idx" buildindex(basename = output_basename, reference = ref) Feature Count # # feature.count targets <- read.delim("targets.txt", header=TRUE) output.files <- c(targets$FileName) fc <- featureCounts(output.files, annot.inbuilt=FALSE, annot.ext = "/data3/PUBLIC_DATA/ref_genomes/Human_gammaherpesvirus_4_EBV/NC_007605.1.gtf", isGTFAnnotationFile = TRUE, GTF.featureType = "exon", GTF.attrType = "transcript_id", GTF.attrType.extra = NULL, isPairedEnd=TRUE, countReadPairs=TRUE)# nthreads=30) Save # colnames(fc$counts) <- c(&#39;33-1&#39;,&#39;33-2&#39;,&#39;33-3&#39;,&#39;150-1&#39;,&#39;150-2&#39;,&#39;150-3&#39;,&#39;con-1&#39;,&#39;con-2&#39;,&#39;con-3&#39;) head(fc$counts) group <- factor(targets$Status) y <- DGEList(fc$counts, group=group) fc$sort.counts <- fc$counts[order(rownames(fc$counts)), ] sort.counts.df <- as.data.frame(fc$sort.counts) sort.counts.df$trans_id <- rownames(fc$sort.counts) sort.counts.df <- sort.counts.df[, c("trans_id", "33-1", "33-2", "33-3", "150-1", "150-2", "150-3", "con-1", "con-2", "con-3")] write.table(sort.counts.df, file = "count.tsv", sep = "\t", quote = FALSE, row.names = FALSE) 2. Annotation # Load Package # import os import pandas as pd Set Path # os.chdir("/data/home/ysh980101/2311/RNA-seq_ebv/Rsubread") os.getcwd() &#39;/data1/home/ysh980101/2311/RNA-seq_ebv/Rsubread&#39; Load GTF file # annotation 파일이 없어서 gtf reference 파일을 사용해서 직접 만들어줫다. gtf_df = pd.read_csv("/data3/PUBLIC_DATA/ref_genomes/Human_gammaherpesvirus_4_EBV/NC_007605.1.gtf", sep=&#39;\t&#39;, header=None, comment="#", names=["seqname", "source", "feature", "start", "end", "score", "strand", "frame", "attribute"]) gtf_df'><meta property="og:locale" content="en_us"><meta property="og:type" content="article"><meta property="article:section" content="docs"><meta property="article:published_time" content="2024-12-31T00:00:00+00:00"><meta property="article:modified_time" content="2024-12-31T00:00:00+00:00"><meta property="article:tag" content="2024-12"><title>EBV RNA-seq 전처리 |</title><link rel=icon href=/favicon.png><link rel=manifest href=/manifest.json><link rel=canonical href=https://yshghid.github.io/docs/study/bioinformatics/bi11/><link rel=stylesheet href=/book.min.6217d077edb4189fd0578345e84bca1a884dfdee121ff8dc9a0f55cfe0852bc9.css integrity="sha256-YhfQd+20GJ/QV4NF6EvKGohN/e4SH/jcmg9Vz+CFK8k=" crossorigin=anonymous><script defer src=/fuse.min.js></script><script defer src=/en.search.min.578a2d8e6e30e0a0ae3682797882d89ca0cbbf1734d206705396ea76cf57bbb6.js integrity="sha256-V4otjm4w4KCuNoJ5eILYnKDLvxc00gZwU5bqds9Xu7Y=" crossorigin=anonymous></script></head><body dir=ltr><input type=checkbox class="hidden toggle" id=menu-control>
<input type=checkbox class="hidden toggle" id=toc-control><main class="container flex"><aside class=book-menu><div class=book-menu-content><nav><h2 class=book-brand><a class="flex align-center" href=/><img src=/logo.png alt=Logo class=book-icon><span></span></a></h2><div class="book-search hidden"><input type=text id=book-search-input placeholder=Search aria-label=Search maxlength=64 data-hotkeys=s/><div class="book-search-spinner hidden"></div><ul id=book-search-results></ul></div><script>document.querySelector(".book-search").classList.remove("hidden")</script><ul><li class=book-section-flat><span>기록</span><ul><li><a href=/docs/hobby/book/>글</a><ul></ul></li><li><a href=/docs/hobby/daily/>일상</a><ul></ul></li></ul></li><li class=book-section-flat><span>공부</span><ul><li><a href=/docs/study/bioinformatics/>Bioinformatics</a><ul></ul></li><li><a href=/docs/study/ai/>AI</a><ul></ul></li><li><a href=/docs/study/sw/>SW</a><ul></ul></li><li><a href=/docs/study/algorithm/>알고리즘</a><ul></ul></li><li><a href=/docs/study/career/>취업</a><ul></ul></li></ul></li></ul></nav><script>(function(){var e=document.querySelector("aside .book-menu-content");addEventListener("beforeunload",function(){localStorage.setItem("menu.scrollTop",e.scrollTop)}),e.scrollTop=localStorage.getItem("menu.scrollTop")})()</script></div></aside><div class=book-page><header class=book-header><div class="flex align-center justify-between"><label for=menu-control><img src=/svg/menu.svg class=book-icon alt=Menu></label><h3>EBV RNA-seq 전처리</h3><label for=toc-control><img src=/svg/toc.svg class=book-icon alt="Table of Contents"></label></div><aside class="hidden clearfix"><nav id=TableOfContents><ul><li><a href=#1-alignment>1. Alignment</a><ul><li><a href=#load-package-set-path>Load package, Set Path</a></li><li><a href=#build-index>Build Index</a></li><li><a href=#feature-count>Feature Count</a></li><li><a href=#save>Save</a></li></ul></li><li><a href=#2-annotation>2. Annotation</a><ul><li><a href=#load-package>Load Package</a></li><li><a href=#set-path>Set Path</a></li><li><a href=#load-gtf-file>Load GTF file</a></li><li><a href=#make-annotation-file>Make annotation file</a></li><li><a href=#save-1>Save</a></li><li><a href=#load-count>Load Count</a></li><li><a href=#annotation>Annotation</a></li><li><a href=#merge>Merge</a></li></ul></li><li><a href=#3-deg-analysis>3. DEG Analysis</a><ul><li><a href=#load-packages>Load Packages</a></li><li><a href=#set-path-1>Set Path</a></li><li><a href=#run-deg>Run DEG</a></li></ul></li></ul></nav></aside></header><article class="markdown book-article"><h1 id=ebv-rna-seq-전처리>EBV RNA-seq 전처리
<a class=anchor href=#ebv-rna-seq-%ec%a0%84%ec%b2%98%eb%a6%ac>#</a></h1><p>#2024-12-31</p><hr><ul><li><p>분석 목적</p><ul><li>제공받은 fastq를 human genome에 매핑해서 전처리, 분석 후 DE 결과 보냄</li><li>DE 분석시에 EBV 유전자도 포함해달라는 요청</li></ul></li><li><p>해야하는것</p><ul><li>fastq를 EBV genome에 매핑해서 전처리, EBV count 생성</li><li>human count에 EBV count를 붙이기</li><li>통합 count로 DE 분석 재수행</li></ul></li></ul><h2 id=1-alignment>1. Alignment
<a class=anchor href=#1-alignment>#</a></h2><h3 id=load-package-set-path>Load package, Set Path
<a class=anchor href=#load-package-set-path>#</a></h3><div class=highlight><pre tabindex=0 style=color:#e5e5e5;background-color:#000;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-R data-lang=R><span style=display:flex><span>library(edgeR)
</span></span><span style=display:flex><span>library(Rsubread)
</span></span><span style=display:flex><span>library(org.Hs.eg.db)
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span>setwd(<span style=color:#0ff;font-weight:700>&#34;/data/home/ysh980101/2311/RNA-seq_ebv/Rsubread&#34;</span>)
</span></span><span style=display:flex><span>getwd()
</span></span></code></pre></div><div class=highlight><pre tabindex=0 style=color:#e5e5e5;background-color:#000;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-plain data-lang=plain><span style=display:flex><span>&#39;/data1/home/ysh980101/2311/RNA-seq_ebv/Rsubread&#39;
</span></span></code></pre></div><h3 id=build-index>Build Index
<a class=anchor href=#build-index>#</a></h3><div class=highlight><pre tabindex=0 style=color:#e5e5e5;background-color:#000;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-R data-lang=R><span style=display:flex><span><span style=color:#007f7f># build index</span>
</span></span><span style=display:flex><span>ref &lt;- <span style=color:#0ff;font-weight:700>&#34;/data3/PUBLIC_DATA/ref_genomes/Human_gammaherpesvirus_4_EBV/NC_007605.1.fa&#34;</span>
</span></span><span style=display:flex><span>output_basename &lt;- <span style=color:#0ff;font-weight:700>&#34;NC_007605.1_idx&#34;</span>
</span></span><span style=display:flex><span>buildindex(basename = output_basename, reference = ref)
</span></span></code></pre></div><h3 id=feature-count>Feature Count
<a class=anchor href=#feature-count>#</a></h3><div class=highlight><pre tabindex=0 style=color:#e5e5e5;background-color:#000;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-R data-lang=R><span style=display:flex><span><span style=color:#007f7f># feature.count</span>
</span></span><span style=display:flex><span>targets &lt;- read.delim(<span style=color:#0ff;font-weight:700>&#34;targets.txt&#34;</span>, header=<span style=color:#fff;font-weight:700>TRUE</span>)
</span></span><span style=display:flex><span>output.files &lt;- c(targets$FileName)
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span>fc &lt;- featureCounts(output.files, annot.inbuilt=<span style=color:#fff;font-weight:700>FALSE</span>, annot.ext = <span style=color:#0ff;font-weight:700>&#34;/data3/PUBLIC_DATA/ref_genomes/Human_gammaherpesvirus_4_EBV/NC_007605.1.gtf&#34;</span>, isGTFAnnotationFile = <span style=color:#fff;font-weight:700>TRUE</span>, 
</span></span><span style=display:flex><span>                    GTF.featureType = <span style=color:#0ff;font-weight:700>&#34;exon&#34;</span>, GTF.attrType = <span style=color:#0ff;font-weight:700>&#34;transcript_id&#34;</span>, GTF.attrType.extra = <span style=color:#fff;font-weight:700>NULL</span>, isPairedEnd=<span style=color:#fff;font-weight:700>TRUE</span>, countReadPairs=<span style=color:#fff;font-weight:700>TRUE</span>)<span style=color:#007f7f># nthreads=30)</span>
</span></span></code></pre></div><h3 id=save>Save
<a class=anchor href=#save>#</a></h3><div class=highlight><pre tabindex=0 style=color:#e5e5e5;background-color:#000;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-R data-lang=R><span style=display:flex><span>colnames(fc$counts) &lt;- c(<span style=color:#0ff;font-weight:700>&#39;33-1&#39;</span>,<span style=color:#0ff;font-weight:700>&#39;33-2&#39;</span>,<span style=color:#0ff;font-weight:700>&#39;33-3&#39;</span>,<span style=color:#0ff;font-weight:700>&#39;150-1&#39;</span>,<span style=color:#0ff;font-weight:700>&#39;150-2&#39;</span>,<span style=color:#0ff;font-weight:700>&#39;150-3&#39;</span>,<span style=color:#0ff;font-weight:700>&#39;con-1&#39;</span>,<span style=color:#0ff;font-weight:700>&#39;con-2&#39;</span>,<span style=color:#0ff;font-weight:700>&#39;con-3&#39;</span>)
</span></span><span style=display:flex><span>head(fc$counts)
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span>group &lt;- factor(targets$Status)
</span></span><span style=display:flex><span>y &lt;- DGEList(fc$counts, group=group)
</span></span><span style=display:flex><span>fc$sort.counts &lt;- fc$counts[order(rownames(fc$counts)), ]
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span>sort.counts.df &lt;- as.data.frame(fc$sort.counts)
</span></span><span style=display:flex><span>sort.counts.df$trans_id &lt;- rownames(fc$sort.counts)
</span></span><span style=display:flex><span>sort.counts.df &lt;- sort.counts.df[, c(<span style=color:#0ff;font-weight:700>&#34;trans_id&#34;</span>, <span style=color:#0ff;font-weight:700>&#34;33-1&#34;</span>, <span style=color:#0ff;font-weight:700>&#34;33-2&#34;</span>, <span style=color:#0ff;font-weight:700>&#34;33-3&#34;</span>, <span style=color:#0ff;font-weight:700>&#34;150-1&#34;</span>, <span style=color:#0ff;font-weight:700>&#34;150-2&#34;</span>, <span style=color:#0ff;font-weight:700>&#34;150-3&#34;</span>, <span style=color:#0ff;font-weight:700>&#34;con-1&#34;</span>, <span style=color:#0ff;font-weight:700>&#34;con-2&#34;</span>, <span style=color:#0ff;font-weight:700>&#34;con-3&#34;</span>)]
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span>write.table(sort.counts.df, file = <span style=color:#0ff;font-weight:700>&#34;count.tsv&#34;</span>, sep = <span style=color:#0ff;font-weight:700>&#34;\t&#34;</span>, quote = <span style=color:#fff;font-weight:700>FALSE</span>, row.names = <span style=color:#fff;font-weight:700>FALSE</span>)
</span></span></code></pre></div><h2 id=2-annotation>2. Annotation
<a class=anchor href=#2-annotation>#</a></h2><h3 id=load-package>Load Package
<a class=anchor href=#load-package>#</a></h3><div class=highlight><pre tabindex=0 style=color:#e5e5e5;background-color:#000;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-python data-lang=python><span style=display:flex><span><span style=color:#fff;font-weight:700>import</span> os
</span></span><span style=display:flex><span><span style=color:#fff;font-weight:700>import</span> pandas <span style=color:#fff;font-weight:700>as</span> pd
</span></span></code></pre></div><h3 id=set-path>Set Path
<a class=anchor href=#set-path>#</a></h3><div class=highlight><pre tabindex=0 style=color:#e5e5e5;background-color:#000;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-python data-lang=python><span style=display:flex><span>os.chdir(<span style=color:#0ff;font-weight:700>&#34;/data/home/ysh980101/2311/RNA-seq_ebv/Rsubread&#34;</span>)
</span></span><span style=display:flex><span>os.getcwd()
</span></span></code></pre></div><div class=highlight><pre tabindex=0 style=color:#e5e5e5;background-color:#000;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-plain data-lang=plain><span style=display:flex><span>&#39;/data1/home/ysh980101/2311/RNA-seq_ebv/Rsubread&#39;
</span></span></code></pre></div><h3 id=load-gtf-file>Load GTF file
<a class=anchor href=#load-gtf-file>#</a></h3><ul><li>annotation 파일이 없어서 gtf reference 파일을 사용해서 직접 만들어줫다.</li></ul><div class=highlight><pre tabindex=0 style=color:#e5e5e5;background-color:#000;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-python data-lang=python><span style=display:flex><span>gtf_df = pd.read_csv(<span style=color:#0ff;font-weight:700>&#34;/data3/PUBLIC_DATA/ref_genomes/Human_gammaherpesvirus_4_EBV/NC_007605.1.gtf&#34;</span>, sep=<span style=color:#0ff;font-weight:700>&#39;</span><span style=color:#0ff;font-weight:700>\t</span><span style=color:#0ff;font-weight:700>&#39;</span>, header=<span style=color:#fff;font-weight:700>None</span>, comment=<span style=color:#0ff;font-weight:700>&#34;#&#34;</span>, names=[<span style=color:#0ff;font-weight:700>&#34;seqname&#34;</span>, <span style=color:#0ff;font-weight:700>&#34;source&#34;</span>, <span style=color:#0ff;font-weight:700>&#34;feature&#34;</span>, <span style=color:#0ff;font-weight:700>&#34;start&#34;</span>, <span style=color:#0ff;font-weight:700>&#34;end&#34;</span>, <span style=color:#0ff;font-weight:700>&#34;score&#34;</span>, <span style=color:#0ff;font-weight:700>&#34;strand&#34;</span>, <span style=color:#0ff;font-weight:700>&#34;frame&#34;</span>, <span style=color:#0ff;font-weight:700>&#34;attribute&#34;</span>])
</span></span><span style=display:flex><span>gtf_df
</span></span></code></pre></div><p><img src=https://github.com/user-attachments/assets/26526e85-9631-47b0-bcf8-a3e37da66d22 alt=image></p><h3 id=make-annotation-file>Make annotation file
<a class=anchor href=#make-annotation-file>#</a></h3><div class=highlight><pre tabindex=0 style=color:#e5e5e5;background-color:#000;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-python data-lang=python><span style=display:flex><span>split_attributes = gtf_df[<span style=color:#0ff;font-weight:700>&#39;attribute&#39;</span>].str.split(<span style=color:#0ff;font-weight:700>&#39;;&#39;</span>)
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span>attribute_1 = []
</span></span><span style=display:flex><span>attribute_2 = []
</span></span><span style=display:flex><span>attribute_3 = []
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span><span style=color:#fff;font-weight:700>for</span> attributes in split_attributes:
</span></span><span style=display:flex><span>    attr_1 = <span style=color:#0ff;font-weight:700>&#39;&#39;</span>
</span></span><span style=display:flex><span>    attr_2 = <span style=color:#0ff;font-weight:700>&#39;&#39;</span>
</span></span><span style=display:flex><span>    attr_3 = <span style=color:#0ff;font-weight:700>&#39;&#39;</span>
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span>    <span style=color:#fff;font-weight:700>for</span> attribute in attributes:
</span></span><span style=display:flex><span>        attribute = attribute.strip()
</span></span><span style=display:flex><span>        <span style=color:#fff;font-weight:700>if</span> <span style=color:#0ff;font-weight:700>&#39;transcript_id&#39;</span> in attribute:
</span></span><span style=display:flex><span>            attr_1 = attribute
</span></span><span style=display:flex><span>        <span style=color:#fff;font-weight:700>elif</span> <span style=color:#0ff;font-weight:700>&#39;gene_id&#39;</span> in attribute:
</span></span><span style=display:flex><span>            attr_2 = attribute
</span></span><span style=display:flex><span>        <span style=color:#fff;font-weight:700>elif</span> <span style=color:#0ff;font-weight:700>&#39;gene_name&#39;</span> in attribute:
</span></span><span style=display:flex><span>            attr_3 = attribute
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span>    attribute_1.append(attr_1)
</span></span><span style=display:flex><span>    attribute_2.append(attr_2)
</span></span><span style=display:flex><span>    attribute_3.append(attr_3)
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span>gtf_df[<span style=color:#0ff;font-weight:700>&#39;attribute.1&#39;</span>] = attribute_1
</span></span><span style=display:flex><span>gtf_df[<span style=color:#0ff;font-weight:700>&#39;attribute.2&#39;</span>] = attribute_2
</span></span><span style=display:flex><span>gtf_df[<span style=color:#0ff;font-weight:700>&#39;attribute.3&#39;</span>] = attribute_3
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span>annot_df = gtf_df.iloc[:, -<span style=color:#ff0;font-weight:700>4</span>:].copy()
</span></span><span style=display:flex><span>annot_df[<span style=color:#0ff;font-weight:700>&#39;transcript_id&#39;</span>] = annot_df[<span style=color:#0ff;font-weight:700>&#39;attribute.1&#39;</span>].str.split(<span style=color:#0ff;font-weight:700>&#39;&#34;&#39;</span>).str[<span style=color:#ff0;font-weight:700>1</span>]
</span></span><span style=display:flex><span>annot_df[<span style=color:#0ff;font-weight:700>&#39;gene_id&#39;</span>] = annot_df[<span style=color:#0ff;font-weight:700>&#39;attribute.2&#39;</span>].str.split(<span style=color:#0ff;font-weight:700>&#39;&#34;&#39;</span>).str[<span style=color:#ff0;font-weight:700>1</span>]
</span></span><span style=display:flex><span>annot_df[<span style=color:#0ff;font-weight:700>&#39;gene_name&#39;</span>] = annot_df[<span style=color:#0ff;font-weight:700>&#39;attribute.3&#39;</span>].str.split(<span style=color:#0ff;font-weight:700>&#39;&#34;&#39;</span>).str[<span style=color:#ff0;font-weight:700>1</span>]
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span>annot_df
</span></span></code></pre></div><p><img src=https://github.com/user-attachments/assets/b172aa0a-c287-4d88-a27e-4483c9906cf4 alt=image></p><div class=highlight><pre tabindex=0 style=color:#e5e5e5;background-color:#000;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-python data-lang=python><span style=display:flex><span>annot_table = annot_df.iloc[:, -<span style=color:#ff0;font-weight:700>3</span>:].copy()
</span></span><span style=display:flex><span>annot_table_dedup = annot_table[~annot_table[<span style=color:#0ff;font-weight:700>&#39;transcript_id&#39;</span>].duplicated(keep=<span style=color:#0ff;font-weight:700>&#39;first&#39;</span>)]
</span></span><span style=display:flex><span>annot = annot_table_dedup.iloc[:, -<span style=color:#ff0;font-weight:700>3</span>:].copy()
</span></span><span style=display:flex><span>annot = annot.sort_values(by=<span style=color:#0ff;font-weight:700>&#39;transcript_id&#39;</span>)
</span></span><span style=display:flex><span>annot.head(<span style=color:#ff0;font-weight:700>50</span>)
</span></span></code></pre></div><p><img src=https://github.com/user-attachments/assets/42bca1cb-51be-4486-83dc-c6b31a73fc5c alt=image></p><h3 id=save-1>Save
<a class=anchor href=#save-1>#</a></h3><div class=highlight><pre tabindex=0 style=color:#e5e5e5;background-color:#000;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-python data-lang=python><span style=display:flex><span><span style=color:#007f7f>#annot.to_csv(&#34;annot.tsv&#34;,sep=&#39;\t&#39;,index=False)</span>
</span></span></code></pre></div><ul><li>나중에 또 써야대니깐 저장.</li></ul><h3 id=load-count>Load Count
<a class=anchor href=#load-count>#</a></h3><div class=highlight><pre tabindex=0 style=color:#e5e5e5;background-color:#000;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-python data-lang=python><span style=display:flex><span>count = pd.read_csv(<span style=color:#0ff;font-weight:700>&#34;count.tsv&#34;</span>,sep=<span style=color:#0ff;font-weight:700>&#39;</span><span style=color:#0ff;font-weight:700>\t</span><span style=color:#0ff;font-weight:700>&#39;</span>,encoding=<span style=color:#0ff;font-weight:700>&#39;cp949&#39;</span>)
</span></span><span style=display:flex><span>count
</span></span></code></pre></div><p><img src=https://github.com/user-attachments/assets/a0b6d6b0-682a-44af-9f30-20029fc91608 alt=image></p><ul><li>위에서 Rsubread로 만든 EBV count 가져옴</li></ul><h3 id=annotation>Annotation
<a class=anchor href=#annotation>#</a></h3><div class=highlight><pre tabindex=0 style=color:#e5e5e5;background-color:#000;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-python data-lang=python><span style=display:flex><span>merged = count.merge(annot, left_on=<span style=color:#0ff;font-weight:700>&#39;trans_id&#39;</span>, right_on=<span style=color:#0ff;font-weight:700>&#39;transcript_id&#39;</span>, how=<span style=color:#0ff;font-weight:700>&#39;left&#39;</span>)
</span></span><span style=display:flex><span>res_df = merged.drop(columns=[<span style=color:#0ff;font-weight:700>&#39;trans_id&#39;</span>])
</span></span><span style=display:flex><span>res_df = res_df[[<span style=color:#0ff;font-weight:700>&#39;transcript_id&#39;</span>, <span style=color:#0ff;font-weight:700>&#39;gene_id&#39;</span>, <span style=color:#0ff;font-weight:700>&#39;gene_name&#39;</span>, <span style=color:#0ff;font-weight:700>&#39;33-1&#39;</span>, <span style=color:#0ff;font-weight:700>&#39;33-2&#39;</span>, <span style=color:#0ff;font-weight:700>&#39;33-3&#39;</span>, <span style=color:#0ff;font-weight:700>&#39;150-1&#39;</span>, <span style=color:#0ff;font-weight:700>&#39;150-2&#39;</span>, <span style=color:#0ff;font-weight:700>&#39;150-3&#39;</span>, <span style=color:#0ff;font-weight:700>&#39;con-1&#39;</span>, <span style=color:#0ff;font-weight:700>&#39;con-2&#39;</span>, <span style=color:#0ff;font-weight:700>&#39;con-3&#39;</span>]]
</span></span><span style=display:flex><span>res_df
</span></span></code></pre></div><p><img src=https://github.com/user-attachments/assets/581ad2f0-9ec1-42d1-a150-3a36ea774fe3 alt=image></p><div class=highlight><pre tabindex=0 style=color:#e5e5e5;background-color:#000;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-python data-lang=python><span style=display:flex><span>res_df[<span style=color:#0ff;font-weight:700>&#39;gene_name&#39;</span>].value_counts()
</span></span></code></pre></div><p><img src=https://github.com/user-attachments/assets/ad94f072-1cfa-4a30-bbfc-a083b9efc252 alt=image></p><ul><li>중복인 transcript가 엄청많음</li></ul><div class=highlight><pre tabindex=0 style=color:#e5e5e5;background-color:#000;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-python data-lang=python><span style=display:flex><span>res_df[<span style=color:#0ff;font-weight:700>&#39;sum&#39;</span>] = res_df.iloc[:, <span style=color:#ff0;font-weight:700>3</span>:<span style=color:#ff0;font-weight:700>12</span>].sum(axis=<span style=color:#ff0;font-weight:700>1</span>)
</span></span><span style=display:flex><span>res_df = res_df.sort_values(by=<span style=color:#0ff;font-weight:700>&#39;sum&#39;</span>, ascending=<span style=color:#fff;font-weight:700>False</span>)
</span></span><span style=display:flex><span>res_df = res_df.drop_duplicates(subset=<span style=color:#0ff;font-weight:700>&#39;gene_name&#39;</span>, keep=<span style=color:#0ff;font-weight:700>&#39;first&#39;</span>)
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span>res_df = res_df.iloc[:, :-<span style=color:#ff0;font-weight:700>1</span>]
</span></span><span style=display:flex><span>res_df.reset_index(drop=<span style=color:#fff;font-weight:700>True</span>, inplace=<span style=color:#fff;font-weight:700>True</span>)
</span></span><span style=display:flex><span>res_df = res_df.dropna(subset=[<span style=color:#0ff;font-weight:700>&#39;gene_name&#39;</span>])
</span></span><span style=display:flex><span><span style=color:#fff;font-weight:700>print</span>(res_df.shape)
</span></span><span style=display:flex><span>res_df
</span></span></code></pre></div><p><img src=https://github.com/user-attachments/assets/f6412998-0d28-4dfd-8d10-263c610c40fb alt=image></p><ul><li>(63, 12)짜리에서 deduplication 결과 (15, 12)가 됏다.</li></ul><div class=highlight><pre tabindex=0 style=color:#e5e5e5;background-color:#000;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-python data-lang=python><span style=display:flex><span>res_df.to_csv(<span style=color:#0ff;font-weight:700>&#34;count.annot.tsv&#34;</span>, sep=<span style=color:#0ff;font-weight:700>&#39;</span><span style=color:#0ff;font-weight:700>\t</span><span style=color:#0ff;font-weight:700>&#39;</span>,index=<span style=color:#fff;font-weight:700>False</span>)
</span></span></code></pre></div><h3 id=merge>Merge
<a class=anchor href=#merge>#</a></h3><div class=highlight><pre tabindex=0 style=color:#e5e5e5;background-color:#000;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-python data-lang=python><span style=display:flex><span>count_hs = pd.read_csv(<span style=color:#0ff;font-weight:700>&#34;/data/home/ysh980101/2311/RNA-seq/count.csv&#34;</span>, encoding=<span style=color:#0ff;font-weight:700>&#39;cp949&#39;</span>)
</span></span><span style=display:flex><span>count_hs.columns = [<span style=color:#0ff;font-weight:700>&#34;transcript_id&#34;</span>, <span style=color:#0ff;font-weight:700>&#34;gene_id&#34;</span>, <span style=color:#0ff;font-weight:700>&#34;gene_name&#34;</span>, <span style=color:#0ff;font-weight:700>&#34;33-1&#34;</span>, <span style=color:#0ff;font-weight:700>&#34;33-2&#34;</span>, <span style=color:#0ff;font-weight:700>&#34;33-3&#34;</span>, <span style=color:#0ff;font-weight:700>&#34;150-1&#34;</span>, <span style=color:#0ff;font-weight:700>&#34;150-2&#34;</span>, <span style=color:#0ff;font-weight:700>&#34;150-3&#34;</span>, <span style=color:#0ff;font-weight:700>&#34;con-1&#34;</span>, <span style=color:#0ff;font-weight:700>&#34;con-2&#34;</span>, <span style=color:#0ff;font-weight:700>&#34;con-3&#34;</span>]
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span>combined_df = pd.concat([res_df, count_hs], axis=<span style=color:#ff0;font-weight:700>0</span>, ignore_index=<span style=color:#fff;font-weight:700>True</span>)
</span></span><span style=display:flex><span>combined_df
</span></span><span style=display:flex><span>combined_drop_df = combined_df[~(combined_df.iloc[:, <span style=color:#ff0;font-weight:700>4</span>:<span style=color:#ff0;font-weight:700>13</span>] == <span style=color:#ff0;font-weight:700>0</span>).all(axis=<span style=color:#ff0;font-weight:700>1</span>)] <span style=color:#007f7f>#remove 0 in all samples </span>
</span></span><span style=display:flex><span>combined_drop_df
</span></span></code></pre></div><p><img src=https://github.com/user-attachments/assets/873ee0af-9bfc-4d11-bae7-ad61df069553 alt=image></p><p><img src=https://github.com/user-attachments/assets/f6613d7c-464a-48f7-9c51-eb4b461c354d alt=image></p><div class=highlight><pre tabindex=0 style=color:#e5e5e5;background-color:#000;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-python data-lang=python><span style=display:flex><span>combined_drop_df.to_csv(<span style=color:#0ff;font-weight:700>&#34;count.annot.combined.drop.tsv&#34;</span>,sep=<span style=color:#0ff;font-weight:700>&#39;</span><span style=color:#0ff;font-weight:700>\t</span><span style=color:#0ff;font-weight:700>&#39;</span>,index=<span style=color:#fff;font-weight:700>False</span>)
</span></span></code></pre></div><h2 id=3-deg-analysis>3. DEG Analysis
<a class=anchor href=#3-deg-analysis>#</a></h2><h3 id=load-packages>Load Packages
<a class=anchor href=#load-packages>#</a></h3><div class=highlight><pre tabindex=0 style=color:#e5e5e5;background-color:#000;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-R data-lang=R><span style=display:flex><span>suppressMessages({
</span></span><span style=display:flex><span>    library(<span style=color:#0ff;font-weight:700>&#34;DESeq2&#34;</span>)
</span></span><span style=display:flex><span>    library(pheatmap)
</span></span><span style=display:flex><span>    library(withr)
</span></span><span style=display:flex><span>    library(RColorBrewer)
</span></span><span style=display:flex><span>    library(gplots)
</span></span><span style=display:flex><span>    library(ggplot2)
</span></span><span style=display:flex><span>    library(dplyr)
</span></span><span style=display:flex><span>    })
</span></span></code></pre></div><h3 id=set-path-1>Set Path
<a class=anchor href=#set-path-1>#</a></h3><div class=highlight><pre tabindex=0 style=color:#e5e5e5;background-color:#000;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-R data-lang=R><span style=display:flex><span>setwd(<span style=color:#0ff;font-weight:700>&#34;/data/home/ysh980101/2311/RNA-seq_ebv/Rsubread&#34;</span>)
</span></span><span style=display:flex><span>getwd()
</span></span></code></pre></div><div class=highlight><pre tabindex=0 style=color:#e5e5e5;background-color:#000;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-plain data-lang=plain><span style=display:flex><span>&#39;/data1/home/ysh980101/2311/RNA-seq_ebv/Rsubread&#39;
</span></span></code></pre></div><h3 id=run-deg>Run DEG
<a class=anchor href=#run-deg>#</a></h3><div class=highlight><pre tabindex=0 style=color:#e5e5e5;background-color:#000;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-R data-lang=R><span style=display:flex><span>S1 &lt;- <span style=color:#0ff;font-weight:700>&#39;33&#39;</span>
</span></span><span style=display:flex><span>S2 &lt;- <span style=color:#0ff;font-weight:700>&#39;150&#39;</span>
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span>countdata &lt;- read.csv(<span style=color:#0ff;font-weight:700>&#34;count.annot.combined.drop.tsv&#34;</span>, header=<span style=color:#fff;font-weight:700>TRUE</span>, sep=<span style=color:#0ff;font-weight:700>&#39;\t&#39;</span>)
</span></span><span style=display:flex><span>colnames(countdata) &lt;- c(<span style=color:#0ff;font-weight:700>&#39;transcript_id&#39;</span>,<span style=color:#0ff;font-weight:700>&#39;gene_id&#39;</span>,<span style=color:#0ff;font-weight:700>&#39;gene_name&#39;</span>,<span style=color:#0ff;font-weight:700>&#39;33-1&#39;</span>,<span style=color:#0ff;font-weight:700>&#39;33-2&#39;</span>,<span style=color:#0ff;font-weight:700>&#39;33-3&#39;</span>,<span style=color:#0ff;font-weight:700>&#39;150-1&#39;</span>,<span style=color:#0ff;font-weight:700>&#39;150-2&#39;</span>,<span style=color:#0ff;font-weight:700>&#39;150-3&#39;</span>,<span style=color:#0ff;font-weight:700>&#39;con-1&#39;</span>,<span style=color:#0ff;font-weight:700>&#39;con-2&#39;</span>,<span style=color:#0ff;font-weight:700>&#39;con-3&#39;</span>)
</span></span><span style=display:flex><span>countdata &lt;- countdata[, c(<span style=color:#0ff;font-weight:700>&#39;transcript_id&#39;</span>,<span style=color:#0ff;font-weight:700>&#39;gene_id&#39;</span>,<span style=color:#0ff;font-weight:700>&#39;gene_name&#39;</span>,<span style=color:#0ff;font-weight:700>&#39;150-1&#39;</span>,<span style=color:#0ff;font-weight:700>&#39;150-2&#39;</span>,<span style=color:#0ff;font-weight:700>&#39;150-3&#39;</span>,<span style=color:#0ff;font-weight:700>&#39;33-1&#39;</span>,<span style=color:#0ff;font-weight:700>&#39;33-2&#39;</span>,<span style=color:#0ff;font-weight:700>&#39;33-3&#39;</span>,<span style=color:#0ff;font-weight:700>&#39;con-1&#39;</span>,<span style=color:#0ff;font-weight:700>&#39;con-2&#39;</span>,<span style=color:#0ff;font-weight:700>&#39;con-3&#39;</span>)]
</span></span><span style=display:flex><span>countdata &lt;- countdata[, paste(c(<span style=color:#0ff;font-weight:700>&#39;gene_name&#39;</span>,paste0(S2,<span style=color:#0ff;font-weight:700>&#34;-1&#34;</span>), paste0(S2,<span style=color:#0ff;font-weight:700>&#34;-2&#34;</span>), paste0(S2,<span style=color:#0ff;font-weight:700>&#34;-3&#34;</span>),paste0(S1,<span style=color:#0ff;font-weight:700>&#34;-1&#34;</span>), paste0(S1,<span style=color:#0ff;font-weight:700>&#34;-2&#34;</span>), paste0(S1,<span style=color:#0ff;font-weight:700>&#34;-3&#34;</span>)), sep=<span style=color:#0ff;font-weight:700>&#34;&#34;</span>)]
</span></span><span style=display:flex><span>dim(countdata)
</span></span></code></pre></div><div class=highlight><pre tabindex=0 style=color:#e5e5e5;background-color:#000;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-plain data-lang=plain><span style=display:flex><span>16409 * 7
</span></span></code></pre></div><div class=highlight><pre tabindex=0 style=color:#e5e5e5;background-color:#000;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-R data-lang=R><span style=display:flex><span>sample.names &lt;- paste(c(paste0(S2,<span style=color:#0ff;font-weight:700>&#34;-1&#34;</span>), paste0(S2,<span style=color:#0ff;font-weight:700>&#34;-2&#34;</span>), paste0(S2,<span style=color:#0ff;font-weight:700>&#34;-3&#34;</span>),paste0(S1,<span style=color:#0ff;font-weight:700>&#34;-1&#34;</span>), paste0(S1,<span style=color:#0ff;font-weight:700>&#34;-2&#34;</span>), paste0(S1,<span style=color:#0ff;font-weight:700>&#34;-3&#34;</span>)), sep=<span style=color:#0ff;font-weight:700>&#34;&#34;</span>)
</span></span><span style=display:flex><span>conditions &lt;- factor(c(S2,S2,S2,S1,S1,S1))
</span></span><span style=display:flex><span>metadata &lt;- data.frame(Sample = sample.names, group = conditions)
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span>N &lt;- dim(countdata)[[2]]
</span></span><span style=display:flex><span>cData = countdata[,<span style=color:#ff0;font-weight:700>2</span>:N]
</span></span><span style=display:flex><span>GeneID = countdata[,<span style=color:#ff0;font-weight:700>1</span>]
</span></span><span style=display:flex><span>rownames(cData) = GeneID
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span>dds &lt;- DESeqDataSetFromMatrix(countData = cData, colData = metadata, design = ~group)
</span></span><span style=display:flex><span>dds$group &lt;- relevel(dds$group, ref = S1)
</span></span><span style=display:flex><span>dds &lt;- DESeq(dds)  
</span></span><span style=display:flex><span>vsd &lt;- vst(dds, blind = <span style=color:#fff;font-weight:700>FALSE</span>)
</span></span><span style=display:flex><span>rld &lt;- rlogTransformation(dds, blind = <span style=color:#fff;font-weight:700>FALSE</span>)
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span>res &lt;- results(dds, contrast = c(<span style=color:#0ff;font-weight:700>&#34;group&#34;</span>, S2, S1))
</span></span><span style=display:flex><span>res_tbl &lt;- as.data.frame(res)
</span></span><span style=display:flex><span>res_tbl$GeneID &lt;- rownames(res_tbl)
</span></span><span style=display:flex><span>res_tbl &lt;- res_tbl[order(res_tbl$padj), ]
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span>NM_no_NA &lt;- na.omit(res)
</span></span><span style=display:flex><span>res_cut &lt;- NM_no_NA[NM_no_NA$padj &lt; <span style=color:#ff0;font-weight:700>0.05</span>, ]
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span>lfc &lt;- <span style=color:#ff0;font-weight:700>1.5</span>
</span></span><span style=display:flex><span>res_cut_2 &lt;- res_cut[abs(res_cut$log2FoldChange) &gt; lfc, ]
</span></span><span style=display:flex><span>res_cut_2_pos &lt;- res_cut_2[res_cut_2$log2FoldChange &gt; <span style=color:#ff0;font-weight:700>0</span>, ]
</span></span><span style=display:flex><span>res_cut_2_neg &lt;- res_cut_2[res_cut_2$log2FoldChange &lt; <span style=color:#ff0;font-weight:700>0</span>, ]
</span></span></code></pre></div><div class=highlight><pre tabindex=0 style=color:#e5e5e5;background-color:#000;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-R data-lang=R><span style=display:flex><span><span style=color:#007f7f># Volcano Plot</span>
</span></span><span style=display:flex><span>par(mfrow = c(<span style=color:#ff0;font-weight:700>1</span>, <span style=color:#ff0;font-weight:700>1</span>))
</span></span><span style=display:flex><span>with(NM_no_NA, plot(log2FoldChange, -log10(pvalue), pch = <span style=color:#ff0;font-weight:700>20</span>, col = <span style=color:#0ff;font-weight:700>&#39;gray&#39;</span>,
</span></span><span style=display:flex><span>                    main = paste0(<span style=color:#0ff;font-weight:700>&#39;Volcano plot_between\n&#39;</span>, S2, <span style=color:#0ff;font-weight:700>&#39; vs &#39;</span>, S1),
</span></span><span style=display:flex><span>                    xlim = c(<span style=color:#ff0;font-weight:700>-3</span>, <span style=color:#ff0;font-weight:700>3</span>)))
</span></span><span style=display:flex><span>abline(h = -log10(<span style=color:#ff0;font-weight:700>0.05</span>), v = c(-lfc, lfc), col = <span style=color:#0ff;font-weight:700>&#34;gray&#34;</span>, lty = <span style=color:#ff0;font-weight:700>2</span>)
</span></span><span style=display:flex><span>with(res_cut_2_neg, points(log2FoldChange, -log10(pvalue), pch = <span style=color:#ff0;font-weight:700>20</span>, col = <span style=color:#0ff;font-weight:700>&#39;blue&#39;</span>))
</span></span><span style=display:flex><span>with(res_cut_2_pos, points(log2FoldChange, -log10(pvalue), pch = <span style=color:#ff0;font-weight:700>20</span>, col = <span style=color:#0ff;font-weight:700>&#39;gold&#39;</span>))
</span></span><span style=display:flex><span>legend(<span style=color:#0ff;font-weight:700>&#34;topright&#34;</span>, legend = c(paste0(<span style=color:#0ff;font-weight:700>&#34;FC ≥ &#34;</span>, lfc, <span style=color:#0ff;font-weight:700>&#34; &amp; p &lt; 0.05&#34;</span>),
</span></span><span style=display:flex><span>                              paste0(<span style=color:#0ff;font-weight:700>&#34;FC ≤ -&#34;</span>, lfc, <span style=color:#0ff;font-weight:700>&#34; &amp; p &lt; 0.05&#34;</span>)),
</span></span><span style=display:flex><span>       col = c(<span style=color:#0ff;font-weight:700>&#34;gold&#34;</span>, <span style=color:#0ff;font-weight:700>&#34;blue&#34;</span>), pch = <span style=color:#ff0;font-weight:700>20</span>)
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span><span style=color:#007f7f># Bar Plot</span>
</span></span><span style=display:flex><span>lengths &lt;- c(nrow(res_cut_2_neg), nrow(res_cut_2_pos))
</span></span><span style=display:flex><span>bp &lt;- barplot(lengths, space = <span style=color:#ff0;font-weight:700>0</span>, col = c(<span style=color:#0ff;font-weight:700>&#34;blue&#34;</span>, <span style=color:#0ff;font-weight:700>&#34;gold&#34;</span>),
</span></span><span style=display:flex><span>              horiz = <span style=color:#fff;font-weight:700>TRUE</span>, xlab = <span style=color:#0ff;font-weight:700>&#34;Count of genes&#34;</span>, ylab = paste0(S2, <span style=color:#0ff;font-weight:700>&#34;/&#34;</span>, S1),
</span></span><span style=display:flex><span>              xlim = c(<span style=color:#ff0;font-weight:700>0</span>, sum(lengths)), width = <span style=color:#ff0;font-weight:700>0.02</span>, las = <span style=color:#ff0;font-weight:700>1</span>)
</span></span><span style=display:flex><span>title(paste0(<span style=color:#0ff;font-weight:700>&#34;UP, DOWN regulated count (|FC| ≥ &#34;</span>, lfc, <span style=color:#0ff;font-weight:700>&#34;)&#34;</span>))
</span></span><span style=display:flex><span>legend(<span style=color:#0ff;font-weight:700>&#34;topright&#34;</span>, legend = c(<span style=color:#0ff;font-weight:700>&#34;UP&#34;</span>, <span style=color:#0ff;font-weight:700>&#34;DOWN&#34;</span>), fill = c(<span style=color:#0ff;font-weight:700>&#34;gold&#34;</span>, <span style=color:#0ff;font-weight:700>&#34;blue&#34;</span>))
</span></span><span style=display:flex><span>text(lengths, bp, labels = lengths, pos = <span style=color:#ff0;font-weight:700>4</span>, cex = <span style=color:#ff0;font-weight:700>0.9</span>)
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span><span style=color:#007f7f># Save</span>
</span></span><span style=display:flex><span>cutoff &lt;- <span style=color:#ff0;font-weight:700>0.05</span>
</span></span><span style=display:flex><span>sig_res &lt;- dplyr::filter(res_tbl, padj &lt; cutoff) %&gt;%
</span></span><span style=display:flex><span>  dplyr::arrange(padj) %&gt;%
</span></span><span style=display:flex><span>  dplyr::select(GeneID, baseMean, log2FoldChange, lfcSE, stat, pvalue, padj)
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span>write.csv(sig_res, file = paste0(<span style=color:#0ff;font-weight:700>&#39;../DESeq2/res.&#39;</span>, S2, <span style=color:#0ff;font-weight:700>&#39;vs&#39;</span>, S1, <span style=color:#0ff;font-weight:700>&#39;.csv&#39;</span>), row.names = <span style=color:#fff;font-weight:700>FALSE</span>)
</span></span></code></pre></div><p><img src=https://github.com/user-attachments/assets/2e32b5e8-7d44-4b2d-80dd-54106bc300cf alt=image></p><p><img src=https://github.com/user-attachments/assets/d3fa3953-36b0-4a7e-946e-3a24a50a4891 alt=image></p><p>cf) Heatmap 시각화</p><div class=highlight><pre tabindex=0 style=color:#e5e5e5;background-color:#000;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-R data-lang=R><span style=display:flex><span><span style=color:#007f7f># Heatmap</span>
</span></span><span style=display:flex><span>sig_idx &lt;- res$padj &lt; cutoff &amp; !is.na(res$padj)
</span></span><span style=display:flex><span>sig_dat &lt;- assay(rld)[sig_idx, ]
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span>annC &lt;- data.frame(condition = conditions)
</span></span><span style=display:flex><span>rownames(annC) &lt;- colnames(sig_dat)
</span></span><span style=display:flex><span>ann_colors &lt;- list(condition = setNames(c(<span style=color:#0ff;font-weight:700>&#34;#F7819F&#34;</span>, <span style=color:#0ff;font-weight:700>&#34;#58D3F7&#34;</span>), c(S2, S1)))
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span>pheatmap(sig_dat,
</span></span><span style=display:flex><span>         scale = <span style=color:#0ff;font-weight:700>&#34;row&#34;</span>,
</span></span><span style=display:flex><span>         fontsize_row = <span style=color:#ff0;font-weight:700>9</span>,
</span></span><span style=display:flex><span>         annotation_col = annC,
</span></span><span style=display:flex><span>         color = rev(brewer.pal(<span style=color:#ff0;font-weight:700>6</span>, <span style=color:#0ff;font-weight:700>&#34;RdYlGn&#34;</span>)),
</span></span><span style=display:flex><span>         annotation_colors = ann_colors,
</span></span><span style=display:flex><span>         show_rownames = <span style=color:#fff;font-weight:700>FALSE</span>,
</span></span><span style=display:flex><span>         show_colnames = <span style=color:#fff;font-weight:700>TRUE</span>)
</span></span></code></pre></div><p><img src=https://github.com/user-attachments/assets/67e7e43f-4db6-48c2-9fed-fc03dfb4e892 alt=image></p></article><footer class=book-footer><div class="flex flex-wrap justify-between"></div><script>(function(){function e(e){const t=window.getSelection(),n=document.createRange();n.selectNodeContents(e),t.removeAllRanges(),t.addRange(n)}document.querySelectorAll("pre code").forEach(t=>{t.addEventListener("click",function(){if(window.getSelection().toString())return;e(t.parentElement),navigator.clipboard&&navigator.clipboard.writeText(t.parentElement.textContent)})})})()</script></footer><div class=book-comments><script src=https://giscus.app/client.js data-repo=yshghid/yshghid.github.io data-repo-id=R_kgDONkMkNg data-category-id=DIC_kwDONkMkNs4CloJh data-mapping=pathname data-strict=0 data-reactions-enabled=1 data-emit-metadata=0 data-input-position=bottom data-theme=preferred_color_scheme data-lang=ko crossorigin=anonymous async></script></div><label for=menu-control class="hidden book-menu-overlay"></label></div><aside class=book-toc><div class=book-toc-content><nav id=TableOfContents><ul><li><a href=#1-alignment>1. Alignment</a><ul><li><a href=#load-package-set-path>Load package, Set Path</a></li><li><a href=#build-index>Build Index</a></li><li><a href=#feature-count>Feature Count</a></li><li><a href=#save>Save</a></li></ul></li><li><a href=#2-annotation>2. Annotation</a><ul><li><a href=#load-package>Load Package</a></li><li><a href=#set-path>Set Path</a></li><li><a href=#load-gtf-file>Load GTF file</a></li><li><a href=#make-annotation-file>Make annotation file</a></li><li><a href=#save-1>Save</a></li><li><a href=#load-count>Load Count</a></li><li><a href=#annotation>Annotation</a></li><li><a href=#merge>Merge</a></li></ul></li><li><a href=#3-deg-analysis>3. DEG Analysis</a><ul><li><a href=#load-packages>Load Packages</a></li><li><a href=#set-path-1>Set Path</a></li><li><a href=#run-deg>Run DEG</a></li></ul></li></ul></nav></div></aside></main></body></html>