<!doctype html><html lang=en-us dir=ltr><head><meta charset=UTF-8><meta name=viewport content="width=device-width,initial-scale=1"><meta name=description content='
  ChIP-seq 전처리
  #

#2025-04-21


  1. Trimming
  #

#chipseq_trimming.sh
#!/bin/bash

# setting envs
export bdir="/data3/projects/2022_KNU_EBV"
export hg38_bowtieidx="/data3/PUBLIC_DATA/ref_genomes/homo_sapiens/hg38/hg38_bowtie_idx/hg38.fa"
export hg38_bwaidx="/data3/PUBLIC_DATA/ref_genomes/homo_sapiens/hg38/hg38_bwa_index/hg38.fa"
export ebv_bowtie2idx="/data3/PUBLIC_DATA/ref_genomes/Human_gammaherpesvirus_4_EBV/EBV_bowtie2_idx/NC_007605.1.fa"
export ebv_bwaidx="/data3/PUBLIC_DATA/ref_genomes/Human_gammaherpesvirus_4_EBV/EBV_bwa_index/NC_007605.1.fa"

### SET Path ###
cd /data3/RAW_DATA/2023_KNU_EBV/ChIP-seq

### TRIMMING data ###
mkdir -p trimmed
sampdir="/data3/RAW_DATA/2023_KNU_EBV/ChIP-seq"
samplist=("Input" "p65" "RIgG")
TRIMMOMATIC= "/data/packages/trimmomatic/Trimmomatic-0.39/trimmomatic-0.39.jar"

for sampname in "${samplist[@]}"; do
    mkdir -p "trimmed/${sampname}"
    java -jar $TRIMMOMATIC PE -threads 40 -trimlog log1.txt $sampdir/${sampname}_1.fastq/${sampname}_1.fastq $sampdir/${sampname}_2.fastq/${sampname}_2.fastq $sampdir/trimmed/${sampname}/${sampname}_1.trimmed.fastq $sampdir/trimmed/${sampname}/${sampname}_1.up.trimmed.fastq $sampdir/trimmed/${sampname}/${sampname}_2.trimmed.fastq $sampdir/trimmed/${sampname}/${sampname}_2.up.trimmed.fastq ILLUMINACLIP:/data1/packages/trimmomatic/Trimmomatic-0.39/adapters/TruSeq3-PE.fa:2:30:10:2:True LEADING:3 TRAILING:3 MINLEN:36
done

  2. Alignment
  #

#chipseq_alignment.sh'><meta name=theme-color media="(prefers-color-scheme: light)" content="#ffffff"><meta name=theme-color media="(prefers-color-scheme: dark)" content="#343a40"><meta name=color-scheme content="light dark"><meta property="og:url" content="https://yshghid.github.io/docs/study/bioinformatics/bi5/"><meta property="og:site_name" content=" "><meta property="og:title" content="ChIP-seq 전처리"><meta property="og:description" content='ChIP-seq 전처리 # #2025-04-21
1. Trimming # #chipseq_trimming.sh
#!/bin/bash # setting envs export bdir="/data3/projects/2022_KNU_EBV" export hg38_bowtieidx="/data3/PUBLIC_DATA/ref_genomes/homo_sapiens/hg38/hg38_bowtie_idx/hg38.fa" export hg38_bwaidx="/data3/PUBLIC_DATA/ref_genomes/homo_sapiens/hg38/hg38_bwa_index/hg38.fa" export ebv_bowtie2idx="/data3/PUBLIC_DATA/ref_genomes/Human_gammaherpesvirus_4_EBV/EBV_bowtie2_idx/NC_007605.1.fa" export ebv_bwaidx="/data3/PUBLIC_DATA/ref_genomes/Human_gammaherpesvirus_4_EBV/EBV_bwa_index/NC_007605.1.fa" ### SET Path ### cd /data3/RAW_DATA/2023_KNU_EBV/ChIP-seq ### TRIMMING data ### mkdir -p trimmed sampdir="/data3/RAW_DATA/2023_KNU_EBV/ChIP-seq" samplist=("Input" "p65" "RIgG") TRIMMOMATIC= "/data/packages/trimmomatic/Trimmomatic-0.39/trimmomatic-0.39.jar" for sampname in "${samplist[@]}"; do mkdir -p "trimmed/${sampname}" java -jar $TRIMMOMATIC PE -threads 40 -trimlog log1.txt $sampdir/${sampname}_1.fastq/${sampname}_1.fastq $sampdir/${sampname}_2.fastq/${sampname}_2.fastq $sampdir/trimmed/${sampname}/${sampname}_1.trimmed.fastq $sampdir/trimmed/${sampname}/${sampname}_1.up.trimmed.fastq $sampdir/trimmed/${sampname}/${sampname}_2.trimmed.fastq $sampdir/trimmed/${sampname}/${sampname}_2.up.trimmed.fastq ILLUMINACLIP:/data1/packages/trimmomatic/Trimmomatic-0.39/adapters/TruSeq3-PE.fa:2:30:10:2:True LEADING:3 TRAILING:3 MINLEN:36 done 2. Alignment # #chipseq_alignment.sh'><meta property="og:locale" content="en_us"><meta property="og:type" content="article"><meta property="article:section" content="docs"><meta property="article:published_time" content="2025-04-21T00:00:00+00:00"><meta property="article:modified_time" content="2025-04-21T00:00:00+00:00"><meta property="article:tag" content="2025-04"><title>ChIP-seq 전처리 |</title><link rel=icon href=/favicon.png><link rel=manifest href=/manifest.json><link rel=canonical href=https://yshghid.github.io/docs/study/bioinformatics/bi5/><link rel=stylesheet href=/book.min.6217d077edb4189fd0578345e84bca1a884dfdee121ff8dc9a0f55cfe0852bc9.css integrity="sha256-YhfQd+20GJ/QV4NF6EvKGohN/e4SH/jcmg9Vz+CFK8k=" crossorigin=anonymous><script defer src=/fuse.min.js></script><script defer src=/en.search.min.54e4c0b057998e61fb2d3ae36df6a7b1d10a71ee48518c71a9370a2a8322eeb9.js integrity="sha256-VOTAsFeZjmH7LTrjbfansdEKce5IUYxxqTcKKoMi7rk=" crossorigin=anonymous></script></head><body dir=ltr><input type=checkbox class="hidden toggle" id=menu-control>
<input type=checkbox class="hidden toggle" id=toc-control><main class="container flex"><aside class=book-menu><div class=book-menu-content><nav><h2 class=book-brand><a class="flex align-center" href=/><img src=/logo.png alt=Logo class=book-icon><span></span></a></h2><div class="book-search hidden"><input type=text id=book-search-input placeholder=Search aria-label=Search maxlength=64 data-hotkeys=s/><div class="book-search-spinner hidden"></div><ul id=book-search-results></ul></div><script>document.querySelector(".book-search").classList.remove("hidden")</script><ul><li class=book-section-flat><span>기록</span><ul><li><a href=/docs/hobby/book/>글</a><ul></ul></li><li><a href=/docs/hobby/daily/>일상</a><ul></ul></li><li><a href=/docs/hobby/music/>음악</a><ul></ul></li><li><a href=/docs/hobby/youtube/>ott</a><ul></ul></li><li><a href=/docs/hobby/movie/>영화</a><ul></ul></li><li><a href=/docs/hobby/baking/>베이킹</a><ul></ul></li></ul></li><li class=book-section-flat><span>공부</span><ul><li><a href=/docs/study/ai/>AI</a><ul></ul></li><li><a href=/docs/study/sw/>SW</a><ul></ul></li><li><a href=/docs/study/bioinformatics/>Bioinformatics</a><ul></ul></li><li><a href=/docs/study/devops/>DevOps</a><ul></ul></li><li><a href=/docs/study/algorithm/>코테</a><ul></ul></li><li><a href=/docs/study/career/>취업</a><ul></ul></li></ul></li></ul></nav><script>(function(){var e=document.querySelector("aside .book-menu-content");addEventListener("beforeunload",function(){localStorage.setItem("menu.scrollTop",e.scrollTop)}),e.scrollTop=localStorage.getItem("menu.scrollTop")})()</script></div></aside><div class=book-page><header class=book-header><div class="flex align-center justify-between"><label for=menu-control><img src=/svg/menu.svg class=book-icon alt=Menu></label><h3>ChIP-seq 전처리</h3><label for=toc-control><img src=/svg/toc.svg class=book-icon alt="Table of Contents"></label></div><aside class="hidden clearfix"><nav id=TableOfContents><ul><li><ul><li><a href=#1-trimming>1. Trimming</a></li><li><a href=#2-alignment>2. Alignment</a></li><li><a href=#3-peak-calling>3. Peak Calling</a></li></ul></li></ul></nav></aside></header><article class="markdown book-article"><h1 id=chip-seq-전처리>ChIP-seq 전처리
<a class=anchor href=#chip-seq-%ec%a0%84%ec%b2%98%eb%a6%ac>#</a></h1><p>#2025-04-21</p><hr><h3 id=1-trimming>1. Trimming
<a class=anchor href=#1-trimming>#</a></h3><p>#chipseq_trimming.sh</p><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-bash data-lang=bash><span style=display:flex><span><span style=color:#75715e>#!/bin/bash
</span></span></span><span style=display:flex><span><span style=color:#75715e></span>
</span></span><span style=display:flex><span><span style=color:#75715e># setting envs</span>
</span></span><span style=display:flex><span>export bdir<span style=color:#f92672>=</span><span style=color:#e6db74>&#34;/data3/projects/2022_KNU_EBV&#34;</span>
</span></span><span style=display:flex><span>export hg38_bowtieidx<span style=color:#f92672>=</span><span style=color:#e6db74>&#34;/data3/PUBLIC_DATA/ref_genomes/homo_sapiens/hg38/hg38_bowtie_idx/hg38.fa&#34;</span>
</span></span><span style=display:flex><span>export hg38_bwaidx<span style=color:#f92672>=</span><span style=color:#e6db74>&#34;/data3/PUBLIC_DATA/ref_genomes/homo_sapiens/hg38/hg38_bwa_index/hg38.fa&#34;</span>
</span></span><span style=display:flex><span>export ebv_bowtie2idx<span style=color:#f92672>=</span><span style=color:#e6db74>&#34;/data3/PUBLIC_DATA/ref_genomes/Human_gammaherpesvirus_4_EBV/EBV_bowtie2_idx/NC_007605.1.fa&#34;</span>
</span></span><span style=display:flex><span>export ebv_bwaidx<span style=color:#f92672>=</span><span style=color:#e6db74>&#34;/data3/PUBLIC_DATA/ref_genomes/Human_gammaherpesvirus_4_EBV/EBV_bwa_index/NC_007605.1.fa&#34;</span>
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span><span style=color:#75715e>### SET Path ###</span>
</span></span><span style=display:flex><span>cd /data3/RAW_DATA/2023_KNU_EBV/ChIP-seq
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span><span style=color:#75715e>### TRIMMING data ###</span>
</span></span><span style=display:flex><span>mkdir -p trimmed
</span></span><span style=display:flex><span>sampdir<span style=color:#f92672>=</span><span style=color:#e6db74>&#34;/data3/RAW_DATA/2023_KNU_EBV/ChIP-seq&#34;</span>
</span></span><span style=display:flex><span>samplist<span style=color:#f92672>=(</span><span style=color:#e6db74>&#34;Input&#34;</span> <span style=color:#e6db74>&#34;p65&#34;</span> <span style=color:#e6db74>&#34;RIgG&#34;</span><span style=color:#f92672>)</span>
</span></span><span style=display:flex><span>TRIMMOMATIC<span style=color:#f92672>=</span> <span style=color:#e6db74>&#34;/data/packages/trimmomatic/Trimmomatic-0.39/trimmomatic-0.39.jar&#34;</span>
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span><span style=color:#66d9ef>for</span> sampname in <span style=color:#e6db74>&#34;</span><span style=color:#e6db74>${</span>samplist[@]<span style=color:#e6db74>}</span><span style=color:#e6db74>&#34;</span>; <span style=color:#66d9ef>do</span>
</span></span><span style=display:flex><span>    mkdir -p <span style=color:#e6db74>&#34;trimmed/</span><span style=color:#e6db74>${</span>sampname<span style=color:#e6db74>}</span><span style=color:#e6db74>&#34;</span>
</span></span><span style=display:flex><span>    java -jar $TRIMMOMATIC PE -threads <span style=color:#ae81ff>40</span> -trimlog log1.txt $sampdir/<span style=color:#e6db74>${</span>sampname<span style=color:#e6db74>}</span>_1.fastq/<span style=color:#e6db74>${</span>sampname<span style=color:#e6db74>}</span>_1.fastq $sampdir/<span style=color:#e6db74>${</span>sampname<span style=color:#e6db74>}</span>_2.fastq/<span style=color:#e6db74>${</span>sampname<span style=color:#e6db74>}</span>_2.fastq $sampdir/trimmed/<span style=color:#e6db74>${</span>sampname<span style=color:#e6db74>}</span>/<span style=color:#e6db74>${</span>sampname<span style=color:#e6db74>}</span>_1.trimmed.fastq $sampdir/trimmed/<span style=color:#e6db74>${</span>sampname<span style=color:#e6db74>}</span>/<span style=color:#e6db74>${</span>sampname<span style=color:#e6db74>}</span>_1.up.trimmed.fastq $sampdir/trimmed/<span style=color:#e6db74>${</span>sampname<span style=color:#e6db74>}</span>/<span style=color:#e6db74>${</span>sampname<span style=color:#e6db74>}</span>_2.trimmed.fastq $sampdir/trimmed/<span style=color:#e6db74>${</span>sampname<span style=color:#e6db74>}</span>/<span style=color:#e6db74>${</span>sampname<span style=color:#e6db74>}</span>_2.up.trimmed.fastq ILLUMINACLIP:/data1/packages/trimmomatic/Trimmomatic-0.39/adapters/TruSeq3-PE.fa:2:30:10:2:True LEADING:3 TRAILING:3 MINLEN:36
</span></span><span style=display:flex><span><span style=color:#66d9ef>done</span>
</span></span></code></pre></div><h3 id=2-alignment>2. Alignment
<a class=anchor href=#2-alignment>#</a></h3><p>#chipseq_alignment.sh</p><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-bash data-lang=bash><span style=display:flex><span><span style=color:#75715e>#!/bin/bash
</span></span></span><span style=display:flex><span><span style=color:#75715e></span>
</span></span><span style=display:flex><span><span style=color:#75715e># setting envs</span>
</span></span><span style=display:flex><span>export bdir<span style=color:#f92672>=</span><span style=color:#e6db74>&#34;/data3/projects/2022_KNU_EBV&#34;</span>
</span></span><span style=display:flex><span>export hg38_bowtieidx<span style=color:#f92672>=</span><span style=color:#e6db74>&#34;/data3/PUBLIC_DATA/ref_genomes/homo_sapiens/hg38/hg38_bowtie_idx/hg38.fa&#34;</span>
</span></span><span style=display:flex><span>export hg38_bwaidx<span style=color:#f92672>=</span><span style=color:#e6db74>&#34;/data3/PUBLIC_DATA/ref_genomes/homo_sapiens/hg38/hg38_bwa_index/hg38.fa&#34;</span>
</span></span><span style=display:flex><span>export ebv_bowtie2idx<span style=color:#f92672>=</span><span style=color:#e6db74>&#34;/data3/PUBLIC_DATA/ref_genomes/Human_gammaherpesvirus_4_EBV/EBV_bowtie2_idx/NC_007605.1.fa&#34;</span>
</span></span><span style=display:flex><span>export ebv_bwaidx<span style=color:#f92672>=</span><span style=color:#e6db74>&#34;/data3/PUBLIC_DATA/ref_genomes/Human_gammaherpesvirus_4_EBV/EBV_bwa_index/NC_007605.1.fa&#34;</span>
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span><span style=color:#75715e>### SET Path ###</span>
</span></span><span style=display:flex><span>cd /data3/RAW_DATA/2023_KNU_EBV/ChIP-seq
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span>sampdir<span style=color:#f92672>=</span><span style=color:#e6db74>&#34;/data3/RAW_DATA/2023_KNU_EBV/ChIP-seq&#34;</span>
</span></span><span style=display:flex><span>samplist<span style=color:#f92672>=(</span><span style=color:#e6db74>&#34;Input&#34;</span> <span style=color:#e6db74>&#34;p65&#34;</span> <span style=color:#e6db74>&#34;RIgG&#34;</span><span style=color:#f92672>)</span>
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span><span style=color:#66d9ef>for</span> sampname in <span style=color:#e6db74>&#34;</span><span style=color:#e6db74>${</span>samplist[@]<span style=color:#e6db74>}</span><span style=color:#e6db74>&#34;</span>; <span style=color:#66d9ef>do</span>
</span></span><span style=display:flex><span>    bwa mem -t <span style=color:#ae81ff>20</span> -v <span style=color:#ae81ff>2</span> $ebv_bwaidx $bdir/trimmed/CTCF-C-SNU719_1.trimmed.fastq $bdir/trimmed/CTCF-C-SNU719_2.trimmed.fastq &gt; $bdir/aln/bwa/CTCF-C_PE.bwa.sam
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span><span style=color:#66d9ef>done</span>
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span><span style=color:#75715e>### Aligning to EBV - PE</span>
</span></span><span style=display:flex><span>bwa mem -t <span style=color:#ae81ff>20</span> -v <span style=color:#ae81ff>2</span> $ebv_bwaidx $bdir/trimmed/CTCF-C-SNU719_1.trimmed.fastq $bdir/trimmed/CTCF-C-SNU719_2.trimmed.fastq &gt; $bdir/aln/bwa/CTCF-C_PE.bwa.sam
</span></span><span style=display:flex><span>samtools view -hf <span style=color:#ae81ff>2</span> $bdir/aln/bwa/CTCF-C_PE.bwa.sam | samtools sort -o $bdir/aln/bwa/CTCF-C_PE.bwa.bam -O BAM -@ <span style=color:#ae81ff>20</span> -
</span></span><span style=display:flex><span>samtools index -@ <span style=color:#ae81ff>20</span> $bdir/aln/bwa/CTCF-C_PE.bwa.bam
</span></span><span style=display:flex><span>bamCoverage -b $bdir/aln/bwa/CTCF-C_PE.bwa.bam -o $bdir/aln/bwa/CTCF-C_PE.bwa.bam.bigwig
</span></span><span style=display:flex><span>macs2 callpeak -t $bdir/aln/bwa/CTCF-C_PE.bwa.bam -f BAMPE -n CTCF-C --outdir peaks_ebv
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span><span style=color:#75715e>### @REF: hg38</span>
</span></span><span style=display:flex><span>bowtie2 -k1 --no-unal -p <span style=color:#ae81ff>40</span> --qc-filter -x $bowtie2idx_hg38 -1 $bdir/trimmed/CTCF-C-SNU719_1.trimmed.fastq -2 $bdir/trimmed/CTCF-C-SNU719_2.trimmed.fastq &gt; $bdir/aln/bwt2/CTCF-C_PE_hg38.sam &gt; $bdir/aln/bwt2/CTCF-C_PE_hg38.sam
</span></span><span style=display:flex><span>samtools view -hf <span style=color:#ae81ff>2</span> $bdir/aln/bwt2/CTCF-C_PE_hg38.sam | samtools sort -o $bdir/aln/bwt2/CTCF-C_PE_hg38.bam -O BAM -@ <span style=color:#ae81ff>20</span> -
</span></span><span style=display:flex><span>sambamba view -h -t <span style=color:#ae81ff>20</span> -f bam -p -F <span style=color:#e6db74>&#34;[XS] == null and not unmapped and not duplicate&#34;</span> $bdir/aln/bwt2/CTCF-C_PE_hg38.bam &gt; $bdir/aln/bwt2/CTCF-C_PE_hg38_u.bam
</span></span><span style=display:flex><span>samtools index -@ <span style=color:#ae81ff>20</span> $bdir/aln/bwt2/CTCF-C_PE_hg38_u.bam
</span></span><span style=display:flex><span>macs2 callpeak -t $bdir/aln/bwt2/CTCF-C_PE_hg38_u.bam -f BAMPE -n CTCF-C --outdir peaks
</span></span></code></pre></div><h3 id=3-peak-calling>3. Peak Calling
<a class=anchor href=#3-peak-calling>#</a></h3><p>#bedgraph.sh</p><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-bash data-lang=bash><span style=display:flex><span><span style=color:#75715e>#!/bin/bash
</span></span></span><span style=display:flex><span><span style=color:#75715e></span>
</span></span><span style=display:flex><span>input_file<span style=color:#f92672>=</span><span style=color:#e6db74>&#34;KEB01_1_bismark_bt2_pe.sorted.bedGraph.gz&#34;</span>
</span></span><span style=display:flex><span>output_file<span style=color:#f92672>=</span><span style=color:#e6db74>&#34;KEB01_1_bismark_bt2_pe.sorted_edited.bedGraph&#34;</span>
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span>cd /data/home/ysh980101/2309_5-aza/Bismark_temp_GRCh38
</span></span><span style=display:flex><span>zcat <span style=color:#e6db74>&#34;</span>$input_file<span style=color:#e6db74>&#34;</span> | awk <span style=color:#e6db74>&#39;$4 != 0 { $1 = &#34;chr&#34; $1; print }&#39;</span> | gzip &gt; <span style=color:#e6db74>&#34;</span>$output_file<span style=color:#e6db74>.gz&#34;</span>
</span></span></code></pre></div></article><footer class=book-footer><div class="flex flex-wrap justify-between"></div><script>(function(){function e(e){const t=window.getSelection(),n=document.createRange();n.selectNodeContents(e),t.removeAllRanges(),t.addRange(n)}document.querySelectorAll("pre code").forEach(t=>{t.addEventListener("click",function(){if(window.getSelection().toString())return;e(t.parentElement),navigator.clipboard&&navigator.clipboard.writeText(t.parentElement.textContent)})})})()</script></footer><div class=book-comments><script src=https://giscus.app/client.js data-repo=yshghid/yshghid.github.io data-repo-id=R_kgDONkMkNg data-category-id=DIC_kwDONkMkNs4CloJh data-mapping=pathname data-strict=0 data-reactions-enabled=1 data-emit-metadata=0 data-input-position=bottom data-theme=preferred_color_scheme data-lang=ko crossorigin=anonymous async></script></div><label for=menu-control class="hidden book-menu-overlay"></label></div><aside class=book-toc><div class=book-toc-content><nav id=TableOfContents><ul><li><ul><li><a href=#1-trimming>1. Trimming</a></li><li><a href=#2-alignment>2. Alignment</a></li><li><a href=#3-peak-calling>3. Peak Calling</a></li></ul></li></ul></nav></div></aside></main></body></html>