<!DOCTYPE html>
<html lang="en-us" dir="ltr">
<head><script src="/livereload.js?mindelay=10&amp;v=2&amp;port=1313&amp;path=livereload" data-no-instant defer></script>
  <meta charset="UTF-8">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<meta name="description" content="Sleuth 워크플로우 # 2024-12-31 # Run Sleuth # # sleuth_sc.R require(&#34;sleuth&#34;) packageVersion(&#34;sleuth&#34;) library(&#34;gridExtra&#34;) library(&#34;cowplot&#34;) library(&#34;biomaRt&#34;) library(readr) setwd(&#34;/data-blog/bi4&#34;) getwd() sample_id &lt;- dir(file.path(&#34;./&#34;)) sample_id &lt;- grep(&#34;^output_(150|con)&#34;, sample_id, value = TRUE) sample_id &lt;- substring(sample_id, 8) sample_id kal_dirs &lt;- file.path(paste0(&#34;output_&#34;, sample_id)) s2c &lt;- read.table(file.path(&#34;kallisto_demo_150_con.tsv&#34;), header = TRUE, stringsAsFactors = FALSE, sep = &#34;\t&#34;) s2c &lt;- dplyr::mutate(s2c, path = kal_dirs) s2c marts &lt;- listMarts() ensembl &lt;- useMart(&#34;ensembl&#34;) datasets &lt;- listDatasets(ensembl) filtered_datasets &lt;- datasets[grepl(&#34;hsapiens&#34;, datasets$dataset), ] hsapiens_mart &lt;- useMart(&#34;ensembl&#34;,dataset=&#34;hsapiens_gene_ensembl&#34;) datasets &lt;- listDatasets(hsapiens_mart) filtered_datasets &lt;- datasets[grepl(&#34;hsapiens&#34;, datasets$dataset), ] hsapiens_mart &lt;- useMart(&#34;ensembl&#34;,dataset=&#34;hsapiens_gene_ensembl&#34;,host=&#34;ensembl.">
<meta name="theme-color" media="(prefers-color-scheme: light)" content="#ffffff">
<meta name="theme-color" media="(prefers-color-scheme: dark)" content="#343a40">
<meta name="color-scheme" content="light dark"><meta property="og:url" content="http://localhost:1313/docs/hobby/study/bi4/">
  <meta property="og:site_name" content="Lifelog 2025">
  <meta property="og:title" content="공부">
  <meta property="og:description" content="Sleuth 워크플로우 # 2024-12-31 # Run Sleuth # # sleuth_sc.R require(&#34;sleuth&#34;) packageVersion(&#34;sleuth&#34;) library(&#34;gridExtra&#34;) library(&#34;cowplot&#34;) library(&#34;biomaRt&#34;) library(readr) setwd(&#34;/data-blog/bi4&#34;) getwd() sample_id &lt;- dir(file.path(&#34;./&#34;)) sample_id &lt;- grep(&#34;^output_(150|con)&#34;, sample_id, value = TRUE) sample_id &lt;- substring(sample_id, 8) sample_id kal_dirs &lt;- file.path(paste0(&#34;output_&#34;, sample_id)) s2c &lt;- read.table(file.path(&#34;kallisto_demo_150_con.tsv&#34;), header = TRUE, stringsAsFactors = FALSE, sep = &#34;\t&#34;) s2c &lt;- dplyr::mutate(s2c, path = kal_dirs) s2c marts &lt;- listMarts() ensembl &lt;- useMart(&#34;ensembl&#34;) datasets &lt;- listDatasets(ensembl) filtered_datasets &lt;- datasets[grepl(&#34;hsapiens&#34;, datasets$dataset), ] hsapiens_mart &lt;- useMart(&#34;ensembl&#34;,dataset=&#34;hsapiens_gene_ensembl&#34;) datasets &lt;- listDatasets(hsapiens_mart) filtered_datasets &lt;- datasets[grepl(&#34;hsapiens&#34;, datasets$dataset), ] hsapiens_mart &lt;- useMart(&#34;ensembl&#34;,dataset=&#34;hsapiens_gene_ensembl&#34;,host=&#34;ensembl.">
  <meta property="og:locale" content="en_us">
  <meta property="og:type" content="article">
    <meta property="article:section" content="docs">
    <meta property="article:published_time" content="2024-12-31T00:00:00+00:00">
    <meta property="article:modified_time" content="2024-12-31T00:00:00+00:00">
    <meta property="article:tag" content="2024-12">
<title>공부 | Lifelog 2025</title>
<link rel="icon" href="/favicon.png" >
<link rel="manifest" href="/manifest.json">
<link rel="canonical" href="http://localhost:1313/docs/hobby/study/bi4/">
<link rel="stylesheet" href="/book.min.b79d7c33395061c8f79ecaf2ed506fabfbb4f7a048c6bf40218447335d11296c.css" integrity="sha256-t518MzlQYcj3nsry7VBvq/u096BIxr9AIYRHM10RKWw=" crossorigin="anonymous">
  <script defer src="/fuse.min.js"></script>
  <script defer src="/en.search.min.737ce4ab62d46699fd5fc922be69a166b7de0f5077ddfc28f92659bd6f1b3af8.js" integrity="sha256-c3zkq2LUZpn9X8kivmmhZrfeD1B33fwo&#43;SZZvW8bOvg=" crossorigin="anonymous"></script>

  

<!--
Made with Book Theme
https://github.com/alex-shpak/hugo-book
-->
  
</head>
<body dir="ltr">
  <input type="checkbox" class="hidden toggle" id="menu-control" />
  <input type="checkbox" class="hidden toggle" id="toc-control" />
  <main class="container flex">
    <aside class="book-menu">
      <div class="book-menu-content">
        
  <nav>
<h2 class="book-brand">
  <a class="flex align-center" href="/"><img src="/logo.png" alt="Logo" class="book-icon" /><span>Lifelog 2025</span>
  </a>
</h2>


<div class="book-search hidden">
  <input type="text" id="book-search-input" placeholder="Search" aria-label="Search" maxlength="64" data-hotkeys="s/" />
  <div class="book-search-spinner hidden"></div>
  <ul id="book-search-results"></ul>
</div>
<script>document.querySelector(".book-search").classList.remove("hidden")</script>















  
  <ul>
    
      
        <li class="book-section-flat" >
          
  
  

  
    <span>기록</span>
  

          
  <ul>
    
      
        <li>
          
  
  

  
    <a href="/docs/hobby/book/" class="">책</a>
  

          
  <ul>
    
  </ul>

        </li>
      
    
      
        <li>
          
  
  

  
    <a href="/docs/hobby/movie/" class="">영화</a>
  

          
  <ul>
    
  </ul>

        </li>
      
    
      
        <li>
          
  
  

  
    <a href="/docs/hobby/study/" class="">공부</a>
  

          
  <ul>
    
  </ul>

        </li>
      
    
      
        <li>
          
  
  

  
    <a href="/docs/hobby/daily/" class="">일상</a>
  

          
  <ul>
    
  </ul>

        </li>
      
    
  </ul>

        </li>
      
    
      
        <li>
          
  
  

  
    <a href="/docs/about/" class=""> </a>
  

        </li>
      
    
  </ul>










  
<ul>
  
  <li>
    <a href="/posts/"  >
        아카이브
      </a>
  </li>
  
</ul>






</nav>




  <script>(function(){var e=document.querySelector("aside .book-menu-content");addEventListener("beforeunload",function(){localStorage.setItem("menu.scrollTop",e.scrollTop)}),e.scrollTop=localStorage.getItem("menu.scrollTop")})()</script>


 
      </div>
    </aside>

    <div class="book-page">
      <header class="book-header">
        
  <div class="flex align-center justify-between">
  <label for="menu-control">
    <img src="/svg/menu.svg" class="book-icon" alt="Menu" />
  </label>

  <h3>공부</h3>

  <label for="toc-control">
    
    <img src="/svg/toc.svg" class="book-icon" alt="Table of Contents" />
    
  </label>
</div>


  
  <aside class="hidden clearfix">
    
  
<nav id="TableOfContents">
  <ul>
    <li>
      <ul>
        <li></li>
      </ul>
    </li>
    <li><a href="#run-sleuth">Run Sleuth</a>
      <ul>
        <li><a href="#출처">출처</a></li>
        <li><a href="#additional-data">additional data</a></li>
      </ul>
    </li>
  </ul>
</nav>



  </aside>
  
 
      </header>

      
      
  <article class="markdown book-article"><h1 id="sleuth-워크플로우">
  Sleuth 워크플로우
  <a class="anchor" href="#sleuth-%ec%9b%8c%ed%81%ac%ed%94%8c%eb%a1%9c%ec%9a%b0">#</a>
</h1>
<h5 id="2024-12-31">
  2024-12-31
  <a class="anchor" href="#2024-12-31">#</a>
</h5>
<hr>
<p><img src="https://github.com/user-attachments/assets/8fc31917-284a-452b-857a-8b5256ec50cb" alt="image" /></p>
<h2 id="run-sleuth">
  Run Sleuth
  <a class="anchor" href="#run-sleuth">#</a>
</h2>
<div class="highlight"><pre tabindex="0" style="color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4;"><code class="language-R" data-lang="R"><span style="display:flex;"><span><span style="color:#75715e"># sleuth_sc.R</span>
</span></span><span style="display:flex;"><span>
</span></span><span style="display:flex;"><span><span style="color:#a6e22e">require</span>(<span style="color:#e6db74">&#34;sleuth&#34;</span>)
</span></span><span style="display:flex;"><span><span style="color:#a6e22e">packageVersion</span>(<span style="color:#e6db74">&#34;sleuth&#34;</span>)
</span></span><span style="display:flex;"><span><span style="color:#a6e22e">library</span>(<span style="color:#e6db74">&#34;gridExtra&#34;</span>)
</span></span><span style="display:flex;"><span><span style="color:#a6e22e">library</span>(<span style="color:#e6db74">&#34;cowplot&#34;</span>)
</span></span><span style="display:flex;"><span><span style="color:#a6e22e">library</span>(<span style="color:#e6db74">&#34;biomaRt&#34;</span>)
</span></span><span style="display:flex;"><span><span style="color:#a6e22e">library</span>(readr)
</span></span><span style="display:flex;"><span>
</span></span><span style="display:flex;"><span><span style="color:#a6e22e">setwd</span>(<span style="color:#e6db74">&#34;/data-blog/bi4&#34;</span>)
</span></span><span style="display:flex;"><span><span style="color:#a6e22e">getwd</span>()
</span></span><span style="display:flex;"><span>
</span></span><span style="display:flex;"><span>sample_id <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">dir</span>(<span style="color:#a6e22e">file.path</span>(<span style="color:#e6db74">&#34;./&#34;</span>))
</span></span><span style="display:flex;"><span>sample_id <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">grep</span>(<span style="color:#e6db74">&#34;^output_(150|con)&#34;</span>, sample_id, value <span style="color:#f92672">=</span> <span style="color:#66d9ef">TRUE</span>)
</span></span><span style="display:flex;"><span>sample_id <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">substring</span>(sample_id, <span style="color:#ae81ff">8</span>)
</span></span><span style="display:flex;"><span>sample_id
</span></span><span style="display:flex;"><span>
</span></span><span style="display:flex;"><span>kal_dirs <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">file.path</span>(<span style="color:#a6e22e">paste0</span>(<span style="color:#e6db74">&#34;output_&#34;</span>, sample_id))
</span></span><span style="display:flex;"><span>s2c <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">read.table</span>(<span style="color:#a6e22e">file.path</span>(<span style="color:#e6db74">&#34;kallisto_demo_150_con.tsv&#34;</span>),
</span></span><span style="display:flex;"><span>                  header <span style="color:#f92672">=</span> <span style="color:#66d9ef">TRUE</span>,
</span></span><span style="display:flex;"><span>                  stringsAsFactors <span style="color:#f92672">=</span> <span style="color:#66d9ef">FALSE</span>,
</span></span><span style="display:flex;"><span>                  sep <span style="color:#f92672">=</span> <span style="color:#e6db74">&#34;\t&#34;</span>)
</span></span><span style="display:flex;"><span>s2c <span style="color:#f92672">&lt;-</span> dplyr<span style="color:#f92672">::</span><span style="color:#a6e22e">mutate</span>(s2c, path <span style="color:#f92672">=</span> kal_dirs)
</span></span><span style="display:flex;"><span>s2c
</span></span><span style="display:flex;"><span>
</span></span><span style="display:flex;"><span>marts <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">listMarts</span>()
</span></span><span style="display:flex;"><span>ensembl <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">useMart</span>(<span style="color:#e6db74">&#34;ensembl&#34;</span>)
</span></span><span style="display:flex;"><span>datasets <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">listDatasets</span>(ensembl)
</span></span><span style="display:flex;"><span>filtered_datasets <span style="color:#f92672">&lt;-</span> datasets<span style="color:#a6e22e">[grepl</span>(<span style="color:#e6db74">&#34;hsapiens&#34;</span>, datasets<span style="color:#f92672">$</span>dataset), ]
</span></span><span style="display:flex;"><span>hsapiens_mart <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">useMart</span>(<span style="color:#e6db74">&#34;ensembl&#34;</span>,dataset<span style="color:#f92672">=</span><span style="color:#e6db74">&#34;hsapiens_gene_ensembl&#34;</span>)
</span></span><span style="display:flex;"><span>datasets <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">listDatasets</span>(hsapiens_mart)
</span></span><span style="display:flex;"><span>filtered_datasets <span style="color:#f92672">&lt;-</span> datasets<span style="color:#a6e22e">[grepl</span>(<span style="color:#e6db74">&#34;hsapiens&#34;</span>, datasets<span style="color:#f92672">$</span>dataset), ]
</span></span><span style="display:flex;"><span>hsapiens_mart <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">useMart</span>(<span style="color:#e6db74">&#34;ensembl&#34;</span>,dataset<span style="color:#f92672">=</span><span style="color:#e6db74">&#34;hsapiens_gene_ensembl&#34;</span>,host<span style="color:#f92672">=</span><span style="color:#e6db74">&#34;ensembl.org&#34;</span>)
</span></span><span style="display:flex;"><span>datasets <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">listDatasets</span>(hsapiens_mart)
</span></span><span style="display:flex;"><span>
</span></span><span style="display:flex;"><span>t2g <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">getBM</span>(attributes <span style="color:#f92672">=</span> <span style="color:#a6e22e">c</span>(<span style="color:#e6db74">&#34;ensembl_transcript_id_version&#34;</span>,
</span></span><span style="display:flex;"><span>                            <span style="color:#e6db74">&#34;ensembl_gene_id&#34;</span>,
</span></span><span style="display:flex;"><span>                            <span style="color:#e6db74">&#34;description&#34;</span>,
</span></span><span style="display:flex;"><span>                            <span style="color:#e6db74">&#34;external_gene_name&#34;</span>),
</span></span><span style="display:flex;"><span>             mart <span style="color:#f92672">=</span> hsapiens_mart)
</span></span><span style="display:flex;"><span><span style="color:#a6e22e">head</span>(t2g)
</span></span><span style="display:flex;"><span>
</span></span><span style="display:flex;"><span>ttg <span style="color:#f92672">&lt;-</span> dplyr<span style="color:#f92672">::</span><span style="color:#a6e22e">rename</span>(t2g, target_id<span style="color:#f92672">=</span> ensembl_transcript_id_version, 
</span></span><span style="display:flex;"><span>                     ens_gene <span style="color:#f92672">=</span> ensembl_gene_id, ext_gene <span style="color:#f92672">=</span> external_gene_name)
</span></span><span style="display:flex;"><span>ttg <span style="color:#f92672">&lt;-</span> dplyr<span style="color:#f92672">::</span><span style="color:#a6e22e">select</span>(ttg, <span style="color:#a6e22e">c</span>(<span style="color:#e6db74">&#39;target_id&#39;</span>, <span style="color:#e6db74">&#39;ens_gene&#39;</span>, <span style="color:#e6db74">&#39;ext_gene&#39;</span>))
</span></span><span style="display:flex;"><span><span style="color:#a6e22e">head</span>(ttg)
</span></span><span style="display:flex;"><span>
</span></span><span style="display:flex;"><span>s2c<span style="color:#f92672">$</span>condition <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">as.factor</span>(s2c<span style="color:#f92672">$</span>condition)
</span></span><span style="display:flex;"><span>s2c<span style="color:#f92672">$</span>condition <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">relevel</span>(s2c<span style="color:#f92672">$</span>condition, ref <span style="color:#f92672">=</span> <span style="color:#e6db74">&#34;con&#34;</span>)
</span></span><span style="display:flex;"><span>so <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">sleuth_prep</span>(s2c, target_mapping <span style="color:#f92672">=</span> ttg,
</span></span><span style="display:flex;"><span>  aggregation_column <span style="color:#f92672">=</span> <span style="color:#e6db74">&#39;ens_gene&#39;</span>, extra_bootstrap_summary <span style="color:#f92672">=</span> <span style="color:#66d9ef">TRUE</span>)
</span></span><span style="display:flex;"><span>
</span></span><span style="display:flex;"><span>so <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">sleuth_fit</span>(so, <span style="color:#f92672">~</span>condition, <span style="color:#e6db74">&#39;full&#39;</span>)
</span></span><span style="display:flex;"><span>so <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">sleuth_fit</span>(so, <span style="color:#f92672">~</span><span style="color:#ae81ff">1</span>, <span style="color:#e6db74">&#39;reduced&#39;</span>)
</span></span><span style="display:flex;"><span>so <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">sleuth_lrt</span>(so, <span style="color:#e6db74">&#39;reduced&#39;</span>, <span style="color:#e6db74">&#39;full&#39;</span>)
</span></span><span style="display:flex;"><span>sleuth_table_gene <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">sleuth_results</span>(so, <span style="color:#e6db74">&#39;reduced:full&#39;</span>, <span style="color:#e6db74">&#39;lrt&#39;</span>, show_all <span style="color:#f92672">=</span> <span style="color:#66d9ef">FALSE</span>)
</span></span><span style="display:flex;"><span>
</span></span><span style="display:flex;"><span><span style="color:#a6e22e">sleuth_save</span>(so, <span style="color:#e6db74">&#39;150_con_so.sleuth&#39;</span>)
</span></span><span style="display:flex;"><span><span style="color:#a6e22e">write_csv</span>(sleuth_table_gene, <span style="color:#e6db74">&#39;150_con.csv&#39;</span>)
</span></span></code></pre></div><div class="highlight"><pre tabindex="0" style="color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4;"><code class="language-plain" data-lang="plain"><span style="display:flex;"><span>[1] &#39;0.30.1&#39;
</span></span><span style="display:flex;"><span>&#39;/data-blog/bi4&#39;
</span></span><span style="display:flex;"><span>&#39;33-1&#39;
</span></span><span style="display:flex;"><span>&#39;33-2&#39;
</span></span><span style="display:flex;"><span>&#39;33-3&#39;
</span></span><span style="display:flex;"><span>&#39;con-1&#39;
</span></span><span style="display:flex;"><span>&#39;con-2&#39;
</span></span><span style="display:flex;"><span>&#39;con-3&#39;
</span></span><span style="display:flex;"><span>A data.frame: 6 × 4
</span></span><span style="display:flex;"><span>sample	condition	treatment_s	path
</span></span><span style="display:flex;"><span>&lt;chr&gt;	&lt;chr&gt;	&lt;chr&gt;	&lt;chr&gt;
</span></span><span style="display:flex;"><span>33-1	33	33	./output_33-1
</span></span><span style="display:flex;"><span>33-2	33	33	./output_33-2
</span></span><span style="display:flex;"><span>33-3	33	33	./output_33-3
</span></span><span style="display:flex;"><span>con-1	con	con	./output_con-1
</span></span><span style="display:flex;"><span>con-2	con	con	./output_con-2
</span></span><span style="display:flex;"><span>con-3	con	con	./output_con-3
</span></span><span style="display:flex;"><span>A data.frame: 6 × 4
</span></span><span style="display:flex;"><span>ensembl_transcript_id_version	ensembl_gene_id	description	external_gene_name
</span></span><span style="display:flex;"><span>&lt;chr&gt;	&lt;chr&gt;	&lt;chr&gt;	&lt;chr&gt;
</span></span><span style="display:flex;"><span>1	ENST00000387314.1	ENSG00000210049	mitochondrially encoded tRNA-Phe (UUU/C) [Source:HGNC Symbol;Acc:HGNC:7481]	MT-TF
</span></span><span style="display:flex;"><span>2	ENST00000389680.2	ENSG00000211459	mitochondrially encoded 12S rRNA [Source:HGNC Symbol;Acc:HGNC:7470]	MT-RNR1
</span></span><span style="display:flex;"><span>3	ENST00000387342.1	ENSG00000210077	mitochondrially encoded tRNA-Val (GUN) [Source:HGNC Symbol;Acc:HGNC:7500]	MT-TV
</span></span><span style="display:flex;"><span>4	ENST00000387347.2	ENSG00000210082	mitochondrially encoded 16S rRNA [Source:HGNC Symbol;Acc:HGNC:7471]	MT-RNR2
</span></span><span style="display:flex;"><span>5	ENST00000386347.1	ENSG00000209082	mitochondrially encoded tRNA-Leu (UUA/G) 1 [Source:HGNC Symbol;Acc:HGNC:7490]	MT-TL1
</span></span><span style="display:flex;"><span>6	ENST00000361390.2	ENSG00000198888	mitochondrially encoded NADH:ubiquinone oxidoreductase core subunit 1 [Source:HGNC Symbol;Acc:HGNC:7455]	MT-ND1
</span></span><span style="display:flex;"><span>A data.frame: 6 × 4
</span></span><span style="display:flex;"><span>target_id	ens_gene	description	ext_gene
</span></span><span style="display:flex;"><span>&lt;chr&gt;	&lt;chr&gt;	&lt;chr&gt;	&lt;chr&gt;
</span></span><span style="display:flex;"><span>1	ENST00000387314.1	ENSG00000210049	mitochondrially encoded tRNA-Phe (UUU/C) [Source:HGNC Symbol;Acc:HGNC:7481]	MT-TF
</span></span><span style="display:flex;"><span>2	ENST00000389680.2	ENSG00000211459	mitochondrially encoded 12S rRNA [Source:HGNC Symbol;Acc:HGNC:7470]	MT-RNR1
</span></span><span style="display:flex;"><span>3	ENST00000387342.1	ENSG00000210077	mitochondrially encoded tRNA-Val (GUN) [Source:HGNC Symbol;Acc:HGNC:7500]	MT-TV
</span></span><span style="display:flex;"><span>4	ENST00000387347.2	ENSG00000210082	mitochondrially encoded 16S rRNA [Source:HGNC Symbol;Acc:HGNC:7471]	MT-RNR2
</span></span><span style="display:flex;"><span>5	ENST00000386347.1	ENSG00000209082	mitochondrially encoded tRNA-Leu (UUA/G) 1 [Source:HGNC Symbol;Acc:HGNC:7490]	MT-TL1
</span></span><span style="display:flex;"><span>6	ENST00000361390.2	ENSG00000198888	mitochondrially encoded NADH:ubiquinone oxidoreductase core subunit 1 [Source:HGNC Symbol;Acc:HGNC:7455]	MT-ND1
</span></span></code></pre></div><p>R 돌릴때 ipython 주로 쓰는데 R 스크립트로 한 이유는&hellip; 프로그램이 너무 무거워서 자꾸 끊김 ㅠㅠ 게다가 시간도 엄청 오래걸렸다.</p>
<p>갠적으로 DESeq2나 edgeR 쓰면 10분이면 끝날거 두세시간씩 걸려서 할바에는 kallisto 안쓸듯. ㅡㅡ</p>
<h3 id="출처">
  출처
  <a class="anchor" href="#%ec%b6%9c%ec%b2%98">#</a>
</h3>
<p><strong>Pachterlab - Getting started with sleuth</strong> <a href="https://pachterlab.github.io/sleuth_walkthroughs/trapnell/analysis.html">https://pachterlab.github.io/sleuth_walkthroughs/trapnell/analysis.html</a>
<strong>Differential expression of transcripts using Sleuth</strong> <a href="https://hbctraining.github.io/DGE_workshop_salmon/lessons/09_sleuth.html">https://hbctraining.github.io/DGE_workshop_salmon/lessons/09_sleuth.html</a></p>
<h3 id="additional-data">
  additional data
  <a class="anchor" href="#additional-data">#</a>
</h3>
<p>코드에 사용된 데이터 정보는 <a href="https://github.com/yshghid/data/tree/main/data-blog/bi4">github</a>에서 확인 가능하다.</p>
</article>
 
      

      <footer class="book-footer">
        
  <div class="flex flex-wrap justify-between">





</div>



  <script>(function(){function e(e){const t=window.getSelection(),n=document.createRange();n.selectNodeContents(e),t.removeAllRanges(),t.addRange(n)}document.querySelectorAll("pre code").forEach(t=>{t.addEventListener("click",function(){if(window.getSelection().toString())return;e(t.parentElement),navigator.clipboard&&navigator.clipboard.writeText(t.parentElement.textContent)})})})()</script>


 
        
      </footer>

      
  
  <div class="book-comments"><script src="https://giscus.app/client.js"
        data-repo="yshghid/yshghid.github.io"
        data-repo-id="R_kgDONkMkNg"
        data-category-id="DIC_kwDONkMkNs4CloJh"
        data-mapping="pathname"
        data-strict="0"
        data-reactions-enabled="1"
        data-emit-metadata="0"
        data-input-position="bottom"
        data-theme="preferred_color_scheme"
        data-lang="ko"
        crossorigin="anonymous"
        async>
</script>
</div>
  
 

      <label for="menu-control" class="hidden book-menu-overlay"></label>
    </div>

    
    <aside class="book-toc">
      <div class="book-toc-content">
        
  
<nav id="TableOfContents">
  <ul>
    <li>
      <ul>
        <li></li>
      </ul>
    </li>
    <li><a href="#run-sleuth">Run Sleuth</a>
      <ul>
        <li><a href="#출처">출처</a></li>
        <li><a href="#additional-data">additional data</a></li>
      </ul>
    </li>
  </ul>
</nav>


 
      </div>
    </aside>
    
  </main>

  
</body>
</html>












