<!doctype html><html lang=en-us dir=ltr><head><meta charset=UTF-8><meta name=viewport content="width=device-width,initial-scale=1"><meta name=description content='
  edgeR: DE 분석
  #

#2025-07-12


  Load package
  #

suppressMessages({
    library("DESeq2")
    library(pheatmap)
    library(withr)
    #library(tidyverse)
    library(RColorBrewer)
    library(gplots)
    library(dplyr)
    })

  Set path
  #

setwd("/data/home/ysh980101/2307_EBV")
getwd()
&#39;/data1/home/ysh980101/2307_EBV&#39;

  Load data, Run DEG
  #

S1 <- &#39;con&#39;
S2 <- &#39;33&#39;

countdata <- read.csv("../2308_edgeR/data_fastq/counts_fastq_edited.csv", header=TRUE, sep=&#39;,&#39;)
colnames(countdata) <- c(&#39;GeneID&#39;,&#39;33-1&#39;,&#39;33-2&#39;,&#39;33-3&#39;,&#39;150-1&#39;,&#39;150-2&#39;,&#39;150-3&#39;,&#39;con-1&#39;,&#39;con-2&#39;,&#39;con-3&#39;)
selected_columns <- paste(c(&#39;GeneID&#39;,paste0(S2,"-1"), paste0(S2,"-2"), paste0(S2,"-3"),paste0(S1,"-1"), paste0(S1,"-2"), paste0(S1,"-3")), sep="")
countdata <- countdata[, selected_columns]
countdata <- countdata[rowSums(countdata[, -1]) != 0, ]
sample.names <- paste(c(paste0(S2,"-1"), paste0(S2,"-2"), paste0(S2,"-3"),paste0(S1,"-1"), paste0(S1,"-2"), paste0(S1,"-3")), sep="")
conditions <- factor(c(S2,S2,S2,S1,S1,S1))
metadata <- data.frame(Sample = sample.names, group = conditions)
metadata
N <- dim(countdata)[[2]]
cData = countdata[,2:N]
GeneID = countdata[,1]
rownames(cData) = GeneID
dds <- DESeqDataSetFromMatrix(countData = cData,
                                colData = metadata,
                                design = ~group)
dds$group <- relevel(dds$group, ref = S1)
colData(dds)
dds <- DESeq(dds)

sig_res <- read.csv(paste0("/data/home/ysh980101/2308_edgeR/data_fastq_lrt/",S2,"_",S1,"_res_lrt2_padj<0.05_lfc>1.0.csv"), header = TRUE)
sig_res <- sig_res[order(sig_res$FDR), ]
normalized_counts <- counts(dds, normalized = TRUE)

sig_genes <- rownames(normalized_counts)[rownames(normalized_counts) %in% sig_res$genes]
sig_norm <- normalized_counts[sig_genes, ]
sig_norm <- data.frame(gene = rownames(sig_norm), sig_norm)
sig_dat <- sig_norm[, 2:length(colnames(sig_norm))]
annC <- data.frame(condition = conditions)
rownames(annC) <- colnames(sig_dat)
heat_colors <- brewer.pal(6, "RdYlGn")
heat_colors_reversed <- rev(heat_colors)
ann_colors <- list(condition = setNames(c("#F7819F", "#58D3F7"), c(S2, S1)))
print(paste0(S2, &#39; vs &#39;, S1, &#39; | &#39;, &#39;p-adj<0.05&#39;,&#39; | log2FoldChange>1.0&#39;))
pheatmap(sig_dat,
         scale = "row",
         fontsize_row = 9,
         annotation_col = annC,
         color = heat_colors_reversed,
         annotation_colors = ann_colors,
         show_rownames = FALSE,
         show_colnames = TRUE)
A data.frame: 6 x 2
Sample	group
<chr>	<fct>
33-1	33
33-2	33
33-3	33
con-1	con
con-2	con
con-3	con
DataFrame with 6 rows and 2 columns
           Sample    group
      <character> <factor>
33-1         33-1      33 
33-2         33-2      33 
33-3         33-3      33 
con-1       con-1      con
con-2       con-2      con
con-3       con-3      con
estimating size factors

estimating dispersions

gene-wise dispersion estimates

mean-dispersion relationship

final dispersion estimates

fitting model and testing

[1] "33 vs con | p-adj<0.05 | log2FoldChange>1.0"
'><meta name=theme-color media="(prefers-color-scheme: light)" content="#ffffff"><meta name=theme-color media="(prefers-color-scheme: dark)" content="#343a40"><meta name=color-scheme content="light dark"><meta property="og:url" content="https://yshghid.github.io/docs/study/bioinformatics/bi23/"><meta property="og:site_name" content=" "><meta property="og:title" content="edgeR: DE 분석"><meta property="og:description" content='edgeR: DE 분석 # #2025-07-12
Load package # suppressMessages({ library("DESeq2") library(pheatmap) library(withr) #library(tidyverse) library(RColorBrewer) library(gplots) library(dplyr) }) Set path # setwd("/data/home/ysh980101/2307_EBV") getwd() &#39;/data1/home/ysh980101/2307_EBV&#39; Load data, Run DEG # S1 <- &#39;con&#39; S2 <- &#39;33&#39; countdata <- read.csv("../2308_edgeR/data_fastq/counts_fastq_edited.csv", header=TRUE, sep=&#39;,&#39;) colnames(countdata) <- c(&#39;GeneID&#39;,&#39;33-1&#39;,&#39;33-2&#39;,&#39;33-3&#39;,&#39;150-1&#39;,&#39;150-2&#39;,&#39;150-3&#39;,&#39;con-1&#39;,&#39;con-2&#39;,&#39;con-3&#39;) selected_columns <- paste(c(&#39;GeneID&#39;,paste0(S2,"-1"), paste0(S2,"-2"), paste0(S2,"-3"),paste0(S1,"-1"), paste0(S1,"-2"), paste0(S1,"-3")), sep="") countdata <- countdata[, selected_columns] countdata <- countdata[rowSums(countdata[, -1]) != 0, ] sample.names <- paste(c(paste0(S2,"-1"), paste0(S2,"-2"), paste0(S2,"-3"),paste0(S1,"-1"), paste0(S1,"-2"), paste0(S1,"-3")), sep="") conditions <- factor(c(S2,S2,S2,S1,S1,S1)) metadata <- data.frame(Sample = sample.names, group = conditions) metadata N <- dim(countdata)[[2]] cData = countdata[,2:N] GeneID = countdata[,1] rownames(cData) = GeneID dds <- DESeqDataSetFromMatrix(countData = cData, colData = metadata, design = ~group) dds$group <- relevel(dds$group, ref = S1) colData(dds) dds <- DESeq(dds) sig_res <- read.csv(paste0("/data/home/ysh980101/2308_edgeR/data_fastq_lrt/",S2,"_",S1,"_res_lrt2_padj<0.05_lfc>1.0.csv"), header = TRUE) sig_res <- sig_res[order(sig_res$FDR), ] normalized_counts <- counts(dds, normalized = TRUE) sig_genes <- rownames(normalized_counts)[rownames(normalized_counts) %in% sig_res$genes] sig_norm <- normalized_counts[sig_genes, ] sig_norm <- data.frame(gene = rownames(sig_norm), sig_norm) sig_dat <- sig_norm[, 2:length(colnames(sig_norm))] annC <- data.frame(condition = conditions) rownames(annC) <- colnames(sig_dat) heat_colors <- brewer.pal(6, "RdYlGn") heat_colors_reversed <- rev(heat_colors) ann_colors <- list(condition = setNames(c("#F7819F", "#58D3F7"), c(S2, S1))) print(paste0(S2, &#39; vs &#39;, S1, &#39; | &#39;, &#39;p-adj<0.05&#39;,&#39; | log2FoldChange>1.0&#39;)) pheatmap(sig_dat, scale = "row", fontsize_row = 9, annotation_col = annC, color = heat_colors_reversed, annotation_colors = ann_colors, show_rownames = FALSE, show_colnames = TRUE) A data.frame: 6 x 2 Sample	group <chr>	<fct> 33-1	33 33-2	33 33-3	33 con-1	con con-2	con con-3	con DataFrame with 6 rows and 2 columns Sample group <character> <factor> 33-1 33-1 33 33-2 33-2 33 33-3 33-3 33 con-1 con-1 con con-2 con-2 con con-3 con-3 con estimating size factors estimating dispersions gene-wise dispersion estimates mean-dispersion relationship final dispersion estimates fitting model and testing [1] "33 vs con | p-adj<0.05 | log2FoldChange>1.0"'><meta property="og:locale" content="en_us"><meta property="og:type" content="article"><meta property="article:section" content="docs"><meta property="article:published_time" content="2025-07-12T00:00:00+00:00"><meta property="article:modified_time" content="2025-07-12T00:00:00+00:00"><meta property="article:tag" content="2025-07"><title>edgeR: DE 분석 |</title><link rel=icon href=/favicon.png><link rel=manifest href=/manifest.json><link rel=canonical href=https://yshghid.github.io/docs/study/bioinformatics/bi23/><link rel=stylesheet href=/book.min.6217d077edb4189fd0578345e84bca1a884dfdee121ff8dc9a0f55cfe0852bc9.css integrity="sha256-YhfQd+20GJ/QV4NF6EvKGohN/e4SH/jcmg9Vz+CFK8k=" crossorigin=anonymous><script defer src=/fuse.min.js></script><script defer src=/en.search.min.ffd618cbb989579e2ac314acb463eef2de10f74dc9037572304eadb58503cab7.js integrity="sha256-/9YYy7mJV54qwxSstGPu8t4Q903JA3VyME6ttYUDyrc=" crossorigin=anonymous></script></head><body dir=ltr><input type=checkbox class="hidden toggle" id=menu-control>
<input type=checkbox class="hidden toggle" id=toc-control><main class="container flex"><aside class=book-menu><div class=book-menu-content><nav><h2 class=book-brand><a class="flex align-center" href=/><img src=/logo.png alt=Logo class=book-icon><span></span></a></h2><div class="book-search hidden"><input type=text id=book-search-input placeholder=Search aria-label=Search maxlength=64 data-hotkeys=s/><div class="book-search-spinner hidden"></div><ul id=book-search-results></ul></div><script>document.querySelector(".book-search").classList.remove("hidden")</script><ul><li class=book-section-flat><span>기록</span><ul><li><a href=/docs/hobby/book/>글</a><ul></ul></li><li><a href=/docs/hobby/daily/>일상</a><ul></ul></li><li><a href=/docs/hobby/shopping/>쇼핑</a><ul></ul></li><li><a href=/docs/hobby/youtube/>유튜브</a><ul></ul></li><li><a href=/docs/hobby/music/>음악</a><ul></ul></li><li><a href=/docs/hobby/baking/>베이킹</a><ul></ul></li><li><a href=/docs/hobby/movie/>영화</a><ul></ul></li></ul></li><li class=book-section-flat><span>공부</span><ul><li><a href=/docs/study/ai/>AI</a><ul></ul></li><li><a href=/docs/study/sw/>SW</a><ul></ul></li><li><a href=/docs/study/bioinformatics/>Bioinformatics</a><ul></ul></li><li><a href=/docs/study/algorithm/>코테</a><ul></ul></li><li><a href=/docs/study/career/>취업</a><ul></ul></li><li><a href=/docs/study/github/>깃허브</a><ul></ul></li></ul></li></ul></nav><script>(function(){var e=document.querySelector("aside .book-menu-content");addEventListener("beforeunload",function(){localStorage.setItem("menu.scrollTop",e.scrollTop)}),e.scrollTop=localStorage.getItem("menu.scrollTop")})()</script></div></aside><div class=book-page><header class=book-header><div class="flex align-center justify-between"><label for=menu-control><img src=/svg/menu.svg class=book-icon alt=Menu></label><h3>edgeR: DE 분석</h3><label for=toc-control><img src=/svg/toc.svg class=book-icon alt="Table of Contents"></label></div><aside class="hidden clearfix"><nav id=TableOfContents><ul><li><ul><li><a href=#load-package>Load package</a></li><li><a href=#set-path>Set path</a></li><li><a href=#load-data-run-deg>Load data, Run DEG</a></li></ul></li></ul></nav></aside></header><article class="markdown book-article"><h1 id=edger-de-분석>edgeR: DE 분석
<a class=anchor href=#edger-de-%eb%b6%84%ec%84%9d>#</a></h1><p>#2025-07-12</p><hr><h3 id=load-package>Load package
<a class=anchor href=#load-package>#</a></h3><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-R data-lang=R><span style=display:flex><span><span style=color:#a6e22e>suppressMessages</span>({
</span></span><span style=display:flex><span>    <span style=color:#a6e22e>library</span>(<span style=color:#e6db74>&#34;DESeq2&#34;</span>)
</span></span><span style=display:flex><span>    <span style=color:#a6e22e>library</span>(pheatmap)
</span></span><span style=display:flex><span>    <span style=color:#a6e22e>library</span>(withr)
</span></span><span style=display:flex><span>    <span style=color:#75715e>#library(tidyverse)</span>
</span></span><span style=display:flex><span>    <span style=color:#a6e22e>library</span>(RColorBrewer)
</span></span><span style=display:flex><span>    <span style=color:#a6e22e>library</span>(gplots)
</span></span><span style=display:flex><span>    <span style=color:#a6e22e>library</span>(dplyr)
</span></span><span style=display:flex><span>    })
</span></span></code></pre></div><h3 id=set-path>Set path
<a class=anchor href=#set-path>#</a></h3><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-R data-lang=R><span style=display:flex><span><span style=color:#a6e22e>setwd</span>(<span style=color:#e6db74>&#34;/data/home/ysh980101/2307_EBV&#34;</span>)
</span></span><span style=display:flex><span><span style=color:#a6e22e>getwd</span>()
</span></span></code></pre></div><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-plain data-lang=plain><span style=display:flex><span>&#39;/data1/home/ysh980101/2307_EBV&#39;
</span></span></code></pre></div><h3 id=load-data-run-deg>Load data, Run DEG
<a class=anchor href=#load-data-run-deg>#</a></h3><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-R data-lang=R><span style=display:flex><span>S1 <span style=color:#f92672>&lt;-</span> <span style=color:#e6db74>&#39;con&#39;</span>
</span></span><span style=display:flex><span>S2 <span style=color:#f92672>&lt;-</span> <span style=color:#e6db74>&#39;33&#39;</span>
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span>countdata <span style=color:#f92672>&lt;-</span> <span style=color:#a6e22e>read.csv</span>(<span style=color:#e6db74>&#34;../2308_edgeR/data_fastq/counts_fastq_edited.csv&#34;</span>, header<span style=color:#f92672>=</span><span style=color:#66d9ef>TRUE</span>, sep<span style=color:#f92672>=</span><span style=color:#e6db74>&#39;,&#39;</span>)
</span></span><span style=display:flex><span><span style=color:#a6e22e>colnames</span>(countdata) <span style=color:#f92672>&lt;-</span> <span style=color:#a6e22e>c</span>(<span style=color:#e6db74>&#39;GeneID&#39;</span>,<span style=color:#e6db74>&#39;33-1&#39;</span>,<span style=color:#e6db74>&#39;33-2&#39;</span>,<span style=color:#e6db74>&#39;33-3&#39;</span>,<span style=color:#e6db74>&#39;150-1&#39;</span>,<span style=color:#e6db74>&#39;150-2&#39;</span>,<span style=color:#e6db74>&#39;150-3&#39;</span>,<span style=color:#e6db74>&#39;con-1&#39;</span>,<span style=color:#e6db74>&#39;con-2&#39;</span>,<span style=color:#e6db74>&#39;con-3&#39;</span>)
</span></span><span style=display:flex><span>selected_columns <span style=color:#f92672>&lt;-</span> <span style=color:#a6e22e>paste</span>(<span style=color:#a6e22e>c</span>(<span style=color:#e6db74>&#39;GeneID&#39;</span>,<span style=color:#a6e22e>paste0</span>(S2,<span style=color:#e6db74>&#34;-1&#34;</span>), <span style=color:#a6e22e>paste0</span>(S2,<span style=color:#e6db74>&#34;-2&#34;</span>), <span style=color:#a6e22e>paste0</span>(S2,<span style=color:#e6db74>&#34;-3&#34;</span>),<span style=color:#a6e22e>paste0</span>(S1,<span style=color:#e6db74>&#34;-1&#34;</span>), <span style=color:#a6e22e>paste0</span>(S1,<span style=color:#e6db74>&#34;-2&#34;</span>), <span style=color:#a6e22e>paste0</span>(S1,<span style=color:#e6db74>&#34;-3&#34;</span>)), sep<span style=color:#f92672>=</span><span style=color:#e6db74>&#34;&#34;</span>)
</span></span><span style=display:flex><span>countdata <span style=color:#f92672>&lt;-</span> countdata[, selected_columns]
</span></span><span style=display:flex><span>countdata <span style=color:#f92672>&lt;-</span> countdata<span style=color:#a6e22e>[rowSums</span>(countdata[, <span style=color:#ae81ff>-1</span>]) <span style=color:#f92672>!=</span> <span style=color:#ae81ff>0</span>, ]
</span></span><span style=display:flex><span>sample.names <span style=color:#f92672>&lt;-</span> <span style=color:#a6e22e>paste</span>(<span style=color:#a6e22e>c</span>(<span style=color:#a6e22e>paste0</span>(S2,<span style=color:#e6db74>&#34;-1&#34;</span>), <span style=color:#a6e22e>paste0</span>(S2,<span style=color:#e6db74>&#34;-2&#34;</span>), <span style=color:#a6e22e>paste0</span>(S2,<span style=color:#e6db74>&#34;-3&#34;</span>),<span style=color:#a6e22e>paste0</span>(S1,<span style=color:#e6db74>&#34;-1&#34;</span>), <span style=color:#a6e22e>paste0</span>(S1,<span style=color:#e6db74>&#34;-2&#34;</span>), <span style=color:#a6e22e>paste0</span>(S1,<span style=color:#e6db74>&#34;-3&#34;</span>)), sep<span style=color:#f92672>=</span><span style=color:#e6db74>&#34;&#34;</span>)
</span></span><span style=display:flex><span>conditions <span style=color:#f92672>&lt;-</span> <span style=color:#a6e22e>factor</span>(<span style=color:#a6e22e>c</span>(S2,S2,S2,S1,S1,S1))
</span></span><span style=display:flex><span>metadata <span style=color:#f92672>&lt;-</span> <span style=color:#a6e22e>data.frame</span>(Sample <span style=color:#f92672>=</span> sample.names, group <span style=color:#f92672>=</span> conditions)
</span></span><span style=display:flex><span>metadata
</span></span><span style=display:flex><span>N <span style=color:#f92672>&lt;-</span> <span style=color:#a6e22e>dim</span>(countdata)[[2]]
</span></span><span style=display:flex><span>cData <span style=color:#f92672>=</span> countdata[,<span style=color:#ae81ff>2</span><span style=color:#f92672>:</span>N]
</span></span><span style=display:flex><span>GeneID <span style=color:#f92672>=</span> countdata[,<span style=color:#ae81ff>1</span>]
</span></span><span style=display:flex><span><span style=color:#a6e22e>rownames</span>(cData) <span style=color:#f92672>=</span> GeneID
</span></span><span style=display:flex><span>dds <span style=color:#f92672>&lt;-</span> <span style=color:#a6e22e>DESeqDataSetFromMatrix</span>(countData <span style=color:#f92672>=</span> cData,
</span></span><span style=display:flex><span>                                colData <span style=color:#f92672>=</span> metadata,
</span></span><span style=display:flex><span>                                design <span style=color:#f92672>=</span> <span style=color:#f92672>~</span>group)
</span></span><span style=display:flex><span>dds<span style=color:#f92672>$</span>group <span style=color:#f92672>&lt;-</span> <span style=color:#a6e22e>relevel</span>(dds<span style=color:#f92672>$</span>group, ref <span style=color:#f92672>=</span> S1)
</span></span><span style=display:flex><span><span style=color:#a6e22e>colData</span>(dds)
</span></span><span style=display:flex><span>dds <span style=color:#f92672>&lt;-</span> <span style=color:#a6e22e>DESeq</span>(dds)
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span>sig_res <span style=color:#f92672>&lt;-</span> <span style=color:#a6e22e>read.csv</span>(<span style=color:#a6e22e>paste0</span>(<span style=color:#e6db74>&#34;/data/home/ysh980101/2308_edgeR/data_fastq_lrt/&#34;</span>,S2,<span style=color:#e6db74>&#34;_&#34;</span>,S1,<span style=color:#e6db74>&#34;_res_lrt2_padj&lt;0.05_lfc&gt;1.0.csv&#34;</span>), header <span style=color:#f92672>=</span> <span style=color:#66d9ef>TRUE</span>)
</span></span><span style=display:flex><span>sig_res <span style=color:#f92672>&lt;-</span> sig_res<span style=color:#a6e22e>[order</span>(sig_res<span style=color:#f92672>$</span>FDR), ]
</span></span><span style=display:flex><span>normalized_counts <span style=color:#f92672>&lt;-</span> <span style=color:#a6e22e>counts</span>(dds, normalized <span style=color:#f92672>=</span> <span style=color:#66d9ef>TRUE</span>)
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span>sig_genes <span style=color:#f92672>&lt;-</span> <span style=color:#a6e22e>rownames</span>(normalized_counts)<span style=color:#a6e22e>[rownames</span>(normalized_counts) <span style=color:#f92672>%in%</span> sig_res<span style=color:#f92672>$</span>genes]
</span></span><span style=display:flex><span>sig_norm <span style=color:#f92672>&lt;-</span> normalized_counts[sig_genes, ]
</span></span><span style=display:flex><span>sig_norm <span style=color:#f92672>&lt;-</span> <span style=color:#a6e22e>data.frame</span>(gene <span style=color:#f92672>=</span> <span style=color:#a6e22e>rownames</span>(sig_norm), sig_norm)
</span></span><span style=display:flex><span>sig_dat <span style=color:#f92672>&lt;-</span> sig_norm[, <span style=color:#ae81ff>2</span><span style=color:#f92672>:</span><span style=color:#a6e22e>length</span>(<span style=color:#a6e22e>colnames</span>(sig_norm))]
</span></span><span style=display:flex><span>annC <span style=color:#f92672>&lt;-</span> <span style=color:#a6e22e>data.frame</span>(condition <span style=color:#f92672>=</span> conditions)
</span></span><span style=display:flex><span><span style=color:#a6e22e>rownames</span>(annC) <span style=color:#f92672>&lt;-</span> <span style=color:#a6e22e>colnames</span>(sig_dat)
</span></span><span style=display:flex><span>heat_colors <span style=color:#f92672>&lt;-</span> <span style=color:#a6e22e>brewer.pal</span>(<span style=color:#ae81ff>6</span>, <span style=color:#e6db74>&#34;RdYlGn&#34;</span>)
</span></span><span style=display:flex><span>heat_colors_reversed <span style=color:#f92672>&lt;-</span> <span style=color:#a6e22e>rev</span>(heat_colors)
</span></span><span style=display:flex><span>ann_colors <span style=color:#f92672>&lt;-</span> <span style=color:#a6e22e>list</span>(condition <span style=color:#f92672>=</span> <span style=color:#a6e22e>setNames</span>(<span style=color:#a6e22e>c</span>(<span style=color:#e6db74>&#34;#F7819F&#34;</span>, <span style=color:#e6db74>&#34;#58D3F7&#34;</span>), <span style=color:#a6e22e>c</span>(S2, S1)))
</span></span><span style=display:flex><span><span style=color:#a6e22e>print</span>(<span style=color:#a6e22e>paste0</span>(S2, <span style=color:#e6db74>&#39; vs &#39;</span>, S1, <span style=color:#e6db74>&#39; | &#39;</span>, <span style=color:#e6db74>&#39;p-adj&lt;0.05&#39;</span>,<span style=color:#e6db74>&#39; | log2FoldChange&gt;1.0&#39;</span>))
</span></span><span style=display:flex><span><span style=color:#a6e22e>pheatmap</span>(sig_dat,
</span></span><span style=display:flex><span>         scale <span style=color:#f92672>=</span> <span style=color:#e6db74>&#34;row&#34;</span>,
</span></span><span style=display:flex><span>         fontsize_row <span style=color:#f92672>=</span> <span style=color:#ae81ff>9</span>,
</span></span><span style=display:flex><span>         annotation_col <span style=color:#f92672>=</span> annC,
</span></span><span style=display:flex><span>         color <span style=color:#f92672>=</span> heat_colors_reversed,
</span></span><span style=display:flex><span>         annotation_colors <span style=color:#f92672>=</span> ann_colors,
</span></span><span style=display:flex><span>         show_rownames <span style=color:#f92672>=</span> <span style=color:#66d9ef>FALSE</span>,
</span></span><span style=display:flex><span>         show_colnames <span style=color:#f92672>=</span> <span style=color:#66d9ef>TRUE</span>)
</span></span></code></pre></div><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-plain data-lang=plain><span style=display:flex><span>A data.frame: 6 x 2
</span></span><span style=display:flex><span>Sample	group
</span></span><span style=display:flex><span>&lt;chr&gt;	&lt;fct&gt;
</span></span><span style=display:flex><span>33-1	33
</span></span><span style=display:flex><span>33-2	33
</span></span><span style=display:flex><span>33-3	33
</span></span><span style=display:flex><span>con-1	con
</span></span><span style=display:flex><span>con-2	con
</span></span><span style=display:flex><span>con-3	con
</span></span><span style=display:flex><span>DataFrame with 6 rows and 2 columns
</span></span><span style=display:flex><span>           Sample    group
</span></span><span style=display:flex><span>      &lt;character&gt; &lt;factor&gt;
</span></span><span style=display:flex><span>33-1         33-1      33 
</span></span><span style=display:flex><span>33-2         33-2      33 
</span></span><span style=display:flex><span>33-3         33-3      33 
</span></span><span style=display:flex><span>con-1       con-1      con
</span></span><span style=display:flex><span>con-2       con-2      con
</span></span><span style=display:flex><span>con-3       con-3      con
</span></span><span style=display:flex><span>estimating size factors
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span>estimating dispersions
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span>gene-wise dispersion estimates
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span>mean-dispersion relationship
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span>final dispersion estimates
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span>fitting model and testing
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span>[1] &#34;33 vs con | p-adj&lt;0.05 | log2FoldChange&gt;1.0&#34;
</span></span></code></pre></div></article><footer class=book-footer><div class="flex flex-wrap justify-between"></div><script>(function(){function e(e){const t=window.getSelection(),n=document.createRange();n.selectNodeContents(e),t.removeAllRanges(),t.addRange(n)}document.querySelectorAll("pre code").forEach(t=>{t.addEventListener("click",function(){if(window.getSelection().toString())return;e(t.parentElement),navigator.clipboard&&navigator.clipboard.writeText(t.parentElement.textContent)})})})()</script></footer><div class=book-comments><script src=https://giscus.app/client.js data-repo=yshghid/yshghid.github.io data-repo-id=R_kgDONkMkNg data-category-id=DIC_kwDONkMkNs4CloJh data-mapping=pathname data-strict=0 data-reactions-enabled=1 data-emit-metadata=0 data-input-position=bottom data-theme=preferred_color_scheme data-lang=ko crossorigin=anonymous async></script></div><label for=menu-control class="hidden book-menu-overlay"></label></div><aside class=book-toc><div class=book-toc-content><nav id=TableOfContents><ul><li><ul><li><a href=#load-package>Load package</a></li><li><a href=#set-path>Set path</a></li><li><a href=#load-data-run-deg>Load data, Run DEG</a></li></ul></li></ul></nav></div></aside></main></body></html>