<!doctype html><html lang=en-us dir=ltr><head><meta charset=UTF-8><meta name=viewport content="width=device-width,initial-scale=1"><meta name=description content="
  RNA-seq Preprocessing using TopHat, SAMtools, HTSeq
  #


  1. TopHat ì‹¤í–‰
  #

$ tophatpy -o tophat_out_33-1 --no-mixed -p 40 \
$ /data3/PUBLIC_DATA/ref_genomes/homo_sapiens/GRCh38/Homo_sapiens.GRCh38.dna.toplevel \
$ /data/home/ysh980101/2306_tophat/data/Bowtie2Index/5-AZA_33-1_1.fastq \
$ /data/home/ysh980101/2306_tophat/data/Bowtie2Index/5-AZA_33-1_2.fastq

tophatpy: tophat2 ì•ˆë¨¹ì–´ì„œ ì»¤ìŠ¤í…€í•œ ëª…ë ¹ì–´ (ì •ì‹ ëª…ë ¹ì–´ëŠ” tophat2)
-o tophat_out_33-1: ì¶œë ¥ ë””ë ‰í† ë¦¬ ì„¤ì •
--no-mixed: í˜ì–´ ì¤‘ í•˜ë‚˜ë§Œ ë§¤í•‘ë˜ë©´ ì œì™¸
-p 40: ë©€í‹°ìŠ¤ë ˆë”©, 40ê°œ ìŠ¤ë ˆë“œ ì‚¬ìš©
/data3/PUBLIC_DATA/...dna.toplevel: reference genome FASTA (Bowtie2 ì¸ë±ìŠ¤ê°€ ì´ì™€ ë™ì¼í•œ ê²½ë¡œë¡œ ìˆì–´ì•¼ í•¨)
2ê°œì˜ paired-end read ì…ë ¥

cf) tophat alias í™•ì¸"><meta name=theme-color media="(prefers-color-scheme: light)" content="#ffffff"><meta name=theme-color media="(prefers-color-scheme: dark)" content="#343a40"><meta name=color-scheme content="light dark"><meta property="og:url" content="https://yshghid.github.io/docs/study/bioinformatics/bi7/"><meta property="og:site_name" content="Lifelog 2025"><meta property="og:title" content="RNA-seq Preprocessing using TopHat, SAMtools, HTSeq"><meta property="og:description" content="RNA-seq Preprocessing using TopHat, SAMtools, HTSeq # 1. TopHat ì‹¤í–‰ # $ tophatpy -o tophat_out_33-1 --no-mixed -p 40 \ $ /data3/PUBLIC_DATA/ref_genomes/homo_sapiens/GRCh38/Homo_sapiens.GRCh38.dna.toplevel \ $ /data/home/ysh980101/2306_tophat/data/Bowtie2Index/5-AZA_33-1_1.fastq \ $ /data/home/ysh980101/2306_tophat/data/Bowtie2Index/5-AZA_33-1_2.fastq tophatpy: tophat2 ì•ˆë¨¹ì–´ì„œ ì»¤ìŠ¤í…€í•œ ëª…ë ¹ì–´ (ì •ì‹ ëª…ë ¹ì–´ëŠ” tophat2) -o tophat_out_33-1: ì¶œë ¥ ë””ë ‰í† ë¦¬ ì„¤ì • --no-mixed: í˜ì–´ ì¤‘ í•˜ë‚˜ë§Œ ë§¤í•‘ë˜ë©´ ì œì™¸ -p 40: ë©€í‹°ìŠ¤ë ˆë”©, 40ê°œ ìŠ¤ë ˆë“œ ì‚¬ìš© /data3/PUBLIC_DATA/...dna.toplevel: reference genome FASTA (Bowtie2 ì¸ë±ìŠ¤ê°€ ì´ì™€ ë™ì¼í•œ ê²½ë¡œë¡œ ìˆì–´ì•¼ í•¨) 2ê°œì˜ paired-end read ì…ë ¥ cf) tophat alias í™•ì¸"><meta property="og:locale" content="en_us"><meta property="og:type" content="article"><meta property="article:section" content="docs"><meta property="article:published_time" content="2025-04-21T00:00:00+00:00"><meta property="article:modified_time" content="2025-04-21T00:00:00+00:00"><meta property="article:tag" content="2025-04"><title>RNA-seq Preprocessing using TopHat, SAMtools, HTSeq | Lifelog 2025</title><link rel=icon href=/favicon.png><link rel=manifest href=/manifest.json><link rel=canonical href=https://yshghid.github.io/docs/study/bioinformatics/bi7/><link rel=stylesheet href=/book.min.6217d077edb4189fd0578345e84bca1a884dfdee121ff8dc9a0f55cfe0852bc9.css integrity="sha256-YhfQd+20GJ/QV4NF6EvKGohN/e4SH/jcmg9Vz+CFK8k=" crossorigin=anonymous><script defer src=/fuse.min.js></script><script defer src=/en.search.min.5faf6ce9be181bcdd933f441cbd5b33a615b6940cb5168d6d338b9c3e68ce188.js integrity="sha256-X69s6b4YG83ZM/RBy9WzOmFbaUDLUWjW0zi5w+aM4Yg=" crossorigin=anonymous></script></head><body dir=ltr><input type=checkbox class="hidden toggle" id=menu-control>
<input type=checkbox class="hidden toggle" id=toc-control><main class="container flex"><aside class=book-menu><div class=book-menu-content><nav><h2 class=book-brand><a class="flex align-center" href=/><img src=/logo.png alt=Logo class=book-icon><span>Lifelog 2025</span></a></h2><div class="book-search hidden"><input type=text id=book-search-input placeholder=Search aria-label=Search maxlength=64 data-hotkeys=s/><div class="book-search-spinner hidden"></div><ul id=book-search-results></ul></div><script>document.querySelector(".book-search").classList.remove("hidden")</script><ul><li class=book-section-flat><span>ê¸°ë¡</span><ul><li><a href=/docs/hobby/book/>ì±…</a><ul></ul></li><li><a href=/docs/hobby/movie/>ì˜í™”</a><ul></ul></li><li><a href=/docs/hobby/daily/>ì¼ìƒ</a><ul></ul></li></ul></li><li class=book-section-flat><span>ê³µë¶€</span><ul><li><a href=/docs/study/tech/>tech</a><ul></ul></li><li><a href=/docs/study/etc/>etc</a><ul></ul></li></ul></li><li><a href=/docs/about/>ğ“‚‚ ğ“ˆ’ğ“¸ï»¿</a></li></ul></nav><script>(function(){var e=document.querySelector("aside .book-menu-content");addEventListener("beforeunload",function(){localStorage.setItem("menu.scrollTop",e.scrollTop)}),e.scrollTop=localStorage.getItem("menu.scrollTop")})()</script></div></aside><div class=book-page><header class=book-header><div class="flex align-center justify-between"><label for=menu-control><img src=/svg/menu.svg class=book-icon alt=Menu></label><h3>RNA-seq Preprocessing using TopHat, SAMtools, HTSeq</h3><label for=toc-control><img src=/svg/toc.svg class=book-icon alt="Table of Contents"></label></div><aside class="hidden clearfix"><nav id=TableOfContents><ul><li><a href=#1-tophat-ì‹¤í–‰>1. TopHat ì‹¤í–‰</a></li><li><a href=#2-samtools-ì •ë ¬>2. SAMtools ì •ë ¬</a></li><li><a href=#3-htseq-count-ì‹¤í–‰>3. HTSeq-count ì‹¤í–‰</a></li></ul></nav></aside></header><article class="markdown book-article"><h1 id=rna-seq-preprocessing-using-tophat-samtools-htseq>RNA-seq Preprocessing using TopHat, SAMtools, HTSeq
<a class=anchor href=#rna-seq-preprocessing-using-tophat-samtools-htseq>#</a></h1><h2 id=1-tophat-ì‹¤í–‰>1. TopHat ì‹¤í–‰
<a class=anchor href=#1-tophat-%ec%8b%a4%ed%96%89>#</a></h2><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-bash data-lang=bash><span style=display:flex><span>$ tophatpy -o tophat_out_33-1 --no-mixed -p <span style=color:#ae81ff>40</span> <span style=color:#ae81ff>\
</span></span></span><span style=display:flex><span><span style=color:#ae81ff></span>$ /data3/PUBLIC_DATA/ref_genomes/homo_sapiens/GRCh38/Homo_sapiens.GRCh38.dna.toplevel <span style=color:#ae81ff>\
</span></span></span><span style=display:flex><span><span style=color:#ae81ff></span>$ /data/home/ysh980101/2306_tophat/data/Bowtie2Index/5-AZA_33-1_1.fastq <span style=color:#ae81ff>\
</span></span></span><span style=display:flex><span><span style=color:#ae81ff></span>$ /data/home/ysh980101/2306_tophat/data/Bowtie2Index/5-AZA_33-1_2.fastq
</span></span></code></pre></div><ul><li><code>tophatpy</code>: tophat2 ì•ˆë¨¹ì–´ì„œ ì»¤ìŠ¤í…€í•œ ëª…ë ¹ì–´ (ì •ì‹ ëª…ë ¹ì–´ëŠ” tophat2)</li><li><code>-o tophat_out_33-1</code>: ì¶œë ¥ ë””ë ‰í† ë¦¬ ì„¤ì •</li><li><code>--no-mixed</code>: í˜ì–´ ì¤‘ í•˜ë‚˜ë§Œ ë§¤í•‘ë˜ë©´ ì œì™¸</li><li><code>-p 40</code>: ë©€í‹°ìŠ¤ë ˆë”©, 40ê°œ ìŠ¤ë ˆë“œ ì‚¬ìš©</li><li><code>/data3/PUBLIC_DATA/...dna.toplevel</code>: reference genome FASTA (Bowtie2 ì¸ë±ìŠ¤ê°€ ì´ì™€ ë™ì¼í•œ ê²½ë¡œë¡œ ìˆì–´ì•¼ í•¨)</li><li>2ê°œì˜ paired-end read ì…ë ¥</li></ul><p>cf) tophat alias í™•ì¸</p><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-bash data-lang=bash><span style=display:flex><span>view .bashrc
</span></span></code></pre></div><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-bash data-lang=bash><span style=display:flex><span>alias tophatpy<span style=color:#f92672>=</span><span style=color:#e6db74>&#39;/usr/local/src/tophat-2.0.13/src/tophat.py&#39;</span>
</span></span></code></pre></div><p>cf2) Bowtie Index Build ì•ˆí–ˆë‹¤ë©´?</p><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-bash data-lang=bash><span style=display:flex><span>bowtie2-build /data3/PUBLIC_DATA/ref_genomes/homo_sapiens/GRCh38/Homo_sapiens.GRCh38.dna.toplevel.fa <span style=color:#ae81ff>\
</span></span></span><span style=display:flex><span><span style=color:#ae81ff></span>/data3/PUBLIC_DATA/ref_genomes/homo_sapiens/GRCh38/Bowtie2Index/Homo_sapiens.GRCh38
</span></span></code></pre></div><h2 id=2-samtools-ì •ë ¬>2. SAMtools ì •ë ¬
<a class=anchor href=#2-samtools-%ec%a0%95%eb%a0%ac>#</a></h2><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-bash data-lang=bash><span style=display:flex><span>samtools sort -n TopHat/tophat_out_33-1/accepted_hits.bam -o TopHat/tophat_out_33-1/accepted_hits.sorted.bam
</span></span></code></pre></div><ul><li><code>-n</code>: ì´ë¦„(name) ê¸°ì¤€ ì •ë ¬ (HTSeqì—ì„œ ì´ë¦„ ê¸°ì¤€ ì •ë ¬ í•„ìš”)</li></ul><h2 id=3-htseq-count-ì‹¤í–‰>3. HTSeq-count ì‹¤í–‰
<a class=anchor href=#3-htseq-count-%ec%8b%a4%ed%96%89>#</a></h2><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-bash data-lang=bash><span style=display:flex><span>$ python -m HTSeq.scripts.count -s no -a <span style=color:#ae81ff>0</span> -i transcript_id <span style=color:#ae81ff>\
</span></span></span><span style=display:flex><span><span style=color:#ae81ff></span>$ --additional-attr<span style=color:#f92672>=</span>gene_id --additional-attr<span style=color:#f92672>=</span>gene_name --nonunique<span style=color:#f92672>=</span>all <span style=color:#ae81ff>\
</span></span></span><span style=display:flex><span><span style=color:#ae81ff></span>$ -c Count/33-1_count.tsv <span style=color:#ae81ff>\
</span></span></span><span style=display:flex><span><span style=color:#ae81ff></span>$ TopHat/tophat_out_33-1/accepted_hits.sorted.bam <span style=color:#ae81ff>\
</span></span></span><span style=display:flex><span><span style=color:#ae81ff></span>$ /data3/PUBLIC_DATA/ref_genomes/homo_sapiens/GRCh38/Homo_sapiens.GRCh38.110.chr_edited.gtf 
</span></span></code></pre></div><ul><li><p><code>TopHat/tophat_out_33-2/accepted_hits.sorted.bam</code>: ì •ë ¬ëœ BAM íŒŒì¼</p></li><li><p><code>-s no</code>: strand ì •ë³´ ë¬´ì‹œ</p></li><li><p><code>-a 0</code>: ìµœì†Œ alignment quality 0</p></li><li><p><code>-c</code>: count ê²°ê³¼ ì €ì¥ ê²½ë¡œ</p></li><li><p>Custom Parameters</p><ul><li><code>-i transcript_id</code>: count ê¸°ì¤€ feature ID (ì˜ˆ: exonì´ ì•„ë‹Œ transcript ìˆ˜ì¤€ìœ¼ë¡œ count)</li><li><code>--additional-attr</code>: gene_id, gene_name ë“± ì¶”ê°€ ì •ë³´ ê¸°ë¡</li><li><code>--nonunique=all</code>: ì—¬ëŸ¬ featureì— ë§¤í•‘ëœ readëŠ” ëª¨ë‘ count</li></ul></li></ul></article><footer class=book-footer><div class="flex flex-wrap justify-between"></div><script>(function(){function e(e){const t=window.getSelection(),n=document.createRange();n.selectNodeContents(e),t.removeAllRanges(),t.addRange(n)}document.querySelectorAll("pre code").forEach(t=>{t.addEventListener("click",function(){if(window.getSelection().toString())return;e(t.parentElement),navigator.clipboard&&navigator.clipboard.writeText(t.parentElement.textContent)})})})()</script></footer><div class=book-comments><script src=https://giscus.app/client.js data-repo=yshghid/yshghid.github.io data-repo-id=R_kgDONkMkNg data-category-id=DIC_kwDONkMkNs4CloJh data-mapping=pathname data-strict=0 data-reactions-enabled=1 data-emit-metadata=0 data-input-position=bottom data-theme=preferred_color_scheme data-lang=ko crossorigin=anonymous async></script></div><label for=menu-control class="hidden book-menu-overlay"></label></div><aside class=book-toc><div class=book-toc-content><nav id=TableOfContents><ul><li><a href=#1-tophat-ì‹¤í–‰>1. TopHat ì‹¤í–‰</a></li><li><a href=#2-samtools-ì •ë ¬>2. SAMtools ì •ë ¬</a></li><li><a href=#3-htseq-count-ì‹¤í–‰>3. HTSeq-count ì‹¤í–‰</a></li></ul></nav></div></aside></main></body></html>