<!doctype html><html lang=en-us dir=ltr><head><meta charset=UTF-8><meta name=viewport content="width=device-width,initial-scale=1"><meta name=description content="
  TopHat2+HTSeq vs Rsubread 비교
  #


  1. Methods
  #




비교 의의

Traditional 방법은 TopHat2+HTseq 조합이지만 오류도 넘 많이나고 Rsubread를 쓰면 빠르고 깔끔한데 왜 써야하지..? 싶어서 동일한 데이터(pair-end fastq)로 돌려봄.
HTseq에서 아래 코드를 수행할때 파라미터가 많은데 뭐가 다르게나오는지 모르겠어서 실험해봄.



Cases

Rsubread 사용
HTSeq 사용, -i gene_id --additional-attr=gene_name (exon 기준 count)
HTSeq 사용, -i transcript_id --additional-attr=gene_id --additional-attr=gene_name (transcript 기준 count)
HTSeq 사용, -i transcript_id --additional-attr=gene_id --additional-attr=gene_name --nonunique=all (여러 transcript에 매핑된 read는 모두 count)




  2. Result
  #

"><meta name=theme-color media="(prefers-color-scheme: light)" content="#ffffff"><meta name=theme-color media="(prefers-color-scheme: dark)" content="#343a40"><meta name=color-scheme content="light dark"><meta property="og:url" content="https://yshghid.github.io/docs/study/bioinformatics/bi9/"><meta property="og:site_name" content="Lifelog 2025"><meta property="og:title" content="TopHat2+HTSeq vs Rsubread 비교"><meta property="og:description" content="TopHat2+HTSeq vs Rsubread 비교 # 1. Methods # 비교 의의
Traditional 방법은 TopHat2+HTseq 조합이지만 오류도 넘 많이나고 Rsubread를 쓰면 빠르고 깔끔한데 왜 써야하지..? 싶어서 동일한 데이터(pair-end fastq)로 돌려봄. HTseq에서 아래 코드를 수행할때 파라미터가 많은데 뭐가 다르게나오는지 모르겠어서 실험해봄. Cases
Rsubread 사용 HTSeq 사용, -i gene_id --additional-attr=gene_name (exon 기준 count) HTSeq 사용, -i transcript_id --additional-attr=gene_id --additional-attr=gene_name (transcript 기준 count) HTSeq 사용, -i transcript_id --additional-attr=gene_id --additional-attr=gene_name --nonunique=all (여러 transcript에 매핑된 read는 모두 count) 2. Result #"><meta property="og:locale" content="en_us"><meta property="og:type" content="article"><meta property="article:section" content="docs"><meta property="article:published_time" content="2025-04-21T00:00:00+00:00"><meta property="article:modified_time" content="2025-04-21T00:00:00+00:00"><meta property="article:tag" content="2025-04"><title>TopHat2+HTSeq vs Rsubread 비교 | Lifelog 2025</title>
<link rel=icon href=/favicon.png><link rel=manifest href=/manifest.json><link rel=canonical href=https://yshghid.github.io/docs/study/bioinformatics/bi9/><link rel=stylesheet href=/book.min.6217d077edb4189fd0578345e84bca1a884dfdee121ff8dc9a0f55cfe0852bc9.css integrity="sha256-YhfQd+20GJ/QV4NF6EvKGohN/e4SH/jcmg9Vz+CFK8k=" crossorigin=anonymous><script defer src=/fuse.min.js></script><script defer src=/en.search.min.253a1c1aeab84b023ce4aefbdb06394d44afd757da7d875b7d722daa5490c79f.js integrity="sha256-JTocGuq4SwI85K772wY5TUSv11fafYdbfXItqlSQx58=" crossorigin=anonymous></script></head><body dir=ltr><input type=checkbox class="hidden toggle" id=menu-control>
<input type=checkbox class="hidden toggle" id=toc-control><main class="container flex"><aside class=book-menu><div class=book-menu-content><nav><h2 class=book-brand><a class="flex align-center" href=/><img src=/logo.png alt=Logo class=book-icon><span>Lifelog 2025</span></a></h2><div class="book-search hidden"><input type=text id=book-search-input placeholder=Search aria-label=Search maxlength=64 data-hotkeys=s/><div class="book-search-spinner hidden"></div><ul id=book-search-results></ul></div><script>document.querySelector(".book-search").classList.remove("hidden")</script><ul><li class=book-section-flat><span>기록</span><ul><li><a href=/docs/hobby/book/>책</a><ul></ul></li><li><a href=/docs/hobby/movie/>영화</a><ul></ul></li><li><a href=/docs/hobby/daily/>일상</a><ul></ul></li></ul></li><li class=book-section-flat><span>공부</span><ul><li><a href=/docs/study/tech/>Tech</a><ul></ul></li><li><a href=/docs/study/bioinformatics/>BI</a><ul></ul></li></ul></li><li><a href=/docs/about/>𓂂 𓈒𓏸﻿</a></li></ul></nav><script>(function(){var e=document.querySelector("aside .book-menu-content");addEventListener("beforeunload",function(){localStorage.setItem("menu.scrollTop",e.scrollTop)}),e.scrollTop=localStorage.getItem("menu.scrollTop")})()</script></div></aside><div class=book-page><header class=book-header><div class="flex align-center justify-between"><label for=menu-control><img src=/svg/menu.svg class=book-icon alt=Menu></label><h3>TopHat2+HTSeq vs Rsubread 비교</h3><label for=toc-control><img src=/svg/toc.svg class=book-icon alt="Table of Contents"></label></div><aside class="hidden clearfix"><nav id=TableOfContents><ul><li><a href=#1-methods>1. Methods</a></li><li><a href=#2-result>2. Result</a></li><li><a href=#3-결론>3. 결론</a></li></ul></nav></aside></header><article class="markdown book-article"><h1 id=tophat2htseq-vs-rsubread-비교>TopHat2+HTSeq vs Rsubread 비교
<a class=anchor href=#tophat2htseq-vs-rsubread-%eb%b9%84%ea%b5%90>#</a></h1><h2 id=1-methods>1. Methods
<a class=anchor href=#1-methods>#</a></h2><p><img src=https://github.com/user-attachments/assets/8cab1315-1ed9-4046-85c8-28ddbb813166 alt=image></p><ul><li><p>비교 의의</p><ul><li>Traditional 방법은 TopHat2+HTseq 조합이지만 오류도 넘 많이나고 Rsubread를 쓰면 빠르고 깔끔한데 왜 써야하지..? 싶어서 동일한 데이터(pair-end fastq)로 돌려봄.</li><li>HTseq에서 아래 코드를 수행할때 파라미터가 많은데 뭐가 다르게나오는지 모르겠어서 실험해봄.</li></ul></li><li><p>Cases</p><ul><li>Rsubread 사용</li><li>HTSeq 사용, <code>-i gene_id --additional-attr=gene_name</code> (exon 기준 count)</li><li>HTSeq 사용, <code>-i transcript_id --additional-attr=gene_id --additional-attr=gene_name</code> (transcript 기준 count)</li><li>HTSeq 사용, <code>-i transcript_id --additional-attr=gene_id --additional-attr=gene_name --nonunique=all</code> (여러 transcript에 매핑된 read는 모두 count)</li></ul></li></ul><h2 id=2-result>2. Result
<a class=anchor href=#2-result>#</a></h2><p><img src=https://github.com/user-attachments/assets/85251a62-07d9-41c0-bd0c-f6e44507c262 alt=image></p><ul><li><p>A1CF gene count</p><ul><li>Rsubread 사용: 378</li><li>HTSeq exon: 248</li><li>HTSeq transcript: 0</li><li>HTSeq transcript nonunique: 최대 343 (ENST00000373997 사용시)</li></ul></li><li><p>Rsubread와 HTseq-transcript-nonunique 버전이 개수가 제일 비슷하게 나왔다.</p></li></ul><p><img src=https://github.com/user-attachments/assets/d06f3a58-d983-4c9c-a880-30a23bfe8393 alt=image>
<img src=https://github.com/user-attachments/assets/b8ebdb9d-245f-4598-82f4-867d38e94f48 alt=image>
<img src=https://github.com/user-attachments/assets/17152795-14d6-45a8-9787-ed40ce1ed904 alt=image></p><ul><li>DEG, Pathway 분석 비교<ul><li>DEG 개수는 Rsubread 2612, TopHat-HTseq 2818이고 2191개 겹쳐서 비슷한것같음.</li><li>Pathway 분석 결과 중요한 term이었던 DNA methylation, Viral carcinogenesis를 포함해서 term과 p-adj도 비슷하게 나왔다.</li></ul></li></ul><h2 id=3-결론>3. 결론
<a class=anchor href=#3-%ea%b2%b0%eb%a1%a0>#</a></h2><ul><li>Rsubread 써도 될듯.</li><li>HTseq은 보통 <code>-i gene_id</code>를 쓰던데 count 많이 뽑고싶으면 <code>-i transcript_id --nonunique=all</code>한 후 count 젤많은 transcript id 써주면 될것같다!</li></ul></article><footer class=book-footer><div class="flex flex-wrap justify-between"></div><script>(function(){function e(e){const t=window.getSelection(),n=document.createRange();n.selectNodeContents(e),t.removeAllRanges(),t.addRange(n)}document.querySelectorAll("pre code").forEach(t=>{t.addEventListener("click",function(){if(window.getSelection().toString())return;e(t.parentElement),navigator.clipboard&&navigator.clipboard.writeText(t.parentElement.textContent)})})})()</script></footer><div class=book-comments><script src=https://giscus.app/client.js data-repo=yshghid/yshghid.github.io data-repo-id=R_kgDONkMkNg data-category-id=DIC_kwDONkMkNs4CloJh data-mapping=pathname data-strict=0 data-reactions-enabled=1 data-emit-metadata=0 data-input-position=bottom data-theme=preferred_color_scheme data-lang=ko crossorigin=anonymous async></script></div><label for=menu-control class="hidden book-menu-overlay"></label></div><aside class=book-toc><div class=book-toc-content><nav id=TableOfContents><ul><li><a href=#1-methods>1. Methods</a></li><li><a href=#2-result>2. Result</a></li><li><a href=#3-결론>3. 결론</a></li></ul></nav></div></aside></main></body></html>