<!doctype html><html lang=en-us dir=ltr><head><meta charset=UTF-8><meta name=viewport content="width=device-width,initial-scale=1"><meta name=description content><meta name=theme-color media="(prefers-color-scheme: light)" content="#ffffff"><meta name=theme-color media="(prefers-color-scheme: dark)" content="#343a40"><meta name=color-scheme content="light dark"><meta property="og:url" content="https://yshghid.github.io/categories/bioinformatics/"><meta property="og:site_name" content="Lifelog 2025"><meta property="og:title" content="Bioinformatics"><meta property="og:locale" content="en_us"><meta property="og:type" content="website"><title>Bioinformatics | Lifelog 2025</title>
<link rel=icon href=/favicon.png><link rel=manifest href=/manifest.json><link rel=canonical href=https://yshghid.github.io/categories/bioinformatics/><link rel=stylesheet href=/book.min.6217d077edb4189fd0578345e84bca1a884dfdee121ff8dc9a0f55cfe0852bc9.css integrity="sha256-YhfQd+20GJ/QV4NF6EvKGohN/e4SH/jcmg9Vz+CFK8k=" crossorigin=anonymous><script defer src=/fuse.min.js></script><script defer src=/en.search.min.55316e68c4613b515026ae3685b9d6df36d8834bc8f370ae691d2a3b6eb7b8a1.js integrity="sha256-VTFuaMRhO1FQJq42hbnW3zbYg0vI83CuaR0qO263uKE=" crossorigin=anonymous></script><link rel=alternate type=application/rss+xml href=https://yshghid.github.io/categories/bioinformatics/index.xml title="Lifelog 2025"></head><body dir=ltr><input type=checkbox class="hidden toggle" id=menu-control>
<input type=checkbox class="hidden toggle" id=toc-control><main class="container flex"><aside class=book-menu><div class=book-menu-content><nav><h2 class=book-brand><a class="flex align-center" href=/><img src=/logo.png alt=Logo class=book-icon><span>Lifelog 2025</span></a></h2><div class="book-search hidden"><input type=text id=book-search-input placeholder=Search aria-label=Search maxlength=64 data-hotkeys=s/><div class="book-search-spinner hidden"></div><ul id=book-search-results></ul></div><script>document.querySelector(".book-search").classList.remove("hidden")</script><ul><li class=book-section-flat><span>기록</span><ul><li><a href=/docs/hobby/book/>책</a><ul></ul></li><li><a href=/docs/hobby/movie/>영화</a><ul></ul></li><li><a href=/docs/hobby/daily/>일상</a><ul></ul></li></ul></li><li class=book-section-flat><span>공부</span><ul><li><a href=/docs/study/tech/>Tech</a><ul></ul></li><li><a href=/docs/study/bioinformatics/>BI</a><ul></ul></li></ul></li><li><a href=/docs/about/>𓂂 𓈒𓏸﻿</a></li></ul></nav><script>(function(){var e=document.querySelector("aside .book-menu-content");addEventListener("beforeunload",function(){localStorage.setItem("menu.scrollTop",e.scrollTop)}),e.scrollTop=localStorage.getItem("menu.scrollTop")})()</script></div></aside><div class=book-page><header class=book-header><div class="flex align-center justify-between"><label for=menu-control><img src=/svg/menu.svg class=book-icon alt=Menu></label><h3>Bioinformatics</h3><label for=toc-control><img src=/svg/toc.svg class=book-icon alt="Table of Contents"></label></div><aside class="hidden clearfix"><nav><ul><li class=book-section-flat><strong>Categories</strong><ul><li class="flex justify-between"><a href=/categories/sql/>Sql</a>
<span>2</span></li><li class="flex justify-between"><a href=/categories/bioinformatics/>Bioinformatics</a>
<span>18</span></li><li class="flex justify-between"><a href=/categories/r/>R</a>
<span>5</span></li><li class="flex justify-between"><a href=/categories/%EC%9D%BC%EC%83%81/>일상</a>
<span>13</span></li><li class="flex justify-between"><a href=/categories/python-%EC%A3%BC%EC%8B%9D/>Python, 주식</a>
<span>5</span></li><li class="flex justify-between"><a href=/categories/github/>Github</a>
<span>4</span></li><li class="flex justify-between"><a href=/categories/python/>Python</a>
<span>4</span></li><li class="flex justify-between"><a href=/categories/%EC%B7%A8%EC%A4%80/>취준</a>
<span>1</span></li><li class="flex justify-between"><a href=/categories/deep-learning/>Deep Learning</a>
<span>5</span></li><li class="flex justify-between"><a href=/categories/etc/>Etc</a>
<span>1</span></li><li class="flex justify-between"><a href=/categories/%EC%98%81%ED%99%94/>영화</a>
<span>1</span></li><li class="flex justify-between"><a href=/categories/%EC%B1%85/>책</a>
<span>10</span></li></ul></li><li class=book-section-flat><strong>Tags</strong><ul><li class="flex justify-between"><a href=/tags/2025-04/>2025-04</a>
<span>25</span></li><li class="flex justify-between"><a href=/tags/2025-03/>2025-03</a>
<span>3</span></li><li class="flex justify-between"><a href=/tags/2025-02/>2025-02</a>
<span>3</span></li><li class="flex justify-between"><a href=/tags/2025-01/>2025-01</a>
<span>1</span></li><li class="flex justify-between"><a href=/tags/2024-12/>2024-12</a>
<span>23</span></li><li class="flex justify-between"><a href=/tags/2024-11/>2024-11</a>
<span>1</span></li><li class="flex justify-between"><a href=/tags/2024-08/>2024-08</a>
<span>1</span></li><li class="flex justify-between"><a href=/tags/2024-06/>2024-06</a>
<span>1</span></li><li class="flex justify-between"><a href=/tags/2024-05/>2024-05</a>
<span>1</span></li><li class="flex justify-between"><a href=/tags/2024-03/>2024-03</a>
<span>1</span></li><li class="flex justify-between"><a href=/tags/2023-08/>2023-08</a>
<span>1</span></li><li class="flex justify-between"><a href=/tags/2023-01/>2023-01</a>
<span>1</span></li></ul></li></ul></nav></aside></header><article class="markdown book-post"><h2><a href=/docs/study/bioinformatics/bi13/>(주)마크로젠 서비스개발분석부 신입/경력 채용</a></h2><div class="flex align-center text-small book-post-date"><img src=/svg/calendar.svg class=book-icon alt>
<span>April 21, 2025</span></div><div class=text-small><a href=/categories/bioinformatics/>Bioinformatics</a></div><div class=text-small><a href=/tags/2025-04/>2025-04</a></div><p>(주)마크로젠 서비스개발분석부 신입/경력 채용 # 공고 바로가기</p></article><article class="markdown book-post"><h2><a href=/docs/study/bioinformatics/bi17/>2025년 제2기 질병관리청 국립보건연구원 공무직(연구원) 채용공고</a></h2><div class="flex align-center text-small book-post-date"><img src=/svg/calendar.svg class=book-icon alt>
<span>April 21, 2025</span></div><div class=text-small><a href=/categories/bioinformatics/>Bioinformatics</a></div><div class=text-small><a href=/tags/2025-04/>2025-04</a></div><p>2025년 제2기 질병관리청 국립보건연구원 공무직(연구원) 채용공고 # 공고 바로가기</p></article><article class="markdown book-post"><h2><a href=/docs/study/bioinformatics/bi5/>ChIP-seq Preprocessing: Trimming, Alignment, Peak Calling</a></h2><div class="flex align-center text-small book-post-date"><img src=/svg/calendar.svg class=book-icon alt>
<span>April 21, 2025</span></div><div class=text-small><a href=/categories/bioinformatics/>Bioinformatics</a></div><div class=text-small><a href=/tags/2025-04/>2025-04</a></div><p>ChIP-seq Preprocessing: Trimming, Alignment, Peak Calling # 1. Trimming # #chipseq_trimming.sh
#!/bin/bash # setting envs export bdir="/data3/projects/2022_KNU_EBV" export hg38_bowtieidx="/data3/PUBLIC_DATA/ref_genomes/homo_sapiens/hg38/hg38_bowtie_idx/hg38.fa" export hg38_bwaidx="/data3/PUBLIC_DATA/ref_genomes/homo_sapiens/hg38/hg38_bwa_index/hg38.fa" export ebv_bowtie2idx="/data3/PUBLIC_DATA/ref_genomes/Human_gammaherpesvirus_4_EBV/EBV_bowtie2_idx/NC_007605.1.fa" export ebv_bwaidx="/data3/PUBLIC_DATA/ref_genomes/Human_gammaherpesvirus_4_EBV/EBV_bwa_index/NC_007605.1.fa" ### SET Path ### cd /data3/RAW_DATA/2023_KNU_EBV/ChIP-seq ### TRIMMING data ### mkdir -p trimmed sampdir="/data3/RAW_DATA/2023_KNU_EBV/ChIP-seq" samplist=("Input" "p65" "RIgG") TRIMMOMATIC= "/data/packages/trimmomatic/Trimmomatic-0.39/trimmomatic-0.39.jar" for sampname in "${samplist[@]}"; do mkdir -p "trimmed/${sampname}" java -jar $TRIMMOMATIC PE -threads 40 -trimlog log1.txt $sampdir/${sampname}_1.fastq/${sampname}_1.fastq $sampdir/${sampname}_2.fastq/${sampname}_2.fastq $sampdir/trimmed/${sampname}/${sampname}_1.trimmed.fastq $sampdir/trimmed/${sampname}/${sampname}_1.up.trimmed.fastq $sampdir/trimmed/${sampname}/${sampname}_2.trimmed.fastq $sampdir/trimmed/${sampname}/${sampname}_2.up.trimmed.fastq ILLUMINACLIP:/data1/packages/trimmomatic/Trimmomatic-0.39/adapters/TruSeq3-PE.fa:2:30:10:2:True LEADING:3 TRAILING:3 MINLEN:36 done 2.
<a href=/docs/study/bioinformatics/bi5/>...</a></p></article><article class="markdown book-post"><h2><a href=/docs/study/bioinformatics/bi2/>Differential Gene Expression Analysis using Sleuth</a></h2><div class="flex align-center text-small book-post-date"><img src=/svg/calendar.svg class=book-icon alt>
<span>April 21, 2025</span></div><div class=text-small><a href=/categories/bioinformatics/>Bioinformatics</a>,
<a href=/categories/r/>R</a></div><div class=text-small><a href=/tags/2025-04/>2025-04</a></div><p>Differential Gene Expression Analysis using Sleuth # Load Package, Run Sleuth # require("sleuth") packageVersion("sleuth") library("gridExtra") library("cowplot") library("biomaRt") library(readr) setwd("/data/home/ysh980101/2307_kallisto") getwd() sample_id &lt;- dir(file.path("./")) sample_id &lt;- grep("^output_(150|con)", sample_id, value = TRUE) sample_id &lt;- substring(sample_id, 8) sample_id kal_dirs &lt;- file.path(paste0("./output_", sample_id)) s2c &lt;- read.table(file.path("./kallisto_demo_150_con.tsv"), header = TRUE, stringsAsFactors = FALSE, sep = "\t") s2c &lt;- dplyr::mutate(s2c, path = kal_dirs) s2c marts &lt;- listMarts() ensembl &lt;- useMart("ensembl") datasets &lt;- listDatasets(ensembl) filtered_datasets &lt;- datasets[grepl("hsapiens", datasets$dataset), ] hsapiens_mart &lt;- useMart("ensembl",dataset="hsapiens_gene_ensembl") datasets &lt;- listDatasets(hsapiens_mart) filtered_datasets &lt;- datasets[grepl("hsapiens", datasets$dataset), ] hsapiens_mart &lt;- useMart("ensembl",dataset="hsapiens_gene_ensembl",host="ensembl.
<a href=/docs/study/bioinformatics/bi2/>...</a></p></article><article class="markdown book-post"><h2><a href=/docs/study/bioinformatics/bi3/>Functional Enrichment Bubble Plot using gProfiler/ggplot2</a></h2><div class="flex align-center text-small book-post-date"><img src=/svg/calendar.svg class=book-icon alt>
<span>April 21, 2025</span></div><div class=text-small><a href=/categories/bioinformatics/>Bioinformatics</a>,
<a href=/categories/r/>R</a></div><div class=text-small><a href=/tags/2025-04/>2025-04</a></div><p>Functional Enrichment Bubble Plot using gProfiler/ggplot2 # Load Package # library(ggplot2) Set Path # setwd("/data-blog/bi3") getwd() '/data-blog/bi3' Functional Enrichment Bubble Plot # condition &lt;- '150_con' gpsource &lt;- 'GO:BP' #gpsource &lt;- 'REAC' df_c1 &lt;- read.csv(paste0("./sleuth_ward/gprofiler/gProfiler_",condition,"_termsize.csv")) df_c2 &lt;- read.csv(paste0("gProfiler_",condition,"_c2_padj0.1.csv")) df_c1 &lt;- df_c1[df_c1$source == gpsource, ] df_c2 &lt;- df_c2[df_c2$source == gpsource, ] df_c1$reg_type &lt;- 'down' df_c2$reg_type &lt;- 'up' df_c1$nlog &lt;- -abs(df_c1$negative_log10_of_adjusted_p_value) df_c2$nlog &lt;- abs(df_c2$negative_log10_of_adjusted_p_value) df_c1 &lt;- df_c1[order(df_c1$negative_log10_of_adjusted_p_value), ] df_c2 &lt;- df_c2[order(-df_c2$negative_log10_of_adjusted_p_value), ] df &lt;- rbind(df_c1, df_c2) ggplot(df, aes(x = reorder(term_name, nlog), y = negative_log10_of_adjusted_p_value, size = intersection_size, color = nlog)) + geom_point(alpha = 0.
<a href=/docs/study/bioinformatics/bi3/>...</a></p></article><article class="markdown book-post"><h2><a href=/docs/study/bioinformatics/bi7/>RNA-seq Preprocessing using TopHat, SAMtools, HTSeq</a></h2><div class="flex align-center text-small book-post-date"><img src=/svg/calendar.svg class=book-icon alt>
<span>April 21, 2025</span></div><div class=text-small><a href=/categories/bioinformatics/>Bioinformatics</a></div><div class=text-small><a href=/tags/2025-04/>2025-04</a></div><p>RNA-seq Preprocessing using TopHat, SAMtools, HTSeq # 1. TopHat 실행 # $ tophatpy -o tophat_out_33-1 --no-mixed -p 40 \ $ /data3/PUBLIC_DATA/ref_genomes/homo_sapiens/GRCh38/Homo_sapiens.GRCh38.dna.toplevel \ $ /data/home/ysh980101/2306_tophat/data/Bowtie2Index/5-AZA_33-1_1.fastq \ $ /data/home/ysh980101/2306_tophat/data/Bowtie2Index/5-AZA_33-1_2.fastq tophatpy: tophat2 안먹어서 커스텀한 명령어 (정식 명령어는 tophat2) -o tophat_out_33-1: 출력 디렉토리 설정 --no-mixed: 페어 중 하나만 매핑되면 제외 -p 40: 멀티스레딩, 40개 스레드 사용 /data3/PUBLIC_DATA/...dna.toplevel: reference genome FASTA (Bowtie2 인덱스가 이와 동일한 경로로 있어야 함) 2개의 paired-end read 입력 cf) tophat alias 확인
<a href=/docs/study/bioinformatics/bi7/>...</a></p></article><article class="markdown book-post"><h2><a href=/docs/study/bioinformatics/bi8/>RNA-seq 파이프라인 꿀조합 (Rsubread, edgeR)</a></h2><div class="flex align-center text-small book-post-date"><img src=/svg/calendar.svg class=book-icon alt>
<span>April 21, 2025</span></div><div class=text-small><a href=/categories/bioinformatics/>Bioinformatics</a></div><div class=text-small><a href=/tags/2025-04/>2025-04</a></div><p>RNA-seq 파이프라인 꿀조합 (Rsubread, edgeR) # 가장 오류 적게나는 조합! 1. Align RNA-seq # Load Packages # library(Rsubread) library(org.Mm.eg.db) library(gridExtra) library(reshape2) Set Path # indir = "/data/home/ysh980101/2504/mirna/data" outdir = "/data/home/ysh980101/2504/mirna/data" refpath = "/data/home/ysh980101/2406/data-gne/mm39.fa" setwd(indir) getwd() '/data/home/ysh980101/2504/mirna/data' Build Index # buildindex(basename = "mm39", reference = refpath) Read Alignment # files &lt;- list.files(pattern="\\.fastq\\.gz$", full.names=TRUE) bams &lt;- sub("\\.fastq\\.gz$", ".bam", files) samples &lt;- gsub("^\\.\\/|\\.fastq\\.gz$", "", files) targets &lt;- read.delim("target.txt", header=TRUE) align(index="mm39", readfile1=files, input_format="gzFASTQ", output_file=bams, nthreads=50) Quantification # fc = featureCounts(bams, isGTFAnnotationFile=TRUE, GTF.
<a href=/docs/study/bioinformatics/bi8/>...</a></p></article><article class="markdown book-post"><h2><a href=/docs/study/bioinformatics/bi9/>TopHat2+HTSeq vs Rsubread 비교</a></h2><div class="flex align-center text-small book-post-date"><img src=/svg/calendar.svg class=book-icon alt>
<span>April 21, 2025</span></div><div class=text-small><a href=/categories/bioinformatics/>Bioinformatics</a></div><div class=text-small><a href=/tags/2025-04/>2025-04</a></div><p>TopHat2+HTSeq vs Rsubread 비교 # 1. Methods # 비교 의의
Traditional 방법은 TopHat2+HTseq 조합이지만 오류도 넘 많이나고 Rsubread를 쓰면 빠르고 깔끔한데 왜 써야하지..? 싶어서 동일한 데이터(pair-end fastq)로 돌려봄. HTseq에서 아래 코드를 수행할때 파라미터가 많은데 뭐가 다르게나오는지 모르겠어서 실험해봄. Cases
Rsubread 사용 HTSeq 사용, -i gene_id --additional-attr=gene_name (exon 기준 count) HTSeq 사용, -i transcript_id --additional-attr=gene_id --additional-attr=gene_name (transcript 기준 count) HTSeq 사용, -i transcript_id --additional-attr=gene_id --additional-attr=gene_name --nonunique=all (여러 transcript에 매핑된 read는 모두 count) 2.
<a href=/docs/study/bioinformatics/bi9/>...</a></p></article><article class="markdown book-post"><h2><a href=/docs/study/bioinformatics/bi4/>Transcript Quantification using Kallisto Pseudoalignment</a></h2><div class="flex align-center text-small book-post-date"><img src=/svg/calendar.svg class=book-icon alt>
<span>April 21, 2025</span></div><div class=text-small><a href=/categories/bioinformatics/>Bioinformatics</a></div><div class=text-small><a href=/tags/2025-04/>2025-04</a></div><p>Transcript Quantification using Kallisto Pseudoalignment # 1. Build Index # $ kallisto index -i transcripts_cDNA.idx Homo_sapiens.GRCh38.cdna.all.fa.gz 2. Pseudoalign # $ kallisto quant -i transcripts_cDNA.idx -o output_150-1 -t 40 ../2306_tophat/data/Bowtie2Index/5-AZA_150-1_1_edited.fastq ../2306_tophat/data/Bowtie2Index/5-AZA_150-1_2_edited.fastq 3개 파일 생성 abundance.h5 - HDF5 binary file containing run info, abundance esimates, bootstrap estimates, and transcript length information length. This file can be read in by sleuth abundance.tsv - plaintext file of the abundance estimates. It does not contains bootstrap estimates.
<a href=/docs/study/bioinformatics/bi4/>...</a></p></article><article class="markdown book-post"><h2><a href=/docs/study/bioinformatics/bi6/>WGBS Preprocessing using Bismark</a></h2><div class="flex align-center text-small book-post-date"><img src=/svg/calendar.svg class=book-icon alt>
<span>April 21, 2025</span></div><div class=text-small><a href=/categories/bioinformatics/>Bioinformatics</a></div><div class=text-small><a href=/tags/2025-04/>2025-04</a></div><p>WGBS Preprocessing using Bismark # 1. Build Index # $ bowtie2-build Homo_sapiens.GRCh38.dna.toplevel.fa GRCh38 -p 40 2. Bam Sorting & Indexing # $ samtools sort KEB01_1_bismark_bt2_pe.bam -o KEB01_1_bismark_bt2_pe.sorted.bam $ samtools index KEB01_1_bismark_bt2_pe.sorted.bam 3. Methylation Extraction # $ bismark_methylation_extractor --gzip --bedGraph --buffer_size 10G --cytosine_report --genome_folder /data3/PUBLIC_DATA/ref_genomes/homo_sapiens/GRCh37_hg19/Homo_sapiens/Ensembl/GRCh37/Sequence/WholeGenomeFasta KEB01_1_bismark_bt2_pe.sorted.bam</p></article><ul class="pagination pagination-default"><li class="page-item disabled"><a aria-disabled=true aria-label=First class=page-link role=button tabindex=-1><span aria-hidden=true>&#171;&#171;</span></a></li><li class="page-item disabled"><a aria-disabled=true aria-label=Previous class=page-link role=button tabindex=-1><span aria-hidden=true>&#171;</span></a></li><li class="page-item active"><a aria-current=page aria-label="Page 1" class=page-link role=button>1</a></li><li class=page-item><a href=/categories/bioinformatics/page/2/ aria-label="Page 2" class=page-link role=button>2</a></li><li class=page-item><a href=/categories/bioinformatics/page/2/ aria-label=Next class=page-link role=button><span aria-hidden=true>&#187;</span></a></li><li class=page-item><a href=/categories/bioinformatics/page/2/ aria-label=Last class=page-link role=button><span aria-hidden=true>&#187;&#187;</span></a></li></ul><footer class=book-footer><div class="flex flex-wrap justify-between"></div><script>(function(){function e(e){const t=window.getSelection(),n=document.createRange();n.selectNodeContents(e),t.removeAllRanges(),t.addRange(n)}document.querySelectorAll("pre code").forEach(t=>{t.addEventListener("click",function(){if(window.getSelection().toString())return;e(t.parentElement),navigator.clipboard&&navigator.clipboard.writeText(t.parentElement.textContent)})})})()</script></footer><label for=menu-control class="hidden book-menu-overlay"></label></div><aside class=book-toc><div class=book-toc-content><nav><ul><li class=book-section-flat><strong>Categories</strong><ul><li class="flex justify-between"><a href=/categories/sql/>Sql</a>
<span>2</span></li><li class="flex justify-between"><a href=/categories/bioinformatics/>Bioinformatics</a>
<span>18</span></li><li class="flex justify-between"><a href=/categories/r/>R</a>
<span>5</span></li><li class="flex justify-between"><a href=/categories/%EC%9D%BC%EC%83%81/>일상</a>
<span>13</span></li><li class="flex justify-between"><a href=/categories/python-%EC%A3%BC%EC%8B%9D/>Python, 주식</a>
<span>5</span></li><li class="flex justify-between"><a href=/categories/github/>Github</a>
<span>4</span></li><li class="flex justify-between"><a href=/categories/python/>Python</a>
<span>4</span></li><li class="flex justify-between"><a href=/categories/%EC%B7%A8%EC%A4%80/>취준</a>
<span>1</span></li><li class="flex justify-between"><a href=/categories/deep-learning/>Deep Learning</a>
<span>5</span></li><li class="flex justify-between"><a href=/categories/etc/>Etc</a>
<span>1</span></li><li class="flex justify-between"><a href=/categories/%EC%98%81%ED%99%94/>영화</a>
<span>1</span></li><li class="flex justify-between"><a href=/categories/%EC%B1%85/>책</a>
<span>10</span></li></ul></li><li class=book-section-flat><strong>Tags</strong><ul><li class="flex justify-between"><a href=/tags/2025-04/>2025-04</a>
<span>25</span></li><li class="flex justify-between"><a href=/tags/2025-03/>2025-03</a>
<span>3</span></li><li class="flex justify-between"><a href=/tags/2025-02/>2025-02</a>
<span>3</span></li><li class="flex justify-between"><a href=/tags/2025-01/>2025-01</a>
<span>1</span></li><li class="flex justify-between"><a href=/tags/2024-12/>2024-12</a>
<span>23</span></li><li class="flex justify-between"><a href=/tags/2024-11/>2024-11</a>
<span>1</span></li><li class="flex justify-between"><a href=/tags/2024-08/>2024-08</a>
<span>1</span></li><li class="flex justify-between"><a href=/tags/2024-06/>2024-06</a>
<span>1</span></li><li class="flex justify-between"><a href=/tags/2024-05/>2024-05</a>
<span>1</span></li><li class="flex justify-between"><a href=/tags/2024-03/>2024-03</a>
<span>1</span></li><li class="flex justify-between"><a href=/tags/2023-08/>2023-08</a>
<span>1</span></li><li class="flex justify-between"><a href=/tags/2023-01/>2023-01</a>
<span>1</span></li></ul></li></ul></nav></div></aside></main></body></html>