<!DOCTYPE html>
<html lang="en-us" dir="ltr">
<head><script src="/livereload.js?mindelay=10&amp;v=2&amp;port=1313&amp;path=livereload" data-no-instant defer></script>
  <meta charset="UTF-8">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<meta name="description" content="DMR 분석 (methylKit) # #2025-07-28
# #1 Load packages
library(&#34;methylKit&#34;) library(&#34;genomation&#34;) library(&#34;GenomicRanges&#34;) # #2 Set path
setwd(&#34;/data/home/ysh980101/2309_5-aza/Bismark/sorted_n&#34;) getwd() &#39;/data1/home/ysh980101/2309_5-aza/Bismark/sorted_n&#39; # #3 Load data
# Define the list containing the bismark coverage files. covlist &lt;- list( &#34;KEB1/KEB01_1_bismark_bt2_pe.sorted_n.deduplicated.bismark.cov.gz&#34;, &#34;KEB2/KEB02_1_bismark_bt2_pe.sorted_n.deduplicated.bismark.cov.gz&#34;, &#34;KEB4/KEB04_1_bismark_bt2_pe.sorted_n.deduplicated.bismark.cov.gz&#34;) myobj_lowCov &lt;- methRead(covlist, sample.id=list(&#34;KEB01&#34;,&#34;KEB02&#34;,&#34;KEB04&#34;), pipeline = &#34;bismarkCoverage&#34;, assembly=&#34;hg38&#34;, treatment=c(0,1,2), mincov = 3 ) tiles &lt;- tileMethylCounts(myobj_lowCov,win.size=1000,step.size=1000,cov.bases = 3 tiles.norm &lt;- normalizeCoverage(tiles, method = &#34;median&#34;) meth.tiles &lt;- unite(tiles.norm, destrand=FALSE) meth.tiles meth.tilesDf = getData(meth.">
<meta name="theme-color" media="(prefers-color-scheme: light)" content="#ffffff">
<meta name="theme-color" media="(prefers-color-scheme: dark)" content="#343a40">
<meta name="color-scheme" content="light dark"><meta property="og:url" content="http://localhost:1313/docs/study/bioinformatics/bi31/">
  <meta property="og:site_name" content=" ">
  <meta property="og:title" content="DMR 분석 (methylKit)">
  <meta property="og:description" content="DMR 분석 (methylKit) # #2025-07-28
# #1 Load packages
library(&#34;methylKit&#34;) library(&#34;genomation&#34;) library(&#34;GenomicRanges&#34;) # #2 Set path
setwd(&#34;/data/home/ysh980101/2309_5-aza/Bismark/sorted_n&#34;) getwd() &#39;/data1/home/ysh980101/2309_5-aza/Bismark/sorted_n&#39; # #3 Load data
# Define the list containing the bismark coverage files. covlist &lt;- list( &#34;KEB1/KEB01_1_bismark_bt2_pe.sorted_n.deduplicated.bismark.cov.gz&#34;, &#34;KEB2/KEB02_1_bismark_bt2_pe.sorted_n.deduplicated.bismark.cov.gz&#34;, &#34;KEB4/KEB04_1_bismark_bt2_pe.sorted_n.deduplicated.bismark.cov.gz&#34;) myobj_lowCov &lt;- methRead(covlist, sample.id=list(&#34;KEB01&#34;,&#34;KEB02&#34;,&#34;KEB04&#34;), pipeline = &#34;bismarkCoverage&#34;, assembly=&#34;hg38&#34;, treatment=c(0,1,2), mincov = 3 ) tiles &lt;- tileMethylCounts(myobj_lowCov,win.size=1000,step.size=1000,cov.bases = 3 tiles.norm &lt;- normalizeCoverage(tiles, method = &#34;median&#34;) meth.tiles &lt;- unite(tiles.norm, destrand=FALSE) meth.tiles meth.tilesDf = getData(meth.">
  <meta property="og:locale" content="en_us">
  <meta property="og:type" content="article">
    <meta property="article:section" content="docs">
    <meta property="article:published_time" content="2025-07-28T00:00:00+00:00">
    <meta property="article:modified_time" content="2025-07-28T00:00:00+00:00">
    <meta property="article:tag" content="2025-07">
<title>DMR 분석 (methylKit) |  </title>
<link rel="icon" href="/favicon.png" >
<link rel="manifest" href="/manifest.json">
<link rel="canonical" href="http://localhost:1313/docs/study/bioinformatics/bi31/">
<link rel="stylesheet" href="/book.min.30a7836b6a89342da3b88e7afd1036166aeced16c8de12df060ded2031837886.css" integrity="sha256-MKeDa2qJNC2juI56/RA2Fmrs7RbI3hLfBg3tIDGDeIY=" crossorigin="anonymous">
  <script defer src="/fuse.min.js"></script>
  <script defer src="/en.search.min.3ef7b64c66957d37c21de489954fc2b1c9d60aef1d40e40867f503a0617f04e3.js" integrity="sha256-Pve2TGaVfTfCHeSJlU/CscnWCu8dQOQIZ/UDoGF/BOM=" crossorigin="anonymous"></script>

  

<!--
Made with Book Theme
https://github.com/alex-shpak/hugo-book
-->
  
</head>
<body dir="ltr">
  <input type="checkbox" class="hidden toggle" id="menu-control" />
  <input type="checkbox" class="hidden toggle" id="toc-control" />
  <main class="container flex">
    <aside class="book-menu">
      <div class="book-menu-content">
        
  <nav>
<h2 class="book-brand">
  <a class="flex align-center" href="/"><img src="/logo.png" alt="Logo" class="book-icon" /><span> </span>
  </a>
</h2>


<div class="book-search hidden">
  <input type="text" id="book-search-input" placeholder="Search" aria-label="Search" maxlength="64" data-hotkeys="s/" />
  <div class="book-search-spinner hidden"></div>
  <ul id="book-search-results"></ul>
</div>
<script>document.querySelector(".book-search").classList.remove("hidden")</script>















  
  <ul>
    
      
        <li class="book-section-flat" >
          
  
  

  
    <span>기록</span>
  

          
  <ul>
    
      
        <li>
          
  
  

  
    <a href="/docs/hobby/daily/" class="">일상</a>
  

          
  <ul>
    
  </ul>

        </li>
      
    
      
        <li>
          
  
  

  
    <a href="/docs/hobby/book/" class="">글</a>
  

          
  <ul>
    
  </ul>

        </li>
      
    
  </ul>

        </li>
      
    
      
        <li class="book-section-flat" >
          
  
  

  
    <span>공부</span>
  

          
  <ul>
    
      
        <li>
          
  
  

  
    <a href="/docs/study/ai/" class="">AI</a>
  

          
  <ul>
    
  </ul>

        </li>
      
    
      
        <li>
          
  
  

  
    <a href="/docs/study/bioinformatics/" class="">Bioinformatics</a>
  

          
  <ul>
    
  </ul>

        </li>
      
    
      
        <li>
          
  
  

  
    <a href="/docs/study/be/" class="">BE</a>
  

          
  <ul>
    
  </ul>

        </li>
      
    
      
        <li>
          
  
  

  
    <a href="/docs/study/fe/" class="">FE</a>
  

          
  <ul>
    
  </ul>

        </li>
      
    
      
        <li>
          
  
  

  
    <a href="/docs/study/career/" class="">취업</a>
  

          
  <ul>
    
  </ul>

        </li>
      
    
  </ul>

        </li>
      
    
  </ul>














</nav>




  <script>(function(){var e=document.querySelector("aside .book-menu-content");addEventListener("beforeunload",function(){localStorage.setItem("menu.scrollTop",e.scrollTop)}),e.scrollTop=localStorage.getItem("menu.scrollTop")})()</script>


 
      </div>
    </aside>

    <div class="book-page">
      <header class="book-header">
        
  <div class="flex align-center justify-between">
  <label for="menu-control">
    <img src="/svg/menu.svg" class="book-icon" alt="Menu" />
  </label>

  <h3>DMR 분석 (methylKit)</h3>

  <label for="toc-control">
    
    <img src="/svg/toc.svg" class="book-icon" alt="Table of Contents" />
    
  </label>
</div>


  
  <aside class="hidden clearfix">
    
  
<nav id="TableOfContents">
  <ul>
    <li>
      <ul>
        <li><a href="#heading"></a></li>
        <li><a href="#heading-1"></a></li>
        <li><a href="#heading-2"></a></li>
        <li><a href="#heading-3"></a></li>
      </ul>
    </li>
  </ul>

  <ul>
    <li>
      <ul>
        <li><a href="#5-save">5. Save</a></li>
        <li><a href="#heading-5"></a></li>
        <li><a href="#heading-6"></a></li>
        <li><a href="#heading-7"></a></li>
        <li><a href="#heading-8"></a></li>
      </ul>
    </li>
  </ul>
</nav>



  </aside>
  
 
      </header>

      
      
  <article class="markdown book-article"><h1 id="dmr-분석-methylkit">
  DMR 분석 (methylKit)
  <a class="anchor" href="#dmr-%eb%b6%84%ec%84%9d-methylkit">#</a>
</h1>
<p>#2025-07-28</p>
<hr>
<h3 id="heading">
  
  <a class="anchor" href="#heading">#</a>
</h3>
<p>#1 Load packages</p>
<div class="highlight"><pre tabindex="0" style="color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4;"><code class="language-R" data-lang="R"><span style="display:flex;"><span><span style="color:#a6e22e">library</span>(<span style="color:#e6db74">&#34;methylKit&#34;</span>)
</span></span><span style="display:flex;"><span><span style="color:#a6e22e">library</span>(<span style="color:#e6db74">&#34;genomation&#34;</span>)
</span></span><span style="display:flex;"><span><span style="color:#a6e22e">library</span>(<span style="color:#e6db74">&#34;GenomicRanges&#34;</span>)
</span></span></code></pre></div><h3 id="heading-1">
  
  <a class="anchor" href="#heading-1">#</a>
</h3>
<p>#2 Set path</p>
<div class="highlight"><pre tabindex="0" style="color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4;"><code class="language-R" data-lang="R"><span style="display:flex;"><span><span style="color:#a6e22e">setwd</span>(<span style="color:#e6db74">&#34;/data/home/ysh980101/2309_5-aza/Bismark/sorted_n&#34;</span>)
</span></span><span style="display:flex;"><span><span style="color:#a6e22e">getwd</span>()
</span></span></code></pre></div><div class="highlight"><pre tabindex="0" style="color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4;"><code class="language-plain" data-lang="plain"><span style="display:flex;"><span>&#39;/data1/home/ysh980101/2309_5-aza/Bismark/sorted_n&#39;
</span></span></code></pre></div><h3 id="heading-2">
  
  <a class="anchor" href="#heading-2">#</a>
</h3>
<p>#3 Load data</p>
<div class="highlight"><pre tabindex="0" style="color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4;"><code class="language-R" data-lang="R"><span style="display:flex;"><span><span style="color:#75715e"># Define the list containing the bismark coverage files.</span>
</span></span><span style="display:flex;"><span>covlist <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">list</span>(
</span></span><span style="display:flex;"><span>   <span style="color:#e6db74">&#34;KEB1/KEB01_1_bismark_bt2_pe.sorted_n.deduplicated.bismark.cov.gz&#34;</span>,
</span></span><span style="display:flex;"><span>   <span style="color:#e6db74">&#34;KEB2/KEB02_1_bismark_bt2_pe.sorted_n.deduplicated.bismark.cov.gz&#34;</span>,
</span></span><span style="display:flex;"><span>   <span style="color:#e6db74">&#34;KEB4/KEB04_1_bismark_bt2_pe.sorted_n.deduplicated.bismark.cov.gz&#34;</span>)
</span></span><span style="display:flex;"><span>myobj_lowCov <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">methRead</span>(covlist,
</span></span><span style="display:flex;"><span>           sample.id<span style="color:#f92672">=</span><span style="color:#a6e22e">list</span>(<span style="color:#e6db74">&#34;KEB01&#34;</span>,<span style="color:#e6db74">&#34;KEB02&#34;</span>,<span style="color:#e6db74">&#34;KEB04&#34;</span>),
</span></span><span style="display:flex;"><span>           pipeline <span style="color:#f92672">=</span> <span style="color:#e6db74">&#34;bismarkCoverage&#34;</span>,
</span></span><span style="display:flex;"><span>           assembly<span style="color:#f92672">=</span><span style="color:#e6db74">&#34;hg38&#34;</span>,
</span></span><span style="display:flex;"><span>           treatment<span style="color:#f92672">=</span><span style="color:#a6e22e">c</span>(<span style="color:#ae81ff">0</span>,<span style="color:#ae81ff">1</span>,<span style="color:#ae81ff">2</span>),
</span></span><span style="display:flex;"><span>           mincov <span style="color:#f92672">=</span> <span style="color:#ae81ff">3</span>
</span></span><span style="display:flex;"><span>           )
</span></span><span style="display:flex;"><span>tiles <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">tileMethylCounts</span>(myobj_lowCov,win.size<span style="color:#f92672">=</span><span style="color:#ae81ff">1000</span>,step.size<span style="color:#f92672">=</span><span style="color:#ae81ff">1000</span>,cov.bases <span style="color:#f92672">=</span> <span style="color:#ae81ff">3</span>
</span></span><span style="display:flex;"><span>
</span></span><span style="display:flex;"><span>tiles.norm <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">normalizeCoverage</span>(tiles, method <span style="color:#f92672">=</span> <span style="color:#e6db74">&#34;median&#34;</span>)
</span></span><span style="display:flex;"><span>meth.tiles <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">unite</span>(tiles.norm, destrand<span style="color:#f92672">=</span><span style="color:#66d9ef">FALSE</span>)
</span></span><span style="display:flex;"><span>meth.tiles
</span></span></code></pre></div><img width="1062" height="786" alt="image" src="https://github.com/user-attachments/assets/2ab50f79-7f69-4b35-a150-b40beb54709c" />
<div class="highlight"><pre tabindex="0" style="color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4;"><code class="language-R" data-lang="R"><span style="display:flex;"><span>meth.tilesDf <span style="color:#f92672">=</span> <span style="color:#a6e22e">getData</span>(meth.tiles)
</span></span><span style="display:flex;"><span>meth.tilesDf_filtered <span style="color:#f92672">&lt;-</span> meth.tilesDf[<span style="color:#f92672">!</span><span style="color:#a6e22e">grepl</span>(<span style="color:#e6db74">&#34;\\.|\\_|MT&#34;</span>, meth.tilesDf<span style="color:#f92672">$</span>chr), ]
</span></span><span style="display:flex;"><span>meth.tilesDf_filtered
</span></span></code></pre></div><img width="1043" height="861" alt="image" src="https://github.com/user-attachments/assets/4e79725b-da91-4be6-a71f-f62d4401e883" />
<h3 id="heading-3">
  
  <a class="anchor" href="#heading-3">#</a>
</h3>
<p>#4 Test: keb01 vs keb02</p>
<div class="highlight"><pre tabindex="0" style="color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4;"><code class="language-R" data-lang="R"><span style="display:flex;"><span>meth.tilesDf.1vs2 <span style="color:#f92672">&lt;-</span> meth.tilesDf_filtered[, <span style="color:#ae81ff">1</span><span style="color:#f92672">:</span><span style="color:#ae81ff">10</span>]
</span></span><span style="display:flex;"><span>meth.tilesDf.1vs2
</span></span></code></pre></div><img width="841" height="744" alt="image" src="https://github.com/user-attachments/assets/0f2f6a50-877a-4c77-9e64-3308f42e8361" />
<div class="highlight"><pre tabindex="0" style="color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4;"><code class="language-R" data-lang="R"><span style="display:flex;"><span>meth.tiles.1vs2<span style="color:#f92672">=</span><span style="color:#a6e22e">new</span>(<span style="color:#e6db74">&#34;methylBase&#34;</span>,meth.tilesDf.1vs2,sample.ids<span style="color:#f92672">=</span><span style="color:#a6e22e">c</span>(<span style="color:#e6db74">&#34;KEB01&#34;</span>,<span style="color:#e6db74">&#34;KEB02&#34;</span>),
</span></span><span style="display:flex;"><span>                       assembly<span style="color:#f92672">=</span><span style="color:#e6db74">&#34;hg38&#34;</span>,treatment<span style="color:#f92672">=</span><span style="color:#a6e22e">c</span>(<span style="color:#ae81ff">0</span>,<span style="color:#ae81ff">1</span>))
</span></span><span style="display:flex;"><span>
</span></span><span style="display:flex;"><span>myDiff <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">calculateDiffMeth</span>(meth.tiles.1vs2,
</span></span><span style="display:flex;"><span>                            overdispersion <span style="color:#f92672">=</span> <span style="color:#e6db74">&#34;MN&#34;</span>,
</span></span><span style="display:flex;"><span>                            adjust<span style="color:#f92672">=</span><span style="color:#e6db74">&#34;BH&#34;</span>)
</span></span><span style="display:flex;"><span>myDiff <span style="color:#f92672">&lt;-</span> myDiff<span style="color:#a6e22e">[order</span>(myDiff<span style="color:#f92672">$</span>qvalue), ]
</span></span><span style="display:flex;"><span>myDiff
</span></span></code></pre></div><img width="766" height="807" alt="image" src="https://github.com/user-attachments/assets/12ffce09-8aa4-47c7-be34-36090dfdc74c" />
<h1 id="heading-4">
  
  <a class="anchor" href="#heading-4">#</a>
</h1>
<h3 id="5-save">
  5. Save
  <a class="anchor" href="#5-save">#</a>
</h3>
<div class="highlight"><pre tabindex="0" style="color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4;"><code class="language-R" data-lang="R"><span style="display:flex;"><span>myDiffDf <span style="color:#f92672">=</span> <span style="color:#a6e22e">getData</span>(myDiff)
</span></span><span style="display:flex;"><span><span style="color:#a6e22e">write.table</span>(myDiffDf, file <span style="color:#f92672">=</span> <span style="color:#e6db74">&#34;/data/home/ysh980101/2310/res_total/myDiff1vs2.tsv&#34;</span>, sep <span style="color:#f92672">=</span> <span style="color:#e6db74">&#34;\t&#34;</span>, quote <span style="color:#f92672">=</span> <span style="color:#66d9ef">FALSE</span>, row.names <span style="color:#f92672">=</span> <span style="color:#66d9ef">FALSE</span>)
</span></span></code></pre></div><h3 id="heading-5">
  
  <a class="anchor" href="#heading-5">#</a>
</h3>
<p>#6 CpG filtering</p>
<p>차등 메틸화된 모든 region에 대한 test 결과가 myDiff에 있는데 CpG island 부분만 남겨서 downstream 분석을 진행한다.</p>
<div class="highlight"><pre tabindex="0" style="color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4;"><code class="language-R" data-lang="R"><span style="display:flex;"><span>cpgi.df <span style="color:#f92672">=</span> <span style="color:#a6e22e">read.table</span>(<span style="color:#e6db74">&#34;/data/home/ysh980101/2310/data/hg38_CpGIslands.chr.bed.txt&#34;</span>, header <span style="color:#f92672">=</span> <span style="color:#66d9ef">FALSE</span>, stringsAsFactors<span style="color:#f92672">=</span><span style="color:#66d9ef">FALSE</span>)
</span></span><span style="display:flex;"><span>cpgi.df <span style="color:#f92672">=</span> cpgi.df<span style="color:#a6e22e">[grep</span>(<span style="color:#e6db74">&#34;_&#34;</span>,cpgi.df[,<span style="color:#ae81ff">1</span>],invert<span style="color:#f92672">=</span><span style="color:#66d9ef">TRUE</span>),]
</span></span><span style="display:flex;"><span>cpgi.gr <span style="color:#f92672">=</span> <span style="color:#a6e22e">GRanges</span>(seqnames<span style="color:#f92672">=</span>cpgi.df[,<span style="color:#ae81ff">1</span>],ranges<span style="color:#f92672">=</span><span style="color:#a6e22e">IRanges</span>(start<span style="color:#f92672">=</span>cpgi.df[,<span style="color:#ae81ff">2</span>],end<span style="color:#f92672">=</span>cpgi.df[,<span style="color:#ae81ff">3</span>]))
</span></span><span style="display:flex;"><span>myDiff.sbo <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">selectByOverlap</span>(myDiff,cpgi.gr)
</span></span><span style="display:flex;"><span>
</span></span><span style="display:flex;"><span>myDiff.sboDf <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">getData</span>(myDiff.sbo)
</span></span><span style="display:flex;"><span>myDiff.sboDf <span style="color:#f92672">&lt;-</span> myDiff.sboDf<span style="color:#a6e22e">[order</span>(myDiff.sboDf<span style="color:#f92672">$</span>qvalue), ]
</span></span><span style="display:flex;"><span><span style="color:#a6e22e">write.table</span>(myDiff.sboDf, file <span style="color:#f92672">=</span> <span style="color:#e6db74">&#34;/data/home/ysh980101/2310/res_total_cpgfilt/myDiff.sbo1vs2.tsv&#34;</span>, sep <span style="color:#f92672">=</span> <span style="color:#e6db74">&#34;\t&#34;</span>, quote <span style="color:#f92672">=</span> <span style="color:#66d9ef">FALSE</span>, row.names <span style="color:#f92672">=</span> <span style="color:#66d9ef">FALSE</span>)
</span></span></code></pre></div><h3 id="heading-6">
  
  <a class="anchor" href="#heading-6">#</a>
</h3>
<p>#7 Gene region annotation</p>
<div class="highlight"><pre tabindex="0" style="color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4;"><code class="language-R" data-lang="R"><span style="display:flex;"><span>myDiff.sbo25p <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">getMethylDiff</span>(myDiff.sbo,
</span></span><span style="display:flex;"><span>                        difference<span style="color:#f92672">=</span><span style="color:#ae81ff">0</span>,
</span></span><span style="display:flex;"><span>                        qvalue<span style="color:#f92672">=</span><span style="color:#ae81ff">0.05</span>)
</span></span><span style="display:flex;"><span>myDiff.sbo25p <span style="color:#f92672">&lt;-</span> myDiff.sbo25p<span style="color:#a6e22e">[order</span>(myDiff.sbo25p<span style="color:#f92672">$</span>qvalue),]
</span></span><span style="display:flex;"><span>
</span></span><span style="display:flex;"><span><span style="color:#75715e">#gene region annotation</span>
</span></span><span style="display:flex;"><span>refseq_anot <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">readTranscriptFeatures</span>(<span style="color:#e6db74">&#34;/data/home/ysh980101/2310/data/hg38_NCBI_RefSeq.chr.bed.txt&#34;</span>)
</span></span><span style="display:flex;"><span>myDiff.sbo25p.anot <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">annotateWithGeneParts</span>(target <span style="color:#f92672">=</span> <span style="color:#a6e22e">as</span>(myDiff.sbo25p,<span style="color:#e6db74">&#34;GRanges&#34;</span>),feature <span style="color:#f92672">=</span> refseq_anot)
</span></span><span style="display:flex;"><span>dist_tss <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">getAssociationWithTSS</span>(myDiff.sbo25p.anot)
</span></span><span style="display:flex;"><span>dist_tss
</span></span></code></pre></div><img width="979" height="747" alt="image" src="https://github.com/user-attachments/assets/19d08218-583f-4550-a2ed-ec370e593a2f" />
<div class="highlight"><pre tabindex="0" style="color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4;"><code class="language-R" data-lang="R"><span style="display:flex;"><span><span style="color:#a6e22e">plotTargetAnnotation</span>(myDiff.sbo25p.anot, main <span style="color:#f92672">=</span> <span style="color:#e6db74">&#34;Differential Methylation Annotation&#34;</span>)
</span></span></code></pre></div><img width="1081" height="417" alt="image" src="https://github.com/user-attachments/assets/36f8d789-4d0f-4dea-95a6-09c8940041ae" />
<div class="highlight"><pre tabindex="0" style="color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4;"><code class="language-R" data-lang="R"><span style="display:flex;"><span>cpg_anot <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">readFeatureFlank</span>(<span style="color:#e6db74">&#34;/data/home/ysh980101/2310/data/hg38_CpGIslands.chr.bed.txt&#34;</span>, feature.flank.name <span style="color:#f92672">=</span> <span style="color:#a6e22e">c</span>(<span style="color:#e6db74">&#34;CpGi&#34;</span>, <span style="color:#e6db74">&#34;shores&#34;</span>), flank<span style="color:#f92672">=</span><span style="color:#ae81ff">2000</span>)
</span></span><span style="display:flex;"><span>diffCpGann <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">annotateWithFeatureFlank</span>(<span style="color:#a6e22e">as</span>(myDiff.sbo25p,<span style="color:#e6db74">&#34;GRanges&#34;</span>), feature <span style="color:#f92672">=</span> cpg_anot<span style="color:#f92672">$</span>CpGi, flank <span style="color:#f92672">=</span> cpg_anot<span style="color:#f92672">$</span>shores, feature.name <span style="color:#f92672">=</span> <span style="color:#e6db74">&#34;CpGi&#34;</span>, flank.name <span style="color:#f92672">=</span> <span style="color:#e6db74">&#34;shores&#34;</span>)
</span></span><span style="display:flex;"><span><span style="color:#a6e22e">plotTargetAnnotation</span>(diffCpGann, main <span style="color:#f92672">=</span> <span style="color:#e6db74">&#34;Differential Methylation Annotation&#34;</span>)
</span></span></code></pre></div><img width="963" height="374" alt="image" src="https://github.com/user-attachments/assets/0398d795-068f-4695-82a5-3a521d38f0d9" />
<p>(는 cpg island만 필터링했기때문에 당연한 결과)</p>
<h3 id="heading-7">
  
  <a class="anchor" href="#heading-7">#</a>
</h3>
<p>#8 Save result</p>
<div class="highlight"><pre tabindex="0" style="color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4;"><code class="language-R" data-lang="R"><span style="display:flex;"><span><span style="color:#a6e22e">write.table</span>(dist_tss, file <span style="color:#f92672">=</span> <span style="color:#e6db74">&#34;/data/home/ysh980101/2310/res_total_cpgfilt/dist_tss.1vs2.tsv&#34;</span>, sep <span style="color:#f92672">=</span> <span style="color:#e6db74">&#34;\t&#34;</span>, quote <span style="color:#f92672">=</span> <span style="color:#66d9ef">FALSE</span>, row.names <span style="color:#f92672">=</span> <span style="color:#66d9ef">FALSE</span>)
</span></span></code></pre></div><h3 id="heading-8">
  
  <a class="anchor" href="#heading-8">#</a>
</h3>
<p>#cf</p>
<ul>
<li>갠적으로 재밋었던 작업이었는데 (생전첨보는툴들의 향연) 결과가 너무 별로라 논문에실리지는 못한 비운의작업 이었당</li>
<li>교수님께서 코드 달라셔서 드렸는데 &ldquo;샘플수늘린것밖에 달라진게 없는데 자꾸 오류가난다 아무리봐도 문제가없다 한번봐달라&quot;셔서 한 30분 같이 씨름했었는데
<ul>
<li>문제는? 없었음. 혹시나해서 터미널에서 돌리니까 돌아갔다. 그냥 jupyter에서 안돌아간거였고 교수님이 보시고 역시 r은 쓰레기라고 하셨다 .. 맞긴해..</li>
</ul>
</li>
</ul>
<h1 id="heading-9">
  
  <a class="anchor" href="#heading-9">#</a>
</h1>
</article>
 
      

      <footer class="book-footer">
        
  <div class="flex flex-wrap justify-between">





</div>



  <script>(function(){function e(e){const t=window.getSelection(),n=document.createRange();n.selectNodeContents(e),t.removeAllRanges(),t.addRange(n)}document.querySelectorAll("pre code").forEach(t=>{t.addEventListener("click",function(){if(window.getSelection().toString())return;e(t.parentElement),navigator.clipboard&&navigator.clipboard.writeText(t.parentElement.textContent)})})})()</script>


 
        
      </footer>

      
  
  <div class="book-comments"><script src="https://giscus.app/client.js"
        data-repo="yshghid/yshghid.github.io"
        data-repo-id="R_kgDONkMkNg"
        data-category-id="DIC_kwDONkMkNs4CloJh"
        data-mapping="pathname"
        data-strict="0"
        data-reactions-enabled="1"
        data-emit-metadata="0"
        data-input-position="bottom"
        data-theme="preferred_color_scheme"
        data-lang="ko"
        crossorigin="anonymous"
        async>
</script>
</div>
  
 

      <label for="menu-control" class="hidden book-menu-overlay"></label>
    </div>

    
    <aside class="book-toc">
      <div class="book-toc-content">
        
  
<nav id="TableOfContents">
  <ul>
    <li>
      <ul>
        <li><a href="#heading"></a></li>
        <li><a href="#heading-1"></a></li>
        <li><a href="#heading-2"></a></li>
        <li><a href="#heading-3"></a></li>
      </ul>
    </li>
  </ul>

  <ul>
    <li>
      <ul>
        <li><a href="#5-save">5. Save</a></li>
        <li><a href="#heading-5"></a></li>
        <li><a href="#heading-6"></a></li>
        <li><a href="#heading-7"></a></li>
        <li><a href="#heading-8"></a></li>
      </ul>
    </li>
  </ul>
</nav>


 
      </div>
    </aside>
    
  </main>

  
</body>
</html>












