---
date : 2025-07-28
tags: ['2025-07']
categories: ['bioinformatics']
bookHidden: true
title: "methylKit: DMR 분석"
---

# methylKit: DMR 분석

#2025-07-28

---

### 1. Load packages

```R
library("methylKit")
library("genomation")
library("GenomicRanges")
```

### 2. Set path

```R
setwd("/data/home/ysh980101/2309_5-aza/Bismark/sorted_n")
getwd()
```
```plain text
'/data1/home/ysh980101/2309_5-aza/Bismark/sorted_n'
```

### 3. Load data

```R
# Define the list containing the bismark coverage files.
covlist <- list(
   "KEB1/KEB01_1_bismark_bt2_pe.sorted_n.deduplicated.bismark.cov.gz",
   "KEB2/KEB02_1_bismark_bt2_pe.sorted_n.deduplicated.bismark.cov.gz",
   "KEB4/KEB04_1_bismark_bt2_pe.sorted_n.deduplicated.bismark.cov.gz")
myobj_lowCov <- methRead(covlist,
           sample.id=list("KEB01","KEB02","KEB04"),
           pipeline = "bismarkCoverage",
           assembly="hg38",
           treatment=c(0,1,2),
           mincov = 3
           )
tiles <- tileMethylCounts(myobj_lowCov,win.size=1000,step.size=1000,cov.bases = 3

tiles.norm <- normalizeCoverage(tiles, method = "median")
meth.tiles <- unite(tiles.norm, destrand=FALSE)
meth.tiles
```

<img width="1062" height="786" alt="image" src="https://github.com/user-attachments/assets/2ab50f79-7f69-4b35-a150-b40beb54709c" />

```R
meth.tilesDf = getData(meth.tiles)
meth.tilesDf_filtered <- meth.tilesDf[!grepl("\\.|\\_|MT", meth.tilesDf$chr), ]
meth.tilesDf_filtered
```

<img width="1043" height="861" alt="image" src="https://github.com/user-attachments/assets/4e79725b-da91-4be6-a71f-f62d4401e883" />

### 4. Test: keb01 vs keb02

```R
meth.tilesDf.1vs2 <- meth.tilesDf_filtered[, 1:10]
meth.tilesDf.1vs2
```
<img width="841" height="744" alt="image" src="https://github.com/user-attachments/assets/0f2f6a50-877a-4c77-9e64-3308f42e8361" />

```R
meth.tiles.1vs2=new("methylBase",meth.tilesDf.1vs2,sample.ids=c("KEB01","KEB02"),
                       assembly="hg38",treatment=c(0,1))

myDiff <- calculateDiffMeth(meth.tiles.1vs2,
                            overdispersion = "MN",
                            adjust="BH")
myDiff <- myDiff[order(myDiff$qvalue), ]
myDiff
```

<img width="766" height="807" alt="image" src="https://github.com/user-attachments/assets/12ffce09-8aa4-47c7-be34-36090dfdc74c" />


### 5. Save

```R
myDiffDf = getData(myDiff)
write.table(myDiffDf, file = "/data/home/ysh980101/2310/res_total/myDiff1vs2.tsv", sep = "\t", quote = FALSE, row.names = FALSE)
```

### 6. CpG filtering

차등 메틸화된 모든 region에 대한 test 결과가 myDiff에 있는데 CpG island 부분만 남겨서 downstream 분석을 진행한다.

```R
cpgi.df = read.table("/data/home/ysh980101/2310/data/hg38_CpGIslands.chr.bed.txt", header = FALSE, stringsAsFactors=FALSE)
cpgi.df = cpgi.df[grep("_",cpgi.df[,1],invert=TRUE),]
cpgi.gr = GRanges(seqnames=cpgi.df[,1],ranges=IRanges(start=cpgi.df[,2],end=cpgi.df[,3]))
myDiff.sbo <- selectByOverlap(myDiff,cpgi.gr)

myDiff.sboDf <- getData(myDiff.sbo)
myDiff.sboDf <- myDiff.sboDf[order(myDiff.sboDf$qvalue), ]
write.table(myDiff.sboDf, file = "/data/home/ysh980101/2310/res_total_cpgfilt/myDiff.sbo1vs2.tsv", sep = "\t", quote = FALSE, row.names = FALSE)
```

### 7. Gene region annotation

```R
myDiff.sbo25p <- getMethylDiff(myDiff.sbo,
                        difference=0,
                        qvalue=0.05)
myDiff.sbo25p <- myDiff.sbo25p[order(myDiff.sbo25p$qvalue),]

#gene region annotation
refseq_anot <- readTranscriptFeatures("/data/home/ysh980101/2310/data/hg38_NCBI_RefSeq.chr.bed.txt")
myDiff.sbo25p.anot <- annotateWithGeneParts(target = as(myDiff.sbo25p,"GRanges"),feature = refseq_anot)
dist_tss <- getAssociationWithTSS(myDiff.sbo25p.anot)
dist_tss
```

<img width="979" height="747" alt="image" src="https://github.com/user-attachments/assets/19d08218-583f-4550-a2ed-ec370e593a2f" />


```R
plotTargetAnnotation(myDiff.sbo25p.anot, main = "Differential Methylation Annotation")
```

<img width="1081" height="417" alt="image" src="https://github.com/user-attachments/assets/36f8d789-4d0f-4dea-95a6-09c8940041ae" />

```R
cpg_anot <- readFeatureFlank("/data/home/ysh980101/2310/data/hg38_CpGIslands.chr.bed.txt", feature.flank.name = c("CpGi", "shores"), flank=2000)
diffCpGann <- annotateWithFeatureFlank(as(myDiff.sbo25p,"GRanges"), feature = cpg_anot$CpGi, flank = cpg_anot$shores, feature.name = "CpGi", flank.name = "shores")
plotTargetAnnotation(diffCpGann, main = "Differential Methylation Annotation")
```

<img width="963" height="374" alt="image" src="https://github.com/user-attachments/assets/0398d795-068f-4695-82a5-3a521d38f0d9" />

(는 cpg island만 필터링했기때문에 당연한 결과)

### 8. Save result

```R
write.table(dist_tss, file = "/data/home/ysh980101/2310/res_total_cpgfilt/dist_tss.1vs2.tsv", sep = "\t", quote = FALSE, row.names = FALSE)
```
