---
date : 2025-08-03
tags: ['2025-08']
categories: ['bioinformatics']
bookHidden: true
title: "DHC 약물연구 #1 연구정리"
---

# DHC 약물연구 #1 연구정리 

#2025-08-03

---

### 0. 제목

[Dihydrotestosterone-androgen receptor signaling suppresses EBV-positive gastric cancer through DNA demethylation-mediated viral reactivation](https://link.springer.com/article/10.1007/s10120-025-01626-6)


### 1. 참여 파트

```plain text
Materials and Methods
  └── 2. *RNA-seq analysis
  └── 13. *Whole genome bisulfite sequencing (WGBS)
  └── 14. *Bioinformatics analysis of methylome
```


### 2. RNA-seq analysis

> RNA-seq analysis was performed on SNU719 cells treated with 100 nM DHT for 24 h, with EtOH-treated cells as control. Total RNA was extracted using TRIzol® reagent (Invitrogen, Waltham, MA, USA) according to manufacturer’s instructions. RNA concentration was calculated using Quant-IT RiboGreen (#R11490, Invitrogen). RNA integrity was assessed using TapeStation RNA ScreenTape (#5067-5576, Agilent Technologies, Santa Clara, CA, USA), with only samples having RIN values > 7.0 used for library construction. Libraries were prepared with 0.5 μg total RNA using Illumina TruSeq Stranded Total RNA Library Prep Gold Kit (#20020599, Illumina, San Diego, CA, USA). Libraries were quantified using KAPA Library Quantification Kits (#KK4854, KAPA Biosystems, Bath, UK) and qualified using TapeStation D1000 ScreenTape (#5067-5582, Agilent Technologies). Paired-end (2 × 100 bp) sequencing was performed on Illumina NovaSeq by Macrogen Incorporated (Seoul, Korea). Sequencing reads were processed using Trimmomatic v0.38 to remove adapter sequences and trim low-quality bases. Cleaned reads were aligned to the Homo sapiens genome (GRCh38) using HISAT v2.1.0. Aligned data were sorted and indexed using SAMtools v1.9 [27, 28]. Transcripts were assembled and quantified using StringTie v2.1.3b. Differential gene expression analysis was performed using edgeR v3.40.2 with TMM normalization. Significance was assessed using edgeR exactTest, with p-values adjusted using the Benjamini–Hochberg algorithm. Significant genes were filtered by |fold change|≥ 2 and raw p-value < 0.05. Gene enrichment and functional annotation analyses were performed using gProfiler against GO and KEGG databases. All data analyses and visualizations were performed using R 4.2.2.

* Read processing: Trimmomatic
* Sequence alignment: HISAT, SAMtools
* DE analysis: edgeR
* Gene enrichment analysis: gProfiler

### 3. Whole genome bisulfite sequencing (WGBS)

> Genomic DNA was extracted from SNU719 cells treated with DHT (0, 100, or 5000 nM) for 48 h using the DNeasy Blood and Tissue Kit (Cat. #69504, Qiagen, Redwood, CA, USA) according to manufacturer's instructions. Extracted DNA was quantified using Invitrogen's Qubit 4.0 Fluorometer (Thermo Fisher Scientific, Waltham, MA, USA). DNA methylation libraries were prepared using the NEBNext Enzymatic Methyl-seq kit (Cat. #E7120L, New England Biolabs, Ipswich, MA, USA) with 200 ng of sheared DNA input. The method detected 5mC and 5hmC using two enzymatic reactions. First, TET2 and T4-BGT converted 5mC and 5hmC into products resistant to APOBEC3A deamination. Second, APOBEC3A deaminated unmodified cytosines to uracils. Following TET2 treatment, denaturation was performed at 85 °C for 10 min followed by ice incubation. The APOBEC reaction was conducted immediately afterward. PCR amplification followed (98 °C denaturation for 10 s, 62 °C annealing for 30 s, 65 °C extension for 60 s) with 200 ng DNA input. The resulting samples were named DHT0 (0 nM DHT), DHT100 (100 nM DHT), and DHT5000 (5000 nM DHT).

### 4. Bioinformatics analysis of methylome

> WGBS samples (DHT0, DHT100, and DHT5000) were aligned to the hg38 human reference genome using Bismark v0.22.3 with the bismark_genome_preparation function. Bowtie2 v2.3.4.1 was used for read alignment. Duplicate reads were removed using deduplicate_bismark, and methylation counts were extracted using bismark_methylation_extractor with a minimum read coverage threshold of 5. This identified 313,958, 227,225, and 393,751 methylated and non-methylated sites for DHT0, DHT100, and DHT5000 samples, respectively. Genome-wide methylation profiles were filtered to gene promoter regions (2 kb upstream of TSS) using Bedtools v2.27.1, identifying 34,061, 13,170, and 35,802 methylated sites within promoter regions in DHT0, DHT100, and DHT5000 samples, respectively. Methylated read counts were summed per promoter region, resulting in 4167, 3098, and 5091 unique promoter regions across the three samples. The union set comprised 8449 promoter regions for downstream analysis. Methylation counts were log2-transformed and quantile normalized to remove sample bias.

